{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "51c34492-c376-48b3-9410-f8ba0b180e97",
   "metadata": {},
   "source": [
    "# Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1a5fac2",
   "metadata": {},
   "source": [
    "This notebook contains the code necessary to conduct all data analyses for the *Pan* 3d Genome project. I have organized this largely by order of appearance in the manuscript; however, a few sections may be out of order. Use the Table of Contents below to navigate to specific analyses."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4dc045cd",
   "metadata": {},
   "source": [
    "## Table of Contents"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f52ef6b4",
   "metadata": {},
   "source": [
    "- [Notebook Setup](#notebooksetup)\n",
    "- [Data Description](#datadescription)\n",
    "    - [N Windows and Comparisons](#nwindowscomparisons)\n",
    "    - [Divergence Score Distribution](#divergencescoredistribution)\n",
    "- [*Pan*-*Homo* Divergence Distribution Comparison](#panhomodivergence)\n",
    "- [Minimally Divergent Windows and Ultraconserved TAD Boundaries](#okhovatetal2023)\n",
    "- [Lineage Comparison](#lineagecomparison)\n",
    "- [Hierarchical Clustering](#hierachicalclustering)\n",
    "    - [Establish N Clusters, Cluster Size, and Cluster Composition](#establishclusters)\n",
    "    - [Single Divergent Individual Clustering Windows](#singledivergentindividualclusteringwindows)\n",
    "    - [Lineage-Specific Clustering Windows](#lineagespecificclusteringwindows)\n",
    "    - [Multiple Divergent Individuals Clustering Windows](#multipledivergentindividualsclusteringwindows)\n",
    "    - [Chr21 Window Topologies](#chr21windowtopologies)\n",
    "    - [Window Topology Divergence Differences](#windowtopologydivergencedifferences)\n",
    "    - [Three, Four, and Five Cluster Window Topologies](#threefourfiveclusterwindowtopologies)\n",
    "- [Bonobo-Chimpanzee Windows](#bonobochimpanzeewindows)\n",
    "    - [Genes Total and Genes per Window](#genestotalgenesperwindow)\n",
    "    - [phastCons Elements](#phastconselements)\n",
    "- [Phenotype Enrichment](#phenotypeenrichment)\n",
    "- [*In Silico* Mutagenesis for Bonobo-Specific Variants](#ppninsilicomutagenesis)\n",
    "    - [Ancestral Allele](#ancestralallele)\n",
    "    - [Explained Divergence](#explaineddivergence)\n",
    "    - [CTCF](#ctcf)\n",
    "    - [Chromatin Contact Effect](#chromatincontacteffect)\n",
    "- [*In Silico* Mutagenesis for Chimpanzee Lineage-Specific Variants](#ptrinsilicomutagenesis)\n",
    "- [Bonobo-Chimpanzee Gene Expression Differences](#geneexpression)\n",
    "- [Cell Type Comparison](#celltypecomparison)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9f8b735",
   "metadata": {},
   "source": [
    "## Notebook Setup <a class = 'anchor' id = 'notebooksetup'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "692752fe",
   "metadata": {},
   "source": [
    "Load all needed packages, change directories, and load the HFF comparisons dataframe that we previously generated in the last notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b4bc6306",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import pybedtools\n",
    "\n",
    "from scipy.stats import fisher_exact\n",
    "from scipy.stats import kstest\n",
    "from scipy.stats import kruskal\n",
    "from scipy.stats import mannwhitneyu\n",
    "from scipy.stats import spearmanr\n",
    "\n",
    "pd.options.display.max_columns = 100\n",
    "pd.options.display.max_rows = 500"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e0abbe9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import scipy.cluster.hierarchy as sch\n",
    "from scipy.spatial.distance import pdist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "94ee43be",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.font_manager as font_manager\n",
    "arial_path = '/wynton/home/capra/cbrand/miniconda3/envs/jupyter/fonts/Arial.ttf'\n",
    "arial = font_manager.FontProperties(fname = arial_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "60a46371",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/wynton/group/capra/projects/pan_3d_genome/data\n"
     ]
    }
   ],
   "source": [
    "cd /wynton/group/capra/projects/pan_3d_genome/data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f6846214",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ind1</th>\n",
       "      <th>ind2</th>\n",
       "      <th>lineages</th>\n",
       "      <th>chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>spearman</th>\n",
       "      <th>divergence</th>\n",
       "      <th>seq_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pte-ptt</td>\n",
       "      <td>chr10</td>\n",
       "      <td>1572864</td>\n",
       "      <td>chr10_1572864</td>\n",
       "      <td>0.000168</td>\n",
       "      <td>0.987655</td>\n",
       "      <td>0.012345</td>\n",
       "      <td>2803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pte-ptt</td>\n",
       "      <td>chr10</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr10_2097152</td>\n",
       "      <td>0.000481</td>\n",
       "      <td>0.969809</td>\n",
       "      <td>0.030191</td>\n",
       "      <td>2715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pte-ptt</td>\n",
       "      <td>chr10</td>\n",
       "      <td>2621440</td>\n",
       "      <td>chr10_2621440</td>\n",
       "      <td>0.001675</td>\n",
       "      <td>0.996398</td>\n",
       "      <td>0.003602</td>\n",
       "      <td>2849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pte-ptt</td>\n",
       "      <td>chr10</td>\n",
       "      <td>3145728</td>\n",
       "      <td>chr10_3145728</td>\n",
       "      <td>0.000323</td>\n",
       "      <td>0.997899</td>\n",
       "      <td>0.002101</td>\n",
       "      <td>2606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pte-ptt</td>\n",
       "      <td>chr10</td>\n",
       "      <td>3670016</td>\n",
       "      <td>chr10_3670016</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>0.996732</td>\n",
       "      <td>0.003268</td>\n",
       "      <td>2594</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ind1    ind2 lineages    chr  window_start         window       mse  \\\n",
       "0  Akwaya-Jean  Alfred  pte-ptt  chr10       1572864  chr10_1572864  0.000168   \n",
       "1  Akwaya-Jean  Alfred  pte-ptt  chr10       2097152  chr10_2097152  0.000481   \n",
       "2  Akwaya-Jean  Alfred  pte-ptt  chr10       2621440  chr10_2621440  0.001675   \n",
       "3  Akwaya-Jean  Alfred  pte-ptt  chr10       3145728  chr10_3145728  0.000323   \n",
       "4  Akwaya-Jean  Alfred  pte-ptt  chr10       3670016  chr10_3670016  0.000143   \n",
       "\n",
       "   spearman  divergence  seq_diff  \n",
       "0  0.987655    0.012345      2803  \n",
       "1  0.969809    0.030191      2715  \n",
       "2  0.996398    0.003602      2849  \n",
       "3  0.997899    0.002101      2606  \n",
       "4  0.996732    0.003268      2594  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/dataframes/HFF_comparisons.txt', sep = '\\t', header = 0)\n",
    "comparisons.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49d34cb1",
   "metadata": {},
   "source": [
    "Double check that the frame is the size that it should be."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "599ce3c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6669390"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51e0075c",
   "metadata": {},
   "source": [
    "## Data Description <a class = 'anchor' id = 'datadescription'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29266968",
   "metadata": {},
   "source": [
    "### N Windows and Comparisons <a class = 'anchor' id = 'nwindowscomparisons'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60ea91d6",
   "metadata": {},
   "source": [
    "Let's generate some basic statistics for this dataset before diving into specific analyses. First, let's confirm the total number of windows and the number of windows per chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "20107d62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4420"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(comparisons['window'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d1d43187",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10</th>\n",
       "      <td>208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11</th>\n",
       "      <td>226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12</th>\n",
       "      <td>210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13</th>\n",
       "      <td>166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14</th>\n",
       "      <td>151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15</th>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr16</th>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17</th>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr18</th>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19</th>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr20</th>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr21</th>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2A</th>\n",
       "      <td>162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2B</th>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3</th>\n",
       "      <td>346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4</th>\n",
       "      <td>324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5</th>\n",
       "      <td>255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr6</th>\n",
       "      <td>292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr7</th>\n",
       "      <td>232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8</th>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9</th>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX</th>\n",
       "      <td>151</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         N\n",
       "chr       \n",
       "chr1   341\n",
       "chr10  208\n",
       "chr11  226\n",
       "chr12  210\n",
       "chr13  166\n",
       "chr14  151\n",
       "chr15  120\n",
       "chr16  102\n",
       "chr17   99\n",
       "chr18  126\n",
       "chr19   70\n",
       "chr20   96\n",
       "chr21   58\n",
       "chr22   48\n",
       "chr2A  162\n",
       "chr2B  223\n",
       "chr3   346\n",
       "chr4   324\n",
       "chr5   255\n",
       "chr6   292\n",
       "chr7   232\n",
       "chr8   236\n",
       "chr9   178\n",
       "chrX   151"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "windows_df = pd.DataFrame(comparisons['window'].unique(), columns = ['window'])\n",
    "windows_df = windows_df['window'].str.split('_', expand = True).rename(columns = {0:'chr', 1:'window_start'})\n",
    "windows_df.groupby(['chr']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84400439",
   "metadata": {},
   "source": [
    "How many comparisons are there per autosomal window? X chromosome window?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f19aa3f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1540"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(comparisons[comparisons['window'] == 'chr1_1048576'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "82e1b9a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "630"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(comparisons[comparisons['window'] == 'chrX_4718592'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4fc9b706",
   "metadata": {},
   "source": [
    "How many comparisons are there per chromosome?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "07b08cb9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>525140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10</th>\n",
       "      <td>320320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11</th>\n",
       "      <td>348040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12</th>\n",
       "      <td>323400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13</th>\n",
       "      <td>255640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14</th>\n",
       "      <td>232540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15</th>\n",
       "      <td>184800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr16</th>\n",
       "      <td>157080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17</th>\n",
       "      <td>152460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr18</th>\n",
       "      <td>194040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19</th>\n",
       "      <td>107800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr20</th>\n",
       "      <td>147840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr21</th>\n",
       "      <td>89320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22</th>\n",
       "      <td>73920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2A</th>\n",
       "      <td>249480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2B</th>\n",
       "      <td>343420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3</th>\n",
       "      <td>532840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4</th>\n",
       "      <td>498960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5</th>\n",
       "      <td>392700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr6</th>\n",
       "      <td>449680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr7</th>\n",
       "      <td>357280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8</th>\n",
       "      <td>363440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9</th>\n",
       "      <td>274120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX</th>\n",
       "      <td>95130</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            N\n",
       "chr          \n",
       "chr1   525140\n",
       "chr10  320320\n",
       "chr11  348040\n",
       "chr12  323400\n",
       "chr13  255640\n",
       "chr14  232540\n",
       "chr15  184800\n",
       "chr16  157080\n",
       "chr17  152460\n",
       "chr18  194040\n",
       "chr19  107800\n",
       "chr20  147840\n",
       "chr21   89320\n",
       "chr22   73920\n",
       "chr2A  249480\n",
       "chr2B  343420\n",
       "chr3   532840\n",
       "chr4   498960\n",
       "chr5   392700\n",
       "chr6   449680\n",
       "chr7   357280\n",
       "chr8   363440\n",
       "chr9   274120\n",
       "chrX    95130"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons.groupby('chr').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d330641c",
   "metadata": {},
   "source": [
    "What is the distribution of lineage comparisons for a single autosomal window?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a42cd8b5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lineages</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ppn-ptr</th>\n",
       "      <td>423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pts-ptt</th>\n",
       "      <td>272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pts-ptv</th>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ptt-ptv</th>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pts-pts</th>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ptt-ptt</th>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-pts</th>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-ptt</th>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-ptv</th>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ppn-ppn</th>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ptv-ptv</th>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-pte</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            N\n",
       "lineages     \n",
       "ppn-ptr   423\n",
       "pts-ptt   272\n",
       "pts-ptv   153\n",
       "ptt-ptv   144\n",
       "pts-pts   136\n",
       "ptt-ptt   120\n",
       "pte-pts    85\n",
       "pte-ptt    80\n",
       "pte-ptv    45\n",
       "ppn-ppn    36\n",
       "ptv-ptv    36\n",
       "pte-pte    10"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[comparisons['window'] == 'chr1_1048576'].groupby('lineages').size().to_frame('N').sort_values(by = 'N', ascending = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19929b9b",
   "metadata": {},
   "source": [
    "For a single chrX window?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f9f2452b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lineages</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ppn-ptr</th>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pts-ptt</th>\n",
       "      <td>121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pts-pts</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pts-ptv</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ptt-ptt</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ptt-ptv</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-pts</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-ptt</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ppn-ppn</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-ptv</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ptv-ptv</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-pte</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            N\n",
       "lineages     \n",
       "ppn-ptr   203\n",
       "pts-ptt   121\n",
       "pts-pts    55\n",
       "pts-ptv    55\n",
       "ptt-ptt    55\n",
       "ptt-ptv    55\n",
       "pte-pts    22\n",
       "pte-ptt    22\n",
       "ppn-ppn    21\n",
       "pte-ptv    10\n",
       "ptv-ptv    10\n",
       "pte-pte     1"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[comparisons['window'] == 'chrX_4718592'].groupby('lineages').size().to_frame('N').sort_values(by = 'N', ascending = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71816ac7",
   "metadata": {},
   "source": [
    "### Divergence Score Distribution <a class = 'anchor' id = 'divergencescoredistribution'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b968ebf6",
   "metadata": {},
   "source": [
    "What does the distribution of divergence scores look like?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "55903eb5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2.492599999737166e-07"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons['divergence'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2248f22f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.873604563894"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons['divergence'].max()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "917dcd7a-97bb-478a-a1e5-c58870ab84a3",
   "metadata": {},
   "source": [
    "How many comparisons have a divergence score < 0.01?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "94a35038-cf57-4d61-bbbe-29feedb516a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5539567"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(comparisons[comparisons['divergence'] < 0.01])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ad1498df-0801-43a5-b698-7404c3c36d41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.830595751635457"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "5539567/6669390"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9310c838",
   "metadata": {},
   "source": [
    "What is the maximum divergence score per window?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "95a25457",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100139008</td>\n",
       "      <td>0.128959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_100663296</td>\n",
       "      <td>0.101497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_101187584</td>\n",
       "      <td>0.033608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_101711872</td>\n",
       "      <td>0.138848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_102236160</td>\n",
       "      <td>0.031867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4415</th>\n",
       "      <td>chrX_94371840</td>\n",
       "      <td>0.076931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4416</th>\n",
       "      <td>chrX_94896128</td>\n",
       "      <td>0.044540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4417</th>\n",
       "      <td>chrX_95420416</td>\n",
       "      <td>0.002984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4418</th>\n",
       "      <td>chrX_99090432</td>\n",
       "      <td>0.030925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4419</th>\n",
       "      <td>chrX_99614720</td>\n",
       "      <td>0.027077</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4420 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               window       max\n",
       "0     chr10_100139008  0.128959\n",
       "1     chr10_100663296  0.101497\n",
       "2     chr10_101187584  0.033608\n",
       "3     chr10_101711872  0.138848\n",
       "4     chr10_102236160  0.031867\n",
       "...               ...       ...\n",
       "4415    chrX_94371840  0.076931\n",
       "4416    chrX_94896128  0.044540\n",
       "4417    chrX_95420416  0.002984\n",
       "4418    chrX_99090432  0.030925\n",
       "4419    chrX_99614720  0.027077\n",
       "\n",
       "[4420 rows x 2 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maxes = comparisons.groupby(['window'])['divergence'].max().to_frame('max').reset_index()\n",
    "maxes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2724fb61",
   "metadata": {},
   "source": [
    "How many windows have a maximum divergence of 0.005 and 0.01? What is the maximum divergence for various quantiles?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "726dd8bd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "245"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(maxes[maxes['max'] < 0.005])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "7ed04614",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "874"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(maxes[maxes['max'] < 0.01])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "d8eb3e7f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.006717990286899901"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maxes['max'].quantile(q = 0.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "d3bfdf8d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0119926032"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maxes['max'].quantile(q = 0.25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "a3fc1ab9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.02476536453449995"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maxes['max'].quantile(q = 0.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "824be24f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.11610571464919983"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maxes['max'].quantile(q = 0.9)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ea234f4",
   "metadata": {},
   "source": [
    "Get divergence scores for Figure 1C examples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "b01c393a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ind1</th>\n",
       "      <th>ind2</th>\n",
       "      <th>lineages</th>\n",
       "      <th>chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>spearman</th>\n",
       "      <th>divergence</th>\n",
       "      <th>seq_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6211267</th>\n",
       "      <td>Maya</td>\n",
       "      <td>Washu</td>\n",
       "      <td>pts-pts</td>\n",
       "      <td>chr4</td>\n",
       "      <td>77594624</td>\n",
       "      <td>chr4_77594624</td>\n",
       "      <td>1.090777e-07</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.526210e-07</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ind1   ind2 lineages   chr  window_start         window  \\\n",
       "6211267  Maya  Washu  pts-pts  chr4      77594624  chr4_77594624   \n",
       "\n",
       "                  mse  spearman    divergence  seq_diff  \n",
       "6211267  1.090777e-07       1.0  4.526210e-07        11  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr4_77594624') & (comparisons['ind1'] == 'Maya') & (comparisons['ind2'] == 'Washu')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "63c8e1ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ind1</th>\n",
       "      <th>ind2</th>\n",
       "      <th>lineages</th>\n",
       "      <th>chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>spearman</th>\n",
       "      <th>divergence</th>\n",
       "      <th>seq_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5136420</th>\n",
       "      <td>Julie-A959</td>\n",
       "      <td>Vincent</td>\n",
       "      <td>pts-ptt</td>\n",
       "      <td>chr18</td>\n",
       "      <td>46137344</td>\n",
       "      <td>chr18_46137344</td>\n",
       "      <td>0.020585</td>\n",
       "      <td>0.350611</td>\n",
       "      <td>0.649389</td>\n",
       "      <td>3049</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               ind1     ind2 lineages    chr  window_start          window  \\\n",
       "5136420  Julie-A959  Vincent  pts-ptt  chr18      46137344  chr18_46137344   \n",
       "\n",
       "              mse  spearman  divergence  seq_diff  \n",
       "5136420  0.020585  0.350611    0.649389      3049  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr18_46137344') & (comparisons['ind1'] == 'Julie-A959') & (comparisons['ind2'] == 'Vincent')]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06219ac2",
   "metadata": {},
   "source": [
    "## *Pan* - *Homo* Divergence Distribution Comparison <a class = 'anchor' id = 'panhomodivergence'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee4397a3",
   "metadata": {},
   "source": [
    "Compare the *Pan* divergence score distribution to the distribution calculated from 130 modern human genomes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "cea75a5b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ind1</th>\n",
       "      <th>ind2</th>\n",
       "      <th>chrm</th>\n",
       "      <th>start_pos</th>\n",
       "      <th>3d_divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AFR_ASW_female_NA19917</td>\n",
       "      <td>AFR_ASW_female_NA19901</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>0.002916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AFR_ASW_female_NA19917</td>\n",
       "      <td>AFR_ASW_female_NA20314</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>0.003368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AFR_ASW_female_NA19917</td>\n",
       "      <td>AFR_ASW_female_NA20317</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>0.005616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AFR_ASW_female_NA19917</td>\n",
       "      <td>AFR_ASW_female_NA19625</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>0.001083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AFR_ASW_female_NA19917</td>\n",
       "      <td>AFR_ACB_female_HG02337</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>0.004772</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     ind1                    ind2  chrm  start_pos  \\\n",
       "0  AFR_ASW_female_NA19917  AFR_ASW_female_NA19901  chr1    1048576   \n",
       "1  AFR_ASW_female_NA19917  AFR_ASW_female_NA20314  chr1    1048576   \n",
       "2  AFR_ASW_female_NA19917  AFR_ASW_female_NA20317  chr1    1048576   \n",
       "3  AFR_ASW_female_NA19917  AFR_ASW_female_NA19625  chr1    1048576   \n",
       "4  AFR_ASW_female_NA19917  AFR_ACB_female_HG02337  chr1    1048576   \n",
       "\n",
       "   3d_divergence  \n",
       "0       0.002916  \n",
       "1       0.003368  \n",
       "2       0.005616  \n",
       "3       0.001083  \n",
       "4       0.004772  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_comparisons = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/comparisons/thousand_genomes_subset_HFF/melted_130v130_1KG_subsample.csv', sep = ',', header = 0)\n",
    "human_comparisons.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "31bca651",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "40860105"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(human_comparisons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "db943480-bdce-4b19-a79c-60357db48d56",
   "metadata": {},
   "outputs": [],
   "source": [
    "pan_comparisons = comparisons[comparisons['chr'] != 'chrX']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "89179588-c64c-419c-b924-b57ac0488d60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6574260"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pan_comparisons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "f05d299e-c706-48c0-8595-a7fd1893c098",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.007906218997677883"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pan_comparisons['divergence'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "bcc938f2-aa5a-424d-9934-c9e99e3e3c2d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.003348655193871765"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_comparisons['3d_divergence'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ff98390d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KstestResult(statistic=0.3294594800502246, pvalue=0.0)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kstest(pan_comparisons['divergence'], human_comparisons['3d_divergence'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26d267e3",
   "metadata": {},
   "source": [
    "## Minimally Divergent Windows and Ultraconserved TAD Boundaries <a class = 'anchor' id = 'okhovatetal2023'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3c75f18",
   "metadata": {},
   "source": [
    "Let's consider the overlap between minimally divergent windows and ultra- and primate-conserved boundaries from Okhovat et al. 2023. First, let's read in the files containing the boundaries in panTro6 coordinates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "e264d0e6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t1098611\t1204332\thg38_n_B00132\n",
      " chr1\t2029035\t2038921\thg38_n_B00190\n",
      " chr1\t5402415\t5412399\thg38_n_B00510\n",
      " chr1\t7054986\t7064987\thg38_n_B00676\n",
      " chr1\t7946002\t7956016\thg38_n_B00765\n",
      " "
     ]
    }
   ],
   "source": [
    "ultraconserved_pbtBED = pybedtools.BedTool('/wynton/group/capra/projects/pan_3d_genome/data/Okhovat_et_al_2023/Okhovat_et_al_2023_ultraconserved_boundaries_panTro6.bed')\n",
    "ultraconserved_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "3b0cb528",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "984"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ultraconserved_pbtBED)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "8f3d0981",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t13254423\t13264412\thg38_n_B01247\n",
      " chr1\t13745195\t13755219\thg38_n_B01296\n",
      " chr1\t17375951\t17385984\thg38_n_B01639\n",
      " chr1\t24694359\t24704375\thg38_n_B02283\n",
      " chr1\t25957982\t25967966\thg38_n_B02394\n",
      " "
     ]
    }
   ],
   "source": [
    "primate_conserved_pbtBED = pybedtools.BedTool('/wynton/group/capra/projects/pan_3d_genome/data/Okhovat_et_al_2023/Okhovat_et_al_2023_primate_conserved_boundaries_panTro6.bed')\n",
    "primate_conserved_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "b4dabc63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "459"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(primate_conserved_pbtBED)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91baba6c",
   "metadata": {},
   "source": [
    "Do these sets overlap at all?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "2ac5e91e",
   "metadata": {},
   "outputs": [],
   "source": [
    "ultraconserved_primate_conserved_intersect = ultraconserved_pbtBED.intersect(primate_conserved_pbtBED)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "793a2477",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ultraconserved_primate_conserved_intersect)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fedaf790",
   "metadata": {},
   "source": [
    "Intersect the TAD boundaries with the available panTro6 windows to determine the maximum amount of overlap."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "58911aef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t1048576\t2097152\n",
      " chr1\t1572864\t2621440\n",
      " chr1\t2097152\t3145728\n",
      " chr1\t2621440\t3670016\n",
      " chr1\t3145728\t4194304\n",
      " "
     ]
    }
   ],
   "source": [
    "windows_pbtBED = pybedtools.BedTool('/wynton/group/capra/projects/pan_3d_genome/data/metadata/panTro6_windows_with_full_coverage.bed')\n",
    "windows_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "7f576676",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "915"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ultraconserved_pbtBED.intersect(windows_pbtBED, u = True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "24abd50a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "415"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(primate_conserved_pbtBED.intersect(windows_pbtBED, u = True))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de819d1d",
   "metadata": {},
   "source": [
    "What is the *Pan* divergence distribution for both sets of regions? Start by creating a pybedtools object for the maxes dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "37543c31",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100139008</td>\n",
       "      <td>101187584</td>\n",
       "      <td>0.128959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100663296</td>\n",
       "      <td>101711872</td>\n",
       "      <td>0.101497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101187584</td>\n",
       "      <td>102236160</td>\n",
       "      <td>0.033608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101711872</td>\n",
       "      <td>102760448</td>\n",
       "      <td>0.138848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102236160</td>\n",
       "      <td>103284736</td>\n",
       "      <td>0.031867</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr      start        end       max\n",
       "0  chr10  100139008  101187584  0.128959\n",
       "1  chr10  100663296  101711872  0.101497\n",
       "2  chr10  101187584  102236160  0.033608\n",
       "3  chr10  101711872  102760448  0.138848\n",
       "4  chr10  102236160  103284736  0.031867"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "window_max_divergence = maxes['max']\n",
    "maxes_BED = maxes['window'].str.split('_', expand = True).rename(columns = {0:'chr', 1:'start'})\n",
    "maxes_BED['start'] = maxes_BED['start'].astype(int)\n",
    "maxes_BED['end'] = maxes_BED['start'] + 1048576\n",
    "maxes_BED['max'] = window_max_divergence\n",
    "maxes_BED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "3bb9cc77",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4420"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(maxes_BED)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "3fd2d8ea",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t1048576\t2097152\t0.036526326318\n",
      " chr1\t1572864\t2621440\t0.006578744897\n",
      " chr1\t2097152\t3145728\t0.015877001035\n",
      " chr1\t2621440\t3670016\t0.079123467404\n",
      " chr1\t3145728\t4194304\t0.192662189319\n",
      " "
     ]
    }
   ],
   "source": [
    "maxes_pbtBED = pybedtools.BedTool().from_dataframe(maxes_BED).sort()\n",
    "maxes_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4f39dc6",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "Now intersect with the ultraconserved regions. Some TAD boundaries may overlie two overlapping windows. We will compute a centrality score and use the maximum divergence from the window in which the TAD boundary is most central."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f13fc70",
   "metadata": {},
   "source": [
    "The centrality score below is computed as:\n",
    "\n",
    "$$ \\mathrm{Centrality\\:score} = |0.5 - \\frac{(\\frac{\\mathrm{TAD\\:Boundary\\:End\\:-\\:TAD\\:Boundary\\:Start}}{2} + \\mathrm{TAD\\:Boundary\\:Start}) - \\mathrm{Window\\:Start}}{1048576}|$$ \n",
    "\n",
    "Scores at or near 0 indicate the TAD boundary is more central to that window where values approaching 0.5 are at the window's edge."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24c1fc8a",
   "metadata": {},
   "source": [
    "Run the intersection for the ultraconserved."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "927d368a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tb_chr</th>\n",
       "      <th>tb_start</th>\n",
       "      <th>tb_end</th>\n",
       "      <th>tb_id</th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>max</th>\n",
       "      <th>centrality_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1098611</td>\n",
       "      <td>1204332</td>\n",
       "      <td>hg38_n_B00132</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>0.036526</td>\n",
       "      <td>0.401871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2029035</td>\n",
       "      <td>2038921</td>\n",
       "      <td>hg38_n_B00190</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1572864</td>\n",
       "      <td>2621440</td>\n",
       "      <td>0.006579</td>\n",
       "      <td>0.060247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2029035</td>\n",
       "      <td>2038921</td>\n",
       "      <td>hg38_n_B00190</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>0.036526</td>\n",
       "      <td>0.439753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5402415</td>\n",
       "      <td>5412399</td>\n",
       "      <td>hg38_n_B00510</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5242880</td>\n",
       "      <td>6291456</td>\n",
       "      <td>0.017641</td>\n",
       "      <td>0.343095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7054986</td>\n",
       "      <td>7064987</td>\n",
       "      <td>hg38_n_B00676</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6291456</td>\n",
       "      <td>7340032</td>\n",
       "      <td>0.002978</td>\n",
       "      <td>0.232928</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  tb_chr  tb_start   tb_end          tb_id window_chr  window_start  \\\n",
       "0   chr1   1098611  1204332  hg38_n_B00132       chr1       1048576   \n",
       "2   chr1   2029035  2038921  hg38_n_B00190       chr1       1572864   \n",
       "1   chr1   2029035  2038921  hg38_n_B00190       chr1       1048576   \n",
       "3   chr1   5402415  5412399  hg38_n_B00510       chr1       5242880   \n",
       "4   chr1   7054986  7064987  hg38_n_B00676       chr1       6291456   \n",
       "\n",
       "   window_end       max  centrality_score  \n",
       "0     2097152  0.036526          0.401871  \n",
       "2     2621440  0.006579          0.060247  \n",
       "1     2097152  0.036526          0.439753  \n",
       "3     6291456  0.017641          0.343095  \n",
       "4     7340032  0.002978          0.232928  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ultraconserved_maxes_intersect = ultraconserved_pbtBED.intersect(maxes_pbtBED, wa = True, wb = True).to_dataframe(names=['tb_chr','tb_start','tb_end','tb_id','window_chr','window_start','window_end','max'])\n",
    "ultraconserved_maxes_intersect['centrality_score'] = abs(0.5 - (((((ultraconserved_maxes_intersect['tb_end'] - ultraconserved_maxes_intersect['tb_start']) / 2) + ultraconserved_maxes_intersect['tb_start']) - ultraconserved_maxes_intersect['window_start']) / 1048576))  # add a center score to assess centrality of TAD boundary when one overlaps multiple 3d windows\n",
    "ultraconserved_maxes_intersect = ultraconserved_maxes_intersect.sort_values(by=['tb_chr','tb_start','centrality_score'], ascending = True)\n",
    "ultraconserved_maxes_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "643260b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tb_chr</th>\n",
       "      <th>tb_start</th>\n",
       "      <th>tb_end</th>\n",
       "      <th>tb_id</th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>max</th>\n",
       "      <th>centrality_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1098611</td>\n",
       "      <td>1204332</td>\n",
       "      <td>hg38_n_B00132</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>0.036526</td>\n",
       "      <td>0.401871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2029035</td>\n",
       "      <td>2038921</td>\n",
       "      <td>hg38_n_B00190</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1572864</td>\n",
       "      <td>2621440</td>\n",
       "      <td>0.006579</td>\n",
       "      <td>0.060247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5402415</td>\n",
       "      <td>5412399</td>\n",
       "      <td>hg38_n_B00510</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5242880</td>\n",
       "      <td>6291456</td>\n",
       "      <td>0.017641</td>\n",
       "      <td>0.343095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7054986</td>\n",
       "      <td>7064987</td>\n",
       "      <td>hg38_n_B00676</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6291456</td>\n",
       "      <td>7340032</td>\n",
       "      <td>0.002978</td>\n",
       "      <td>0.232928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7946002</td>\n",
       "      <td>7956016</td>\n",
       "      <td>hg38_n_B00765</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7340032</td>\n",
       "      <td>8388608</td>\n",
       "      <td>0.021275</td>\n",
       "      <td>0.082673</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  tb_chr  tb_start   tb_end          tb_id window_chr  window_start  \\\n",
       "0   chr1   1098611  1204332  hg38_n_B00132       chr1       1048576   \n",
       "2   chr1   2029035  2038921  hg38_n_B00190       chr1       1572864   \n",
       "3   chr1   5402415  5412399  hg38_n_B00510       chr1       5242880   \n",
       "4   chr1   7054986  7064987  hg38_n_B00676       chr1       6291456   \n",
       "6   chr1   7946002  7956016  hg38_n_B00765       chr1       7340032   \n",
       "\n",
       "   window_end       max  centrality_score  \n",
       "0     2097152  0.036526          0.401871  \n",
       "2     2621440  0.006579          0.060247  \n",
       "3     6291456  0.017641          0.343095  \n",
       "4     7340032  0.002978          0.232928  \n",
       "6     8388608  0.021275          0.082673  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ultraconserved_maxes_intersect = ultraconserved_maxes_intersect.drop_duplicates(subset = ['tb_chr','tb_start','tb_end'], keep = 'first')\n",
    "ultraconserved_maxes_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "20e5d7fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "915"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ultraconserved_maxes_intersect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "843133a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "ultraconserved_maxes_intersect['max'].to_csv('/wynton/group/capra/projects/pan_3d_genome/data/Okhovat_et_al_2023/ultraconserved_window_maxes.txt', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dce97fc8",
   "metadata": {},
   "source": [
    "Now the primate-conserved."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "63fe14cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tb_chr</th>\n",
       "      <th>tb_start</th>\n",
       "      <th>tb_end</th>\n",
       "      <th>tb_id</th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>max</th>\n",
       "      <th>centrality_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13254423</td>\n",
       "      <td>13264412</td>\n",
       "      <td>hg38_n_B01247</td>\n",
       "      <td>chr1</td>\n",
       "      <td>12582912</td>\n",
       "      <td>13631488</td>\n",
       "      <td>0.140253</td>\n",
       "      <td>0.145166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13745195</td>\n",
       "      <td>13755219</td>\n",
       "      <td>hg38_n_B01296</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13107200</td>\n",
       "      <td>14155776</td>\n",
       "      <td>0.039724</td>\n",
       "      <td>0.113219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17375951</td>\n",
       "      <td>17385984</td>\n",
       "      <td>hg38_n_B01639</td>\n",
       "      <td>chr1</td>\n",
       "      <td>16777216</td>\n",
       "      <td>17825792</td>\n",
       "      <td>0.028705</td>\n",
       "      <td>0.075782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24694359</td>\n",
       "      <td>24704375</td>\n",
       "      <td>hg38_n_B02283</td>\n",
       "      <td>chr1</td>\n",
       "      <td>24117248</td>\n",
       "      <td>25165824</td>\n",
       "      <td>0.015984</td>\n",
       "      <td>0.055152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25957982</td>\n",
       "      <td>25967966</td>\n",
       "      <td>hg38_n_B02394</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25690112</td>\n",
       "      <td>26738688</td>\n",
       "      <td>0.009441</td>\n",
       "      <td>0.239779</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  tb_chr  tb_start    tb_end          tb_id window_chr  window_start  \\\n",
       "0   chr1  13254423  13264412  hg38_n_B01247       chr1      12582912   \n",
       "3   chr1  13745195  13755219  hg38_n_B01296       chr1      13107200   \n",
       "4   chr1  17375951  17385984  hg38_n_B01639       chr1      16777216   \n",
       "6   chr1  24694359  24704375  hg38_n_B02283       chr1      24117248   \n",
       "9   chr1  25957982  25967966  hg38_n_B02394       chr1      25690112   \n",
       "\n",
       "   window_end       max  centrality_score  \n",
       "0    13631488  0.140253          0.145166  \n",
       "3    14155776  0.039724          0.113219  \n",
       "4    17825792  0.028705          0.075782  \n",
       "6    25165824  0.015984          0.055152  \n",
       "9    26738688  0.009441          0.239779  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primate_conserved_maxes_intersect = primate_conserved_pbtBED.intersect(maxes_pbtBED, wa = True, wb = True).to_dataframe(names=['tb_chr','tb_start','tb_end','tb_id','window_chr','window_start','window_end','max'])\n",
    "primate_conserved_maxes_intersect['centrality_score'] = abs(0.5 - (((((primate_conserved_maxes_intersect['tb_end'] - primate_conserved_maxes_intersect['tb_start']) / 2) + primate_conserved_maxes_intersect['tb_start']) - primate_conserved_maxes_intersect['window_start']) / 1048576))  # add a center score to assess centrality of TAD boundary when one overlaps multiple 3d windows\n",
    "primate_conserved_maxes_intersect = primate_conserved_maxes_intersect.sort_values(by=['tb_chr','tb_start','centrality_score'], ascending = True)\n",
    "primate_conserved_maxes_intersect = primate_conserved_maxes_intersect.drop_duplicates(subset = ['tb_chr','tb_start','tb_end'], keep = 'first')\n",
    "primate_conserved_maxes_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "7baf0372",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "415"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(primate_conserved_maxes_intersect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "49843d54",
   "metadata": {},
   "outputs": [],
   "source": [
    "primate_conserved_maxes_intersect['max'].to_csv('/wynton/group/capra/projects/pan_3d_genome/data/Okhovat_et_al_2023/primate_conserved_window_maxes.txt', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "605a3ea7",
   "metadata": {},
   "source": [
    "Are these distributions different? Run a KS test. First the ultraconserved."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "6d361bd7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KstestResult(statistic=0.18024379991593106, pvalue=5.261687464016495e-22)"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kstest(maxes['max'], ultraconserved_maxes_intersect['max'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "f22165bd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.05016095426516081"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maxes['max'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "fd274d54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.02460737114364039"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ultraconserved_maxes_intersect['max'].mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c26fd681",
   "metadata": {},
   "source": [
    "Now the primate-conserved."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "12abf6c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KstestResult(statistic=0.12690127023932835, pvalue=8.713086838339734e-06)"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kstest(maxes['max'], primate_conserved_maxes_intersect['max'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "995e7193",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.03320046814213971"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primate_conserved_maxes_intersect['max'].mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "57ad5152",
   "metadata": {},
   "source": [
    "Export the window maxes as a genome-wide background for visualization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "edf80d36",
   "metadata": {},
   "outputs": [],
   "source": [
    "maxes['max'].to_csv('/wynton/group/capra/projects/pan_3d_genome/data/Okhovat_et_al_2023/window_maxes.txt', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f45fb9cf",
   "metadata": {},
   "source": [
    "## Lineage Comparison <a class = 'anchor' id = 'lineagecomparison'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ca2e61d",
   "metadata": {},
   "source": [
    "Let's look at 3d divergence by lineages represented in the comparison. We will simplify things by collapsing when chimpanzees of different subspecies are compared."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "667c0466",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ind1</th>\n",
       "      <th>ind2</th>\n",
       "      <th>lineages</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pt-pt</td>\n",
       "      <td>0.012345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pt-pt</td>\n",
       "      <td>0.030191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pt-pt</td>\n",
       "      <td>0.003602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pt-pt</td>\n",
       "      <td>0.002101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>pt-pt</td>\n",
       "      <td>0.003268</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ind1    ind2 lineages  divergence\n",
       "0  Akwaya-Jean  Alfred    pt-pt    0.012345\n",
       "1  Akwaya-Jean  Alfred    pt-pt    0.030191\n",
       "2  Akwaya-Jean  Alfred    pt-pt    0.003602\n",
       "3  Akwaya-Jean  Alfred    pt-pt    0.002101\n",
       "4  Akwaya-Jean  Alfred    pt-pt    0.003268"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "simple_dyad_comparisons = comparisons[['ind1','ind2','lineages','divergence']].copy()\n",
    "simple_dyad_comparisons['lineages'] = simple_dyad_comparisons['lineages'].replace({'pte-pts':'pt-pt', 'pte-ptt':'pt-pt', 'pte-ptv':'pt-pt', 'pts-ptt':'pt-pt', 'pts-ptv':'pt-pt', 'ptt-ptv':'pt-pt'})\n",
    "simple_dyad_comparisons.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "a3b279cd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>median</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lineages</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ppn-ppn</th>\n",
       "      <td>0.000829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ppn-ptr</th>\n",
       "      <td>0.004130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pt-pt</th>\n",
       "      <td>0.002331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pte-pte</th>\n",
       "      <td>0.001282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pts-pts</th>\n",
       "      <td>0.001629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ptt-ptt</th>\n",
       "      <td>0.002165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ptv-ptv</th>\n",
       "      <td>0.000590</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            median\n",
       "lineages          \n",
       "ppn-ppn   0.000829\n",
       "ppn-ptr   0.004130\n",
       "pt-pt     0.002331\n",
       "pte-pte   0.001282\n",
       "pts-pts   0.001629\n",
       "ptt-ptt   0.002165\n",
       "ptv-ptv   0.000590"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "simple_dyad_comparisons.groupby('lineages')['divergence'].median().to_frame('median')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d76bbd54",
   "metadata": {},
   "source": [
    "## Hierarchical Clustering <a class = 'anchor' id = 'hierachicalclustering'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9cbcac50-0ece-4c07-a4c8-cc5ebf58b0ad",
   "metadata": {},
   "source": [
    "### Establish N Clusters, Cluster Size, and Cluster Composition <a class = 'anchor' id = 'establishclusters'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85116f0c",
   "metadata": {},
   "source": [
    "Run a script to generate trees per window, grab the cluster identity per individual, and catch the y-coordinates of the top four nodes in the tree. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "03a5d5b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "complete_linkage_trees = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/window_topologies/complete_linkage_clustering_per_window.txt', sep = '\\t', header = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "75a8ea87",
   "metadata": {},
   "outputs": [],
   "source": [
    "autosomal_complete_linkage_trees = complete_linkage_trees[~complete_linkage_trees[0].str.startswith('chrX_')]\n",
    "chrX_complete_linkage_trees = complete_linkage_trees[complete_linkage_trees[0].str.startswith('chrX_')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "8a4497ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "autosomal_trees_header = ['window','Akwaya-Jean','Alfred','Alice','Andromeda','Athanga','Berta','Bihati','Blanquita','Bono','Bosco','Brigitta','Bwamble','Cindy-schwein','Cindy-troglodytes','Cindy-verus','Cleo','Clint','Coco-chimp','Damian','Desmond','Doris','Dzeeta','Frederike','Gamin','Hermien','Hortense','Ikuru','Jimmie','Julie-A959','Julie-LWC21','Kidongo','Koby','Kombote','Kosana','Koto','Kumbuka','Lara','Linda','Luky','Marlin','Maya','Mgbadolite','Mirinda','Nakuu','Natalie','Negrita','SeppToni','Taweh','Tibe','Tongo','Trixie','Ula','Vaillant','Vincent','Washu','Yogui','node_0','node_1','node_2','node_3']\n",
    "chrX_trees_header = ['window','Alice','Andromeda','Berta','Bihati','Blanquita','Cindy-schwein','Cindy-troglodytes','Cindy-verus','Cleo','Coco-chimp','Doris','Dzeeta','Frederike','Hermien','Hortense','Ikuru','Jimmie','Julie-A959','Julie-LWC21','Kidongo','Kombote','Kosana','Kumbuka','Lara','Linda','Luky','Marlin','Maya','Mirinda','Nakuu','Natalie','Negrita','Taweh','Tibe','Trixie','Ula','node_0','node_1','node_2','node_3']\n",
    "\n",
    "autosomal_complete_linkage_trees.columns = autosomal_trees_header\n",
    "chrX_complete_linkage_trees = chrX_complete_linkage_trees.dropna(axis = 1)\n",
    "chrX_complete_linkage_trees.columns = chrX_trees_header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "70c1c801",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <th>Alfred</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Athanga</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Bono</th>\n",
       "      <th>Bosco</th>\n",
       "      <th>Brigitta</th>\n",
       "      <th>Bwamble</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Clint</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Damian</th>\n",
       "      <th>Desmond</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Gamin</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Koby</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Koto</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mgbadolite</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>SeppToni</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Tongo</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>Vaillant</th>\n",
       "      <th>Vincent</th>\n",
       "      <th>Washu</th>\n",
       "      <th>Yogui</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_1572864</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.014522</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>0.067514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_2097152</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.038838</td>\n",
       "      <td>0.269334</td>\n",
       "      <td>0.269334</td>\n",
       "      <td>0.115208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_2621440</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005674</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>0.006737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_3145728</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002904</td>\n",
       "      <td>0.010145</td>\n",
       "      <td>0.010145</td>\n",
       "      <td>0.006418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_3670016</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005140</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>0.048080</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          window Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati  \\\n",
       "0  chr10_1572864          C3     C3    C3        C2      C2    C3     C3   \n",
       "1  chr10_2097152          C2     C2    C2        C2      C2    C2     C2   \n",
       "2  chr10_2621440          C2     C2    C2        C2      C2    C2     C2   \n",
       "3  chr10_3145728          C2     C2    C1        C2      C2    C2     C2   \n",
       "4  chr10_3670016          C2     C2    C2        C2      C2    C2     C2   \n",
       "\n",
       "  Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes  \\\n",
       "0        C2   C1    C3       C3      C2            C2                C3   \n",
       "1        C2   C2    C2       C2      C2            C2                C1   \n",
       "2        C2   C2    C2       C2      C2            C2                C1   \n",
       "3        C2   C2    C2       C2      C2            C2                C2   \n",
       "4        C2   C1    C2       C2      C2            C2                C2   \n",
       "\n",
       "  Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike  \\\n",
       "0          C3   C2    C3         C2     C3      C1    C2     C1        C3   \n",
       "1          C2   C2    C2         C2     C2      C2    C1     C2        C2   \n",
       "2          C2   C2    C2         C2     C2      C2    C1     C2        C2   \n",
       "3          C1   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "4          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "\n",
       "  Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby  \\\n",
       "0    C3      C1       C1    C3     C3         C3          C3      C2   C3   \n",
       "1    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "2    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "3    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "4    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "\n",
       "  Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda  \\\n",
       "0      C1     C1   C3      C1   C2    C3   C3     C3   C2         C2      C2   \n",
       "1      C2     C2   C2      C2   C2    C2   C1     C1   C2         C2      C2   \n",
       "2      C2     C2   C2      C2   C2    C2   C1     C1   C2         C2      C2   \n",
       "3      C2     C2   C2      C2   C2    C2   C1     C2   C2         C2      C2   \n",
       "4      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2      C2   \n",
       "\n",
       "  Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent  \\\n",
       "0    C2      C1      C2       C3    C3   C3    C2     C2  C3       C3      C2   \n",
       "1    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "2    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "3    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "4    C2      C1      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "\n",
       "  Washu Yogui    node_0    node_1    node_2    node_3  \n",
       "0    C2    C2  0.014522  0.091748  0.091748  0.067514  \n",
       "1    C2    C2  0.038838  0.269334  0.269334  0.115208  \n",
       "2    C2    C2  0.005674  0.012824  0.012824  0.006737  \n",
       "3    C2    C2  0.002904  0.010145  0.010145  0.006418  \n",
       "4    C2    C2  0.005140  0.096248  0.096248  0.048080  "
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "autosomal_complete_linkage_trees.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "616ec528",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4269</th>\n",
       "      <td>chrX_4718592</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000386870517</td>\n",
       "      <td>0.0942329238279999</td>\n",
       "      <td>0.0942329238279999</td>\n",
       "      <td>0.0244069347169999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4270</th>\n",
       "      <td>chrX_5242880</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C0</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.018755481251</td>\n",
       "      <td>0.018755481251</td>\n",
       "      <td>0.0144252191229999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4271</th>\n",
       "      <td>chrX_6815744</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>0.0004348450219999</td>\n",
       "      <td>0.0176649495409999</td>\n",
       "      <td>0.0176649495409999</td>\n",
       "      <td>0.014919062537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4272</th>\n",
       "      <td>chrX_9437184</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0653681878299999</td>\n",
       "      <td>0.0653681878299999</td>\n",
       "      <td>0.020195340661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4273</th>\n",
       "      <td>chrX_11010048</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>7.667215399997929e-05</td>\n",
       "      <td>0.063108737967</td>\n",
       "      <td>0.063108737967</td>\n",
       "      <td>0.0414393167169999</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             window Alice Andromeda Berta Bihati Blanquita Cindy-schwein  \\\n",
       "4269   chrX_4718592    C1        C2    C1     C2        C2            C2   \n",
       "4270   chrX_5242880    C2        C2    C2     C2        C2            C2   \n",
       "4271   chrX_6815744    C1        C3    C3     C3        C3            C3   \n",
       "4272   chrX_9437184    C1        C1    C1     C1        C1            C1   \n",
       "4273  chrX_11010048    C2        C1    C2     C2        C2            C2   \n",
       "\n",
       "     Cindy-troglodytes Cindy-verus Cleo Coco-chimp Doris Dzeeta Frederike  \\\n",
       "4269                C2          C2   C2         C2    C2     C2        C2   \n",
       "4270                C2          C2   C2         C2    C1     C2        C2   \n",
       "4271                C3          C3   C3         C3    C3     C3        C3   \n",
       "4272                C1          C1   C1         C1    C1     C1        C1   \n",
       "4273                C2          C2   C2         C2    C2     C2        C2   \n",
       "\n",
       "     Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Kombote  \\\n",
       "4269      C2       C2    C2     C2         C2          C2      C2      C2   \n",
       "4270      C2       C2    C2     C2         C0          C2      C2      C2   \n",
       "4271      C3       C3    C3     C1         C3          C3      C3      C3   \n",
       "4272      C1       C1    C1     C1         C1          C1      C1      C1   \n",
       "4273      C2       C2    C2     C2         C2          C2      C2      C2   \n",
       "\n",
       "     Kosana Kumbuka Lara Linda Luky Marlin Maya Mirinda Nakuu Natalie Negrita  \\\n",
       "4269     C2      C2   C2    C2   C2     C2   C2      C2    C2      C2      C2   \n",
       "4270     C2      C2   C2    C2   C1     C2   C2      C2    C2      C2      C2   \n",
       "4271     C2      C3   C3    C1   C3     C3   C3      C3    C3      C2      C3   \n",
       "4272     C1      C1   C1    C1   C0     C1   C1      C1    C1      C1      C1   \n",
       "4273     C2      C2   C2    C2   C2     C2   C1      C2    C2      C2      C2   \n",
       "\n",
       "     Taweh Tibe Trixie Ula                 node_0              node_1  \\\n",
       "4269    C2   C2     C2  C2         0.000386870517  0.0942329238279999   \n",
       "4270    C2   C2     C2  C2                    0.0      0.018755481251   \n",
       "4271    C3   C3     C3  C3     0.0004348450219999  0.0176649495409999   \n",
       "4272    C1   C1     C1  C1                    0.0  0.0653681878299999   \n",
       "4273    C2   C2     C2  C2  7.667215399997929e-05      0.063108737967   \n",
       "\n",
       "                  node_2              node_3  \n",
       "4269  0.0942329238279999  0.0244069347169999  \n",
       "4270      0.018755481251  0.0144252191229999  \n",
       "4271  0.0176649495409999      0.014919062537  \n",
       "4272  0.0653681878299999      0.020195340661  \n",
       "4273      0.063108737967  0.0414393167169999  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrX_complete_linkage_trees.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06ffe929",
   "metadata": {},
   "source": [
    "Write a function that creates a cluster composition string. We will sort clusters by increasing size and delineate them by a '/'."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "5b4fa6ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_autosomal_clusters_lists(dataframe, window):\n",
    "    row_index = dataframe[dataframe['window'] == window].index\n",
    "    row = dataframe.loc[row_index,['Akwaya-Jean','Alfred','Alice','Andromeda','Athanga','Berta','Bihati','Blanquita','Bono','Bosco','Brigitta','Bwamble','Cindy-schwein','Cindy-troglodytes','Cindy-verus','Cleo','Clint','Coco-chimp','Damian','Desmond','Doris','Dzeeta','Frederike','Gamin','Hermien','Hortense','Ikuru','Jimmie','Julie-A959','Julie-LWC21','Kidongo','Koby','Kombote','Kosana','Koto','Kumbuka','Lara','Linda','Luky','Marlin','Maya','Mgbadolite','Mirinda','Nakuu','Natalie','Negrita','SeppToni','Taweh','Tibe','Tongo','Trixie','Ula','Vaillant','Vincent','Washu','Yogui']].values[0].tolist()\n",
    "    cluster_values = sorted(list(set(row)))\n",
    "    clusters_list = []\n",
    "    single_clusters = 0\n",
    "    \n",
    "    for c in cluster_values:\n",
    "        cluster_samples = dataframe.loc[row_index].apply(lambda row: row[row == c].index, axis = 1).values[0].tolist()\n",
    "        if len(cluster_samples) == 1:\n",
    "            single_clusters += 1\n",
    "        cluster_samples = ' '.join(cluster_samples)\n",
    "        clusters_list.append(cluster_samples)\n",
    "\n",
    "    def count_spaces(s):\n",
    "        return s.count(' ')\n",
    "\n",
    "    sorted_data = sorted(clusters_list, key = count_spaces)\n",
    "    cluster_composition = '/'.join(sorted_data)\n",
    "    return len(clusters_list), single_clusters, cluster_composition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "e0b183a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_chrX_clusters_lists(dataframe, window):\n",
    "    row_index = dataframe[dataframe['window'] == window].index\n",
    "    row = dataframe.loc[row_index,['Alice','Andromeda','Berta','Bihati','Blanquita','Cindy-schwein','Cindy-troglodytes','Cindy-verus','Cleo','Coco-chimp','Doris','Dzeeta','Frederike','Hermien','Hortense','Ikuru','Jimmie','Julie-A959','Julie-LWC21','Kidongo','Kombote','Kosana','Kumbuka','Lara','Linda','Luky','Marlin','Maya','Mirinda','Nakuu','Natalie','Negrita','Taweh','Tibe','Trixie','Ula']].values[0].tolist()\n",
    "    cluster_values = sorted(list(set(row)))\n",
    "    clusters_list = []\n",
    "    single_clusters = 0\n",
    "    \n",
    "    for c in cluster_values:\n",
    "        cluster_samples = dataframe.loc[row_index].apply(lambda row: row[row == c].index, axis = 1).values[0].tolist()\n",
    "        if len(cluster_samples) == 1:\n",
    "            single_clusters += 1\n",
    "        cluster_samples = ' '.join(cluster_samples)\n",
    "        clusters_list.append(cluster_samples)\n",
    "    \n",
    "    def count_spaces(s):\n",
    "        return s.count(' ')\n",
    "\n",
    "    sorted_data = sorted(clusters_list, key = count_spaces)\n",
    "    cluster_composition = '/'.join(sorted_data)\n",
    "    return len(clusters_list), single_clusters, cluster_composition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "f84fd8fe",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/cbrand/ipykernel_2007303/587978040.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  autosomal_complete_linkage_trees[['n_clusters','n_single_clusters','cluster_composition']] = pd.DataFrame(autosomal_complete_linkage_trees.apply(lambda row: generate_autosomal_clusters_lists(autosomal_complete_linkage_trees, row['window']), axis=1).tolist(), index=autosomal_complete_linkage_trees.index)\n",
      "/scratch/cbrand/ipykernel_2007303/587978040.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  autosomal_complete_linkage_trees[['n_clusters','n_single_clusters','cluster_composition']] = pd.DataFrame(autosomal_complete_linkage_trees.apply(lambda row: generate_autosomal_clusters_lists(autosomal_complete_linkage_trees, row['window']), axis=1).tolist(), index=autosomal_complete_linkage_trees.index)\n",
      "/scratch/cbrand/ipykernel_2007303/587978040.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  autosomal_complete_linkage_trees[['n_clusters','n_single_clusters','cluster_composition']] = pd.DataFrame(autosomal_complete_linkage_trees.apply(lambda row: generate_autosomal_clusters_lists(autosomal_complete_linkage_trees, row['window']), axis=1).tolist(), index=autosomal_complete_linkage_trees.index)\n"
     ]
    }
   ],
   "source": [
    "autosomal_complete_linkage_trees[['n_clusters','n_single_clusters','cluster_composition']] = pd.DataFrame(autosomal_complete_linkage_trees.apply(lambda row: generate_autosomal_clusters_lists(autosomal_complete_linkage_trees, row['window']), axis=1).tolist(), index=autosomal_complete_linkage_trees.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "43a3d656",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <th>Alfred</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Athanga</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Bono</th>\n",
       "      <th>Bosco</th>\n",
       "      <th>Brigitta</th>\n",
       "      <th>Bwamble</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Clint</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Damian</th>\n",
       "      <th>Desmond</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Gamin</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Koby</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Koto</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mgbadolite</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>SeppToni</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Tongo</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>Vaillant</th>\n",
       "      <th>Vincent</th>\n",
       "      <th>Washu</th>\n",
       "      <th>Yogui</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "      <th>n_clusters</th>\n",
       "      <th>n_single_clusters</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_1572864</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.014522</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>0.067514</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_2097152</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.038838</td>\n",
       "      <td>0.269334</td>\n",
       "      <td>0.269334</td>\n",
       "      <td>0.115208</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_2621440</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005674</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>0.006737</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_3145728</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002904</td>\n",
       "      <td>0.010145</td>\n",
       "      <td>0.010145</td>\n",
       "      <td>0.006418</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Cindy-verus Luky/Akwaya-Jean Alfred Andr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_3670016</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005140</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>0.048080</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          window Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati  \\\n",
       "0  chr10_1572864          C3     C3    C3        C2      C2    C3     C3   \n",
       "1  chr10_2097152          C2     C2    C2        C2      C2    C2     C2   \n",
       "2  chr10_2621440          C2     C2    C2        C2      C2    C2     C2   \n",
       "3  chr10_3145728          C2     C2    C1        C2      C2    C2     C2   \n",
       "4  chr10_3670016          C2     C2    C2        C2      C2    C2     C2   \n",
       "\n",
       "  Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes  \\\n",
       "0        C2   C1    C3       C3      C2            C2                C3   \n",
       "1        C2   C2    C2       C2      C2            C2                C1   \n",
       "2        C2   C2    C2       C2      C2            C2                C1   \n",
       "3        C2   C2    C2       C2      C2            C2                C2   \n",
       "4        C2   C1    C2       C2      C2            C2                C2   \n",
       "\n",
       "  Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike  \\\n",
       "0          C3   C2    C3         C2     C3      C1    C2     C1        C3   \n",
       "1          C2   C2    C2         C2     C2      C2    C1     C2        C2   \n",
       "2          C2   C2    C2         C2     C2      C2    C1     C2        C2   \n",
       "3          C1   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "4          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "\n",
       "  Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby  \\\n",
       "0    C3      C1       C1    C3     C3         C3          C3      C2   C3   \n",
       "1    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "2    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "3    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "4    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "\n",
       "  Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda  \\\n",
       "0      C1     C1   C3      C1   C2    C3   C3     C3   C2         C2      C2   \n",
       "1      C2     C2   C2      C2   C2    C2   C1     C1   C2         C2      C2   \n",
       "2      C2     C2   C2      C2   C2    C2   C1     C1   C2         C2      C2   \n",
       "3      C2     C2   C2      C2   C2    C2   C1     C2   C2         C2      C2   \n",
       "4      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2      C2   \n",
       "\n",
       "  Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent  \\\n",
       "0    C2      C1      C2       C3    C3   C3    C2     C2  C3       C3      C2   \n",
       "1    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "2    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "3    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "4    C2      C1      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "\n",
       "  Washu Yogui    node_0    node_1    node_2    node_3  n_clusters  \\\n",
       "0    C2    C2  0.014522  0.091748  0.091748  0.067514           3   \n",
       "1    C2    C2  0.038838  0.269334  0.269334  0.115208           2   \n",
       "2    C2    C2  0.005674  0.012824  0.012824  0.006737           2   \n",
       "3    C2    C2  0.002904  0.010145  0.010145  0.006418           2   \n",
       "4    C2    C2  0.005140  0.096248  0.096248  0.048080           2   \n",
       "\n",
       "   n_single_clusters                                cluster_composition  \n",
       "0                  0  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1                  0  Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...  \n",
       "2                  0  Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...  \n",
       "3                  0  Alice Cindy-verus Luky/Akwaya-Jean Alfred Andr...  \n",
       "4                  0  Bono Desmond Dzeeta Hermien Hortense Kombote K...  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "autosomal_complete_linkage_trees.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "cd43403e",
   "metadata": {},
   "outputs": [],
   "source": [
    "chrX_complete_linkage_trees[['n_clusters','n_single_clusters','cluster_composition']]= pd.DataFrame(chrX_complete_linkage_trees.apply(lambda row: generate_chrX_clusters_lists(chrX_complete_linkage_trees, row['window']), axis=1).tolist(), index=chrX_complete_linkage_trees.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "5ef14a84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "      <th>n_clusters</th>\n",
       "      <th>n_single_clusters</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4269</th>\n",
       "      <td>chrX_4718592</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000386870517</td>\n",
       "      <td>0.0942329238279999</td>\n",
       "      <td>0.0942329238279999</td>\n",
       "      <td>0.0244069347169999</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta/Andromeda Bihati Blanquita Cindy-s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4270</th>\n",
       "      <td>chrX_5242880</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C0</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.018755481251</td>\n",
       "      <td>0.018755481251</td>\n",
       "      <td>0.0144252191229999</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>Julie-A959/Doris Luky/Alice Andromeda Berta Bi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4271</th>\n",
       "      <td>chrX_6815744</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>0.0004348450219999</td>\n",
       "      <td>0.0176649495409999</td>\n",
       "      <td>0.0176649495409999</td>\n",
       "      <td>0.014919062537</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Kosana Natalie/Alice Jimmie Linda/Andromeda Be...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4272</th>\n",
       "      <td>chrX_9437184</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0653681878299999</td>\n",
       "      <td>0.0653681878299999</td>\n",
       "      <td>0.020195340661</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Luky/Alice Andromeda Berta Bihati Blanquita Ci...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4273</th>\n",
       "      <td>chrX_11010048</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>7.667215399997929e-05</td>\n",
       "      <td>0.063108737967</td>\n",
       "      <td>0.063108737967</td>\n",
       "      <td>0.0414393167169999</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Andromeda Maya/Alice Berta Bihati Blanquita Ci...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             window Alice Andromeda Berta Bihati Blanquita Cindy-schwein  \\\n",
       "4269   chrX_4718592    C1        C2    C1     C2        C2            C2   \n",
       "4270   chrX_5242880    C2        C2    C2     C2        C2            C2   \n",
       "4271   chrX_6815744    C1        C3    C3     C3        C3            C3   \n",
       "4272   chrX_9437184    C1        C1    C1     C1        C1            C1   \n",
       "4273  chrX_11010048    C2        C1    C2     C2        C2            C2   \n",
       "\n",
       "     Cindy-troglodytes Cindy-verus Cleo Coco-chimp Doris Dzeeta Frederike  \\\n",
       "4269                C2          C2   C2         C2    C2     C2        C2   \n",
       "4270                C2          C2   C2         C2    C1     C2        C2   \n",
       "4271                C3          C3   C3         C3    C3     C3        C3   \n",
       "4272                C1          C1   C1         C1    C1     C1        C1   \n",
       "4273                C2          C2   C2         C2    C2     C2        C2   \n",
       "\n",
       "     Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Kombote  \\\n",
       "4269      C2       C2    C2     C2         C2          C2      C2      C2   \n",
       "4270      C2       C2    C2     C2         C0          C2      C2      C2   \n",
       "4271      C3       C3    C3     C1         C3          C3      C3      C3   \n",
       "4272      C1       C1    C1     C1         C1          C1      C1      C1   \n",
       "4273      C2       C2    C2     C2         C2          C2      C2      C2   \n",
       "\n",
       "     Kosana Kumbuka Lara Linda Luky Marlin Maya Mirinda Nakuu Natalie Negrita  \\\n",
       "4269     C2      C2   C2    C2   C2     C2   C2      C2    C2      C2      C2   \n",
       "4270     C2      C2   C2    C2   C1     C2   C2      C2    C2      C2      C2   \n",
       "4271     C2      C3   C3    C1   C3     C3   C3      C3    C3      C2      C3   \n",
       "4272     C1      C1   C1    C1   C0     C1   C1      C1    C1      C1      C1   \n",
       "4273     C2      C2   C2    C2   C2     C2   C1      C2    C2      C2      C2   \n",
       "\n",
       "     Taweh Tibe Trixie Ula                 node_0              node_1  \\\n",
       "4269    C2   C2     C2  C2         0.000386870517  0.0942329238279999   \n",
       "4270    C2   C2     C2  C2                    0.0      0.018755481251   \n",
       "4271    C3   C3     C3  C3     0.0004348450219999  0.0176649495409999   \n",
       "4272    C1   C1     C1  C1                    0.0  0.0653681878299999   \n",
       "4273    C2   C2     C2  C2  7.667215399997929e-05      0.063108737967   \n",
       "\n",
       "                  node_2              node_3  n_clusters  n_single_clusters  \\\n",
       "4269  0.0942329238279999  0.0244069347169999           2                  0   \n",
       "4270      0.018755481251  0.0144252191229999           3                  1   \n",
       "4271  0.0176649495409999      0.014919062537           3                  0   \n",
       "4272  0.0653681878299999      0.020195340661           2                  1   \n",
       "4273      0.063108737967  0.0414393167169999           2                  0   \n",
       "\n",
       "                                    cluster_composition  \n",
       "4269  Alice Berta/Andromeda Bihati Blanquita Cindy-s...  \n",
       "4270  Julie-A959/Doris Luky/Alice Andromeda Berta Bi...  \n",
       "4271  Kosana Natalie/Alice Jimmie Linda/Andromeda Be...  \n",
       "4272  Luky/Alice Andromeda Berta Bihati Blanquita Ci...  \n",
       "4273  Andromeda Maya/Alice Berta Bihati Blanquita Ci...  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrX_complete_linkage_trees.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f23e407f",
   "metadata": {},
   "source": [
    "Concat the dataframes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "18645e37",
   "metadata": {},
   "outputs": [],
   "source": [
    "complete_linkage_trees = pd.concat([autosomal_complete_linkage_trees, chrX_complete_linkage_trees])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "d41be449",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4420"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(complete_linkage_trees)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e87bb922",
   "metadata": {},
   "source": [
    "Take a look at the number of clusters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "30b18b27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n_clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               N\n",
       "n_clusters      \n",
       "2           3622\n",
       "3            748\n",
       "4             46\n",
       "5              4"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "complete_linkage_trees.groupby(['n_clusters'])['window'].count().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "624501d2",
   "metadata": {},
   "source": [
    "Let's get a table of unique cluster compositions for each set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "3c7b1c7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "autosomal_complete_summary = autosomal_complete_linkage_trees.groupby(['cluster_composition'])['window'].count().to_frame('N')\n",
    "chrX_complete_summary = chrX_complete_linkage_trees.groupby(['cluster_composition'])['window'].count().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "3a8ce281",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2745"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(autosomal_complete_summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "1c620bc5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "104"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(chrX_complete_summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "15ce2c40",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_composition</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</th>\n",
       "      <td>315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Luky/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Lara Linda Marlin Maya Mgbadolite Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</th>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Desmond/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Dzeeta Frederike Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</th>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Doris/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Desmond Dzeeta Frederike Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</th>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Marlin/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Lara Linda Luky Maya Mgbadolite Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</th>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      N\n",
       "cluster_composition                                    \n",
       "Bono Desmond Dzeeta Hermien Hortense Kombote Ko...  315\n",
       "Luky/Akwaya-Jean Alfred Alice Andromeda Athanga...   33\n",
       "Desmond/Akwaya-Jean Alfred Alice Andromeda Atha...   30\n",
       "Doris/Akwaya-Jean Alfred Alice Andromeda Athang...   29\n",
       "Marlin/Akwaya-Jean Alfred Alice Andromeda Athan...   29"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "autosomal_complete_summary.sort_values(by = 'N', ascending = False).head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a722e69-ea90-4ade-917a-56bd926ed03a",
   "metadata": {},
   "source": [
    "### Single Divergent Individual Clustering Windows <a class = 'anchor' id = 'singledivergentindividualclusteringwindows'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "631319fd",
   "metadata": {},
   "source": [
    "Let's look for different topologies. Starting with windows where a single individual is divergent to all others."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "f0b76ca9",
   "metadata": {},
   "outputs": [],
   "source": [
    "SDI_windows = complete_linkage_trees[(complete_linkage_trees['n_clusters'] == 2) & (complete_linkage_trees['n_single_clusters'] == 1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "edfb264a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "800"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(SDI_windows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "2fd10337",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <th>Alfred</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Athanga</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Bono</th>\n",
       "      <th>Bosco</th>\n",
       "      <th>Brigitta</th>\n",
       "      <th>Bwamble</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Clint</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Damian</th>\n",
       "      <th>Desmond</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Gamin</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Koby</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Koto</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mgbadolite</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>SeppToni</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Tongo</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>Vaillant</th>\n",
       "      <th>Vincent</th>\n",
       "      <th>Washu</th>\n",
       "      <th>Yogui</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "      <th>n_clusters</th>\n",
       "      <th>n_single_clusters</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr10_8388608</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.144765</td>\n",
       "      <td>0.144765</td>\n",
       "      <td>0.093802</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SeppToni/Akwaya-Jean Alfred Alice Andromeda At...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr10_10485760</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.056945</td>\n",
       "      <td>0.056945</td>\n",
       "      <td>0.01485</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Desmond/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr10_11010048</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.019138</td>\n",
       "      <td>0.019138</td>\n",
       "      <td>0.011459</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Desmond/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr10_20971520</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.024704</td>\n",
       "      <td>0.024704</td>\n",
       "      <td>0.01044</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Kidongo/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr10_21495808</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.017877</td>\n",
       "      <td>0.017877</td>\n",
       "      <td>0.005218</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Kidongo/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati  \\\n",
       "11   chr10_8388608          C1     C1    C1        C1      C1    C1     C1   \n",
       "15  chr10_10485760          C1     C1    C1        C1      C1    C1     C1   \n",
       "16  chr10_11010048          C1     C1    C1        C1      C1    C1     C1   \n",
       "29  chr10_20971520          C1     C1    C1        C1      C1    C1     C1   \n",
       "30  chr10_21495808          C1     C1    C1        C1      C1    C1     C1   \n",
       "\n",
       "   Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes  \\\n",
       "11        C1   C1    C1       C1      C1            C1                C1   \n",
       "15        C1   C1    C1       C1      C1            C1                C1   \n",
       "16        C1   C1    C1       C1      C1            C1                C1   \n",
       "29        C1   C1    C1       C1      C1            C1                C1   \n",
       "30        C1   C1    C1       C1      C1            C1                C1   \n",
       "\n",
       "   Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike  \\\n",
       "11          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "15          C1   C1    C1         C1     C1      C0    C1     C1        C1   \n",
       "16          C1   C1    C1         C1     C1      C0    C1     C1        C1   \n",
       "29          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "30          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "\n",
       "   Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby  \\\n",
       "11    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "15    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "16    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "29    C1      C1       C1    C1     C1         C1          C1      C0   C1   \n",
       "30    C1      C1       C1    C1     C1         C1          C1      C0   C1   \n",
       "\n",
       "   Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda  \\\n",
       "11      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1      C1   \n",
       "15      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1      C1   \n",
       "16      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1      C1   \n",
       "29      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1      C1   \n",
       "30      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1      C1   \n",
       "\n",
       "   Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant  \\\n",
       "11    C1      C1      C1       C0    C1   C1    C1     C1  C1       C1   \n",
       "15    C1      C1      C1       C1    C1   C1    C1     C1  C1       C1   \n",
       "16    C1      C1      C1       C1    C1   C1    C1     C1  C1       C1   \n",
       "29    C1      C1      C1       C1    C1   C1    C1     C1  C1       C1   \n",
       "30    C1      C1      C1       C1    C1   C1    C1     C1  C1       C1   \n",
       "\n",
       "   Vincent Washu Yogui node_0    node_1    node_2    node_3  n_clusters  \\\n",
       "11      C1    C1    C1    0.0  0.144765  0.144765  0.093802           2   \n",
       "15      C1    C1    C1    0.0  0.056945  0.056945   0.01485           2   \n",
       "16      C1    C1    C1    0.0  0.019138  0.019138  0.011459           2   \n",
       "29      C1    C1    C1    0.0  0.024704  0.024704   0.01044           2   \n",
       "30      C1    C1    C1    0.0  0.017877  0.017877  0.005218           2   \n",
       "\n",
       "    n_single_clusters                                cluster_composition  \n",
       "11                  1  SeppToni/Akwaya-Jean Alfred Alice Andromeda At...  \n",
       "15                  1  Desmond/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "16                  1  Desmond/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "29                  1  Kidongo/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "30                  1  Kidongo/Akwaya-Jean Alfred Alice Andromeda Ath...  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_windows.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21c691c6",
   "metadata": {},
   "source": [
    "What is the distribution of individuals in the single individual clusters?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "d4891305",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>SDI_ind</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr10_8388608</td>\n",
       "      <td>SeppToni</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr10_10485760</td>\n",
       "      <td>Desmond</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr10_11010048</td>\n",
       "      <td>Desmond</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr10_20971520</td>\n",
       "      <td>Kidongo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr10_21495808</td>\n",
       "      <td>Kidongo</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window   SDI_ind\n",
       "11   chr10_8388608  SeppToni\n",
       "15  chr10_10485760   Desmond\n",
       "16  chr10_11010048   Desmond\n",
       "29  chr10_20971520   Kidongo\n",
       "30  chr10_21495808   Kidongo"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_individuals = SDI_windows['window'].copy().to_frame('window')\n",
    "SDI_individuals['SDI_ind'] = SDI_windows['cluster_composition'].str.split('/').str[0]\n",
    "SDI_individuals.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "c2e1dbbd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDI_ind</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Alfred</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Alice</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Andromeda</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Athanga</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              N\n",
       "SDI_ind        \n",
       "Akwaya-Jean  16\n",
       "Alfred       21\n",
       "Alice         9\n",
       "Andromeda    12\n",
       "Athanga      13"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_individuals_summary = SDI_individuals.groupby(['SDI_ind']).size().to_frame('N')\n",
    "SDI_individuals_summary.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "4010aec9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "56"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(SDI_individuals_summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "0620a782",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDI_ind</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Mgbadolite</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            N\n",
       "SDI_ind      \n",
       "Mgbadolite  1"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_individuals_summary[SDI_individuals_summary['N'] == SDI_individuals_summary['N'].min()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "01d0dfe0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDI_ind</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Luky</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          N\n",
       "SDI_ind    \n",
       "Luky     34"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_individuals_summary[SDI_individuals_summary['N'] == SDI_individuals_summary['N'].max()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "b28e0995",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "N    14.285714\n",
       "dtype: float64"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_individuals_summary.mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29d7acc0",
   "metadata": {},
   "source": [
    "There is some inter-individual variation in the number of windows where an individual is highly divergent. These windows may also vary considerably in their maximum depth so let's assess the range of max divergence per individual. We will also confirm that the individual is part of the pair with the maximum."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "a966620a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>ind1</th>\n",
       "      <th>ind2</th>\n",
       "      <th>divergence</th>\n",
       "      <th>SDI_ind</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100663296</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>Lara</td>\n",
       "      <td>0.101497</td>\n",
       "      <td>Alfred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_102236160</td>\n",
       "      <td>Frederike</td>\n",
       "      <td>Lara</td>\n",
       "      <td>0.031867</td>\n",
       "      <td>Lara</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_102760448</td>\n",
       "      <td>Bono</td>\n",
       "      <td>Bwamble</td>\n",
       "      <td>0.082444</td>\n",
       "      <td>Bwamble</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_10485760</td>\n",
       "      <td>Desmond</td>\n",
       "      <td>Julie-A959</td>\n",
       "      <td>0.056945</td>\n",
       "      <td>Desmond</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_106954752</td>\n",
       "      <td>Kumbuka</td>\n",
       "      <td>Washu</td>\n",
       "      <td>0.340326</td>\n",
       "      <td>Kumbuka</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window       ind1        ind2  divergence  SDI_ind\n",
       "0  chr10_100663296     Alfred        Lara    0.101497   Alfred\n",
       "1  chr10_102236160  Frederike        Lara    0.031867     Lara\n",
       "2  chr10_102760448       Bono     Bwamble    0.082444  Bwamble\n",
       "3   chr10_10485760    Desmond  Julie-A959    0.056945  Desmond\n",
       "4  chr10_106954752    Kumbuka       Washu    0.340326  Kumbuka"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maxes_with_pair_IDs = comparisons.loc[comparisons.groupby('window')['divergence'].idxmax(), ['window', 'ind1', 'ind2', 'divergence']]\n",
    "SDI_maxes_with_pair_IDs = maxes_with_pair_IDs[maxes_with_pair_IDs['window'].isin(SDI_windows['window'])]\n",
    "SDI_maxes_with_pair_IDs = SDI_maxes_with_pair_IDs.merge(SDI_individuals, on = 'window')\n",
    "SDI_maxes_with_pair_IDs.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bcc0f53",
   "metadata": {},
   "source": [
    "Let's check if there are any rows where the SDI individual does not match the most divergent comparison."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "52212b71",
   "metadata": {},
   "outputs": [],
   "source": [
    "SDI_maxes_with_pair_IDs = SDI_maxes_with_pair_IDs[(SDI_maxes_with_pair_IDs['SDI_ind'] == SDI_maxes_with_pair_IDs['ind1']) | (SDI_maxes_with_pair_IDs['SDI_ind'] == SDI_maxes_with_pair_IDs['ind2'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "f5b74d37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "800"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(SDI_maxes_with_pair_IDs)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1515309",
   "metadata": {},
   "source": [
    "Everything looks good! Let's identify the min, average, and max divergence score per individual in windows where they are highly divergent."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "eb362f1f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>ind1</th>\n",
       "      <th>ind2</th>\n",
       "      <th>divergence</th>\n",
       "      <th>SDI_ind</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100663296</td>\n",
       "      <td>Alfred</td>\n",
       "      <td>Lara</td>\n",
       "      <td>0.101497</td>\n",
       "      <td>Alfred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_102236160</td>\n",
       "      <td>Frederike</td>\n",
       "      <td>Lara</td>\n",
       "      <td>0.031867</td>\n",
       "      <td>Lara</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_102760448</td>\n",
       "      <td>Bono</td>\n",
       "      <td>Bwamble</td>\n",
       "      <td>0.082444</td>\n",
       "      <td>Bwamble</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_10485760</td>\n",
       "      <td>Desmond</td>\n",
       "      <td>Julie-A959</td>\n",
       "      <td>0.056945</td>\n",
       "      <td>Desmond</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_106954752</td>\n",
       "      <td>Kumbuka</td>\n",
       "      <td>Washu</td>\n",
       "      <td>0.340326</td>\n",
       "      <td>Kumbuka</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window       ind1        ind2  divergence  SDI_ind\n",
       "0  chr10_100663296     Alfred        Lara    0.101497   Alfred\n",
       "1  chr10_102236160  Frederike        Lara    0.031867     Lara\n",
       "2  chr10_102760448       Bono     Bwamble    0.082444  Bwamble\n",
       "3   chr10_10485760    Desmond  Julie-A959    0.056945  Desmond\n",
       "4  chr10_106954752    Kumbuka       Washu    0.340326  Kumbuka"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_maxes_with_pair_IDs.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "f84b09f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "SDI_n_per_ind = SDI_maxes_with_pair_IDs.groupby(['SDI_ind'])['divergence'].count()\n",
    "SDI_min_per_ind = SDI_maxes_with_pair_IDs.groupby(['SDI_ind'])['divergence'].min()\n",
    "SDI_mean_per_ind = SDI_maxes_with_pair_IDs.groupby(['SDI_ind'])['divergence'].mean()\n",
    "SDI_max_per_ind = SDI_maxes_with_pair_IDs.groupby(['SDI_ind'])['divergence'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "5e8fcb60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SDI_ind</th>\n",
       "      <th>N</th>\n",
       "      <th>min</th>\n",
       "      <th>mean</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>16</td>\n",
       "      <td>0.011237</td>\n",
       "      <td>0.049478</td>\n",
       "      <td>0.204380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Alfred</td>\n",
       "      <td>21</td>\n",
       "      <td>0.004539</td>\n",
       "      <td>0.055706</td>\n",
       "      <td>0.201247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Alice</td>\n",
       "      <td>9</td>\n",
       "      <td>0.008469</td>\n",
       "      <td>0.118712</td>\n",
       "      <td>0.502525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Andromeda</td>\n",
       "      <td>12</td>\n",
       "      <td>0.013257</td>\n",
       "      <td>0.050905</td>\n",
       "      <td>0.107627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Athanga</td>\n",
       "      <td>13</td>\n",
       "      <td>0.006031</td>\n",
       "      <td>0.126002</td>\n",
       "      <td>0.463006</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       SDI_ind   N       min      mean       max\n",
       "0  Akwaya-Jean  16  0.011237  0.049478  0.204380\n",
       "1       Alfred  21  0.004539  0.055706  0.201247\n",
       "2        Alice   9  0.008469  0.118712  0.502525\n",
       "3    Andromeda  12  0.013257  0.050905  0.107627\n",
       "4      Athanga  13  0.006031  0.126002  0.463006"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_stats_per_ind = pd.DataFrame({'N':SDI_n_per_ind, 'min':SDI_min_per_ind, 'mean':SDI_mean_per_ind, 'max':SDI_max_per_ind}).reset_index()\n",
    "SDI_stats_per_ind.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd739a85",
   "metadata": {},
   "source": [
    "Map on lineage for plotting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "d43bbceb",
   "metadata": {},
   "outputs": [],
   "source": [
    "lineage_dict = {'Bono':'ppn','Desmond':'ppn','Dzeeta':'ppn','Hermien':'ppn','Hortense':'ppn','Kombote':'ppn','Kosana':'ppn','Kumbuka':'ppn','Natalie':'ppn',\n",
    "                'Akwaya-Jean':'pte', 'Damian':'pte', 'Julie-LWC21':'pte', 'Koto':'pte', 'Taweh':'pte',\n",
    "                'Andromeda':'pts','Athanga':'pts','Bihati':'pts','Bwamble':'pts','Cindy-schwein':'pts','Cleo':'pts','Coco-chimp':'pts','Frederike':'pts','Ikuru':'pts','Kidongo':'pts','Maya':'pts','Mgbadolite':'pts','Nakuu':'pts','Tongo':'pts','Trixie':'pts','Vincent':'pts','Washu':'pts',\n",
    "                'Alfred':'ptt','Blanquita':'ptt','Brigitta':'ptt','Cindy-troglodytes':'ptt','Doris':'ptt','Gamin':'ptt','Julie-A959':'ptt','Lara':'ptt','Luky':'ptt','Marlin':'ptt','Mirinda':'ptt','Negrita':'ptt','Tibe':'ptt','Ula':'ptt','Vaillant':'ptt','Yogui':'ptt',\n",
    "                'Alice':'ptv','Berta':'ptv','Bosco':'ptv','Cindy-verus':'ptv','Clint':'ptv','Jimmie':'ptv','Koby':'ptv','Linda':'ptv','SeppToni':'ptv'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "29424ac2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SDI_ind</th>\n",
       "      <th>lineage</th>\n",
       "      <th>N</th>\n",
       "      <th>min</th>\n",
       "      <th>mean</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Akwaya-Jean</td>\n",
       "      <td>pte</td>\n",
       "      <td>16</td>\n",
       "      <td>0.011237</td>\n",
       "      <td>0.049478</td>\n",
       "      <td>0.204380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Alfred</td>\n",
       "      <td>ptt</td>\n",
       "      <td>21</td>\n",
       "      <td>0.004539</td>\n",
       "      <td>0.055706</td>\n",
       "      <td>0.201247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Alice</td>\n",
       "      <td>ptv</td>\n",
       "      <td>9</td>\n",
       "      <td>0.008469</td>\n",
       "      <td>0.118712</td>\n",
       "      <td>0.502525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Andromeda</td>\n",
       "      <td>pts</td>\n",
       "      <td>12</td>\n",
       "      <td>0.013257</td>\n",
       "      <td>0.050905</td>\n",
       "      <td>0.107627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Athanga</td>\n",
       "      <td>pts</td>\n",
       "      <td>13</td>\n",
       "      <td>0.006031</td>\n",
       "      <td>0.126002</td>\n",
       "      <td>0.463006</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       SDI_ind lineage   N       min      mean       max\n",
       "0  Akwaya-Jean     pte  16  0.011237  0.049478  0.204380\n",
       "1       Alfred     ptt  21  0.004539  0.055706  0.201247\n",
       "2        Alice     ptv   9  0.008469  0.118712  0.502525\n",
       "3    Andromeda     pts  12  0.013257  0.050905  0.107627\n",
       "4      Athanga     pts  13  0.006031  0.126002  0.463006"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_stats_per_ind['lineage'] = SDI_stats_per_ind['SDI_ind'].map(lineage_dict)\n",
    "SDI_stats_per_ind = SDI_stats_per_ind[['SDI_ind','lineage','N','min','mean','max']]\n",
    "SDI_stats_per_ind.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "24492445",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['pte', 'ptt', 'ptv', 'pts', 'ppn'], dtype=object)"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_stats_per_ind['lineage'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "12f03038",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "53"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(SDI_stats_per_ind[SDI_stats_per_ind['max'] > 0.05])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "0b4841c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(SDI_stats_per_ind[SDI_stats_per_ind['max'] > 0.25])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "d828da28",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.5"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "28/56"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d787bb2f",
   "metadata": {},
   "source": [
    "Save the dataframe for plotting and inclusion in supplemental data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "10825354",
   "metadata": {},
   "outputs": [],
   "source": [
    "SDI_stats_per_ind.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/window_topologies/SDI_stats_per_ind.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e1df57d-2b10-484d-87f3-4ef4ed42f131",
   "metadata": {},
   "source": [
    "### Lineage-Specific Clustering Windows <a class = 'anchor' id = 'lineagespecificclusteringwindows'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c319a06b-a606-49de-bdcd-c19da9652051",
   "metadata": {},
   "source": [
    "Now let's look for windows where all members of a given lineage cluster separately from all other individuals. We'll write a function to do this and then pass in lists of IDs for the autosomes and chrX. We can use \"startswith\" here because the clusters are ordered from smallest to largest, thus, the smaller, lineage-specific cluster will be listed first."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "74fb27d7-3207-449c-a109-adf92e6862b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_lineage_specific_clustering_windows(IDs, female_IDs):\n",
    "    autosomal_windows = complete_linkage_trees[(complete_linkage_trees['n_clusters'] == 2) & (~complete_linkage_trees['window'].str.startswith('chrX')) & (complete_linkage_trees['cluster_composition'].str.startswith(IDs))]\n",
    "    chrX_windows = complete_linkage_trees[(complete_linkage_trees['n_clusters'] == 2) & (complete_linkage_trees['window'].str.startswith('chrX')) & (complete_linkage_trees['cluster_composition'].str.startswith(female_IDs))]\n",
    "    return len(autosomal_windows + chrX_windows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "5c7670ae-d0b4-4564-8d33-a73e91a08b24",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_IDs = 'Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/'\n",
    "ppn_female_IDs = 'Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/'\n",
    "\n",
    "pts_IDs = 'Andromeda Athanga Bihati Cindy-schwein Cleo Coco-chimp Frederike Ikuru Kidongo Maya Mgbadolite Nakuu Tongo Trixie Vincent Washu/'\n",
    "pts_female_IDs = 'Andromeda Bihati Cindy-schwein Cleo Coco-chimp Frederike Ikuru Kidongo Maya Nakuu Trixie/'\n",
    "\n",
    "ptt_IDs = 'Alfred Blanquita Brigitta Cindy-troglodytes Doris Gamin Julie-A959 Lara Luky Marlin Mirinda Negrita Tibe Ula Vaillant Yogui/'\n",
    "ptt_female_IDs = 'Blanquita Cindy-troglodytes Doris Julie-A959 Lara Luky Marlin Mirinda Negrita Tibe Ula/'\n",
    "\n",
    "ptv_IDs = 'Alice Berta Bosco Cindy-verus Clint Jimmie Koby Linda SeppToni/'\n",
    "ptv_female_IDs = 'Alice Berta Cindy-verus Jimmie Linda/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "a99fdc76-0d82-4e99-b43f-a207456dac4f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "339"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_lineage_specific_clustering_windows(ppn_IDs, ppn_female_IDs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "1652d0cc-e452-4b2a-86e7-f36d6e5ab80d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_lineage_specific_clustering_windows(pts_IDs, pts_female_IDs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "8872fe01-b6c8-43c2-a70e-b6ec88570568",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_lineage_specific_clustering_windows(ptt_IDs, ptt_female_IDs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "2587348b-15e6-4ec1-9ed4-8fef6907cffa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_lineage_specific_clustering_windows(ptv_IDs, ptv_female_IDs)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "862be985-e181-4314-802f-9f088503b240",
   "metadata": {},
   "source": [
    "Let's do this manually for Nigeria-Cameroon chimpanzees. There is only one female; therefore, any lineage-specific chr X patterns are designated as rare contact pattern windows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "e986e9d4-c474-4de6-a8a1-c7fcfbbd9b98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(complete_linkage_trees[(complete_linkage_trees['n_clusters'] == 2) & (~complete_linkage_trees['window'].str.startswith('chrX')) & (complete_linkage_trees['cluster_composition'].str.startswith('Akwaya-Jean Damian Julie-LWC21 Koto Taweh/'))])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60f2f3d5-8d4d-42ff-bea9-646ad6632e8e",
   "metadata": {},
   "source": [
    "Grab the bonobo-chimpanzee clustering windows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "ad410cd0-635a-49e8-b86d-f6f8b0ad6218",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_windows = complete_linkage_trees[(complete_linkage_trees['n_clusters'] == 2) & (~complete_linkage_trees['window'].str.startswith('chrX')) & (complete_linkage_trees['cluster_composition'].str.startswith('Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/')) | (complete_linkage_trees['n_clusters'] == 2) & (complete_linkage_trees['window'].str.startswith('chrX')) & (complete_linkage_trees['cluster_composition'].str.startswith('Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/'))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "71adff8d-074e-478b-91e2-b3b4f813697a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "339"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_windows)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "647acb8b-1292-4e79-82b9-dac6a4bb5fd8",
   "metadata": {},
   "source": [
    "Now let's take a closer look at the western chimpanzee windows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "298f23da-6f08-4f94-a72c-64061f2d93d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <th>Alfred</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Athanga</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Bono</th>\n",
       "      <th>Bosco</th>\n",
       "      <th>Brigitta</th>\n",
       "      <th>Bwamble</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Clint</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Damian</th>\n",
       "      <th>Desmond</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Gamin</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Koby</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Koto</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mgbadolite</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>SeppToni</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Tongo</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>Vaillant</th>\n",
       "      <th>Vincent</th>\n",
       "      <th>Washu</th>\n",
       "      <th>Yogui</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "      <th>n_clusters</th>\n",
       "      <th>n_single_clusters</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>chr15_28835840</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000993</td>\n",
       "      <td>0.007565</td>\n",
       "      <td>0.007565</td>\n",
       "      <td>0.004479</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Cindy-verus Clint Jimmie Kob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020</th>\n",
       "      <td>chr15_40370176</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002281</td>\n",
       "      <td>0.009507</td>\n",
       "      <td>0.009507</td>\n",
       "      <td>0.006044</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Cindy-verus Clint Jimmie Kob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1798</th>\n",
       "      <td>chr1_210239488</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.00071</td>\n",
       "      <td>0.019751</td>\n",
       "      <td>0.019751</td>\n",
       "      <td>0.013096</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Cindy-verus Clint Jimmie Kob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2708</th>\n",
       "      <td>chr3_169345024</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002744</td>\n",
       "      <td>0.014599</td>\n",
       "      <td>0.014599</td>\n",
       "      <td>0.007954</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Cindy-verus Clint Jimmie Kob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3153</th>\n",
       "      <td>chr5_44564480</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000428</td>\n",
       "      <td>0.011706</td>\n",
       "      <td>0.011706</td>\n",
       "      <td>0.004777</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Cindy-verus Clint Jimmie Kob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3154</th>\n",
       "      <td>chr5_45088768</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000929</td>\n",
       "      <td>0.023077</td>\n",
       "      <td>0.023077</td>\n",
       "      <td>0.014528</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Cindy-verus Clint Jimmie Kob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3425</th>\n",
       "      <td>chr6_52428800</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001113</td>\n",
       "      <td>0.009093</td>\n",
       "      <td>0.009093</td>\n",
       "      <td>0.003563</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Cindy-verus Clint Jimmie Kob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3426</th>\n",
       "      <td>chr6_52953088</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.006607</td>\n",
       "      <td>0.030093</td>\n",
       "      <td>0.030093</td>\n",
       "      <td>0.016993</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Cindy-verus Clint Jimmie Kob...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              window Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati  \\\n",
       "998   chr15_28835840          C2     C2    C1        C2      C2    C1     C2   \n",
       "1020  chr15_40370176          C2     C2    C1        C2      C2    C1     C2   \n",
       "1798  chr1_210239488          C2     C2    C1        C2      C2    C1     C2   \n",
       "2708  chr3_169345024          C2     C2    C1        C2      C2    C1     C2   \n",
       "3153   chr5_44564480          C2     C2    C1        C2      C2    C1     C2   \n",
       "3154   chr5_45088768          C2     C2    C1        C2      C2    C1     C2   \n",
       "3425   chr6_52428800          C2     C2    C1        C2      C2    C1     C2   \n",
       "3426   chr6_52953088          C2     C2    C1        C2      C2    C1     C2   \n",
       "\n",
       "     Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes  \\\n",
       "998         C2   C2    C1       C2      C2            C2                C2   \n",
       "1020        C2   C2    C1       C2      C2            C2                C2   \n",
       "1798        C2   C2    C1       C2      C2            C2                C2   \n",
       "2708        C2   C2    C1       C2      C2            C2                C2   \n",
       "3153        C2   C2    C1       C2      C2            C2                C2   \n",
       "3154        C2   C2    C1       C2      C2            C2                C2   \n",
       "3425        C2   C2    C1       C2      C2            C2                C2   \n",
       "3426        C2   C2    C1       C2      C2            C2                C2   \n",
       "\n",
       "     Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike  \\\n",
       "998           C1   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "1020          C1   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "1798          C1   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "2708          C1   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "3153          C1   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "3154          C1   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "3425          C1   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "3426          C1   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "\n",
       "     Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby  \\\n",
       "998     C2      C2       C2    C2     C1         C2          C2      C2   C1   \n",
       "1020    C2      C2       C2    C2     C1         C2          C2      C2   C1   \n",
       "1798    C2      C2       C2    C2     C1         C2          C2      C2   C1   \n",
       "2708    C2      C2       C2    C2     C1         C2          C2      C2   C1   \n",
       "3153    C2      C2       C2    C2     C1         C2          C2      C2   C1   \n",
       "3154    C2      C2       C2    C2     C1         C2          C2      C2   C1   \n",
       "3425    C2      C2       C2    C2     C1         C2          C2      C2   C1   \n",
       "3426    C2      C2       C2    C2     C1         C2          C2      C2   C1   \n",
       "\n",
       "     Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite  \\\n",
       "998       C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "1020      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "1798      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "2708      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "3153      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "3154      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "3425      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "3426      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "\n",
       "     Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula  \\\n",
       "998       C2    C2      C2      C2       C1    C2   C2    C2     C2  C2   \n",
       "1020      C2    C2      C2      C2       C1    C2   C2    C2     C2  C2   \n",
       "1798      C2    C2      C2      C2       C1    C2   C2    C2     C2  C2   \n",
       "2708      C2    C2      C2      C2       C1    C2   C2    C2     C2  C2   \n",
       "3153      C2    C2      C2      C2       C1    C2   C2    C2     C2  C2   \n",
       "3154      C2    C2      C2      C2       C1    C2   C2    C2     C2  C2   \n",
       "3425      C2    C2      C2      C2       C1    C2   C2    C2     C2  C2   \n",
       "3426      C2    C2      C2      C2       C1    C2   C2    C2     C2  C2   \n",
       "\n",
       "     Vaillant Vincent Washu Yogui    node_0    node_1    node_2    node_3  \\\n",
       "998        C2      C2    C2    C2  0.000993  0.007565  0.007565  0.004479   \n",
       "1020       C2      C2    C2    C2  0.002281  0.009507  0.009507  0.006044   \n",
       "1798       C2      C2    C2    C2   0.00071  0.019751  0.019751  0.013096   \n",
       "2708       C2      C2    C2    C2  0.002744  0.014599  0.014599  0.007954   \n",
       "3153       C2      C2    C2    C2  0.000428  0.011706  0.011706  0.004777   \n",
       "3154       C2      C2    C2    C2  0.000929  0.023077  0.023077  0.014528   \n",
       "3425       C2      C2    C2    C2  0.001113  0.009093  0.009093  0.003563   \n",
       "3426       C2      C2    C2    C2  0.006607  0.030093  0.030093  0.016993   \n",
       "\n",
       "      n_clusters  n_single_clusters  \\\n",
       "998            2                  0   \n",
       "1020           2                  0   \n",
       "1798           2                  0   \n",
       "2708           2                  0   \n",
       "3153           2                  0   \n",
       "3154           2                  0   \n",
       "3425           2                  0   \n",
       "3426           2                  0   \n",
       "\n",
       "                                    cluster_composition  \n",
       "998   Alice Berta Bosco Cindy-verus Clint Jimmie Kob...  \n",
       "1020  Alice Berta Bosco Cindy-verus Clint Jimmie Kob...  \n",
       "1798  Alice Berta Bosco Cindy-verus Clint Jimmie Kob...  \n",
       "2708  Alice Berta Bosco Cindy-verus Clint Jimmie Kob...  \n",
       "3153  Alice Berta Bosco Cindy-verus Clint Jimmie Kob...  \n",
       "3154  Alice Berta Bosco Cindy-verus Clint Jimmie Kob...  \n",
       "3425  Alice Berta Bosco Cindy-verus Clint Jimmie Kob...  \n",
       "3426  Alice Berta Bosco Cindy-verus Clint Jimmie Kob...  "
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "western_chimpanzee_divergent_windows = complete_linkage_trees[(complete_linkage_trees['cluster_composition'] == 'Alice Berta Bosco Cindy-verus Clint Jimmie Koby Linda SeppToni/Akwaya-Jean Alfred Andromeda Athanga Bihati Blanquita Bono Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Coco-chimp Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Ikuru Julie-A959 Julie-LWC21 Kidongo Kombote Kosana Koto Kumbuka Lara Luky Marlin Maya Mgbadolite Mirinda Nakuu Natalie Negrita Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui') | (complete_linkage_trees['cluster_composition'] == 'Alice Berta Cindy-verus Jimmie Linda/Andromeda Bihati Blanquita Cindy-schwein Cindy-troglodytes Cleo Coco-chimp Doris Dzeeta Frederike Hermien Hortense Ikuru Julie-A959 Julie-LWC21 Kidongo Kombote Kosana Kumbuka Lara Luky Marlin Maya Mirinda Nakuu Natalie Negrita Taweh Tibe Trixie Ula')]\n",
    "western_chimpanzee_divergent_windows"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa48563f-97f3-40d4-838b-b610c721882f",
   "metadata": {},
   "source": [
    "Do any of these windows intersect interesting genes? First, create a BED file of the windows above and then create a pybedtools object."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "e8f7332e-bc8a-4311-a8e2-8ca1b9a402ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1798</th>\n",
       "      <td>chr1</td>\n",
       "      <td>210239488</td>\n",
       "      <td>211288064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>chr15</td>\n",
       "      <td>28835840</td>\n",
       "      <td>29884416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020</th>\n",
       "      <td>chr15</td>\n",
       "      <td>40370176</td>\n",
       "      <td>41418752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2708</th>\n",
       "      <td>chr3</td>\n",
       "      <td>169345024</td>\n",
       "      <td>170393600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3153</th>\n",
       "      <td>chr5</td>\n",
       "      <td>44564480</td>\n",
       "      <td>45613056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3154</th>\n",
       "      <td>chr5</td>\n",
       "      <td>45088768</td>\n",
       "      <td>46137344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3425</th>\n",
       "      <td>chr6</td>\n",
       "      <td>52428800</td>\n",
       "      <td>53477376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3426</th>\n",
       "      <td>chr6</td>\n",
       "      <td>52953088</td>\n",
       "      <td>54001664</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr  window_start  window_end\n",
       "1798   chr1     210239488   211288064\n",
       "998   chr15      28835840    29884416\n",
       "1020  chr15      40370176    41418752\n",
       "2708   chr3     169345024   170393600\n",
       "3153   chr5      44564480    45613056\n",
       "3154   chr5      45088768    46137344\n",
       "3425   chr6      52428800    53477376\n",
       "3426   chr6      52953088    54001664"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "western_chimpanzee_divergent_windows_BED = pd.DataFrame()\n",
    "western_chimpanzee_divergent_windows_BED['window_split'] = western_chimpanzee_divergent_windows['window']\n",
    "western_chimpanzee_divergent_windows_BED = western_chimpanzee_divergent_windows_BED['window_split'].str.split('_', expand=True)\n",
    "western_chimpanzee_divergent_windows_BED.rename(columns = {0:'chr', 1:'window_start'}, inplace = True)\n",
    "western_chimpanzee_divergent_windows_BED['window_start'] = western_chimpanzee_divergent_windows_BED['window_start'].astype(int)\n",
    "western_chimpanzee_divergent_windows_BED['window_end'] = western_chimpanzee_divergent_windows_BED['window_start'] + 1048576\n",
    "western_chimpanzee_divergent_windows_BED = western_chimpanzee_divergent_windows_BED.sort_values(by = ['chr','window_start'])\n",
    "western_chimpanzee_divergent_windows_BED.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "de3a8dce-c228-450d-94c0-c900336631d9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t210239488\t211288064\n",
      " chr15\t28835840\t29884416\n",
      " chr15\t40370176\t41418752\n",
      " chr3\t169345024\t170393600\n",
      " chr5\t44564480\t45613056\n",
      " chr5\t45088768\t46137344\n",
      " chr6\t52428800\t53477376\n",
      " chr6\t52953088\t54001664\n",
      " "
     ]
    }
   ],
   "source": [
    "western_chimpanzee_divergent_windows_pbtBED = pybedtools.BedTool().from_dataframe(western_chimpanzee_divergent_windows_BED).sort()\n",
    "western_chimpanzee_divergent_windows_pbtBED.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99a2096a-e9da-43d5-8c7d-3a5b35422b68",
   "metadata": {},
   "source": [
    "Now read in the panTro6 genes as a pybedtools object."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "61c58557-72c5-4b11-b3ba-144ce9fddf2d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t344\t12299\tNM_001280424.1\tINTS11\n",
      " chr1\t12400\t16513\tXM_016952033.2\tCPTP\n",
      " chr1\t17887\t23426\tXM_003307748.4\tTAS1R3\n",
      " chr1\t22943\t36942\tXM_016958290.2\tDVL1\n",
      " chr1\t39619\t45507\tNM_001280245.1\tMXRA8\n",
      " "
     ]
    }
   ],
   "source": [
    "genes_pbtBED = pybedtools.BedTool('/wynton/group/capra/projects/pan_3d_genome/data/annotations/panTro6_genes.bed')\n",
    "genes_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "385d68c6-ad01-46a7-b6f7-db5b59187aa2",
   "metadata": {},
   "source": [
    "Intersect the western chimpanzee divergent windows with panTro6 genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "c27be111-b871-4996-8090-ab99693572e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>gene_chr</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>gene_transcript</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>210239488</td>\n",
       "      <td>211288064</td>\n",
       "      <td>chr1</td>\n",
       "      <td>210270967</td>\n",
       "      <td>210286436</td>\n",
       "      <td>XM_016941282.1</td>\n",
       "      <td>GGPS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>210239488</td>\n",
       "      <td>211288064</td>\n",
       "      <td>chr1</td>\n",
       "      <td>210393829</td>\n",
       "      <td>210457855</td>\n",
       "      <td>XM_016941305.2</td>\n",
       "      <td>B3GALNT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>210239488</td>\n",
       "      <td>211288064</td>\n",
       "      <td>chr1</td>\n",
       "      <td>210504199</td>\n",
       "      <td>210604057</td>\n",
       "      <td>XM_009441703.3</td>\n",
       "      <td>GNG4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>210239488</td>\n",
       "      <td>211288064</td>\n",
       "      <td>chr1</td>\n",
       "      <td>210927111</td>\n",
       "      <td>211015721</td>\n",
       "      <td>XM_001156001.5</td>\n",
       "      <td>NID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>210239488</td>\n",
       "      <td>211288064</td>\n",
       "      <td>chr1</td>\n",
       "      <td>211167578</td>\n",
       "      <td>211234322</td>\n",
       "      <td>XM_016927789.2</td>\n",
       "      <td>ERO1B</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  window_chr  window_start  window_end gene_chr  gene_start   gene_end  \\\n",
       "0       chr1     210239488   211288064     chr1   210270967  210286436   \n",
       "1       chr1     210239488   211288064     chr1   210393829  210457855   \n",
       "2       chr1     210239488   211288064     chr1   210504199  210604057   \n",
       "3       chr1     210239488   211288064     chr1   210927111  211015721   \n",
       "4       chr1     210239488   211288064     chr1   211167578  211234322   \n",
       "\n",
       "  gene_transcript      gene  \n",
       "0  XM_016941282.1     GGPS1  \n",
       "1  XM_016941305.2  B3GALNT2  \n",
       "2  XM_009441703.3      GNG4  \n",
       "3  XM_001156001.5      NID1  \n",
       "4  XM_016927789.2     ERO1B  "
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "western_chimpanzee_divergent_windows_genes_intersect = western_chimpanzee_divergent_windows_pbtBED.intersect(genes_pbtBED, wa = True, wb = True).to_dataframe(names=['window_chr','window_start','window_end','gene_chr','gene_start','gene_end','gene_transcript','gene'])\n",
    "western_chimpanzee_divergent_windows_genes_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "bdf43392-af6c-438e-99d7-25d4dc7e1ad6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "52"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(western_chimpanzee_divergent_windows_genes_intersect['gene'].drop_duplicates())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82346e4c-859e-45fd-b51b-6afbc0b2620f",
   "metadata": {},
   "source": [
    "Let's grab the identities of the pair for the most divergent comparison per western chimpanzee divergent window."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "340c486d-1e91-451a-ab27-99b7c4b49c41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                     Gamin\n",
       "ind2                  SeppToni\n",
       "lineages               ptt-ptv\n",
       "chr                       chr1\n",
       "window_start         210239488\n",
       "window          chr1_210239488\n",
       "mse                   0.013139\n",
       "spearman              0.980249\n",
       "divergence            0.019751\n",
       "seq_diff                  2283\n",
       "Name: 4472871, dtype: object"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr1_210239488')].sort_values(by = 'divergence', ascending = False).iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "27d2b15e-eb2e-4e22-9bb2-34a8a5f5cbe2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                    Jimmie\n",
       "ind2                     Tongo\n",
       "lineages               pts-ptv\n",
       "chr                       chr3\n",
       "window_start         169345024\n",
       "window          chr3_169345024\n",
       "mse                   0.003829\n",
       "spearman              0.985401\n",
       "divergence            0.014599\n",
       "seq_diff                  2550\n",
       "Name: 5002410, dtype: object"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr3_169345024')].sort_values(by = 'divergence', ascending = False).iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "6b9ec9f7-df03-404d-aad5-535916c8d79d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                  Hermien\n",
       "ind2                     Koby\n",
       "lineages              ppn-ptr\n",
       "chr                      chr5\n",
       "window_start         44564480\n",
       "window          chr5_44564480\n",
       "mse                  0.009547\n",
       "spearman             0.988294\n",
       "divergence           0.011706\n",
       "seq_diff                 2870\n",
       "Name: 4543436, dtype: object"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr5_44564480')].sort_values(by = 'divergence', ascending = False).iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "0719ca4f-5328-404d-ac20-e23d33111321",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1            Cindy-schwein\n",
       "ind2                   Jimmie\n",
       "lineages              pts-ptv\n",
       "chr                      chr5\n",
       "window_start         45088768\n",
       "window          chr5_45088768\n",
       "mse                  0.006429\n",
       "spearman             0.976923\n",
       "divergence           0.023077\n",
       "seq_diff                 2542\n",
       "Name: 2625131, dtype: object"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr5_45088768')].sort_values(by = 'divergence', ascending = False).iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "4ffb3f7b-15c1-4ce9-8e50-d1394c7f6fc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                    Bosco\n",
       "ind2               Julie-A959\n",
       "lineages              ptt-ptv\n",
       "chr                      chr6\n",
       "window_start         52428800\n",
       "window          chr6_52428800\n",
       "mse                  0.005152\n",
       "spearman             0.990907\n",
       "divergence           0.009093\n",
       "seq_diff                 2202\n",
       "Name: 2064653, dtype: object"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr6_52428800')].sort_values(by = 'divergence', ascending = False).iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "3fb8cb20-6e92-45dc-8f4a-214b92ffddd8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                    Bosco\n",
       "ind2                  Kumbuka\n",
       "lineages              ppn-ptr\n",
       "chr                      chr6\n",
       "window_start         52953088\n",
       "window          chr6_52953088\n",
       "mse                  0.009424\n",
       "spearman             0.969907\n",
       "divergence           0.030093\n",
       "seq_diff                 3791\n",
       "Name: 2094537, dtype: object"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr6_52953088')].sort_values(by = 'divergence', ascending = False).iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "750e3b63-8005-40a4-b094-759613da20de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                     Bosco\n",
       "ind2                   Mirinda\n",
       "lineages               ptt-ptv\n",
       "chr                      chr15\n",
       "window_start          28835840\n",
       "window          chr15_28835840\n",
       "mse                   0.001035\n",
       "spearman              0.992435\n",
       "divergence            0.007565\n",
       "seq_diff                  1963\n",
       "Name: 2121992, dtype: object"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr15_28835840')].sort_values(by = 'divergence', ascending = False).iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "c3869648-444c-42e3-ad70-cc63b1be5a98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                 Andromeda\n",
       "ind2                     Clint\n",
       "lineages               pts-ptv\n",
       "chr                      chr15\n",
       "window_start          40370176\n",
       "window          chr15_40370176\n",
       "mse                   0.002546\n",
       "spearman              0.990493\n",
       "divergence            0.009507\n",
       "seq_diff                  2485\n",
       "Name: 750168, dtype: object"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr15_40370176')].sort_values(by = 'divergence', ascending = False).iloc[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0cfd99e-54b5-4580-bae9-cf1a85475af1",
   "metadata": {},
   "source": [
    "### Multiple Divergent Individuals Clustering Windows <a class = 'anchor' id = 'multipledivergentindividualsclusteringwindows'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90315905",
   "metadata": {},
   "source": [
    "Thus far, we have identified 800 single divergent individual clustering windows, 339 bonobo-chimpanzee clustering windows, and 8 Western chimpanzee clustering windows. However, this still leaves 2475 two cluster windows: 3622 - (800 + 339 + 8). Based on the autosomal and X chromosome dataframes above, the remaining two cluster windows appear to be topologies where a smaller cluster of individuals from the same lineage or multiple lineages cluster together."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "36368844",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "MDI_windows = complete_linkage_trees[complete_linkage_trees['n_clusters'] == 2]\n",
    "windows_to_exclude = pd.concat([SDI_windows, ppn_ptr_windows, western_chimpanzee_divergent_windows])['window']\n",
    "MDI_windows = MDI_windows[~MDI_windows['window'].isin(windows_to_exclude)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "a03b0bb8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2475"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(MDI_windows)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c884a1d0",
   "metadata": {},
   "source": [
    "What is the range of size and lineage composition of the smaller clusters in the MDIs?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "e27138e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1                [Cindy-troglodytes, Doris, Luky, Marlin]\n",
       "2                [Cindy-troglodytes, Doris, Luky, Marlin]\n",
       "3                              [Alice, Cindy-verus, Luky]\n",
       "6       [Alfred, Andromeda, Bono, Coco-chimp, Doris, D...\n",
       "8       [Bono, Desmond, Dzeeta, Hermien, Hortense, Kos...\n",
       "                              ...                        \n",
       "4411                                    [Dzeeta, Kumbuka]\n",
       "4414                              [Bihati, Cindy-schwein]\n",
       "4415    [Cindy-troglodytes, Luky, Marlin, Negrita, Tib...\n",
       "4418    [Blanquita, Dzeeta, Hermien, Hortense, Kombote...\n",
       "4419    [Andromeda, Cindy-schwein, Cindy-troglodytes, ...\n",
       "Name: cluster_composition, Length: 2475, dtype: object"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MDI_windows_small_cluster = MDI_windows['cluster_composition'].str.split('/').apply(lambda x: x[0].split())\n",
    "MDI_windows_small_cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "01d69824",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>small_cluster_N</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>9</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>11</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>13</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>14</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>15</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>16</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>17</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>18</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>19</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>20</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>21</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>22</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>23</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>24</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>26</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>27</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>28</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    small_cluster_N    N\n",
       "0                 2  436\n",
       "1                 3  264\n",
       "2                 4  166\n",
       "3                 5  154\n",
       "4                 6  122\n",
       "8                 7   98\n",
       "7                 8  101\n",
       "11                9   68\n",
       "5                10  116\n",
       "6                11  102\n",
       "10               12   69\n",
       "9                13   71\n",
       "17               14   53\n",
       "18               15   52\n",
       "12               16   62\n",
       "14               17   55\n",
       "23               18   42\n",
       "21               19   44\n",
       "25               20   39\n",
       "22               21   42\n",
       "13               22   57\n",
       "15               23   55\n",
       "19               24   52\n",
       "24               25   40\n",
       "16               26   54\n",
       "20               27   45\n",
       "26               28   16"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MDI_windows_small_cluster_counts = MDI_windows_small_cluster.apply(lambda x: len(x)).to_frame('small_cluster_N')\n",
    "MDI_windows_small_cluster_counts_summary = MDI_windows_small_cluster_counts['small_cluster_N'].value_counts().to_frame('N').reset_index().rename(columns = {'index':'small_cluster_N'}).sort_values(by=['small_cluster_N'])\n",
    "MDI_windows_small_cluster_counts_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "fc31d2ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "small_cluster_N    7.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MDI_windows_small_cluster_counts.median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "712330ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "MDI_windows_small_cluster_counts_summary.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/window_topologies/MDI_windows_small_cluster_counts_summary.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "c28c8923",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[ptt, ptt, ptt, ptt]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[ptt, ptt, ptt, ptt]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[ptv, ptv, ptt]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>[ptt, pts, ppn, pts, ptt, ppn, ptt, ppn, ppn, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>[ppn, ppn, ppn, ppn, ppn, ppn, ppn, ppn]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         composition\n",
       "1                               [ptt, ptt, ptt, ptt]\n",
       "2                               [ptt, ptt, ptt, ptt]\n",
       "3                                    [ptv, ptv, ptt]\n",
       "6  [ptt, pts, ppn, pts, ptt, ppn, ptt, ppn, ppn, ...\n",
       "8           [ppn, ppn, ppn, ppn, ppn, ppn, ppn, ppn]"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MDI_windows_small_cluster_composition = MDI_windows_small_cluster.map(lambda x: [lineage_dict.get(item, item) for item in x]).to_frame('composition')\n",
    "MDI_windows_small_cluster_composition.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "0b8eb202",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>composition</th>\n",
       "      <th>ppn_present</th>\n",
       "      <th>pte_present</th>\n",
       "      <th>pts_present</th>\n",
       "      <th>ptt_present</th>\n",
       "      <th>ptv_present</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[ptt, ptt, ptt, ptt]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[ptt, ptt, ptt, ptt]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[ptv, ptv, ptt]</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>[ptt, pts, ppn, pts, ptt, ppn, ptt, ppn, ppn, ...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>[ppn, ppn, ppn, ppn, ppn, ppn, ppn, ppn]</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         composition  ppn_present  \\\n",
       "1                               [ptt, ptt, ptt, ptt]        False   \n",
       "2                               [ptt, ptt, ptt, ptt]        False   \n",
       "3                                    [ptv, ptv, ptt]        False   \n",
       "6  [ptt, pts, ppn, pts, ptt, ppn, ptt, ppn, ppn, ...         True   \n",
       "8           [ppn, ppn, ppn, ppn, ppn, ppn, ppn, ppn]         True   \n",
       "\n",
       "   pte_present  pts_present  ptt_present  ptv_present  \n",
       "1        False        False         True        False  \n",
       "2        False        False         True        False  \n",
       "3        False        False         True         True  \n",
       "6         True         True         True        False  \n",
       "8        False        False        False        False  "
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pan_lineages = ['ppn','pte','pts','ptt','ptv']\n",
    "for lineage in pan_lineages:\n",
    "    MDI_windows_small_cluster_composition[lineage + '_present'] = MDI_windows_small_cluster_composition['composition'].apply(lambda x: lineage in x)\n",
    "MDI_windows_small_cluster_composition.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "946bb82a",
   "metadata": {},
   "outputs": [],
   "source": [
    "MDI_windows_small_cluster_composition[['ppn_present','pte_present','pts_present','ptt_present','ptv_present']].to_csv('/wynton/group/capra/projects/pan_3d_genome/data/window_topologies/MDI_windows_small_cluster_composition.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec65987d-3180-4ec6-b8b3-732de4428584",
   "metadata": {},
   "source": [
    "### Chr21 Window Topologies <a class = 'anchor' id = 'chr21windowtopologies'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09fee4b4",
   "metadata": {},
   "source": [
    "Now let's look at the topologies along an entire chromosome for Figure 2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "67ba71b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <th>Alfred</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Athanga</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Bono</th>\n",
       "      <th>Bosco</th>\n",
       "      <th>Brigitta</th>\n",
       "      <th>Bwamble</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Clint</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Damian</th>\n",
       "      <th>Desmond</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Gamin</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Koby</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Koto</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mgbadolite</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>SeppToni</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Tongo</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>Vaillant</th>\n",
       "      <th>Vincent</th>\n",
       "      <th>Washu</th>\n",
       "      <th>Yogui</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "      <th>n_clusters</th>\n",
       "      <th>n_single_clusters</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1915</th>\n",
       "      <td>chr21_1048576</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.007273</td>\n",
       "      <td>0.101743</td>\n",
       "      <td>0.101743</td>\n",
       "      <td>0.018713</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cindy-troglodytes Kosana/Akwaya-Jean Alfred Al...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1916</th>\n",
       "      <td>chr21_1572864</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.009908</td>\n",
       "      <td>0.092433</td>\n",
       "      <td>0.092433</td>\n",
       "      <td>0.030027</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cindy-troglodytes Kosana/Akwaya-Jean Alfred Al...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1917</th>\n",
       "      <td>chr21_2097152</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.008533</td>\n",
       "      <td>0.016795</td>\n",
       "      <td>0.016795</td>\n",
       "      <td>0.013133</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Andromeda Cleo Coco-chimp Frederike Ikuru Juli...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1918</th>\n",
       "      <td>chr21_2621440</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001896</td>\n",
       "      <td>0.012374</td>\n",
       "      <td>0.012374</td>\n",
       "      <td>0.008402</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1919</th>\n",
       "      <td>chr21_3145728</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.058120</td>\n",
       "      <td>0.058120</td>\n",
       "      <td>0.038627</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Hortense/Akwaya-Jean Alfred Alice Andromeda At...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1920</th>\n",
       "      <td>chr21_3670016</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005336</td>\n",
       "      <td>0.025069</td>\n",
       "      <td>0.025069</td>\n",
       "      <td>0.007850</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hortense Kombote Kosana Ku...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1921</th>\n",
       "      <td>chr21_4194304</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001153</td>\n",
       "      <td>0.017037</td>\n",
       "      <td>0.017037</td>\n",
       "      <td>0.009157</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Blanquita Cindy-troglodytes Ula/Akwaya-Jean Al...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1922</th>\n",
       "      <td>chr21_4718592</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.056423</td>\n",
       "      <td>0.056423</td>\n",
       "      <td>0.021865</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SeppToni/Akwaya-Jean Alfred Alice Andromeda At...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1923</th>\n",
       "      <td>chr21_5242880</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.119308</td>\n",
       "      <td>0.119308</td>\n",
       "      <td>0.041856</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SeppToni/Akwaya-Jean Alfred Alice Andromeda At...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1924</th>\n",
       "      <td>chr21_5767168</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.049416</td>\n",
       "      <td>0.123899</td>\n",
       "      <td>0.123899</td>\n",
       "      <td>0.108459</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>Luky/Bihati Bwamble Frederike Julie-A959 Mgbad...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1925</th>\n",
       "      <td>chr21_6291456</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.029523</td>\n",
       "      <td>0.155177</td>\n",
       "      <td>0.155177</td>\n",
       "      <td>0.064509</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bihati Bwamble Frederike Julie-A959 Mgbadolite...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1926</th>\n",
       "      <td>chr21_6815744</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002659</td>\n",
       "      <td>0.064121</td>\n",
       "      <td>0.064121</td>\n",
       "      <td>0.027898</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Dzeeta Hortense Kombote Kumbuka/Akwaya-Je...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1927</th>\n",
       "      <td>chr21_7340032</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.075647</td>\n",
       "      <td>0.075647</td>\n",
       "      <td>0.026706</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Natalie/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1928</th>\n",
       "      <td>chr21_7864320</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.183761</td>\n",
       "      <td>0.183761</td>\n",
       "      <td>0.053167</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Natalie/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1929</th>\n",
       "      <td>chr21_8388608</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.142003</td>\n",
       "      <td>0.142003</td>\n",
       "      <td>0.023950</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Vaillant/Akwaya-Jean Alfred Alice Andromeda At...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1930</th>\n",
       "      <td>chr21_8912896</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.037891</td>\n",
       "      <td>0.037891</td>\n",
       "      <td>0.011049</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Vaillant/Akwaya-Jean Alfred Alice Andromeda At...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1931</th>\n",
       "      <td>chr21_9437184</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025588</td>\n",
       "      <td>0.025588</td>\n",
       "      <td>0.004655</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Bosco/Akwaya-Jean Alfred Alice Andromeda Athan...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1932</th>\n",
       "      <td>chr21_9961472</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.022009</td>\n",
       "      <td>0.075191</td>\n",
       "      <td>0.075191</td>\n",
       "      <td>0.048403</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Doris Dzeeta Hermien Hortense Kom...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1933</th>\n",
       "      <td>chr21_10485760</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.143209</td>\n",
       "      <td>0.143209</td>\n",
       "      <td>0.067397</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Cindy-troglodytes/Akwaya-Jean Alfred Alice And...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1934</th>\n",
       "      <td>chr21_11010048</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.179972</td>\n",
       "      <td>0.179972</td>\n",
       "      <td>0.084036</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Cindy-troglodytes/Akwaya-Jean Alfred Alice And...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1935</th>\n",
       "      <td>chr21_11534336</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000866</td>\n",
       "      <td>0.006946</td>\n",
       "      <td>0.006946</td>\n",
       "      <td>0.004649</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1936</th>\n",
       "      <td>chr21_12058624</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>0.004243</td>\n",
       "      <td>0.007764</td>\n",
       "      <td>0.007764</td>\n",
       "      <td>0.007669</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1937</th>\n",
       "      <td>chr21_12582912</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>0.006425</td>\n",
       "      <td>0.009628</td>\n",
       "      <td>0.009628</td>\n",
       "      <td>0.007451</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Cindy-troglodytes Mirinda/Alfred Alice Athanga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1938</th>\n",
       "      <td>chr21_13107200</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001945</td>\n",
       "      <td>0.011701</td>\n",
       "      <td>0.011701</td>\n",
       "      <td>0.009078</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Julie-LWC21 Tibe/Alice Athanga Berta Bihati Bl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1939</th>\n",
       "      <td>chr21_13631488</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.007718</td>\n",
       "      <td>0.030535</td>\n",
       "      <td>0.030535</td>\n",
       "      <td>0.016379</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Athanga Berta Bosco Brigitta Cindy-trogl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1940</th>\n",
       "      <td>chr21_14155776</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002078</td>\n",
       "      <td>0.016118</td>\n",
       "      <td>0.016118</td>\n",
       "      <td>0.011492</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Akwaya-Jean Athanga Bosco Clint Mgbadolite Vin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1941</th>\n",
       "      <td>chr21_14680064</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000191</td>\n",
       "      <td>0.052577</td>\n",
       "      <td>0.052577</td>\n",
       "      <td>0.013309</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Linda/Akwaya-Jean Alfred And...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1942</th>\n",
       "      <td>chr21_15204352</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000181</td>\n",
       "      <td>0.019700</td>\n",
       "      <td>0.019700</td>\n",
       "      <td>0.012599</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Linda/Akwaya-Jean Alfred And...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1943</th>\n",
       "      <td>chr21_15728640</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>0.001821</td>\n",
       "      <td>0.004103</td>\n",
       "      <td>0.004103</td>\n",
       "      <td>0.003296</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Akwaya-Jean Alfred Alice Berta Bosco Cindy-tro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1944</th>\n",
       "      <td>chr21_16252928</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.004971</td>\n",
       "      <td>0.049261</td>\n",
       "      <td>0.049261</td>\n",
       "      <td>0.018393</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cleo Ula Washu/Akwaya-Jean Alfred Alice Androm...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1945</th>\n",
       "      <td>chr21_16777216</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.044758</td>\n",
       "      <td>0.105713</td>\n",
       "      <td>0.105713</td>\n",
       "      <td>0.046932</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alfred Andromeda Blanquita Bono Brigitta Cindy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1946</th>\n",
       "      <td>chr21_17301504</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.004207</td>\n",
       "      <td>0.025882</td>\n",
       "      <td>0.025882</td>\n",
       "      <td>0.012505</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Blanquita Bono Cindy-troglodytes Desmond Dzeet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1947</th>\n",
       "      <td>chr21_18874368</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001572</td>\n",
       "      <td>0.006887</td>\n",
       "      <td>0.006887</td>\n",
       "      <td>0.003752</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Andromeda Athanga Bihati Brigitta Bwamble Cind...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1948</th>\n",
       "      <td>chr21_19398656</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.003290</td>\n",
       "      <td>0.006391</td>\n",
       "      <td>0.006391</td>\n",
       "      <td>0.004359</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1949</th>\n",
       "      <td>chr21_19922944</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002029</td>\n",
       "      <td>0.024514</td>\n",
       "      <td>0.024514</td>\n",
       "      <td>0.005010</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1950</th>\n",
       "      <td>chr21_20447232</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002951</td>\n",
       "      <td>0.028111</td>\n",
       "      <td>0.028111</td>\n",
       "      <td>0.016831</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1951</th>\n",
       "      <td>chr21_20971520</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001637</td>\n",
       "      <td>0.010048</td>\n",
       "      <td>0.010048</td>\n",
       "      <td>0.005598</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Brigitta Bwamble Cindy-schwein Tibe Tongo/Akwa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1952</th>\n",
       "      <td>chr21_21495808</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>0.007069</td>\n",
       "      <td>0.034957</td>\n",
       "      <td>0.034957</td>\n",
       "      <td>0.031945</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Berta Bosco Clint Jimmie Koby Linda Sepp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1953</th>\n",
       "      <td>chr21_22020096</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>0.001332</td>\n",
       "      <td>0.004596</td>\n",
       "      <td>0.004596</td>\n",
       "      <td>0.003950</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Gamin Julie-A959/Akwaya-Jean Alice Berta Bosco...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1954</th>\n",
       "      <td>chr21_22544384</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001901</td>\n",
       "      <td>0.007092</td>\n",
       "      <td>0.007092</td>\n",
       "      <td>0.004085</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bihati Cleo Coco-chimp Doris Ikuru Kidongo May...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1955</th>\n",
       "      <td>chr21_23068672</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>0.009465</td>\n",
       "      <td>0.018354</td>\n",
       "      <td>0.018354</td>\n",
       "      <td>0.016051</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Bihati Cleo Lara Luky Maya Negrita/Bono Desmon...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1956</th>\n",
       "      <td>chr21_23592960</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002276</td>\n",
       "      <td>0.009771</td>\n",
       "      <td>0.009771</td>\n",
       "      <td>0.006670</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Akwaya-Jean Bono Bosco Clint Jimmie Vincent Wa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1957</th>\n",
       "      <td>chr21_24117248</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005625</td>\n",
       "      <td>0.034644</td>\n",
       "      <td>0.034644</td>\n",
       "      <td>0.011145</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1958</th>\n",
       "      <td>chr21_24641536</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C0</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.015595</td>\n",
       "      <td>0.015595</td>\n",
       "      <td>0.013081</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>Marlin/Bihati Bono Desmond Dzeeta Hermien Hort...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1959</th>\n",
       "      <td>chr21_25165824</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009962</td>\n",
       "      <td>0.009962</td>\n",
       "      <td>0.003332</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Negrita/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1960</th>\n",
       "      <td>chr21_25690112</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.031645</td>\n",
       "      <td>0.031645</td>\n",
       "      <td>0.013932</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Negrita/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1961</th>\n",
       "      <td>chr21_26214400</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.006477</td>\n",
       "      <td>0.032148</td>\n",
       "      <td>0.032148</td>\n",
       "      <td>0.022181</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Damian Taweh/Akwaya-Jean Alfred Alice Andromed...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1962</th>\n",
       "      <td>chr21_26738688</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.046547</td>\n",
       "      <td>0.046547</td>\n",
       "      <td>0.018381</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Damian Taweh/Akwaya-Jean Alfred Alice Andromed...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1963</th>\n",
       "      <td>chr21_27262976</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002623</td>\n",
       "      <td>0.008421</td>\n",
       "      <td>0.008421</td>\n",
       "      <td>0.005637</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Akwaya-Jean Alice Berta Bihati Bosco Cindy-ver...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1964</th>\n",
       "      <td>chr21_27787264</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.021100</td>\n",
       "      <td>0.048625</td>\n",
       "      <td>0.048625</td>\n",
       "      <td>0.031750</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1965</th>\n",
       "      <td>chr21_28311552</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.055626</td>\n",
       "      <td>0.055626</td>\n",
       "      <td>0.029203</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Negrita/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1966</th>\n",
       "      <td>chr21_28835840</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.029190</td>\n",
       "      <td>0.029190</td>\n",
       "      <td>0.017642</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Negrita/Akwaya-Jean Alfred Alice Andromeda Ath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1967</th>\n",
       "      <td>chr21_29360128</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002349</td>\n",
       "      <td>0.004950</td>\n",
       "      <td>0.004950</td>\n",
       "      <td>0.002586</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Akwaya-Jean Alfred Bihati Blanquita Bwamble Ci...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1968</th>\n",
       "      <td>chr21_29884416</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001061</td>\n",
       "      <td>0.010507</td>\n",
       "      <td>0.010507</td>\n",
       "      <td>0.005920</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Frederike Maya Mgbadolite Trixie/Akwaya-Jean A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1969</th>\n",
       "      <td>chr21_30408704</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.003379</td>\n",
       "      <td>0.019816</td>\n",
       "      <td>0.019816</td>\n",
       "      <td>0.007892</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Frederike Maya Mgbadolite Trixie/Akwaya-Jean A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1970</th>\n",
       "      <td>chr21_30932992</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.043778</td>\n",
       "      <td>0.043778</td>\n",
       "      <td>0.025046</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Damian/Akwaya-Jean Alfred Alice Andromeda Atha...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1971</th>\n",
       "      <td>chr21_31457280</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C0</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.028066</td>\n",
       "      <td>0.028066</td>\n",
       "      <td>0.020042</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Damian Kumbuka/Akwaya-Jean Alfred Alice Androm...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1972</th>\n",
       "      <td>chr21_31981568</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001392</td>\n",
       "      <td>0.028204</td>\n",
       "      <td>0.028204</td>\n",
       "      <td>0.019390</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Kumbuka/Akwaya-Jean Alfred Alice Andromed...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              window Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati  \\\n",
       "1915   chr21_1048576          C2     C2    C2        C2      C2    C2     C2   \n",
       "1916   chr21_1572864          C2     C2    C2        C2      C2    C2     C2   \n",
       "1917   chr21_2097152          C3     C1    C1        C2      C3    C1     C3   \n",
       "1918   chr21_2621440          C2     C2    C2        C2      C2    C2     C2   \n",
       "1919   chr21_3145728          C1     C1    C1        C1      C1    C1     C1   \n",
       "1920   chr21_3670016          C2     C2    C2        C2      C2    C2     C2   \n",
       "1921   chr21_4194304          C2     C2    C2        C2      C2    C2     C2   \n",
       "1922   chr21_4718592          C1     C1    C1        C1      C1    C1     C1   \n",
       "1923   chr21_5242880          C1     C1    C1        C1      C1    C1     C1   \n",
       "1924   chr21_5767168          C1     C1    C1        C1      C1    C1     C2   \n",
       "1925   chr21_6291456          C2     C2    C2        C2      C2    C2     C1   \n",
       "1926   chr21_6815744          C2     C2    C2        C2      C2    C2     C2   \n",
       "1927   chr21_7340032          C1     C1    C1        C1      C1    C1     C1   \n",
       "1928   chr21_7864320          C1     C1    C1        C1      C1    C1     C1   \n",
       "1929   chr21_8388608          C1     C1    C1        C1      C1    C1     C1   \n",
       "1930   chr21_8912896          C1     C1    C1        C1      C1    C1     C1   \n",
       "1931   chr21_9437184          C1     C1    C1        C1      C1    C1     C1   \n",
       "1932   chr21_9961472          C2     C2    C2        C2      C2    C2     C2   \n",
       "1933  chr21_10485760          C1     C1    C1        C1      C1    C1     C1   \n",
       "1934  chr21_11010048          C1     C1    C1        C1      C1    C1     C1   \n",
       "1935  chr21_11534336          C2     C2    C2        C2      C2    C2     C2   \n",
       "1936  chr21_12058624          C1     C1    C3        C3      C1    C3     C3   \n",
       "1937  chr21_12582912          C3     C1    C1        C3      C1    C1     C3   \n",
       "1938  chr21_13107200          C3     C3    C2        C3      C2    C2     C2   \n",
       "1939  chr21_13631488          C2     C2    C1        C2      C1    C1     C2   \n",
       "1940  chr21_14155776          C1     C2    C3        C2      C1    C2     C3   \n",
       "1941  chr21_14680064          C2     C2    C1        C2      C2    C1     C2   \n",
       "1942  chr21_15204352          C2     C2    C1        C2      C2    C1     C2   \n",
       "1943  chr21_15728640          C1     C1    C1        C2      C3    C1     C2   \n",
       "1944  chr21_16252928          C2     C2    C2        C2      C2    C2     C2   \n",
       "1945  chr21_16777216          C2     C1    C2        C1      C2    C2     C2   \n",
       "1946  chr21_17301504          C2     C2    C2        C2      C2    C2     C2   \n",
       "1947  chr21_18874368          C2     C2    C2        C1      C1    C2     C1   \n",
       "1948  chr21_19398656          C2     C2    C2        C2      C2    C2     C2   \n",
       "1949  chr21_19922944          C2     C2    C2        C2      C2    C2     C2   \n",
       "1950  chr21_20447232          C2     C2    C2        C2      C2    C2     C2   \n",
       "1951  chr21_20971520          C2     C2    C2        C2      C2    C2     C2   \n",
       "1952  chr21_21495808          C3     C3    C1        C3      C3    C1     C3   \n",
       "1953  chr21_22020096          C2     C3    C2        C3      C3    C2     C3   \n",
       "1954  chr21_22544384          C2     C2    C2        C2      C2    C2     C1   \n",
       "1955  chr21_23068672          C3     C3    C3        C3      C3    C3     C1   \n",
       "1956  chr21_23592960          C1     C2    C2        C2      C2    C2     C2   \n",
       "1957  chr21_24117248          C2     C2    C2        C2      C2    C2     C2   \n",
       "1958  chr21_24641536          C2     C2    C2        C2      C2    C2     C1   \n",
       "1959  chr21_25165824          C1     C1    C1        C1      C1    C1     C1   \n",
       "1960  chr21_25690112          C1     C1    C1        C1      C1    C1     C1   \n",
       "1961  chr21_26214400          C2     C2    C2        C2      C2    C2     C2   \n",
       "1962  chr21_26738688          C2     C2    C2        C2      C2    C2     C2   \n",
       "1963  chr21_27262976          C1     C2    C1        C2      C2    C1     C1   \n",
       "1964  chr21_27787264          C1     C1    C1        C1      C1    C1     C1   \n",
       "1965  chr21_28311552          C1     C1    C1        C1      C1    C1     C1   \n",
       "1966  chr21_28835840          C1     C1    C1        C1      C1    C1     C1   \n",
       "1967  chr21_29360128          C2     C2    C1        C1      C1    C1     C2   \n",
       "1968  chr21_29884416          C2     C2    C2        C2      C2    C2     C2   \n",
       "1969  chr21_30408704          C2     C2    C2        C2      C2    C2     C2   \n",
       "1970  chr21_30932992          C1     C1    C1        C1      C1    C1     C1   \n",
       "1971  chr21_31457280          C1     C1    C1        C1      C1    C1     C1   \n",
       "1972  chr21_31981568          C2     C2    C2        C2      C2    C2     C2   \n",
       "\n",
       "     Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes  \\\n",
       "1915        C2   C2    C2       C2      C2            C2                C1   \n",
       "1916        C2   C2    C2       C2      C2            C2                C1   \n",
       "1917        C3   C3    C1       C1      C1            C3                C3   \n",
       "1918        C2   C1    C2       C2      C2            C2                C2   \n",
       "1919        C1   C1    C1       C1      C1            C1                C1   \n",
       "1920        C2   C1    C2       C2      C2            C2                C2   \n",
       "1921        C1   C2    C2       C2      C2            C2                C1   \n",
       "1922        C1   C1    C1       C1      C1            C1                C1   \n",
       "1923        C1   C1    C1       C1      C1            C1                C1   \n",
       "1924        C1   C1    C1       C1      C2            C1                C1   \n",
       "1925        C2   C2    C2       C2      C1            C2                C2   \n",
       "1926        C2   C1    C2       C2      C2            C2                C2   \n",
       "1927        C1   C1    C1       C1      C1            C1                C1   \n",
       "1928        C1   C1    C1       C1      C1            C1                C1   \n",
       "1929        C1   C1    C1       C1      C1            C1                C1   \n",
       "1930        C1   C1    C1       C1      C1            C1                C1   \n",
       "1931        C1   C1    C0       C1      C1            C1                C1   \n",
       "1932        C2   C1    C2       C2      C2            C2                C2   \n",
       "1933        C1   C1    C1       C1      C1            C1                C0   \n",
       "1934        C1   C1    C1       C1      C1            C1                C0   \n",
       "1935        C2   C1    C2       C2      C2            C2                C2   \n",
       "1936        C3   C2    C3       C1      C3            C3                C3   \n",
       "1937        C3   C1    C3       C3      C3            C3                C2   \n",
       "1938        C2   C3    C2       C2      C3            C3                C2   \n",
       "1939        C2   C2    C1       C1      C2            C2                C1   \n",
       "1940        C3   C2    C1       C3      C2            C2                C3   \n",
       "1941        C2   C2    C1       C2      C2            C2                C2   \n",
       "1942        C2   C2    C1       C2      C2            C2                C2   \n",
       "1943        C3   C2    C1       C3      C2            C2                C1   \n",
       "1944        C2   C2    C2       C2      C2            C2                C2   \n",
       "1945        C1   C1    C2       C1      C2            C1                C1   \n",
       "1946        C1   C1    C2       C2      C2            C2                C1   \n",
       "1947        C2   C2    C2       C1      C1            C1                C2   \n",
       "1948        C2   C1    C2       C2      C2            C2                C2   \n",
       "1949        C2   C1    C2       C2      C2            C2                C2   \n",
       "1950        C2   C1    C2       C2      C2            C2                C2   \n",
       "1951        C2   C2    C2       C1      C1            C1                C2   \n",
       "1952        C3   C2    C1       C3      C3            C3                C3   \n",
       "1953        C3   C3    C2       C3      C3            C3                C3   \n",
       "1954        C2   C2    C2       C2      C2            C2                C2   \n",
       "1955        C3   C2    C3       C3      C3            C3                C3   \n",
       "1956        C2   C1    C1       C2      C2            C2                C2   \n",
       "1957        C2   C1    C2       C2      C2            C2                C2   \n",
       "1958        C2   C1    C2       C2      C2            C2                C2   \n",
       "1959        C1   C1    C1       C1      C1            C1                C1   \n",
       "1960        C1   C1    C1       C1      C1            C1                C1   \n",
       "1961        C2   C2    C2       C2      C2            C2                C2   \n",
       "1962        C2   C2    C2       C2      C2            C2                C2   \n",
       "1963        C2   C2    C1       C2      C2            C2                C2   \n",
       "1964        C1   C2    C1       C1      C1            C1                C1   \n",
       "1965        C1   C1    C1       C1      C1            C1                C1   \n",
       "1966        C1   C1    C1       C1      C1            C1                C1   \n",
       "1967        C2   C1    C1       C1      C2            C2                C1   \n",
       "1968        C2   C2    C2       C2      C2            C2                C2   \n",
       "1969        C2   C2    C2       C2      C2            C2                C2   \n",
       "1970        C1   C1    C1       C1      C1            C1                C1   \n",
       "1971        C1   C1    C1       C1      C1            C1                C1   \n",
       "1972        C2   C1    C2       C2      C2            C2                C2   \n",
       "\n",
       "     Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike  \\\n",
       "1915          C2   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "1916          C2   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "1917          C3   C2    C1         C2     C3      C3    C3     C3        C2   \n",
       "1918          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1919          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1920          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1921          C2   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "1922          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1923          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1924          C1   C1    C1         C1     C1      C1    C1     C1        C2   \n",
       "1925          C2   C2    C2         C2     C2      C2    C2     C2        C1   \n",
       "1926          C2   C2    C2         C2     C2      C2    C2     C1        C2   \n",
       "1927          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1928          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1929          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1930          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1931          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1932          C2   C2    C2         C2     C2      C1    C1     C1        C2   \n",
       "1933          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1934          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1935          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1936          C3   C3    C3         C1     C3      C2    C3     C2        C3   \n",
       "1937          C1   C3    C3         C3     C3      C1    C3     C1        C3   \n",
       "1938          C2   C3    C3         C3     C3      C3    C3     C3        C3   \n",
       "1939          C1   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "1940          C3   C2    C1         C3     C3      C2    C2     C2        C2   \n",
       "1941          C2   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "1942          C2   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "1943          C1   C3    C1         C2     C1      C2    C3     C2        C2   \n",
       "1944          C2   C1    C2         C2     C2      C2    C2     C2        C2   \n",
       "1945          C2   C2    C2         C2     C2      C2    C1     C1        C1   \n",
       "1946          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1947          C2   C1    C2         C2     C2      C2    C2     C2        C1   \n",
       "1948          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1949          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1950          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1951          C2   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "1952          C3   C3    C1         C3     C2      C2    C3     C2        C3   \n",
       "1953          C2   C3    C2         C3     C3      C3    C3     C3        C3   \n",
       "1954          C2   C1    C2         C1     C2      C2    C1     C2        C2   \n",
       "1955          C3   C1    C3         C3     C3      C2    C3     C2        C3   \n",
       "1956          C2   C2    C1         C2     C2      C2    C2     C2        C2   \n",
       "1957          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1958          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1959          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1960          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1961          C2   C2    C2         C2     C1      C2    C2     C2        C2   \n",
       "1962          C2   C2    C2         C2     C1      C2    C2     C2        C2   \n",
       "1963          C1   C1    C1         C2     C2      C2    C2     C2        C2   \n",
       "1964          C1   C1    C1         C1     C1      C2    C1     C2        C1   \n",
       "1965          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1966          C1   C1    C1         C1     C1      C1    C1     C1        C1   \n",
       "1967          C1   C1    C2         C2     C2      C1    C2     C1        C2   \n",
       "1968          C2   C2    C2         C2     C2      C2    C2     C2        C1   \n",
       "1969          C2   C2    C2         C2     C2      C2    C2     C2        C1   \n",
       "1970          C1   C1    C1         C1     C0      C1    C1     C1        C1   \n",
       "1971          C1   C1    C1         C1     C0      C1    C1     C1        C1   \n",
       "1972          C2   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "\n",
       "     Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby  \\\n",
       "1915    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1916    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1917    C3      C3       C3    C2     C1         C2          C2      C2   C1   \n",
       "1918    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1919    C1      C1       C0    C1     C1         C1          C1      C1   C1   \n",
       "1920    C2      C2       C1    C2     C2         C2          C2      C2   C2   \n",
       "1921    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1922    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1923    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1924    C1      C1       C1    C1     C1         C2          C1      C1   C1   \n",
       "1925    C2      C2       C2    C2     C2         C1          C2      C2   C2   \n",
       "1926    C2      C2       C1    C2     C2         C2          C2      C2   C2   \n",
       "1927    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1928    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1929    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1930    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1931    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1932    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1933    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1934    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1935    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1936    C3      C2       C2    C3     C3         C1          C1      C3   C1   \n",
       "1937    C3      C1       C1    C3     C3         C3          C3      C3   C1   \n",
       "1938    C2      C3       C3    C2     C2         C3          C1      C2   C2   \n",
       "1939    C1      C2       C2    C1     C1         C2          C2      C2   C1   \n",
       "1940    C3      C2       C2    C3     C3         C2          C2      C3   C3   \n",
       "1941    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1942    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1943    C1      C2       C2    C2     C1         C3          C3      C2   C1   \n",
       "1944    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1945    C1      C2       C1    C1     C2         C2          C2      C2   C1   \n",
       "1946    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1947    C2      C2       C2    C1     C2         C2          C2      C1   C2   \n",
       "1948    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1949    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1950    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1951    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1952    C3      C2       C2    C3     C1         C3          C2      C3   C1   \n",
       "1953    C1      C3       C3    C3     C2         C1          C3      C3   C2   \n",
       "1954    C2      C2       C2    C1     C2         C2          C2      C1   C2   \n",
       "1955    C3      C2       C2    C3     C3         C3          C3      C3   C3   \n",
       "1956    C2      C2       C2    C2     C1         C2          C2      C2   C2   \n",
       "1957    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1958    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1959    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1960    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1961    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1962    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1963    C2      C2       C2    C2     C1         C1          C2      C2   C1   \n",
       "1964    C1      C2       C2    C1     C1         C1          C1      C1   C1   \n",
       "1965    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1966    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1967    C2      C1       C1    C1     C2         C2          C2      C1   C1   \n",
       "1968    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1969    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "1970    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1971    C1      C1       C1    C1     C1         C1          C1      C1   C1   \n",
       "1972    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "\n",
       "     Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite  \\\n",
       "1915      C2     C1   C2      C2   C2    C2   C2     C2   C2         C2   \n",
       "1916      C2     C1   C2      C2   C2    C2   C2     C2   C2         C2   \n",
       "1917      C3     C3   C3      C3   C2    C1   C3     C3   C1         C2   \n",
       "1918      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1919      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1920      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1921      C2     C2   C2      C2   C2    C2   C2     C2   C2         C2   \n",
       "1922      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1923      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1924      C1     C1   C1      C1   C1    C1   C0     C1   C1         C2   \n",
       "1925      C2     C2   C2      C2   C2    C2   C2     C2   C2         C1   \n",
       "1926      C1     C2   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1927      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1928      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1929      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1930      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1931      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1932      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1933      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1934      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1935      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1936      C2     C2   C1      C2   C3    C3   C1     C3   C3         C1   \n",
       "1937      C1     C1   C1      C1   C3    C1   C1     C3   C3         C3   \n",
       "1938      C3     C3   C3      C3   C3    C2   C2     C3   C2         C2   \n",
       "1939      C2     C2   C2      C2   C2    C1   C2     C2   C1         C1   \n",
       "1940      C2     C2   C3      C2   C3    C3   C3     C3   C2         C1   \n",
       "1941      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "1942      C2     C2   C2      C2   C2    C1   C2     C2   C2         C2   \n",
       "1943      C2     C2   C3      C2   C3    C1   C3     C3   C2         C2   \n",
       "1944      C2     C2   C2      C2   C2    C2   C2     C2   C2         C2   \n",
       "1945      C1     C1   C2      C1   C2    C2   C1     C1   C1         C2   \n",
       "1946      C1     C1   C2      C1   C2    C2   C1     C2   C2         C2   \n",
       "1947      C2     C2   C2      C2   C1    C2   C2     C1   C1         C1   \n",
       "1948      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1949      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1950      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1951      C2     C2   C2      C2   C2    C2   C2     C2   C2         C2   \n",
       "1952      C2     C3   C2      C2   C3    C1   C3     C3   C3         C3   \n",
       "1953      C3     C3   C3      C3   C3    C2   C3     C3   C3         C3   \n",
       "1954      C2     C2   C2      C2   C2    C2   C2     C2   C1         C1   \n",
       "1955      C2     C2   C3      C2   C1    C3   C1     C3   C1         C3   \n",
       "1956      C2     C2   C2      C2   C2    C2   C2     C2   C2         C2   \n",
       "1957      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1958      C1     C1   C1      C1   C2    C2   C2     C0   C2         C2   \n",
       "1959      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1960      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1961      C2     C2   C2      C2   C2    C2   C2     C2   C2         C2   \n",
       "1962      C2     C2   C2      C2   C2    C2   C2     C2   C2         C2   \n",
       "1963      C2     C2   C2      C2   C2    C1   C2     C1   C2         C2   \n",
       "1964      C2     C2   C1      C2   C1    C1   C1     C1   C1         C1   \n",
       "1965      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1966      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1967      C1     C1   C2      C1   C1    C1   C1     C2   C2         C2   \n",
       "1968      C2     C2   C2      C2   C2    C2   C2     C2   C1         C1   \n",
       "1969      C2     C2   C2      C2   C2    C2   C2     C2   C1         C1   \n",
       "1970      C1     C1   C1      C1   C1    C1   C1     C1   C1         C1   \n",
       "1971      C1     C1   C1      C0   C1    C1   C1     C1   C1         C1   \n",
       "1972      C2     C2   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "\n",
       "     Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula  \\\n",
       "1915      C2    C2      C2      C2       C2    C2   C2    C2     C2  C2   \n",
       "1916      C2    C2      C2      C2       C2    C2   C2    C2     C2  C2   \n",
       "1917      C3    C3      C3      C1       C1    C3   C1    C1     C2  C1   \n",
       "1918      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1919      C1    C1      C1      C1       C1    C1   C1    C1     C1  C1   \n",
       "1920      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1921      C2    C2      C2      C2       C2    C2   C2    C2     C2  C1   \n",
       "1922      C1    C1      C1      C1       C0    C1   C1    C1     C1  C1   \n",
       "1923      C1    C1      C1      C1       C0    C1   C1    C1     C1  C1   \n",
       "1924      C1    C1      C1      C1       C1    C1   C1    C2     C1  C1   \n",
       "1925      C2    C2      C2      C2       C2    C2   C2    C1     C2  C2   \n",
       "1926      C2    C2      C2      C2       C2    C2   C2    C2     C2  C2   \n",
       "1927      C1    C1      C0      C1       C1    C1   C1    C1     C1  C1   \n",
       "1928      C1    C1      C0      C1       C1    C1   C1    C1     C1  C1   \n",
       "1929      C1    C1      C1      C1       C1    C1   C1    C1     C1  C1   \n",
       "1930      C1    C1      C1      C1       C1    C1   C1    C1     C1  C1   \n",
       "1931      C1    C1      C1      C1       C1    C1   C1    C1     C1  C1   \n",
       "1932      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1933      C1    C1      C1      C1       C1    C1   C1    C1     C1  C1   \n",
       "1934      C1    C1      C1      C1       C1    C1   C1    C1     C1  C1   \n",
       "1935      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1936      C3    C3      C2      C3       C3    C3   C3    C3     C3  C1   \n",
       "1937      C2    C3      C1      C3       C3    C1   C3    C1     C3  C1   \n",
       "1938      C2    C3      C3      C2       C2    C2   C1    C3     C3  C2   \n",
       "1939      C2    C2      C2      C2       C1    C2   C1    C2     C2  C1   \n",
       "1940      C2    C3      C2      C3       C2    C2   C3    C2     C2  C3   \n",
       "1941      C2    C2      C2      C2       C2    C2   C2    C2     C2  C2   \n",
       "1942      C2    C2      C2      C2       C2    C2   C2    C2     C2  C2   \n",
       "1943      C3    C3      C2      C1       C3    C1   C3    C2     C2  C3   \n",
       "1944      C2    C2      C2      C2       C2    C2   C2    C2     C2  C1   \n",
       "1945      C1    C2      C1      C1       C2    C2   C1    C2     C2  C1   \n",
       "1946      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1947      C2    C2      C2      C2       C2    C2   C2    C1     C1  C2   \n",
       "1948      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1949      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1950      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1951      C2    C2      C2      C2       C2    C2   C1    C1     C2  C2   \n",
       "1952      C3    C3      C2      C3       C1    C2   C3    C3     C3  C3   \n",
       "1953      C3    C3      C3      C3       C2    C3   C3    C3     C3  C3   \n",
       "1954      C2    C1      C2      C2       C2    C2   C2    C2     C1  C2   \n",
       "1955      C3    C3      C2      C1       C3    C3   C3    C3     C3  C3   \n",
       "1956      C2    C2      C2      C2       C2    C2   C2    C2     C2  C2   \n",
       "1957      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1958      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1959      C1    C1      C1      C0       C1    C1   C1    C1     C1  C1   \n",
       "1960      C1    C1      C1      C0       C1    C1   C1    C1     C1  C1   \n",
       "1961      C2    C2      C2      C2       C2    C1   C2    C2     C2  C2   \n",
       "1962      C2    C2      C2      C2       C2    C1   C2    C2     C2  C2   \n",
       "1963      C2    C2      C2      C2       C1    C2   C2    C1     C2  C2   \n",
       "1964      C1    C1      C2      C1       C1    C1   C1    C1     C1  C1   \n",
       "1965      C1    C1      C1      C0       C1    C1   C1    C1     C1  C1   \n",
       "1966      C1    C1      C1      C0       C1    C1   C1    C1     C1  C1   \n",
       "1967      C2    C2      C1      C2       C1    C1   C1    C1     C2  C2   \n",
       "1968      C2    C2      C2      C2       C2    C2   C2    C2     C1  C2   \n",
       "1969      C2    C2      C2      C2       C2    C2   C2    C2     C1  C2   \n",
       "1970      C1    C1      C1      C1       C1    C1   C1    C1     C1  C1   \n",
       "1971      C1    C1      C1      C1       C1    C1   C1    C1     C1  C1   \n",
       "1972      C2    C2      C2      C2       C2    C2   C2    C2     C2  C2   \n",
       "\n",
       "     Vaillant Vincent Washu Yogui    node_0    node_1    node_2    node_3  \\\n",
       "1915       C2      C2    C2    C2  0.007273  0.101743  0.101743  0.018713   \n",
       "1916       C2      C2    C2    C2  0.009908  0.092433  0.092433  0.030027   \n",
       "1917       C3      C2    C3    C1  0.008533  0.016795  0.016795  0.013133   \n",
       "1918       C2      C2    C2    C2  0.001896  0.012374  0.012374  0.008402   \n",
       "1919       C1      C1    C1    C1  0.000000  0.058120  0.058120  0.038627   \n",
       "1920       C2      C2    C2    C2  0.005336  0.025069  0.025069  0.007850   \n",
       "1921       C2      C2    C2    C2  0.001153  0.017037  0.017037  0.009157   \n",
       "1922       C1      C1    C1    C1  0.000000  0.056423  0.056423  0.021865   \n",
       "1923       C1      C1    C1    C1  0.000000  0.119308  0.119308  0.041856   \n",
       "1924       C1      C2    C2    C1  0.049416  0.123899  0.123899  0.108459   \n",
       "1925       C2      C1    C1    C2  0.029523  0.155177  0.155177  0.064509   \n",
       "1926       C2      C2    C2    C2  0.002659  0.064121  0.064121  0.027898   \n",
       "1927       C1      C1    C1    C1  0.000000  0.075647  0.075647  0.026706   \n",
       "1928       C1      C1    C1    C1  0.000000  0.183761  0.183761  0.053167   \n",
       "1929       C0      C1    C1    C1  0.000000  0.142003  0.142003  0.023950   \n",
       "1930       C0      C1    C1    C1  0.000000  0.037891  0.037891  0.011049   \n",
       "1931       C1      C1    C1    C1  0.000000  0.025588  0.025588  0.004655   \n",
       "1932       C2      C2    C2    C2  0.022009  0.075191  0.075191  0.048403   \n",
       "1933       C1      C1    C1    C1  0.000000  0.143209  0.143209  0.067397   \n",
       "1934       C1      C1    C1    C1  0.000000  0.179972  0.179972  0.084036   \n",
       "1935       C2      C2    C2    C2  0.000866  0.006946  0.006946  0.004649   \n",
       "1936       C3      C3    C3    C3  0.004243  0.007764  0.007764  0.007669   \n",
       "1937       C3      C3    C3    C3  0.006425  0.009628  0.009628  0.007451   \n",
       "1938       C3      C2    C3    C2  0.001945  0.011701  0.011701  0.009078   \n",
       "1939       C2      C2    C2    C1  0.007718  0.030535  0.030535  0.016379   \n",
       "1940       C3      C1    C2    C2  0.002078  0.016118  0.016118  0.011492   \n",
       "1941       C2      C2    C2    C2  0.000191  0.052577  0.052577  0.013309   \n",
       "1942       C2      C2    C2    C2  0.000181  0.019700  0.019700  0.012599   \n",
       "1943       C2      C2    C2    C3  0.001821  0.004103  0.004103  0.003296   \n",
       "1944       C2      C2    C1    C2  0.004971  0.049261  0.049261  0.018393   \n",
       "1945       C2      C2    C2    C1  0.044758  0.105713  0.105713  0.046932   \n",
       "1946       C2      C2    C2    C2  0.004207  0.025882  0.025882  0.012505   \n",
       "1947       C2      C1    C1    C2  0.001572  0.006887  0.006887  0.003752   \n",
       "1948       C2      C2    C2    C2  0.003290  0.006391  0.006391  0.004359   \n",
       "1949       C2      C2    C2    C2  0.002029  0.024514  0.024514  0.005010   \n",
       "1950       C2      C2    C2    C2  0.002951  0.028111  0.028111  0.016831   \n",
       "1951       C2      C2    C2    C2  0.001637  0.010048  0.010048  0.005598   \n",
       "1952       C3      C3    C3    C3  0.007069  0.034957  0.034957  0.031945   \n",
       "1953       C3      C3    C3    C3  0.001332  0.004596  0.004596  0.003950   \n",
       "1954       C2      C2    C2    C2  0.001901  0.007092  0.007092  0.004085   \n",
       "1955       C3      C3    C3    C3  0.009465  0.018354  0.018354  0.016051   \n",
       "1956       C2      C1    C1    C2  0.002276  0.009771  0.009771  0.006670   \n",
       "1957       C2      C2    C2    C2  0.005625  0.034644  0.034644  0.011145   \n",
       "1958       C2      C2    C2    C2  0.000000  0.015595  0.015595  0.013081   \n",
       "1959       C1      C1    C1    C1  0.000000  0.009962  0.009962  0.003332   \n",
       "1960       C1      C1    C1    C1  0.000000  0.031645  0.031645  0.013932   \n",
       "1961       C2      C2    C2    C2  0.006477  0.032148  0.032148  0.022181   \n",
       "1962       C2      C2    C2    C2  0.000385  0.046547  0.046547  0.018381   \n",
       "1963       C2      C1    C1    C2  0.002623  0.008421  0.008421  0.005637   \n",
       "1964       C1      C1    C1    C1  0.021100  0.048625  0.048625  0.031750   \n",
       "1965       C1      C1    C1    C1  0.000000  0.055626  0.055626  0.029203   \n",
       "1966       C1      C1    C1    C1  0.000000  0.029190  0.029190  0.017642   \n",
       "1967       C2      C1    C1    C2  0.002349  0.004950  0.004950  0.002586   \n",
       "1968       C2      C2    C2    C2  0.001061  0.010507  0.010507  0.005920   \n",
       "1969       C2      C2    C2    C2  0.003379  0.019816  0.019816  0.007892   \n",
       "1970       C1      C1    C1    C1  0.000000  0.043778  0.043778  0.025046   \n",
       "1971       C1      C1    C1    C1  0.000000  0.028066  0.028066  0.020042   \n",
       "1972       C2      C2    C2    C2  0.001392  0.028204  0.028204  0.019390   \n",
       "\n",
       "      n_clusters  n_single_clusters  \\\n",
       "1915           2                  0   \n",
       "1916           2                  0   \n",
       "1917           3                  0   \n",
       "1918           2                  0   \n",
       "1919           2                  1   \n",
       "1920           2                  0   \n",
       "1921           2                  0   \n",
       "1922           2                  1   \n",
       "1923           2                  1   \n",
       "1924           3                  1   \n",
       "1925           2                  0   \n",
       "1926           2                  0   \n",
       "1927           2                  1   \n",
       "1928           2                  1   \n",
       "1929           2                  1   \n",
       "1930           2                  1   \n",
       "1931           2                  1   \n",
       "1932           2                  0   \n",
       "1933           2                  1   \n",
       "1934           2                  1   \n",
       "1935           2                  0   \n",
       "1936           3                  0   \n",
       "1937           3                  0   \n",
       "1938           3                  0   \n",
       "1939           2                  0   \n",
       "1940           3                  0   \n",
       "1941           2                  0   \n",
       "1942           2                  0   \n",
       "1943           3                  0   \n",
       "1944           2                  0   \n",
       "1945           2                  0   \n",
       "1946           2                  0   \n",
       "1947           2                  0   \n",
       "1948           2                  0   \n",
       "1949           2                  0   \n",
       "1950           2                  0   \n",
       "1951           2                  0   \n",
       "1952           3                  0   \n",
       "1953           3                  0   \n",
       "1954           2                  0   \n",
       "1955           3                  0   \n",
       "1956           2                  0   \n",
       "1957           2                  0   \n",
       "1958           3                  1   \n",
       "1959           2                  1   \n",
       "1960           2                  1   \n",
       "1961           2                  0   \n",
       "1962           2                  0   \n",
       "1963           2                  0   \n",
       "1964           2                  0   \n",
       "1965           2                  1   \n",
       "1966           2                  1   \n",
       "1967           2                  0   \n",
       "1968           2                  0   \n",
       "1969           2                  0   \n",
       "1970           2                  1   \n",
       "1971           2                  0   \n",
       "1972           2                  0   \n",
       "\n",
       "                                    cluster_composition  \n",
       "1915  Cindy-troglodytes Kosana/Akwaya-Jean Alfred Al...  \n",
       "1916  Cindy-troglodytes Kosana/Akwaya-Jean Alfred Al...  \n",
       "1917  Andromeda Cleo Coco-chimp Frederike Ikuru Juli...  \n",
       "1918  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1919  Hortense/Akwaya-Jean Alfred Alice Andromeda At...  \n",
       "1920  Bono Desmond Dzeeta Hortense Kombote Kosana Ku...  \n",
       "1921  Blanquita Cindy-troglodytes Ula/Akwaya-Jean Al...  \n",
       "1922  SeppToni/Akwaya-Jean Alfred Alice Andromeda At...  \n",
       "1923  SeppToni/Akwaya-Jean Alfred Alice Andromeda At...  \n",
       "1924  Luky/Bihati Bwamble Frederike Julie-A959 Mgbad...  \n",
       "1925  Bihati Bwamble Frederike Julie-A959 Mgbadolite...  \n",
       "1926  Bono Dzeeta Hortense Kombote Kumbuka/Akwaya-Je...  \n",
       "1927  Natalie/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "1928  Natalie/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "1929  Vaillant/Akwaya-Jean Alfred Alice Andromeda At...  \n",
       "1930  Vaillant/Akwaya-Jean Alfred Alice Andromeda At...  \n",
       "1931  Bosco/Akwaya-Jean Alfred Alice Andromeda Athan...  \n",
       "1932  Bono Desmond Doris Dzeeta Hermien Hortense Kom...  \n",
       "1933  Cindy-troglodytes/Akwaya-Jean Alfred Alice And...  \n",
       "1934  Cindy-troglodytes/Akwaya-Jean Alfred Alice And...  \n",
       "1935  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1936  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1937  Cindy-troglodytes Mirinda/Alfred Alice Athanga...  \n",
       "1938  Julie-LWC21 Tibe/Alice Athanga Berta Bihati Bl...  \n",
       "1939  Alice Athanga Berta Bosco Brigitta Cindy-trogl...  \n",
       "1940  Akwaya-Jean Athanga Bosco Clint Mgbadolite Vin...  \n",
       "1941  Alice Berta Bosco Linda/Akwaya-Jean Alfred And...  \n",
       "1942  Alice Berta Bosco Linda/Akwaya-Jean Alfred And...  \n",
       "1943  Akwaya-Jean Alfred Alice Berta Bosco Cindy-tro...  \n",
       "1944  Cleo Ula Washu/Akwaya-Jean Alfred Alice Androm...  \n",
       "1945  Alfred Andromeda Blanquita Bono Brigitta Cindy...  \n",
       "1946  Blanquita Bono Cindy-troglodytes Desmond Dzeet...  \n",
       "1947  Andromeda Athanga Bihati Brigitta Bwamble Cind...  \n",
       "1948  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1949  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1950  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1951  Brigitta Bwamble Cindy-schwein Tibe Tongo/Akwa...  \n",
       "1952  Alice Berta Bosco Clint Jimmie Koby Linda Sepp...  \n",
       "1953  Gamin Julie-A959/Akwaya-Jean Alice Berta Bosco...  \n",
       "1954  Bihati Cleo Coco-chimp Doris Ikuru Kidongo May...  \n",
       "1955  Bihati Cleo Lara Luky Maya Negrita/Bono Desmon...  \n",
       "1956  Akwaya-Jean Bono Bosco Clint Jimmie Vincent Wa...  \n",
       "1957  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1958  Marlin/Bihati Bono Desmond Dzeeta Hermien Hort...  \n",
       "1959  Negrita/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "1960  Negrita/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "1961  Damian Taweh/Akwaya-Jean Alfred Alice Andromed...  \n",
       "1962  Damian Taweh/Akwaya-Jean Alfred Alice Andromed...  \n",
       "1963  Akwaya-Jean Alice Berta Bihati Bosco Cindy-ver...  \n",
       "1964  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1965  Negrita/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "1966  Negrita/Akwaya-Jean Alfred Alice Andromeda Ath...  \n",
       "1967  Akwaya-Jean Alfred Bihati Blanquita Bwamble Ci...  \n",
       "1968  Frederike Maya Mgbadolite Trixie/Akwaya-Jean A...  \n",
       "1969  Frederike Maya Mgbadolite Trixie/Akwaya-Jean A...  \n",
       "1970  Damian/Akwaya-Jean Alfred Alice Andromeda Atha...  \n",
       "1971  Damian Kumbuka/Akwaya-Jean Alfred Alice Androm...  \n",
       "1972  Bono Kumbuka/Akwaya-Jean Alfred Alice Andromed...  "
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "autosomal_complete_linkage_trees[autosomal_complete_linkage_trees['window'].str.startswith('chr21_')]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5524c247",
   "metadata": {},
   "source": [
    "Let's subset the species divergent windows for this chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "2e2c0a71",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <th>Alfred</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Athanga</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Bono</th>\n",
       "      <th>Bosco</th>\n",
       "      <th>Brigitta</th>\n",
       "      <th>Bwamble</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Clint</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Damian</th>\n",
       "      <th>Desmond</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Gamin</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Koby</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Koto</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mgbadolite</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>SeppToni</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Tongo</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>Vaillant</th>\n",
       "      <th>Vincent</th>\n",
       "      <th>Washu</th>\n",
       "      <th>Yogui</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "      <th>n_clusters</th>\n",
       "      <th>n_single_clusters</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1918</th>\n",
       "      <td>chr21_2621440</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001896</td>\n",
       "      <td>0.012374</td>\n",
       "      <td>0.012374</td>\n",
       "      <td>0.008402</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1935</th>\n",
       "      <td>chr21_11534336</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.000866</td>\n",
       "      <td>0.006946</td>\n",
       "      <td>0.006946</td>\n",
       "      <td>0.004649</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1948</th>\n",
       "      <td>chr21_19398656</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.003290</td>\n",
       "      <td>0.006391</td>\n",
       "      <td>0.006391</td>\n",
       "      <td>0.004359</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1949</th>\n",
       "      <td>chr21_19922944</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002029</td>\n",
       "      <td>0.024514</td>\n",
       "      <td>0.024514</td>\n",
       "      <td>0.005010</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1950</th>\n",
       "      <td>chr21_20447232</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002951</td>\n",
       "      <td>0.028111</td>\n",
       "      <td>0.028111</td>\n",
       "      <td>0.016831</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1957</th>\n",
       "      <td>chr21_24117248</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005625</td>\n",
       "      <td>0.034644</td>\n",
       "      <td>0.034644</td>\n",
       "      <td>0.011145</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1964</th>\n",
       "      <td>chr21_27787264</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>0.021100</td>\n",
       "      <td>0.048625</td>\n",
       "      <td>0.048625</td>\n",
       "      <td>0.031750</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              window Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati  \\\n",
       "1918   chr21_2621440          C2     C2    C2        C2      C2    C2     C2   \n",
       "1935  chr21_11534336          C2     C2    C2        C2      C2    C2     C2   \n",
       "1948  chr21_19398656          C2     C2    C2        C2      C2    C2     C2   \n",
       "1949  chr21_19922944          C2     C2    C2        C2      C2    C2     C2   \n",
       "1950  chr21_20447232          C2     C2    C2        C2      C2    C2     C2   \n",
       "1957  chr21_24117248          C2     C2    C2        C2      C2    C2     C2   \n",
       "1964  chr21_27787264          C1     C1    C1        C1      C1    C1     C1   \n",
       "\n",
       "     Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes  \\\n",
       "1918        C2   C1    C2       C2      C2            C2                C2   \n",
       "1935        C2   C1    C2       C2      C2            C2                C2   \n",
       "1948        C2   C1    C2       C2      C2            C2                C2   \n",
       "1949        C2   C1    C2       C2      C2            C2                C2   \n",
       "1950        C2   C1    C2       C2      C2            C2                C2   \n",
       "1957        C2   C1    C2       C2      C2            C2                C2   \n",
       "1964        C1   C2    C1       C1      C1            C1                C1   \n",
       "\n",
       "     Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike  \\\n",
       "1918          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1935          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1948          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1949          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1950          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1957          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "1964          C1   C1    C1         C1     C1      C2    C1     C2        C1   \n",
       "\n",
       "     Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby  \\\n",
       "1918    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1935    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1948    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1949    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1950    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1957    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "1964    C1      C2       C2    C1     C1         C1          C1      C1   C1   \n",
       "\n",
       "     Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite  \\\n",
       "1918      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1935      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1948      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1949      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1950      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1957      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2   \n",
       "1964      C2     C2   C1      C2   C1    C1   C1     C1   C1         C1   \n",
       "\n",
       "     Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula  \\\n",
       "1918      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1935      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1948      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1949      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1950      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1957      C2    C2      C1      C2       C2    C2   C2    C2     C2  C2   \n",
       "1964      C1    C1      C2      C1       C1    C1   C1    C1     C1  C1   \n",
       "\n",
       "     Vaillant Vincent Washu Yogui    node_0    node_1    node_2    node_3  \\\n",
       "1918       C2      C2    C2    C2  0.001896  0.012374  0.012374  0.008402   \n",
       "1935       C2      C2    C2    C2  0.000866  0.006946  0.006946  0.004649   \n",
       "1948       C2      C2    C2    C2  0.003290  0.006391  0.006391  0.004359   \n",
       "1949       C2      C2    C2    C2  0.002029  0.024514  0.024514  0.005010   \n",
       "1950       C2      C2    C2    C2  0.002951  0.028111  0.028111  0.016831   \n",
       "1957       C2      C2    C2    C2  0.005625  0.034644  0.034644  0.011145   \n",
       "1964       C1      C1    C1    C1  0.021100  0.048625  0.048625  0.031750   \n",
       "\n",
       "      n_clusters  n_single_clusters  \\\n",
       "1918           2                  0   \n",
       "1935           2                  0   \n",
       "1948           2                  0   \n",
       "1949           2                  0   \n",
       "1950           2                  0   \n",
       "1957           2                  0   \n",
       "1964           2                  0   \n",
       "\n",
       "                                    cluster_composition  \n",
       "1918  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1935  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1948  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1949  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1950  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1957  Bono Desmond Dzeeta Hermien Hortense Kombote K...  \n",
       "1964  Bono Desmond Dzeeta Hermien Hortense Kombote K...  "
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "autosomal_complete_linkage_trees[(autosomal_complete_linkage_trees['window'].str.startswith('chr21_')) & (autosomal_complete_linkage_trees['cluster_composition'] == 'Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui')]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2445e5c8-ea6c-4136-89bb-99de0b4d7fb4",
   "metadata": {},
   "source": [
    "### Window Topology Divergence Differences <a class = 'anchor' id = 'windowtopologydivergencedifferences'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1af39dec",
   "metadata": {},
   "source": [
    "Does the distribution for window maxima vary by the different topologies?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "e7584227",
   "metadata": {},
   "outputs": [],
   "source": [
    "SDI_window_maxes = maxes[maxes['window'].isin(SDI_windows['window'])]['max']\n",
    "MDI_window_maxes = maxes[maxes['window'].isin(MDI_windows['window'])]['max']\n",
    "ppn_ptr_window_maxes = maxes[maxes['window'].isin(ppn_ptr_windows['window'])]['max']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "c50add04",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "800"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(SDI_window_maxes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "64515a47",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2475"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(MDI_window_maxes )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "caa34807",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "339"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_window_maxes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "a6da3f25",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.06748740115086496"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SDI_window_maxes.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "aa229d1f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.053014378694518735"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MDI_window_maxes.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "4cf7695a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.049107185857274284"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_window_maxes.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "884d6fc3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=31.09985751100612, pvalue=1.7650286021802972e-07)"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(SDI_window_maxes, MDI_window_maxes, ppn_ptr_window_maxes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7db82990",
   "metadata": {},
   "source": [
    "Save these results for visualization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "2ecc7c64",
   "metadata": {},
   "outputs": [],
   "source": [
    "SDI_window_maxes.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/window_topologies/SDI_window_maxes.txt', sep = '\\t', header = False, index = False)\n",
    "MDI_window_maxes.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/window_topologies/MDI_window_maxes.txt', sep = '\\t', header = False, index = False)\n",
    "ppn_ptr_window_maxes.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/window_topologies/ppn_ptr_window_maxes.txt', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65ac0fe4-914a-49e1-9075-9370d50da771",
   "metadata": {},
   "source": [
    "### Three, Four, and Five Cluster Window Topologies <a class = 'anchor' id = 'threefourfiveclusterwindowtopologies'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e625a8df",
   "metadata": {},
   "source": [
    "Let's take a peak at the three, four, and five cluster windows. Change the maximum column width display setting to get a better look."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "44962f4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.options.display.max_colwidth = 500"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "67c30401",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_1572864</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Andromeda Athanga Blanquita Bwamble Cindy-schwein Cleo Coco-chimp Doris Kidongo Lara Maya Mgbadolite Mirinda Nakuu Negrita Tongo Trixie Vincent Washu Yogui/Akwaya-Jean Alfred Alice Berta Bihati Bosco Brigitta Cindy-troglodytes Cindy-verus Clint Damian Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Koto Linda Luky Marlin SeppToni Taweh Tibe Ula Vaillant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr10_5242880</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Andromeda Brigitta Coco-chimp Frederike Julie-LWC21 Lara Maya Mirinda Nakuu Tongo Vincent/Akwaya-Jean Alfred Alice Athanga Berta Bihati Blanquita Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Damian Doris Gamin Ikuru Jimmie Julie-A959 Kidongo Koby Koto Linda Luky Marlin Mgbadolite Negrita SeppToni Taweh Tibe Trixie Ula Vaillant Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr10_9437184</td>\n",
       "      <td>Ikuru/Cleo Kidongo Maya Mgbadolite Washu/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Jimmie Julie-A959 Julie-LWC21 Koby Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr10_13107200</td>\n",
       "      <td>Frederike Julie-A959 Tibe Ula/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie Negrita/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Gamin Ikuru Jimmie Julie-LWC21 Kidongo Koby Koto Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu SeppToni Taweh Tongo Trixie Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr10_15728640</td>\n",
       "      <td>Kosana Kumbuka/Akwaya-Jean Andromeda Brigitta Cindy-troglodytes Damian Gamin Ikuru Julie-A959 Julie-LWC21 Koto Lara Nakuu Ula Yogui/Alfred Alice Athanga Berta Bihati Blanquita Bono Bosco Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Desmond Doris Dzeeta Frederike Hermien Hortense Jimmie Kidongo Koby Kombote Linda Luky Marlin Maya Mgbadolite Mirinda Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Vaillant Vincent Washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr10_16252928</td>\n",
       "      <td>Julie-A959/Bono Brigitta Desmond Dzeeta Gamin Hermien Hortense Kombote Kosana Kumbuka Linda Mirinda Natalie/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Ikuru Jimmie Julie-LWC21 Kidongo Koby Koto Lara Luky Marlin Maya Mgbadolite Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr10_22020096</td>\n",
       "      <td>Bono Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alfred Andromeda Athanga Blanquita Brigitta Coco-chimp Damian Ikuru Julie-LWC21 Kidongo Koto Lara Luky Mgbadolite Taweh Tongo Trixie Ula/Akwaya-Jean Alice Berta Bihati Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Desmond Doris Frederike Gamin Jimmie Julie-A959 Koby Linda Marlin Maya Mirinda Nakuu Negrita SeppToni Tibe Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr10_22544384</td>\n",
       "      <td>Lara/Cindy-schwein Gamin Hortense Julie-A959 Kosana Kumbuka Vaillant/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bono Bosco Brigitta Bwamble Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike Hermien Ikuru Jimmie Julie-LWC21 Kidongo Koby Kombote Koto Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr10_23068672</td>\n",
       "      <td>Lara/Alfred Athanga Brigitta Cindy-schwein Cleo Coco-chimp Ikuru Marlin Mgbadolite Mirinda Negrita Trixie Vaillant Vincent/Akwaya-Jean Alice Andromeda Berta Bihati Blanquita Bono Bosco Bwamble Cindy-troglodytes Cindy-verus Clint Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Linda Luky Maya Nakuu Natalie SeppToni Taweh Tibe Tongo Ula Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr10_26214400</td>\n",
       "      <td>Cindy-troglodytes Lara/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window  \\\n",
       "0    chr10_1572864   \n",
       "7    chr10_5242880   \n",
       "13   chr10_9437184   \n",
       "20  chr10_13107200   \n",
       "21  chr10_15728640   \n",
       "22  chr10_16252928   \n",
       "31  chr10_22020096   \n",
       "32  chr10_22544384   \n",
       "33  chr10_23068672   \n",
       "39  chr10_26214400   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                 cluster_composition  \n",
       "0   Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Andromeda Athanga Blanquita Bwamble Cindy-schwein Cleo Coco-chimp Doris Kidongo Lara Maya Mgbadolite Mirinda Nakuu Negrita Tongo Trixie Vincent Washu Yogui/Akwaya-Jean Alfred Alice Berta Bihati Bosco Brigitta Cindy-troglodytes Cindy-verus Clint Damian Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Koto Linda Luky Marlin SeppToni Taweh Tibe Ula Vaillant  \n",
       "7   Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Andromeda Brigitta Coco-chimp Frederike Julie-LWC21 Lara Maya Mirinda Nakuu Tongo Vincent/Akwaya-Jean Alfred Alice Athanga Berta Bihati Blanquita Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Damian Doris Gamin Ikuru Jimmie Julie-A959 Kidongo Koby Koto Linda Luky Marlin Mgbadolite Negrita SeppToni Taweh Tibe Trixie Ula Vaillant Washu Yogui  \n",
       "13  Ikuru/Cleo Kidongo Maya Mgbadolite Washu/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Jimmie Julie-A959 Julie-LWC21 Koby Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Yogui  \n",
       "20  Frederike Julie-A959 Tibe Ula/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie Negrita/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Gamin Ikuru Jimmie Julie-LWC21 Kidongo Koby Koto Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu SeppToni Taweh Tongo Trixie Vaillant Vincent Washu Yogui  \n",
       "21  Kosana Kumbuka/Akwaya-Jean Andromeda Brigitta Cindy-troglodytes Damian Gamin Ikuru Julie-A959 Julie-LWC21 Koto Lara Nakuu Ula Yogui/Alfred Alice Athanga Berta Bihati Blanquita Bono Bosco Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Desmond Doris Dzeeta Frederike Hermien Hortense Jimmie Kidongo Koby Kombote Linda Luky Marlin Maya Mgbadolite Mirinda Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Vaillant Vincent Washu  \n",
       "22  Julie-A959/Bono Brigitta Desmond Dzeeta Gamin Hermien Hortense Kombote Kosana Kumbuka Linda Mirinda Natalie/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Ikuru Jimmie Julie-LWC21 Kidongo Koby Koto Lara Luky Marlin Maya Mgbadolite Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui  \n",
       "31  Bono Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alfred Andromeda Athanga Blanquita Brigitta Coco-chimp Damian Ikuru Julie-LWC21 Kidongo Koto Lara Luky Mgbadolite Taweh Tongo Trixie Ula/Akwaya-Jean Alice Berta Bihati Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Desmond Doris Frederike Gamin Jimmie Julie-A959 Koby Linda Marlin Maya Mirinda Nakuu Negrita SeppToni Tibe Vaillant Vincent Washu Yogui  \n",
       "32  Lara/Cindy-schwein Gamin Hortense Julie-A959 Kosana Kumbuka Vaillant/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bono Bosco Brigitta Bwamble Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike Hermien Ikuru Jimmie Julie-LWC21 Kidongo Koby Kombote Koto Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vincent Washu Yogui  \n",
       "33  Lara/Alfred Athanga Brigitta Cindy-schwein Cleo Coco-chimp Ikuru Marlin Mgbadolite Mirinda Negrita Trixie Vaillant Vincent/Akwaya-Jean Alice Andromeda Berta Bihati Blanquita Bono Bosco Bwamble Cindy-troglodytes Cindy-verus Clint Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Linda Luky Maya Nakuu Natalie SeppToni Taweh Tibe Tongo Ula Washu Yogui  \n",
       "39  Cindy-troglodytes Lara/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui  "
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "complete_linkage_trees[complete_linkage_trees['n_clusters'] == 3][['window','cluster_composition']].head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "58e56d41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr10_31457280</td>\n",
       "      <td>Blanquita Bosco Coco-chimp Lara Trixie/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alice Berta Cindy-verus Clint Damian Jimmie Julie-LWC21 Koby Koto Linda SeppToni Taweh Tibe Tongo/Alfred Andromeda Athanga Bihati Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Doris Frederike Gamin Ikuru Julie-A959 Kidongo Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>chr10_65536000</td>\n",
       "      <td>Doris Ula Yogui/Athanga Bono Cindy-schwein Cleo Coco-chimp Desmond Dzeeta Hermien Hortense Kidongo Kombote Kosana Kumbuka Nakuu Natalie Washu/Akwaya-Jean Alice Bosco Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Koto Lara Linda Marlin SeppToni Tibe Vaillant Vincent/Alfred Andromeda Berta Bihati Blanquita Brigitta Bwamble Cindy-troglodytes Cindy-verus Clint Damian Frederike Luky Maya Mgbadolite Mirinda Negrita Taweh Tongo Trixie</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>chr10_121110528</td>\n",
       "      <td>Brigitta Hermien/Doris Gamin SeppToni/Athanga Bihati Cindy-schwein Coco-chimp Damian Ikuru Julie-A959 Nakuu Tibe Vaillant/Akwaya-Jean Alfred Alice Andromeda Berta Blanquita Bono Bosco Bwamble Cindy-troglodytes Cindy-verus Cleo Clint Desmond Dzeeta Frederike Hortense Jimmie Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda Natalie Negrita Taweh Tongo Trixie Ula Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>chr11_2097152</td>\n",
       "      <td>Cleo Ikuru/Coco-chimp Kidongo Nakuu/Alfred Blanquita Cindy-troglodytes Gamin Julie-A959 Lara Marlin Negrita Ula Yogui/Akwaya-Jean Alice Andromeda Athanga Berta Bihati Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-verus Clint Damian Desmond Doris Dzeeta Frederike Hermien Hortense Jimmie Julie-LWC21 Koby Kombote Kosana Koto Kumbuka Linda Luky Maya Mgbadolite Mirinda Natalie SeppToni Taweh Tibe Tongo Trixie Vaillant Vincent Washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>chr11_41418752</td>\n",
       "      <td>Mirinda Negrita/Bono Dzeeta Hortense Kombote Kosana/Alice Berta Bosco Cindy-verus Koby Linda SeppToni/Akwaya-Jean Alfred Andromeda Athanga Bihati Blanquita Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Clint Coco-chimp Damian Desmond Doris Frederike Gamin Hermien Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koto Kumbuka Lara Luky Marlin Maya Mgbadolite Nakuu Natalie Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>chr11_87031808</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Brigitta Doris Julie-LWC21 Lara Luky Marlin Negrita Tibe Ula Vaillant/Andromeda Athanga Bihati Bwamble Cindy-schwein Cleo Coco-chimp Ikuru Julie-A959 Kidongo Mgbadolite Nakuu Tongo Trixie Vincent Yogui/Akwaya-Jean Alfred Alice Berta Blanquita Bosco Cindy-troglodytes Cindy-verus Clint Damian Frederike Gamin Jimmie Koby Koto Linda Maya Mirinda SeppToni Taweh Washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>chr12_95420416</td>\n",
       "      <td>Akwaya-Jean Alfred Koto Nakuu Taweh/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Andromeda Athanga Cindy-schwein Cleo Coco-chimp Ikuru Lara Marlin Negrita Tibe Tongo Trixie/Alice Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-troglodytes Cindy-verus Clint Damian Doris Frederike Gamin Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Linda Luky Maya Mgbadolite Mirinda SeppToni Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>690</th>\n",
       "      <td>chr13_26214400</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alice Berta Bosco Brigitta Cindy-verus Cleo Clint Jimmie Julie-A959 Koby Linda Nakuu SeppToni/Alfred Athanga Bihati Blanquita Bwamble Cindy-schwein Coco-chimp Kidongo Maya Mgbadolite Mirinda Negrita Tongo Trixie Ula Vincent Yogui/Akwaya-Jean Andromeda Cindy-troglodytes Damian Doris Frederike Gamin Ikuru Julie-LWC21 Koto Lara Luky Marlin Taweh Tibe Vaillant Washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>757</th>\n",
       "      <td>chr13_65011712</td>\n",
       "      <td>Cindy-troglodytes/Akwaya-Jean Alfred Brigitta Damian Kidongo Koto Taweh Vaillant/Berta Bono Desmond Dzeeta Gamin Hermien Hortense Jimmie Julie-LWC21 Koby Kombote Kosana Kumbuka Luky Marlin Natalie Negrita Ula Yogui/Alice Andromeda Athanga Bihati Blanquita Bosco Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Doris Frederike Ikuru Julie-A959 Lara Linda Maya Mgbadolite Mirinda Nakuu SeppToni Tibe Tongo Trixie Vincent Washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>958</th>\n",
       "      <td>chr14_83361792</td>\n",
       "      <td>Blanquita Negrita/Bono Cindy-troglodytes Cleo Desmond Doris Dzeeta Hermien Hortense Kombote Kumbuka Lara Mirinda Natalie Tibe Trixie Yogui/Alfred Andromeda Athanga Brigitta Bwamble Cindy-schwein Coco-chimp Gamin Ikuru Kosana Marlin Maya Mgbadolite Tongo Ula Vaillant Washu/Akwaya-Jean Alice Berta Bihati Bosco Cindy-verus Clint Damian Frederike Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Linda Luky Nakuu SeppToni Taweh Vincent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1076</th>\n",
       "      <td>chr15_77070336</td>\n",
       "      <td>Kidongo Mgbadolite/Blanquita Cindy-troglodytes Koto/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Bosco Brigitta Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Lara Linda Luky Marlin Maya Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1130</th>\n",
       "      <td>chr16_40370176</td>\n",
       "      <td>Akwaya-Jean Koto/Brigitta Julie-A959 Tibe/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-LWC21 Kidongo Koby Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita SeppToni Taweh Tongo Trixie Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1261</th>\n",
       "      <td>chr17_60817408</td>\n",
       "      <td>Cindy-verus/Bono Dzeeta Hermien Hortense Kombote Kosana Kumbuka/Akwaya-Jean Alfred Andromeda Bihati Blanquita Bwamble Cleo Damian Frederike Kidongo Maya Mgbadolite Mirinda Nakuu Taweh Tongo Trixie Ula Washu/Alice Athanga Berta Bosco Brigitta Cindy-schwein Cindy-troglodytes Clint Coco-chimp Desmond Doris Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Koto Lara Linda Luky Marlin Natalie Negrita SeppToni Tibe Vaillant Vincent Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1277</th>\n",
       "      <td>chr17_71303168</td>\n",
       "      <td>Ula/Bihati Coco-chimp Maya Mgbadolite Washu/Alfred Andromeda Blanquita Bono Bwamble Cindy-schwein Cleo Desmond Doris Gamin Hortense Kombote Koto Kumbuka Lara Luky Mirinda Nakuu Natalie Tibe Tongo Vaillant Yogui/Akwaya-Jean Alice Athanga Berta Bosco Brigitta Cindy-troglodytes Cindy-verus Clint Damian Dzeeta Frederike Hermien Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Kosana Linda Marlin Negrita SeppToni Taweh Trixie Vincent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1309</th>\n",
       "      <td>chr18_17301504</td>\n",
       "      <td>Cindy-troglodytes Mgbadolite/Bihati Coco-chimp Kidongo Nakuu Trixie/Alice Andromeda Athanga Berta Bosco Cindy-schwein Cindy-verus Cleo Clint Jimmie Koby Linda Marlin SeppToni Washu/Akwaya-Jean Alfred Blanquita Bono Brigitta Bwamble Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Ikuru Julie-A959 Julie-LWC21 Kombote Kosana Koto Kumbuka Lara Luky Maya Mirinda Natalie Negrita Taweh Tibe Tongo Ula Vaillant Vincent Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1466</th>\n",
       "      <td>chr19_45088768</td>\n",
       "      <td>Clint Jimmie/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Bosco Cindy-verus Coco-chimp Ikuru Julie-A959 Kidongo Koby Linda Marlin Negrita Vincent Washu/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Damian Doris Frederike Gamin Julie-LWC21 Koto Lara Luky Maya Mgbadolite Mirinda Nakuu SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1505</th>\n",
       "      <td>chr1_20447232</td>\n",
       "      <td>Hortense/Doris Julie-A959 Marlin Vaillant/Blanquita Brigitta Lara Tibe Yogui/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Bono Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Desmond Dzeeta Frederike Gamin Hermien Ikuru Jimmie Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Linda Luky Maya Mgbadolite Mirinda Nakuu Natalie Negrita SeppToni Taweh Tongo Trixie Ula Vincent Washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1512</th>\n",
       "      <td>chr1_25165824</td>\n",
       "      <td>Bihati Mgbadolite/Bono Desmond Dzeeta Hermien/Akwaya-Jean Hortense Kombote Kosana Kumbuka Natalie/Alfred Alice Andromeda Athanga Berta Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Lara Linda Luky Marlin Maya Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1618</th>\n",
       "      <td>chr1_90701824</td>\n",
       "      <td>Cindy-schwein Nakuu Tongo/Cindy-verus Jimmie Linda/Alfred Athanga Bihati Bono Bwamble Coco-chimp Desmond Dzeeta Frederike Gamin Hermien Hortense Ikuru Kombote Kosana Kumbuka Maya Mgbadolite Natalie Vaillant Vincent Washu/Akwaya-Jean Alice Andromeda Berta Blanquita Bosco Brigitta Cindy-troglodytes Cleo Clint Damian Doris Julie-A959 Julie-LWC21 Kidongo Koby Koto Lara Luky Marlin Mirinda Negrita SeppToni Taweh Tibe Trixie Ula Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1813</th>\n",
       "      <td>chr1_219676672</td>\n",
       "      <td>Brigitta Negrita/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Athanga Bihati Bwamble Cindy-schwein Cindy-troglodytes Doris Gamin Ikuru Julie-A959 Kidongo Koto Lara Luky Maya Mgbadolite Nakuu Tongo Trixie Ula Vaillant Washu Yogui/Akwaya-Jean Alfred Alice Andromeda Berta Blanquita Bosco Cindy-verus Cleo Clint Coco-chimp Damian Frederike Jimmie Julie-LWC21 Koby Linda Marlin Mirinda SeppToni Taweh Tibe Vincent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2047</th>\n",
       "      <td>chr2A_14680064</td>\n",
       "      <td>Akwaya-Jean Koto/Alice Athanga Bosco Brigitta Cleo Clint Frederike Jimmie Luky Marlin Mgbadolite Trixie Ula Washu/Andromeda Bihati Bono Coco-chimp Desmond Dzeeta Hermien Hortense Julie-A959 Julie-LWC21 Kombote Kosana Kumbuka Lara Maya Natalie Tongo/Alfred Berta Blanquita Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Damian Doris Gamin Ikuru Kidongo Koby Linda Mirinda Nakuu Negrita SeppToni Taweh Tibe Vaillant Vincent Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2105</th>\n",
       "      <td>chr2A_48234496</td>\n",
       "      <td>Blanquita Tibe/Andromeda Bihati Coco-chimp Kidongo Maya Mgbadolite Trixie Ula Vincent Washu/Cindy-verus Desmond Dzeeta Hermien Hortense Koby Kosana Mirinda Natalie Tongo/Akwaya-Jean Alfred Alice Athanga Berta Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Clint Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Kombote Koto Kumbuka Lara Linda Luky Marlin Nakuu Negrita SeppToni Taweh Vaillant Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2225</th>\n",
       "      <td>chr2B_27262976</td>\n",
       "      <td>Bono Brigitta/Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Andromeda Bihati Bwamble Cindy-schwein Frederike Julie-A959 Maya Mgbadolite Mirinda Tibe Tongo Trixie Ula Vaillant Vincent/Akwaya-Jean Alfred Alice Athanga Berta Blanquita Bosco Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Gamin Ikuru Jimmie Julie-LWC21 Kidongo Koby Koto Lara Linda Luky Marlin Nakuu Negrita SeppToni Taweh Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2267</th>\n",
       "      <td>chr2B_49283072</td>\n",
       "      <td>Andromeda Maya/Bono Desmond Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Athanga Bihati Doris Ikuru Kidongo Koto Marlin Mgbadolite Taweh Tongo Trixie Vincent Washu/Alfred Alice Berta Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Dzeeta Frederike Gamin Jimmie Julie-A959 Julie-LWC21 Koby Lara Linda Luky Mirinda Nakuu Negrita SeppToni Tibe Ula Vaillant Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2339</th>\n",
       "      <td>chr2B_88604672</td>\n",
       "      <td>Julie-A959 Julie-LWC21/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Athanga Bihati Blanquita Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Doris Gamin Kidongo Lara Luky Nakuu Tibe Ula Vaillant Vincent Yogui/Akwaya-Jean Alfred Alice Andromeda Berta Bosco Cindy-verus Clint Coco-chimp Damian Frederike Ikuru Jimmie Koby Koto Linda Marlin Maya Mgbadolite Mirinda Negrita SeppToni Taweh Tongo Trixie Washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2626</th>\n",
       "      <td>chr3_124256256</td>\n",
       "      <td>Desmond Kosana Natalie/Akwaya-Jean Alice Berta Bosco Cindy-verus Clint Jimmie Koby Koto Linda Mirinda Nakuu SeppToni Taweh Tibe Vincent/Bihati Blanquita Bono Brigitta Bwamble Coco-chimp Damian Doris Dzeeta Frederike Gamin Hermien Hortense Kombote Kumbuka Marlin Negrita Ula/Alfred Andromeda Athanga Cindy-schwein Cindy-troglodytes Cleo Ikuru Julie-A959 Julie-LWC21 Kidongo Lara Luky Maya Mgbadolite Tongo Trixie Vaillant Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2632</th>\n",
       "      <td>chr3_128450560</td>\n",
       "      <td>Berta Blanquita Linda Luky SeppToni/Andromeda Athanga Bwamble Cindy-schwein Ikuru/Alfred Bono Cindy-troglodytes Coco-chimp Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alice Bihati Bosco Brigitta Cindy-verus Cleo Clint Damian Doris Frederike Gamin Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Lara Marlin Maya Mgbadolite Mirinda Nakuu Negrita Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2683</th>\n",
       "      <td>chr3_156237824</td>\n",
       "      <td>Akwaya-Jean Athanga Brigitta Damian Julie-LWC21 Koto Linda Nakuu/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alfred Andromeda Blanquita Cindy-troglodytes Doris Gamin Lara Luky Maya Mirinda Negrita Taweh Tibe Ula Vaillant Washu/Alice Berta Bihati Bosco Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Frederike Ikuru Jimmie Julie-A959 Kidongo Koby Marlin Mgbadolite SeppToni Tongo Trixie Vincent Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2694</th>\n",
       "      <td>chr3_162004992</td>\n",
       "      <td>Hortense Natalie Vaillant/Athanga Bono Desmond Dzeeta Frederike Hermien Trixie/Damian Kidongo Kombote Kosana Kumbuka Vincent Washu/Akwaya-Jean Alfred Alice Andromeda Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Doris Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Koto Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Ula Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2879</th>\n",
       "      <td>chr4_71827456</td>\n",
       "      <td>Athanga Kidongo Mgbadolite Nakuu Tongo Vincent/Cindy-schwein Cleo Dzeeta Hermien Hortense Ikuru Kombote Kumbuka Natalie Washu/Andromeda Bihati Blanquita Bono Bwamble Cindy-troglodytes Coco-chimp Desmond Doris Kosana Luky Marlin Maya Mirinda Negrita Trixie Vaillant Yogui/Akwaya-Jean Alfred Alice Berta Bosco Brigitta Cindy-verus Clint Damian Frederike Gamin Jimmie Julie-A959 Julie-LWC21 Koby Koto Lara Linda SeppToni Taweh Tibe Ula</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               window  \\\n",
       "49     chr10_31457280   \n",
       "101    chr10_65536000   \n",
       "199   chr10_121110528   \n",
       "212     chr11_2097152   \n",
       "282    chr11_41418752   \n",
       "358    chr11_87031808   \n",
       "584    chr12_95420416   \n",
       "690    chr13_26214400   \n",
       "757    chr13_65011712   \n",
       "958    chr14_83361792   \n",
       "1076   chr15_77070336   \n",
       "1130   chr16_40370176   \n",
       "1261   chr17_60817408   \n",
       "1277   chr17_71303168   \n",
       "1309   chr18_17301504   \n",
       "1466   chr19_45088768   \n",
       "1505    chr1_20447232   \n",
       "1512    chr1_25165824   \n",
       "1618    chr1_90701824   \n",
       "1813   chr1_219676672   \n",
       "2047   chr2A_14680064   \n",
       "2105   chr2A_48234496   \n",
       "2225   chr2B_27262976   \n",
       "2267   chr2B_49283072   \n",
       "2339   chr2B_88604672   \n",
       "2626   chr3_124256256   \n",
       "2632   chr3_128450560   \n",
       "2683   chr3_156237824   \n",
       "2694   chr3_162004992   \n",
       "2879    chr4_71827456   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                   cluster_composition  \n",
       "49    Blanquita Bosco Coco-chimp Lara Trixie/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alice Berta Cindy-verus Clint Damian Jimmie Julie-LWC21 Koby Koto Linda SeppToni Taweh Tibe Tongo/Alfred Andromeda Athanga Bihati Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Doris Frederike Gamin Ikuru Julie-A959 Kidongo Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita Ula Vaillant Vincent Washu Yogui  \n",
       "101   Doris Ula Yogui/Athanga Bono Cindy-schwein Cleo Coco-chimp Desmond Dzeeta Hermien Hortense Kidongo Kombote Kosana Kumbuka Nakuu Natalie Washu/Akwaya-Jean Alice Bosco Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Koto Lara Linda Marlin SeppToni Tibe Vaillant Vincent/Alfred Andromeda Berta Bihati Blanquita Brigitta Bwamble Cindy-troglodytes Cindy-verus Clint Damian Frederike Luky Maya Mgbadolite Mirinda Negrita Taweh Tongo Trixie  \n",
       "199   Brigitta Hermien/Doris Gamin SeppToni/Athanga Bihati Cindy-schwein Coco-chimp Damian Ikuru Julie-A959 Nakuu Tibe Vaillant/Akwaya-Jean Alfred Alice Andromeda Berta Blanquita Bono Bosco Bwamble Cindy-troglodytes Cindy-verus Cleo Clint Desmond Dzeeta Frederike Hortense Jimmie Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda Natalie Negrita Taweh Tongo Trixie Ula Vincent Washu Yogui  \n",
       "212   Cleo Ikuru/Coco-chimp Kidongo Nakuu/Alfred Blanquita Cindy-troglodytes Gamin Julie-A959 Lara Marlin Negrita Ula Yogui/Akwaya-Jean Alice Andromeda Athanga Berta Bihati Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-verus Clint Damian Desmond Doris Dzeeta Frederike Hermien Hortense Jimmie Julie-LWC21 Koby Kombote Kosana Koto Kumbuka Linda Luky Maya Mgbadolite Mirinda Natalie SeppToni Taweh Tibe Tongo Trixie Vaillant Vincent Washu  \n",
       "282   Mirinda Negrita/Bono Dzeeta Hortense Kombote Kosana/Alice Berta Bosco Cindy-verus Koby Linda SeppToni/Akwaya-Jean Alfred Andromeda Athanga Bihati Blanquita Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Clint Coco-chimp Damian Desmond Doris Frederike Gamin Hermien Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koto Kumbuka Lara Luky Marlin Maya Mgbadolite Nakuu Natalie Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui  \n",
       "358   Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Brigitta Doris Julie-LWC21 Lara Luky Marlin Negrita Tibe Ula Vaillant/Andromeda Athanga Bihati Bwamble Cindy-schwein Cleo Coco-chimp Ikuru Julie-A959 Kidongo Mgbadolite Nakuu Tongo Trixie Vincent Yogui/Akwaya-Jean Alfred Alice Berta Blanquita Bosco Cindy-troglodytes Cindy-verus Clint Damian Frederike Gamin Jimmie Koby Koto Linda Maya Mirinda SeppToni Taweh Washu  \n",
       "584   Akwaya-Jean Alfred Koto Nakuu Taweh/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Andromeda Athanga Cindy-schwein Cleo Coco-chimp Ikuru Lara Marlin Negrita Tibe Tongo Trixie/Alice Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-troglodytes Cindy-verus Clint Damian Doris Frederike Gamin Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Linda Luky Maya Mgbadolite Mirinda SeppToni Ula Vaillant Vincent Washu Yogui  \n",
       "690   Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alice Berta Bosco Brigitta Cindy-verus Cleo Clint Jimmie Julie-A959 Koby Linda Nakuu SeppToni/Alfred Athanga Bihati Blanquita Bwamble Cindy-schwein Coco-chimp Kidongo Maya Mgbadolite Mirinda Negrita Tongo Trixie Ula Vincent Yogui/Akwaya-Jean Andromeda Cindy-troglodytes Damian Doris Frederike Gamin Ikuru Julie-LWC21 Koto Lara Luky Marlin Taweh Tibe Vaillant Washu  \n",
       "757   Cindy-troglodytes/Akwaya-Jean Alfred Brigitta Damian Kidongo Koto Taweh Vaillant/Berta Bono Desmond Dzeeta Gamin Hermien Hortense Jimmie Julie-LWC21 Koby Kombote Kosana Kumbuka Luky Marlin Natalie Negrita Ula Yogui/Alice Andromeda Athanga Bihati Blanquita Bosco Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Doris Frederike Ikuru Julie-A959 Lara Linda Maya Mgbadolite Mirinda Nakuu SeppToni Tibe Tongo Trixie Vincent Washu  \n",
       "958   Blanquita Negrita/Bono Cindy-troglodytes Cleo Desmond Doris Dzeeta Hermien Hortense Kombote Kumbuka Lara Mirinda Natalie Tibe Trixie Yogui/Alfred Andromeda Athanga Brigitta Bwamble Cindy-schwein Coco-chimp Gamin Ikuru Kosana Marlin Maya Mgbadolite Tongo Ula Vaillant Washu/Akwaya-Jean Alice Berta Bihati Bosco Cindy-verus Clint Damian Frederike Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Linda Luky Nakuu SeppToni Taweh Vincent  \n",
       "1076  Kidongo Mgbadolite/Blanquita Cindy-troglodytes Koto/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Bosco Brigitta Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Lara Linda Luky Marlin Maya Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui  \n",
       "1130  Akwaya-Jean Koto/Brigitta Julie-A959 Tibe/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alfred Alice Andromeda Athanga Berta Bihati Blanquita Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-LWC21 Kidongo Koby Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita SeppToni Taweh Tongo Trixie Ula Vaillant Vincent Washu Yogui  \n",
       "1261  Cindy-verus/Bono Dzeeta Hermien Hortense Kombote Kosana Kumbuka/Akwaya-Jean Alfred Andromeda Bihati Blanquita Bwamble Cleo Damian Frederike Kidongo Maya Mgbadolite Mirinda Nakuu Taweh Tongo Trixie Ula Washu/Alice Athanga Berta Bosco Brigitta Cindy-schwein Cindy-troglodytes Clint Coco-chimp Desmond Doris Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Koto Lara Linda Luky Marlin Natalie Negrita SeppToni Tibe Vaillant Vincent Yogui  \n",
       "1277  Ula/Bihati Coco-chimp Maya Mgbadolite Washu/Alfred Andromeda Blanquita Bono Bwamble Cindy-schwein Cleo Desmond Doris Gamin Hortense Kombote Koto Kumbuka Lara Luky Mirinda Nakuu Natalie Tibe Tongo Vaillant Yogui/Akwaya-Jean Alice Athanga Berta Bosco Brigitta Cindy-troglodytes Cindy-verus Clint Damian Dzeeta Frederike Hermien Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Kosana Linda Marlin Negrita SeppToni Taweh Trixie Vincent  \n",
       "1309  Cindy-troglodytes Mgbadolite/Bihati Coco-chimp Kidongo Nakuu Trixie/Alice Andromeda Athanga Berta Bosco Cindy-schwein Cindy-verus Cleo Clint Jimmie Koby Linda Marlin SeppToni Washu/Akwaya-Jean Alfred Blanquita Bono Brigitta Bwamble Damian Desmond Doris Dzeeta Frederike Gamin Hermien Hortense Ikuru Julie-A959 Julie-LWC21 Kombote Kosana Koto Kumbuka Lara Luky Maya Mirinda Natalie Negrita Taweh Tibe Tongo Ula Vaillant Vincent Yogui  \n",
       "1466  Clint Jimmie/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Bosco Cindy-verus Coco-chimp Ikuru Julie-A959 Kidongo Koby Linda Marlin Negrita Vincent Washu/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Blanquita Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Damian Doris Frederike Gamin Julie-LWC21 Koto Lara Luky Maya Mgbadolite Mirinda Nakuu SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Yogui  \n",
       "1505  Hortense/Doris Julie-A959 Marlin Vaillant/Blanquita Brigitta Lara Tibe Yogui/Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati Bono Bosco Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Desmond Dzeeta Frederike Gamin Hermien Ikuru Jimmie Julie-LWC21 Kidongo Koby Kombote Kosana Koto Kumbuka Linda Luky Maya Mgbadolite Mirinda Nakuu Natalie Negrita SeppToni Taweh Tongo Trixie Ula Vincent Washu  \n",
       "1512  Bihati Mgbadolite/Bono Desmond Dzeeta Hermien/Akwaya-Jean Hortense Kombote Kosana Kumbuka Natalie/Alfred Alice Andromeda Athanga Berta Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Lara Linda Luky Marlin Maya Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui  \n",
       "1618  Cindy-schwein Nakuu Tongo/Cindy-verus Jimmie Linda/Alfred Athanga Bihati Bono Bwamble Coco-chimp Desmond Dzeeta Frederike Gamin Hermien Hortense Ikuru Kombote Kosana Kumbuka Maya Mgbadolite Natalie Vaillant Vincent Washu/Akwaya-Jean Alice Andromeda Berta Blanquita Bosco Brigitta Cindy-troglodytes Cleo Clint Damian Doris Julie-A959 Julie-LWC21 Kidongo Koby Koto Lara Luky Marlin Mirinda Negrita SeppToni Taweh Tibe Trixie Ula Yogui  \n",
       "1813  Brigitta Negrita/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Athanga Bihati Bwamble Cindy-schwein Cindy-troglodytes Doris Gamin Ikuru Julie-A959 Kidongo Koto Lara Luky Maya Mgbadolite Nakuu Tongo Trixie Ula Vaillant Washu Yogui/Akwaya-Jean Alfred Alice Andromeda Berta Blanquita Bosco Cindy-verus Cleo Clint Coco-chimp Damian Frederike Jimmie Julie-LWC21 Koby Linda Marlin Mirinda SeppToni Taweh Tibe Vincent  \n",
       "2047  Akwaya-Jean Koto/Alice Athanga Bosco Brigitta Cleo Clint Frederike Jimmie Luky Marlin Mgbadolite Trixie Ula Washu/Andromeda Bihati Bono Coco-chimp Desmond Dzeeta Hermien Hortense Julie-A959 Julie-LWC21 Kombote Kosana Kumbuka Lara Maya Natalie Tongo/Alfred Berta Blanquita Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Damian Doris Gamin Ikuru Kidongo Koby Linda Mirinda Nakuu Negrita SeppToni Taweh Tibe Vaillant Vincent Yogui  \n",
       "2105  Blanquita Tibe/Andromeda Bihati Coco-chimp Kidongo Maya Mgbadolite Trixie Ula Vincent Washu/Cindy-verus Desmond Dzeeta Hermien Hortense Koby Kosana Mirinda Natalie Tongo/Akwaya-Jean Alfred Alice Athanga Berta Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Clint Damian Doris Frederike Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Kombote Koto Kumbuka Lara Linda Luky Marlin Nakuu Negrita SeppToni Taweh Vaillant Yogui  \n",
       "2225  Bono Brigitta/Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Andromeda Bihati Bwamble Cindy-schwein Frederike Julie-A959 Maya Mgbadolite Mirinda Tibe Tongo Trixie Ula Vaillant Vincent/Akwaya-Jean Alfred Alice Athanga Berta Blanquita Bosco Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Doris Gamin Ikuru Jimmie Julie-LWC21 Kidongo Koby Koto Lara Linda Luky Marlin Nakuu Negrita SeppToni Taweh Washu Yogui  \n",
       "2267  Andromeda Maya/Bono Desmond Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Athanga Bihati Doris Ikuru Kidongo Koto Marlin Mgbadolite Taweh Tongo Trixie Vincent Washu/Alfred Alice Berta Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Damian Dzeeta Frederike Gamin Jimmie Julie-A959 Julie-LWC21 Koby Lara Linda Luky Mirinda Nakuu Negrita SeppToni Tibe Ula Vaillant Yogui  \n",
       "2339  Julie-A959 Julie-LWC21/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Athanga Bihati Blanquita Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cleo Doris Gamin Kidongo Lara Luky Nakuu Tibe Ula Vaillant Vincent Yogui/Akwaya-Jean Alfred Alice Andromeda Berta Bosco Cindy-verus Clint Coco-chimp Damian Frederike Ikuru Jimmie Koby Koto Linda Marlin Maya Mgbadolite Mirinda Negrita SeppToni Taweh Tongo Trixie Washu  \n",
       "2626  Desmond Kosana Natalie/Akwaya-Jean Alice Berta Bosco Cindy-verus Clint Jimmie Koby Koto Linda Mirinda Nakuu SeppToni Taweh Tibe Vincent/Bihati Blanquita Bono Brigitta Bwamble Coco-chimp Damian Doris Dzeeta Frederike Gamin Hermien Hortense Kombote Kumbuka Marlin Negrita Ula/Alfred Andromeda Athanga Cindy-schwein Cindy-troglodytes Cleo Ikuru Julie-A959 Julie-LWC21 Kidongo Lara Luky Maya Mgbadolite Tongo Trixie Vaillant Washu Yogui  \n",
       "2632  Berta Blanquita Linda Luky SeppToni/Andromeda Athanga Bwamble Cindy-schwein Ikuru/Alfred Bono Cindy-troglodytes Coco-chimp Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alice Bihati Bosco Brigitta Cindy-verus Cleo Clint Damian Doris Frederike Gamin Jimmie Julie-A959 Julie-LWC21 Kidongo Koby Koto Lara Marlin Maya Mgbadolite Mirinda Nakuu Negrita Taweh Tibe Tongo Trixie Ula Vaillant Vincent Washu Yogui  \n",
       "2683  Akwaya-Jean Athanga Brigitta Damian Julie-LWC21 Koto Linda Nakuu/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alfred Andromeda Blanquita Cindy-troglodytes Doris Gamin Lara Luky Maya Mirinda Negrita Taweh Tibe Ula Vaillant Washu/Alice Berta Bihati Bosco Bwamble Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Frederike Ikuru Jimmie Julie-A959 Kidongo Koby Marlin Mgbadolite SeppToni Tongo Trixie Vincent Yogui  \n",
       "2694  Hortense Natalie Vaillant/Athanga Bono Desmond Dzeeta Frederike Hermien Trixie/Damian Kidongo Kombote Kosana Kumbuka Vincent Washu/Akwaya-Jean Alfred Alice Andromeda Berta Bihati Blanquita Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes Cindy-verus Cleo Clint Coco-chimp Doris Gamin Ikuru Jimmie Julie-A959 Julie-LWC21 Koby Koto Lara Linda Luky Marlin Maya Mgbadolite Mirinda Nakuu Negrita SeppToni Taweh Tibe Tongo Ula Yogui  \n",
       "2879  Athanga Kidongo Mgbadolite Nakuu Tongo Vincent/Cindy-schwein Cleo Dzeeta Hermien Hortense Ikuru Kombote Kumbuka Natalie Washu/Andromeda Bihati Blanquita Bono Bwamble Cindy-troglodytes Coco-chimp Desmond Doris Kosana Luky Marlin Maya Mirinda Negrita Trixie Vaillant Yogui/Akwaya-Jean Alfred Alice Berta Bosco Brigitta Cindy-verus Clint Damian Frederike Gamin Jimmie Julie-A959 Julie-LWC21 Koby Koto Lara Linda SeppToni Taweh Tibe Ula  "
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "complete_linkage_trees[complete_linkage_trees['n_clusters'] == 4][['window','cluster_composition']].head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "f3c26938",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>cluster_composition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>chr12_42467328</td>\n",
       "      <td>Hermien Kombote Vaillant/Bono Brigitta Julie-LWC21 Koby Kosana Koto Linda Yogui/Alice Berta Bosco Cindy-verus Clint Desmond Jimmie Natalie SeppToni/Alfred Bihati Bwamble Cindy-schwein Doris Frederike Gamin Ikuru Kidongo Luky Mgbadolite Nakuu Tongo Trixie Vincent Washu/Akwaya-Jean Andromeda Athanga Blanquita Cindy-troglodytes Cleo Coco-chimp Damian Dzeeta Hortense Julie-A959 Kumbuka Lara Marlin Maya Mirinda Negrita Taweh Tibe Ula</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1494</th>\n",
       "      <td>chr1_12058624</td>\n",
       "      <td>Athanga/Cindy-troglodytes Frederike Maya/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alice Bosco Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Doris Ikuru Jimmie Lara Linda Marlin Mgbadolite Nakuu SeppToni Taweh Ula Vincent/Akwaya-Jean Alfred Andromeda Berta Bihati Blanquita Brigitta Bwamble Damian Gamin Julie-A959 Julie-LWC21 Kidongo Koby Koto Luky Mirinda Negrita Tibe Tongo Trixie Vaillant Washu Yogui</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2817</th>\n",
       "      <td>chr4_39321600</td>\n",
       "      <td>Brigitta Cleo Negrita Vincent/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka/Bihati Blanquita Kidongo Marlin Mirinda Trixie Ula Vaillant Yogui/Alfred Alice Berta Bosco Cindy-troglodytes Cindy-verus Clint Jimmie Koby Linda SeppToni Taweh/Akwaya-Jean Andromeda Athanga Bwamble Cindy-schwein Coco-chimp Damian Doris Frederike Gamin Ikuru Julie-A959 Julie-LWC21 Koto Lara Luky Maya Mgbadolite Nakuu Natalie Tibe Tongo Washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2889</th>\n",
       "      <td>chr4_78118912</td>\n",
       "      <td>Brigitta/Bihati Cindy-schwein Cleo Coco-chimp Frederike Ikuru Tibe Tongo/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alice Cindy-troglodytes Damian Julie-A959 Koto Luky Marlin Mirinda SeppToni Taweh/Alfred Andromeda Athanga Berta Blanquita Bosco Bwamble Cindy-verus Clint Doris Gamin Jimmie Julie-LWC21 Kidongo Koby Lara Linda Maya Mgbadolite Nakuu Negrita Trixie Ula Vaillant Vincent Washu Yogui</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              window  \\\n",
       "497   chr12_42467328   \n",
       "1494   chr1_12058624   \n",
       "2817   chr4_39321600   \n",
       "2889   chr4_78118912   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                   cluster_composition  \n",
       "497   Hermien Kombote Vaillant/Bono Brigitta Julie-LWC21 Koby Kosana Koto Linda Yogui/Alice Berta Bosco Cindy-verus Clint Desmond Jimmie Natalie SeppToni/Alfred Bihati Bwamble Cindy-schwein Doris Frederike Gamin Ikuru Kidongo Luky Mgbadolite Nakuu Tongo Trixie Vincent Washu/Akwaya-Jean Andromeda Athanga Blanquita Cindy-troglodytes Cleo Coco-chimp Damian Dzeeta Hortense Julie-A959 Kumbuka Lara Marlin Maya Mirinda Negrita Taweh Tibe Ula  \n",
       "1494  Athanga/Cindy-troglodytes Frederike Maya/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Alice Bosco Cindy-schwein Cindy-verus Cleo Clint Coco-chimp Doris Ikuru Jimmie Lara Linda Marlin Mgbadolite Nakuu SeppToni Taweh Ula Vincent/Akwaya-Jean Alfred Andromeda Berta Bihati Blanquita Brigitta Bwamble Damian Gamin Julie-A959 Julie-LWC21 Kidongo Koby Koto Luky Mirinda Negrita Tibe Tongo Trixie Vaillant Washu Yogui  \n",
       "2817  Brigitta Cleo Negrita Vincent/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka/Bihati Blanquita Kidongo Marlin Mirinda Trixie Ula Vaillant Yogui/Alfred Alice Berta Bosco Cindy-troglodytes Cindy-verus Clint Jimmie Koby Linda SeppToni Taweh/Akwaya-Jean Andromeda Athanga Bwamble Cindy-schwein Coco-chimp Damian Doris Frederike Gamin Ikuru Julie-A959 Julie-LWC21 Koto Lara Luky Maya Mgbadolite Nakuu Natalie Tibe Tongo Washu  \n",
       "2889  Brigitta/Bihati Cindy-schwein Cleo Coco-chimp Frederike Ikuru Tibe Tongo/Bono Desmond Dzeeta Hermien Hortense Kombote Kosana Kumbuka Natalie/Akwaya-Jean Alice Cindy-troglodytes Damian Julie-A959 Koto Luky Marlin Mirinda SeppToni Taweh/Alfred Andromeda Athanga Berta Blanquita Bosco Bwamble Cindy-verus Clint Doris Gamin Jimmie Julie-LWC21 Kidongo Koby Lara Linda Maya Mgbadolite Nakuu Negrita Trixie Ula Vaillant Vincent Washu Yogui  "
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "complete_linkage_trees[complete_linkage_trees['n_clusters'] == 5][['window','cluster_composition']].head(4)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c032ba89",
   "metadata": {},
   "source": [
    "Reset the display setting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "fe5271c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.options.display.max_colwidth = 50"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26209c3e-3c01-4873-8b4d-06882f634d4b",
   "metadata": {},
   "source": [
    "Let's map the topological classification of each window as well as genes in that window onto the hierarchical clustering dataframe. First, topology classification."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "e77d3065-a57a-441d-a490-7ec191df1a53",
   "metadata": {},
   "outputs": [],
   "source": [
    "def topology(complete_linkage_trees):\n",
    "    if complete_linkage_trees['window'] in SDI_windows['window'].values:\n",
    "        return 'Highly Divergent Individual'\n",
    "    elif complete_linkage_trees['window'] in MDI_windows['window'].values:\n",
    "        return 'Multiple Divergent Individuals'\n",
    "    elif complete_linkage_trees['window'] in ppn_ptr_windows['window'].values:\n",
    "        return 'Bonobo-Chimpanzee Clustering'\n",
    "    elif complete_linkage_trees['window'] in western_chimpanzee_divergent_windows['window'].values:\n",
    "        return 'Western Chimpanzee Clustering'\n",
    "    elif complete_linkage_trees['n_clusters'] == 3:\n",
    "        return 'Three Cluster'\n",
    "    elif complete_linkage_trees['n_clusters'] == 4:\n",
    "        return 'Four Cluster'\n",
    "    elif complete_linkage_trees['n_clusters'] == 5:\n",
    "        return 'Five Cluster'\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "complete_linkage_trees['topology'] = complete_linkage_trees.apply(topology, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "954226ae-3f74-4101-ae4e-8acd633df0b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <th>Alfred</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Athanga</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Bono</th>\n",
       "      <th>Bosco</th>\n",
       "      <th>Brigitta</th>\n",
       "      <th>Bwamble</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Clint</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Damian</th>\n",
       "      <th>Desmond</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Gamin</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Koby</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Koto</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mgbadolite</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>SeppToni</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Tongo</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>Vaillant</th>\n",
       "      <th>Vincent</th>\n",
       "      <th>Washu</th>\n",
       "      <th>Yogui</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "      <th>n_clusters</th>\n",
       "      <th>n_single_clusters</th>\n",
       "      <th>cluster_composition</th>\n",
       "      <th>topology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_1572864</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.014522</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>0.067514</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "      <td>Three Cluster</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_2097152</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.038838</td>\n",
       "      <td>0.269334</td>\n",
       "      <td>0.269334</td>\n",
       "      <td>0.115208</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...</td>\n",
       "      <td>Multiple Divergent Individuals</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_2621440</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005674</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>0.006737</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...</td>\n",
       "      <td>Multiple Divergent Individuals</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_3145728</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002904</td>\n",
       "      <td>0.010145</td>\n",
       "      <td>0.010145</td>\n",
       "      <td>0.006418</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Cindy-verus Luky/Akwaya-Jean Alfred Andr...</td>\n",
       "      <td>Multiple Divergent Individuals</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_3670016</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.00514</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>0.04808</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "      <td>Bonobo-Chimpanzee Clustering</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          window Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati  \\\n",
       "0  chr10_1572864          C3     C3    C3        C2      C2    C3     C3   \n",
       "1  chr10_2097152          C2     C2    C2        C2      C2    C2     C2   \n",
       "2  chr10_2621440          C2     C2    C2        C2      C2    C2     C2   \n",
       "3  chr10_3145728          C2     C2    C1        C2      C2    C2     C2   \n",
       "4  chr10_3670016          C2     C2    C2        C2      C2    C2     C2   \n",
       "\n",
       "  Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes  \\\n",
       "0        C2   C1    C3       C3      C2            C2                C3   \n",
       "1        C2   C2    C2       C2      C2            C2                C1   \n",
       "2        C2   C2    C2       C2      C2            C2                C1   \n",
       "3        C2   C2    C2       C2      C2            C2                C2   \n",
       "4        C2   C1    C2       C2      C2            C2                C2   \n",
       "\n",
       "  Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike  \\\n",
       "0          C3   C2    C3         C2     C3      C1    C2     C1        C3   \n",
       "1          C2   C2    C2         C2     C2      C2    C1     C2        C2   \n",
       "2          C2   C2    C2         C2     C2      C2    C1     C2        C2   \n",
       "3          C1   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "4          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "\n",
       "  Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby  \\\n",
       "0    C3      C1       C1    C3     C3         C3          C3      C2   C3   \n",
       "1    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "2    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "3    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "4    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "\n",
       "  Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda  \\\n",
       "0      C1     C1   C3      C1   C2    C3   C3     C3   C2         C2      C2   \n",
       "1      C2     C2   C2      C2   C2    C2   C1     C1   C2         C2      C2   \n",
       "2      C2     C2   C2      C2   C2    C2   C1     C1   C2         C2      C2   \n",
       "3      C2     C2   C2      C2   C2    C2   C1     C2   C2         C2      C2   \n",
       "4      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2      C2   \n",
       "\n",
       "  Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent  \\\n",
       "0    C2      C1      C2       C3    C3   C3    C2     C2  C3       C3      C2   \n",
       "1    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "2    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "3    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "4    C2      C1      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "\n",
       "  Washu Yogui    node_0    node_1    node_2    node_3  n_clusters  \\\n",
       "0    C2    C2  0.014522  0.091748  0.091748  0.067514           3   \n",
       "1    C2    C2  0.038838  0.269334  0.269334  0.115208           2   \n",
       "2    C2    C2  0.005674  0.012824  0.012824  0.006737           2   \n",
       "3    C2    C2  0.002904  0.010145  0.010145  0.006418           2   \n",
       "4    C2    C2   0.00514  0.096248  0.096248   0.04808           2   \n",
       "\n",
       "   n_single_clusters                                cluster_composition  \\\n",
       "0                  0  Bono Desmond Dzeeta Hermien Hortense Kombote K...   \n",
       "1                  0  Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...   \n",
       "2                  0  Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...   \n",
       "3                  0  Alice Cindy-verus Luky/Akwaya-Jean Alfred Andr...   \n",
       "4                  0  Bono Desmond Dzeeta Hermien Hortense Kombote K...   \n",
       "\n",
       "                         topology  \n",
       "0                   Three Cluster  \n",
       "1  Multiple Divergent Individuals  \n",
       "2  Multiple Divergent Individuals  \n",
       "3  Multiple Divergent Individuals  \n",
       "4    Bonobo-Chimpanzee Clustering  "
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "complete_linkage_trees.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dca4c8d8-e645-4c7c-a0b1-fe17315bc73c",
   "metadata": {},
   "source": [
    "On to genes per window. First, read in an intersection of the panTro6 windows used in analysis and genes BED files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "87a2d009-ca5b-4d40-9e00-743105d390c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>gene_chr</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>gene_transcript</th>\n",
       "      <th>gene</th>\n",
       "      <th>window</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1088438</td>\n",
       "      <td>1096220</td>\n",
       "      <td>XM_513729.6</td>\n",
       "      <td>PEX10</td>\n",
       "      <td>chr1_1048576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1220059</td>\n",
       "      <td>1221565</td>\n",
       "      <td>XM_009446771.2</td>\n",
       "      <td>HES5</td>\n",
       "      <td>chr1_1048576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1248417</td>\n",
       "      <td>1256443</td>\n",
       "      <td>XM_009446795.3</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>chr1_1048576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1279618</td>\n",
       "      <td>1284329</td>\n",
       "      <td>XM_001151207.3</td>\n",
       "      <td>FAM213B</td>\n",
       "      <td>chr1_1048576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1606738</td>\n",
       "      <td>1608184</td>\n",
       "      <td>XM_524851.6</td>\n",
       "      <td>ACTRT2</td>\n",
       "      <td>chr1_1048576</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  window_chr  window_start  window_end gene_chr  gene_start  gene_end  \\\n",
       "0       chr1       1048576     2097152     chr1     1088438   1096220   \n",
       "1       chr1       1048576     2097152     chr1     1220059   1221565   \n",
       "2       chr1       1048576     2097152     chr1     1248417   1256443   \n",
       "3       chr1       1048576     2097152     chr1     1279618   1284329   \n",
       "4       chr1       1048576     2097152     chr1     1606738   1608184   \n",
       "\n",
       "  gene_transcript      gene        window  \n",
       "0     XM_513729.6     PEX10  chr1_1048576  \n",
       "1  XM_009446771.2      HES5  chr1_1048576  \n",
       "2  XM_009446795.3  TNFRSF14  chr1_1048576  \n",
       "3  XM_001151207.3   FAM213B  chr1_1048576  \n",
       "4     XM_524851.6    ACTRT2  chr1_1048576  "
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "windows_genes_intersect = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/annotations/windows_genes_intersect.bed', sep = '\\t', names = ['window_chr','window_start','window_end','gene_chr','gene_start','gene_end','gene_transcript','gene'])\n",
    "windows_genes_intersect['window'] = windows_genes_intersect['window_chr'] + '_' + windows_genes_intersect['window_start'].astype(str)\n",
    "windows_genes_intersect.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb745274-109e-4cae-bbe6-8f8d6c04ff08",
   "metadata": {},
   "source": [
    "Next, define a function to create an alphabetized list of genes per window."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "de62488c-5007-4e58-8740-f39f4dba51fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_gene_list_per_window(window):\n",
    "    sorted_genes = sorted(window)\n",
    "    genes = [gene for gene in sorted_genes]\n",
    "    return ', '.join(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "e74bc3a7-3e5e-41b3-b2d4-9a7803b483c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100139008</td>\n",
       "      <td>CFAP58, GSTO2, ITPRIP, SORCS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_100663296</td>\n",
       "      <td>SORCS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_101711872</td>\n",
       "      <td>SORCS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_102236160</td>\n",
       "      <td>SORCS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_102760448</td>\n",
       "      <td>SORCS1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window                           gene\n",
       "0  chr10_100139008  CFAP58, GSTO2, ITPRIP, SORCS3\n",
       "1  chr10_100663296                         SORCS3\n",
       "2  chr10_101711872                         SORCS1\n",
       "3  chr10_102236160                         SORCS1\n",
       "4  chr10_102760448                         SORCS1"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_per_window = windows_genes_intersect.groupby('window')['gene'].apply(create_gene_list_per_window).reset_index()\n",
    "genes_per_window.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "c8ea7feb-8c8b-4b34-9895-c8563fa0377f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>Akwaya-Jean</th>\n",
       "      <th>Alfred</th>\n",
       "      <th>Alice</th>\n",
       "      <th>Andromeda</th>\n",
       "      <th>Athanga</th>\n",
       "      <th>Berta</th>\n",
       "      <th>Bihati</th>\n",
       "      <th>Blanquita</th>\n",
       "      <th>Bono</th>\n",
       "      <th>Bosco</th>\n",
       "      <th>Brigitta</th>\n",
       "      <th>Bwamble</th>\n",
       "      <th>Cindy-schwein</th>\n",
       "      <th>Cindy-troglodytes</th>\n",
       "      <th>Cindy-verus</th>\n",
       "      <th>Cleo</th>\n",
       "      <th>Clint</th>\n",
       "      <th>Coco-chimp</th>\n",
       "      <th>Damian</th>\n",
       "      <th>Desmond</th>\n",
       "      <th>Doris</th>\n",
       "      <th>Dzeeta</th>\n",
       "      <th>Frederike</th>\n",
       "      <th>Gamin</th>\n",
       "      <th>Hermien</th>\n",
       "      <th>Hortense</th>\n",
       "      <th>Ikuru</th>\n",
       "      <th>Jimmie</th>\n",
       "      <th>Julie-A959</th>\n",
       "      <th>Julie-LWC21</th>\n",
       "      <th>Kidongo</th>\n",
       "      <th>Koby</th>\n",
       "      <th>Kombote</th>\n",
       "      <th>Kosana</th>\n",
       "      <th>Koto</th>\n",
       "      <th>Kumbuka</th>\n",
       "      <th>Lara</th>\n",
       "      <th>Linda</th>\n",
       "      <th>Luky</th>\n",
       "      <th>Marlin</th>\n",
       "      <th>Maya</th>\n",
       "      <th>Mgbadolite</th>\n",
       "      <th>Mirinda</th>\n",
       "      <th>Nakuu</th>\n",
       "      <th>Natalie</th>\n",
       "      <th>Negrita</th>\n",
       "      <th>SeppToni</th>\n",
       "      <th>Taweh</th>\n",
       "      <th>Tibe</th>\n",
       "      <th>Tongo</th>\n",
       "      <th>Trixie</th>\n",
       "      <th>Ula</th>\n",
       "      <th>Vaillant</th>\n",
       "      <th>Vincent</th>\n",
       "      <th>Washu</th>\n",
       "      <th>Yogui</th>\n",
       "      <th>node_0</th>\n",
       "      <th>node_1</th>\n",
       "      <th>node_2</th>\n",
       "      <th>node_3</th>\n",
       "      <th>n_clusters</th>\n",
       "      <th>n_single_clusters</th>\n",
       "      <th>cluster_composition</th>\n",
       "      <th>topology</th>\n",
       "      <th>genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_1572864</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C3</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C3</td>\n",
       "      <td>C3</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.014522</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>0.067514</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "      <td>Three Cluster</td>\n",
       "      <td>ADARB2, LOC112204541, LOC736356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_2621440</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.005674</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>0.012824</td>\n",
       "      <td>0.006737</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...</td>\n",
       "      <td>Multiple Divergent Individuals</td>\n",
       "      <td>LOC107976892, PFKP, PITRM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_3145728</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.002904</td>\n",
       "      <td>0.010145</td>\n",
       "      <td>0.010145</td>\n",
       "      <td>0.006418</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Alice Cindy-verus Luky/Akwaya-Jean Alfred Andr...</td>\n",
       "      <td>Multiple Divergent Individuals</td>\n",
       "      <td>KLF6, LOC107976892, PFKP, PITRM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_3670016</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.00514</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>0.04808</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "      <td>Bonobo-Chimpanzee Clustering</td>\n",
       "      <td>KLF6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_4194304</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C1</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>C2</td>\n",
       "      <td>0.001365</td>\n",
       "      <td>0.012365</td>\n",
       "      <td>0.012365</td>\n",
       "      <td>0.007215</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Bono Desmond Dzeeta Hermien Hortense Kombote K...</td>\n",
       "      <td>Bonobo-Chimpanzee Clustering</td>\n",
       "      <td>AKR1C2, AKR1C3, AKR1E2, LOC100607996</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          window Akwaya-Jean Alfred Alice Andromeda Athanga Berta Bihati  \\\n",
       "0  chr10_1572864          C3     C3    C3        C2      C2    C3     C3   \n",
       "1  chr10_2621440          C2     C2    C2        C2      C2    C2     C2   \n",
       "2  chr10_3145728          C2     C2    C1        C2      C2    C2     C2   \n",
       "3  chr10_3670016          C2     C2    C2        C2      C2    C2     C2   \n",
       "4  chr10_4194304          C2     C2    C2        C2      C2    C2     C2   \n",
       "\n",
       "  Blanquita Bono Bosco Brigitta Bwamble Cindy-schwein Cindy-troglodytes  \\\n",
       "0        C2   C1    C3       C3      C2            C2                C3   \n",
       "1        C2   C2    C2       C2      C2            C2                C1   \n",
       "2        C2   C2    C2       C2      C2            C2                C2   \n",
       "3        C2   C1    C2       C2      C2            C2                C2   \n",
       "4        C2   C1    C2       C2      C2            C2                C2   \n",
       "\n",
       "  Cindy-verus Cleo Clint Coco-chimp Damian Desmond Doris Dzeeta Frederike  \\\n",
       "0          C3   C2    C3         C2     C3      C1    C2     C1        C3   \n",
       "1          C2   C2    C2         C2     C2      C2    C1     C2        C2   \n",
       "2          C1   C2    C2         C2     C2      C2    C2     C2        C2   \n",
       "3          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "4          C2   C2    C2         C2     C2      C1    C2     C1        C2   \n",
       "\n",
       "  Gamin Hermien Hortense Ikuru Jimmie Julie-A959 Julie-LWC21 Kidongo Koby  \\\n",
       "0    C3      C1       C1    C3     C3         C3          C3      C2   C3   \n",
       "1    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "2    C2      C2       C2    C2     C2         C2          C2      C2   C2   \n",
       "3    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "4    C2      C1       C1    C2     C2         C2          C2      C2   C2   \n",
       "\n",
       "  Kombote Kosana Koto Kumbuka Lara Linda Luky Marlin Maya Mgbadolite Mirinda  \\\n",
       "0      C1     C1   C3      C1   C2    C3   C3     C3   C2         C2      C2   \n",
       "1      C2     C2   C2      C2   C2    C2   C1     C1   C2         C2      C2   \n",
       "2      C2     C2   C2      C2   C2    C2   C1     C2   C2         C2      C2   \n",
       "3      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2      C2   \n",
       "4      C1     C1   C2      C1   C2    C2   C2     C2   C2         C2      C2   \n",
       "\n",
       "  Nakuu Natalie Negrita SeppToni Taweh Tibe Tongo Trixie Ula Vaillant Vincent  \\\n",
       "0    C2      C1      C2       C3    C3   C3    C2     C2  C3       C3      C2   \n",
       "1    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "2    C2      C2      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "3    C2      C1      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "4    C2      C1      C2       C2    C2   C2    C2     C2  C2       C2      C2   \n",
       "\n",
       "  Washu Yogui    node_0    node_1    node_2    node_3  n_clusters  \\\n",
       "0    C2    C2  0.014522  0.091748  0.091748  0.067514           3   \n",
       "1    C2    C2  0.005674  0.012824  0.012824  0.006737           2   \n",
       "2    C2    C2  0.002904  0.010145  0.010145  0.006418           2   \n",
       "3    C2    C2   0.00514  0.096248  0.096248   0.04808           2   \n",
       "4    C2    C2  0.001365  0.012365  0.012365  0.007215           2   \n",
       "\n",
       "   n_single_clusters                                cluster_composition  \\\n",
       "0                  0  Bono Desmond Dzeeta Hermien Hortense Kombote K...   \n",
       "1                  0  Cindy-troglodytes Doris Luky Marlin/Akwaya-Jea...   \n",
       "2                  0  Alice Cindy-verus Luky/Akwaya-Jean Alfred Andr...   \n",
       "3                  0  Bono Desmond Dzeeta Hermien Hortense Kombote K...   \n",
       "4                  0  Bono Desmond Dzeeta Hermien Hortense Kombote K...   \n",
       "\n",
       "                         topology                                 genes  \n",
       "0                   Three Cluster       ADARB2, LOC112204541, LOC736356  \n",
       "1  Multiple Divergent Individuals            LOC107976892, PFKP, PITRM1  \n",
       "2  Multiple Divergent Individuals      KLF6, LOC107976892, PFKP, PITRM1  \n",
       "3    Bonobo-Chimpanzee Clustering                                  KLF6  \n",
       "4    Bonobo-Chimpanzee Clustering  AKR1C2, AKR1C3, AKR1E2, LOC100607996  "
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "complete_linkage_trees_with_genes = pd.merge(complete_linkage_trees, genes_per_window, on = ['window'])\n",
    "complete_linkage_trees_with_genes = complete_linkage_trees_with_genes.rename(columns = {'gene':'genes'})\n",
    "complete_linkage_trees_with_genes.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eeb9160f-78b2-4d69-af88-d5407b132efd",
   "metadata": {},
   "source": [
    "Save this dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "5cabb8b8-a8eb-446a-9727-b9032070dd4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "complete_linkage_trees_with_genes.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/window_topologies/complete_linkage_trees_with_topologies_and_genes.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a6586e3",
   "metadata": {},
   "source": [
    "## Bonobo-Chimpanzee Windows <a class = 'anchor' id = 'bonobochimpanzeewindows'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa172ba6",
   "metadata": {},
   "source": [
    "Let's take a closer look at the 339 windows with bonobo-chimpanzee clustering. Start with some basic statistics on the divergence scores in these windows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "81fe4c2e-b5d5-463c-955c-b1c3c1f10c3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_ppn_ptr_window_stats(window):\n",
    "    stats = []\n",
    "    subset = comparisons[comparisons['window'] == window]\n",
    "    subset = subset[subset['lineages'] == 'ppn-ptr']\n",
    "    minimum = subset['divergence'].min()\n",
    "    mean = subset['divergence'].mean()\n",
    "    maximum = subset['divergence'].max()\n",
    "    variance = subset['divergence'].var()\n",
    "    stats.append(subset['window'].iloc[0])\n",
    "    stats.append(minimum)\n",
    "    stats.append(mean)\n",
    "    stats.append(maximum)\n",
    "    stats.append(variance)\n",
    "    \n",
    "    return stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "84757b55-afd8-4774-ae94-0ff8ebe37c03",
   "metadata": {},
   "outputs": [],
   "source": [
    "#ppn_ptr_window_stats = [get_ppn_ptr_window_stats(window) for window in ppn_ptr_windows['window']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "88ffae5f-57bd-404c-a456-d5ae6b7dc1b1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#ppn_ptr_window_stats = pd.DataFrame(ppn_ptr_window_stats, columns = ['window', 'minimum', 'mean', 'maximum', 'variance'])\n",
    "#ppn_ptr_window_stats.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35e8b592-11cf-4874-8524-c1ea31a12a00",
   "metadata": {},
   "source": [
    "Save this dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "c00c54c1-bb90-42d8-bac3-bc7eafd3888b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#ppn_ptr_window_stats.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_window_stats.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d513b7ea-0001-4ab6-b313-5bd07f26f761",
   "metadata": {},
   "source": [
    "Read in dataframe once it's been created."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "e0a8b92d-a887-4733-8d29-309bb304df2b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>minimum</th>\n",
       "      <th>mean</th>\n",
       "      <th>maximum</th>\n",
       "      <th>variance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_3670016</td>\n",
       "      <td>0.008749</td>\n",
       "      <td>0.029608</td>\n",
       "      <td>0.096248</td>\n",
       "      <td>2.549325e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_4194304</td>\n",
       "      <td>0.001334</td>\n",
       "      <td>0.003769</td>\n",
       "      <td>0.012365</td>\n",
       "      <td>3.552572e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_23592960</td>\n",
       "      <td>0.003272</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>0.055726</td>\n",
       "      <td>9.082293e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_28311552</td>\n",
       "      <td>0.000317</td>\n",
       "      <td>0.001306</td>\n",
       "      <td>0.004317</td>\n",
       "      <td>4.994337e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_30932992</td>\n",
       "      <td>0.003739</td>\n",
       "      <td>0.011764</td>\n",
       "      <td>0.024301</td>\n",
       "      <td>1.398396e-05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           window   minimum      mean   maximum      variance\n",
       "0   chr10_3670016  0.008749  0.029608  0.096248  2.549325e-04\n",
       "1   chr10_4194304  0.001334  0.003769  0.012365  3.552572e-06\n",
       "2  chr10_23592960  0.003272  0.019802  0.055726  9.082293e-05\n",
       "3  chr10_28311552  0.000317  0.001306  0.004317  4.994337e-07\n",
       "4  chr10_30932992  0.003739  0.011764  0.024301  1.398396e-05"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_window_stats = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_window_stats.txt', sep = '\\t', header = 0)\n",
    "ppn_ptr_window_stats.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2f27c65-b3a1-46a8-a614-b6066efb2afb",
   "metadata": {},
   "source": [
    "The bonobo-chimpanzee clustering windows include quite a few with minimum divergence. Let's get a count of the number of windows using different thresholds."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "6250bd49-d09a-456b-8f32-fef969a8fbac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "89"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_window_stats[ppn_ptr_window_stats['minimum'] > 0.01])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "63ddfc44-fb0a-426a-ab24-6dd2f75d4578",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "56"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_window_stats[ppn_ptr_window_stats['minimum'] > 0.02])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "56fc2a29-a037-4c79-b4e2-3bba7bd6c710",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "39"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_window_stats[ppn_ptr_window_stats['minimum'] > 0.03])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "9bb4f423-8527-41e3-9211-3a8199180d8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_window_stats[ppn_ptr_window_stats['minimum'] > 0.04])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "51934641-79d9-4a44-af44-a3d3d4eb778b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_window_stats[ppn_ptr_window_stats['minimum'] > 0.05])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4153fde4-3ff1-447e-9b0d-08f058e6c48a",
   "metadata": {},
   "source": [
    "Let's examine the divergence for the example in Figure 3A."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "17982fcb-72e4-4ac9-b6d7-ca444703e75a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ind1</th>\n",
       "      <th>ind2</th>\n",
       "      <th>lineages</th>\n",
       "      <th>chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>spearman</th>\n",
       "      <th>divergence</th>\n",
       "      <th>seq_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3323824</th>\n",
       "      <td>Clint</td>\n",
       "      <td>Hortense</td>\n",
       "      <td>ppn-ptr</td>\n",
       "      <td>chr5</td>\n",
       "      <td>16252928</td>\n",
       "      <td>chr5_16252928</td>\n",
       "      <td>0.018395</td>\n",
       "      <td>0.96789</td>\n",
       "      <td>0.03211</td>\n",
       "      <td>4441</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ind1      ind2 lineages   chr  window_start         window  \\\n",
       "3323824  Clint  Hortense  ppn-ptr  chr5      16252928  chr5_16252928   \n",
       "\n",
       "              mse  spearman  divergence  seq_diff  \n",
       "3323824  0.018395   0.96789     0.03211      4441  "
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr5_16252928') & (comparisons['ind1'] == 'Clint') & (comparisons['ind2'] == 'Hortense')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "ab04fbe5-4e58-41f7-91d2-ea94cc1e6e41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.02491708517"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr5_16252928') & (comparisons['lineages'] == 'ppn-ptr')]['divergence'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "ed7c21b1-0c52-4346-9924-cd6e3c6d631c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.03670547712"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr5_16252928') & (comparisons['lineages'] == 'ppn-ptr')]['divergence'].max()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "344c6872-e174-44c7-8521-791140cbe839",
   "metadata": {},
   "source": [
    "Let's collect a random sample of windows at the lower thresholds to visualize map differences at those cutoffs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "3e5e51e3-10df-4b89-950d-fc24ccfaf94a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                      Bono\n",
       "ind2                      Koto\n",
       "lineages               ppn-ptr\n",
       "chr                      chr21\n",
       "window_start          24117248\n",
       "window          chr21_24117248\n",
       "mse                   0.000601\n",
       "spearman              0.998292\n",
       "divergence            0.001708\n",
       "seq_diff                  4174\n",
       "Name: 1892425, dtype: object"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr21_24117248') & (comparisons['lineages'] == 'ppn-ptr')].sort_values(by = 'divergence').iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "0795ae55-2c0e-4055-aae1-24526de41324",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>minimum</th>\n",
       "      <th>mean</th>\n",
       "      <th>maximum</th>\n",
       "      <th>variance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>chr21_24117248</td>\n",
       "      <td>0.001708</td>\n",
       "      <td>0.011038</td>\n",
       "      <td>0.034644</td>\n",
       "      <td>0.00003</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             window   minimum      mean   maximum  variance\n",
       "139  chr21_24117248  0.001708  0.011038  0.034644   0.00003"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_window_stats[ppn_ptr_window_stats['window'] == 'chr21_24117248']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "d8ffafdf-d138-4fa5-a226-bd5fe003d5de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                  Kidongo\n",
       "ind2                   Kosana\n",
       "lineages              ppn-ptr\n",
       "chr                      chr3\n",
       "window_start         88604672\n",
       "window          chr3_88604672\n",
       "mse                  0.000306\n",
       "spearman             0.994241\n",
       "divergence           0.005759\n",
       "seq_diff                 4150\n",
       "Name: 5272682, dtype: object"
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr3_88604672') & (comparisons['lineages'] == 'ppn-ptr')].sort_values(by = 'divergence').iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "8ab821bc-7c50-4cca-91d4-379500835341",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>minimum</th>\n",
       "      <th>mean</th>\n",
       "      <th>maximum</th>\n",
       "      <th>variance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>chr3_88604672</td>\n",
       "      <td>0.005759</td>\n",
       "      <td>0.01364</td>\n",
       "      <td>0.030025</td>\n",
       "      <td>0.000016</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window   minimum     mean   maximum  variance\n",
       "187  chr3_88604672  0.005759  0.01364  0.030025  0.000016"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_window_stats[ppn_ptr_window_stats['window'] == 'chr3_88604672']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "c1af5d0f-b73f-44d8-9aef-a062a8a6ffd3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                   Hermien\n",
       "ind2                     Linda\n",
       "lineages               ppn-ptr\n",
       "chr                       chr7\n",
       "window_start         137887744\n",
       "window          chr7_137887744\n",
       "mse                   0.003671\n",
       "spearman              0.991005\n",
       "divergence            0.008995\n",
       "seq_diff                  3731\n",
       "Name: 4570328, dtype: object"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr7_137887744') & (comparisons['lineages'] == 'ppn-ptr')].sort_values(by = 'divergence').iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "3e75e0fd-c583-45ae-bdd8-8f6c6e918e8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>minimum</th>\n",
       "      <th>mean</th>\n",
       "      <th>maximum</th>\n",
       "      <th>variance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>chr7_137887744</td>\n",
       "      <td>0.008995</td>\n",
       "      <td>0.015233</td>\n",
       "      <td>0.031131</td>\n",
       "      <td>0.000013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             window   minimum      mean   maximum  variance\n",
       "280  chr7_137887744  0.008995  0.015233  0.031131  0.000013"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_window_stats[ppn_ptr_window_stats['window'] == 'chr7_137887744']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "af4cb67d-682d-4836-9fdc-f0662227f3e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ind1                   Dzeeta\n",
       "ind2                     Luky\n",
       "lineages              ppn-ptr\n",
       "chr                      chr9\n",
       "window_start         62914560\n",
       "window          chr9_62914560\n",
       "mse                   0.00152\n",
       "spearman             0.988534\n",
       "divergence           0.011466\n",
       "seq_diff                 3190\n",
       "Name: 4158497, dtype: object"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr9_62914560') & (comparisons['lineages'] == 'ppn-ptr')].sort_values(by = 'divergence').iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "0543f060-ec7c-4946-ba1d-621cb496e949",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>minimum</th>\n",
       "      <th>mean</th>\n",
       "      <th>maximum</th>\n",
       "      <th>variance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>chr9_62914560</td>\n",
       "      <td>0.011466</td>\n",
       "      <td>0.019278</td>\n",
       "      <td>0.035595</td>\n",
       "      <td>0.000013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window   minimum      mean   maximum  variance\n",
       "304  chr9_62914560  0.011466  0.019278  0.035595  0.000013"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_window_stats[ppn_ptr_window_stats['window'] == 'chr9_62914560']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c684e7a-3845-4d51-ba80-fbc23670c341",
   "metadata": {},
   "source": [
    "Let's define \"bonobo-chimpanzee divergent windows\" as those with a minimum interspecific 3d divergence of 0.01."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "893ad321-e8fa-445b-976e-594cf0c476e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr10_78643200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr10_87556096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr11_22544384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr11_77070336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr11_77594624</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window\n",
       "9   chr10_78643200\n",
       "13  chr10_87556096\n",
       "20  chr11_22544384\n",
       "28  chr11_77070336\n",
       "29  chr11_77594624"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows = ppn_ptr_window_stats[ppn_ptr_window_stats['minimum'] >= 0.01]['window'].to_frame('window')\n",
    "ppn_ptr_divergent_windows.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79a2be02-735e-4215-ab38-16cd84b5791f",
   "metadata": {},
   "source": [
    "Create and saving a BED file of the divergent bonobo-chimpanzee windows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "5dee8ef5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153092096</td>\n",
       "      <td>154140672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153616384</td>\n",
       "      <td>154664960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>chr1</td>\n",
       "      <td>173015040</td>\n",
       "      <td>174063616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr10</td>\n",
       "      <td>78643200</td>\n",
       "      <td>79691776</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr  window_start  window_end\n",
       "115   chr1     126353408   127401984\n",
       "116   chr1     153092096   154140672\n",
       "117   chr1     153616384   154664960\n",
       "119   chr1     173015040   174063616\n",
       "9    chr10      78643200    79691776"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_BED = pd.DataFrame()\n",
    "ppn_ptr_divergent_windows_BED['window_split'] = ppn_ptr_divergent_windows['window']\n",
    "ppn_ptr_divergent_windows_BED = ppn_ptr_divergent_windows_BED['window_split'].str.split('_', expand=True)\n",
    "ppn_ptr_divergent_windows_BED.rename(columns = {0:'chr', 1:'window_start'}, inplace = True)\n",
    "ppn_ptr_divergent_windows_BED['window_start'] = ppn_ptr_divergent_windows_BED['window_start'].astype(int)\n",
    "ppn_ptr_divergent_windows_BED['window_end'] = ppn_ptr_divergent_windows_BED['window_start'] + 1048576\n",
    "ppn_ptr_divergent_windows_BED = ppn_ptr_divergent_windows_BED.sort_values(by = ['chr','window_start'])\n",
    "ppn_ptr_divergent_windows_BED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "898e5353",
   "metadata": {
    "editable": true,
    "slideshow": {
     "slide_type": ""
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_windows_BED.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_divergent_windows.bed', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7aa72342",
   "metadata": {},
   "source": [
    "How are these distributed among the chromosomes?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "e3661be6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr16</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr18</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr20</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr21</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2A</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2B</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr6</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr7</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        N\n",
       "chr      \n",
       "chr1    4\n",
       "chr10   2\n",
       "chr11   3\n",
       "chr12   2\n",
       "chr13   7\n",
       "chr14   4\n",
       "chr15   3\n",
       "chr16   1\n",
       "chr18   2\n",
       "chr19   3\n",
       "chr20   1\n",
       "chr21   2\n",
       "chr2A   3\n",
       "chr2B   4\n",
       "chr3   13\n",
       "chr4    8\n",
       "chr5    2\n",
       "chr6    6\n",
       "chr7    5\n",
       "chr8    3\n",
       "chr9    4\n",
       "chrX    7"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_BED.groupby(['chr']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "914a198a-d639-4e11-8564-e29a46058621",
   "metadata": {},
   "source": [
    "How many of these loci are unique (i.e., non-overlapping)? First, create a pybedtools object (we'll use this later) and then perform a merge."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "98bd5e09-a233-4dc8-a7e5-552c3c7c5ec9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t126353408\t127401984\n",
      " chr1\t153092096\t154140672\n",
      " chr1\t153616384\t154664960\n",
      " chr1\t173015040\t174063616\n",
      " chr10\t78643200\t79691776\n",
      " chr10\t87556096\t88604672\n",
      " chr11\t22544384\t23592960\n",
      " chr11\t77070336\t78118912\n",
      " chr11\t77594624\t78643200\n",
      " chr12\t37748736\t38797312\n",
      " "
     ]
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_pbtBED = pybedtools.BedTool().from_dataframe(ppn_ptr_divergent_windows_BED).sort()\n",
    "ppn_ptr_divergent_windows_pbtBED.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "558f27bd-9410-49bf-96bb-5128aa7dd3f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "71"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_windows_pbtBED.merge())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6eebf775-b91c-4fa7-95e6-3cb6c4edb2e9",
   "metadata": {},
   "source": [
    "### Genes Total and Genes per Window <a class = 'anchor' id = 'genestotalgenesperwindow'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b9c002ab",
   "metadata": {},
   "source": [
    "Let's intersect this object with genes in the panTro6 genome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "51a4d7de",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t344\t12299\tNM_001280424.1\tINTS11\n",
      " chr1\t12400\t16513\tXM_016952033.2\tCPTP\n",
      " chr1\t17887\t23426\tXM_003307748.4\tTAS1R3\n",
      " chr1\t22943\t36942\tXM_016958290.2\tDVL1\n",
      " chr1\t39619\t45507\tNM_001280245.1\tMXRA8\n",
      " "
     ]
    }
   ],
   "source": [
    "genes_pbtBED = pybedtools.BedTool('/wynton/group/capra/projects/pan_3d_genome/data/annotations/panTro6_genes.bed')\n",
    "genes_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "df2697f0-95aa-4e43-8fac-94427518c1ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>gene_chr</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>gene_transcript</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126517594</td>\n",
       "      <td>126522374</td>\n",
       "      <td>XM_009431515.3</td>\n",
       "      <td>MRPL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126524628</td>\n",
       "      <td>126529347</td>\n",
       "      <td>NM_001172571.2</td>\n",
       "      <td>OAZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126558327</td>\n",
       "      <td>126563506</td>\n",
       "      <td>XM_524871.3</td>\n",
       "      <td>LINGO4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126595864</td>\n",
       "      <td>126597165</td>\n",
       "      <td>XM_016927235.2</td>\n",
       "      <td>C2CD4D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126603960</td>\n",
       "      <td>126612050</td>\n",
       "      <td>XM_016927244.1</td>\n",
       "      <td>THEM5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  window_chr  window_start  window_end gene_chr  gene_start   gene_end  \\\n",
       "0       chr1     126353408   127401984     chr1   126517594  126522374   \n",
       "1       chr1     126353408   127401984     chr1   126524628  126529347   \n",
       "2       chr1     126353408   127401984     chr1   126558327  126563506   \n",
       "3       chr1     126353408   127401984     chr1   126595864  126597165   \n",
       "4       chr1     126353408   127401984     chr1   126603960  126612050   \n",
       "\n",
       "  gene_transcript    gene  \n",
       "0  XM_009431515.3   MRPL9  \n",
       "1  NM_001172571.2    OAZ3  \n",
       "2     XM_524871.3  LINGO4  \n",
       "3  XM_016927235.2  C2CD4D  \n",
       "4  XM_016927244.1   THEM5  "
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes_intersect = ppn_ptr_divergent_windows_pbtBED.intersect(genes_pbtBED, wa = True, wb = True).to_dataframe(names=['window_chr','window_start','window_end','gene_chr','gene_start','gene_end','gene_transcript','gene'])\n",
    "ppn_ptr_divergent_windows_genes_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "d1ceea91-2954-4e35-83ec-91aa18fa9120",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "454"
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_windows_genes_intersect)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "259aa11a-54f5-4950-8bbf-0c6a2d10f981",
   "metadata": {},
   "source": [
    "Let's get a list of genes in these windows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "c91afb02-9346-4f99-ad4c-35233cb1a271",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43         ABL2\n",
       "415         ABO\n",
       "436        ACE2\n",
       "346      ADAM22\n",
       "420    ADAMTS13\n",
       "Name: gene, dtype: object"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes = ppn_ptr_divergent_windows_genes_intersect['gene'].drop_duplicates().sort_values()\n",
    "ppn_ptr_divergent_windows_genes.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "c8b71361-0fa9-4a50-bd8c-d7f3f2ae86ec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "431"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_windows_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "59c55f06-9fe5-4802-a6e9-e4f990439fec",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_windows_genes_set = set(ppn_ptr_divergent_windows_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "8bf1d95a-8dee-4da9-bc76-763c4fc2e252",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "431"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_windows_genes_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c32c079-d4f0-497e-a5b5-30e623f1e53e",
   "metadata": {},
   "source": [
    "We will save this file to look for gene enrichment of particular phenotypes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "71658684-14ca-43a6-a1e6-59dd4d4c152d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_windows_genes.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/phenotype_enrichment/data/ppn_ptr_divergent_windows_genes.txt', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "123b3093-94ae-4653-b852-c23849adc54a",
   "metadata": {},
   "source": [
    "Let's run another intersection and count the number of genes in each divergent window."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "4ccf6b45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>gene_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153092096</td>\n",
       "      <td>154140672</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153616384</td>\n",
       "      <td>154664960</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>173015040</td>\n",
       "      <td>174063616</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>78643200</td>\n",
       "      <td>79691776</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  window_chr  window_start  window_end  gene_count\n",
       "0       chr1     126353408   127401984          28\n",
       "1       chr1     153092096   154140672           7\n",
       "2       chr1     153616384   154664960          10\n",
       "3       chr1     173015040   174063616           5\n",
       "4      chr10      78643200    79691776           1"
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes_count_intersect = ppn_ptr_divergent_windows_pbtBED.intersect(genes_pbtBED, c = True).to_dataframe(names=['window_chr','window_start','window_end','gene_count'])\n",
    "ppn_ptr_divergent_windows_genes_count_intersect.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e80db777-0461-4768-9757-572812df9e58",
   "metadata": {},
   "source": [
    "Save this dataframe for visualization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "a47a8a42-050b-425e-b911-1c4f17cd9a2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_windows_genes_count_intersect.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_divergent_windows_genes_counts.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "8f8b93aa-eec2-420d-b251-3414be658344",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes_count_intersect['gene_count'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "eeb0ffd3-a592-42ad-85ce-8037719704cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes_count_intersect['gene_count'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "36ea4786",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5.101123595505618"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes_count_intersect['gene_count'].mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5b385b5-6cd2-4abf-b906-ebc85dfc0c82",
   "metadata": {},
   "source": [
    "Let's take a look at the distribution of gene counts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "2d0db723-3bde-4f43-8612-de7d4337a1a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_count</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             N\n",
       "gene_count    \n",
       "0           17\n",
       "1           18\n",
       "2           13\n",
       "3            8\n",
       "4            4\n",
       "5            4\n",
       "6            2\n",
       "7            3\n",
       "8            4\n",
       "9            1\n",
       "10           2\n",
       "12           2\n",
       "13           2\n",
       "14           3\n",
       "17           1\n",
       "19           1\n",
       "28           1\n",
       "29           1\n",
       "35           1\n",
       "36           1"
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes_count_intersect.groupby(['gene_count']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72e955b3-14ea-4cdb-9abd-c8480ad85f57",
   "metadata": {},
   "source": [
    "We should compare these to the genome-wide background."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "482cf8ba-196a-4c7b-9dd3-a5fccdd274f2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>gene_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1572864</td>\n",
       "      <td>2621440</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2097152</td>\n",
       "      <td>3145728</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2621440</td>\n",
       "      <td>3670016</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3145728</td>\n",
       "      <td>4194304</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  window_chr  window_start  window_end  gene_count\n",
       "0       chr1       1048576     2097152          14\n",
       "1       chr1       1572864     2621440          13\n",
       "2       chr1       2097152     3145728          10\n",
       "3       chr1       2621440     3670016           1\n",
       "4       chr1       3145728     4194304           2"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_windows_genes_count_intersect = windows_pbtBED.intersect(genes_pbtBED, c = True).to_dataframe(names=['window_chr','window_start','window_end','gene_count'])\n",
    "all_windows_genes_count_intersect.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b4a25dfd-30f6-4e47-a47b-ba5707682e08",
   "metadata": {},
   "source": [
    "Save this dataframe for visualization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "14eeda7e-0e5e-4d2f-8196-b3e0bf10ffc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_windows_genes_count_intersect.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/all_windows_genes_count_intersect.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "6b761147-3c6a-4d1e-ab28-dcab02dec806",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_windows_genes_count_intersect['gene_count'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "6c84c031-e2e5-4d29-9014-f632b2213f2a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "67"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_windows_genes_count_intersect['gene_count'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "f4a59e46-b596-412e-9900-8a3d5b8219b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7.511764705882353"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_windows_genes_count_intersect['gene_count'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "f9e96499-f34d-407d-941e-3e2fe617995c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=252261.5, pvalue=4.590052657934087e-06)"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(all_windows_genes_count_intersect['gene_count'], ppn_ptr_divergent_windows_genes_count_intersect['gene_count'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4445c0f2-4b4a-4be1-96af-8b0227b02189",
   "metadata": {},
   "source": [
    "What about the overall sum of genes? Let's load in the randomization results where we sampled 89 random windows and summed the number of genes in all windows 10,000 times."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "52767a3b-bc43-4607-a32e-deb5f7e0656f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>629</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     N\n",
       "0  705\n",
       "1  685\n",
       "2  571\n",
       "3  745\n",
       "4  629"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_windows_n_genes_randomization = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_windows_n_genes_randomization.txt', sep = '\\t', names = ['N'])\n",
    "ppn_ptr_windows_n_genes_randomization.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "875151c4-a4fe-404c-af0c-22d0a8bbcbdd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10000"
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_windows_n_genes_randomization)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "932c4f79-3891-4116-88ce-3971ab83204f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "418"
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_windows_n_genes_randomization['N'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "fc008546-187b-4921-9ced-cc89721a06db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "967"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_windows_n_genes_randomization['N'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "dba77ace-efd2-49f1-afb7-5a00d052f743",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "659.6825"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_windows_n_genes_randomization['N'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "07d4b943-ef40-4b47-8152-3abcc40f1004",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_windows_n_genes_randomization[ppn_ptr_windows_n_genes_randomization['N'] <= 431])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "b36fd4a0-0cab-44e5-b0ef-2b3abc354a7f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0002"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "2/10000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "eacc1ab8-7692-4479-ada5-c7a86774731b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.6533446013802094"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "431/659.6825"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc8f5a7e-89fb-4b21-a53d-7cb5ccad744f",
   "metadata": {},
   "source": [
    "What if we rank the most divergent bonobo-chimpanzee windows? Is there an association between divergence score and number of genes?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "a7ef13de-b41a-44a7-b326-bb2bf26036a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "89"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_stats = ppn_ptr_window_stats[ppn_ptr_window_stats['window'].isin(ppn_ptr_divergent_windows['window'])]\n",
    "len(ppn_ptr_divergent_window_stats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "cdd3bf16-6209-46e6-ad96-27387eb6371a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>gene_count</th>\n",
       "      <th>window</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>28</td>\n",
       "      <td>chr1_126353408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153092096</td>\n",
       "      <td>154140672</td>\n",
       "      <td>7</td>\n",
       "      <td>chr1_153092096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153616384</td>\n",
       "      <td>154664960</td>\n",
       "      <td>10</td>\n",
       "      <td>chr1_153616384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>173015040</td>\n",
       "      <td>174063616</td>\n",
       "      <td>5</td>\n",
       "      <td>chr1_173015040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>78643200</td>\n",
       "      <td>79691776</td>\n",
       "      <td>1</td>\n",
       "      <td>chr10_78643200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  window_chr  window_start  window_end  gene_count          window\n",
       "0       chr1     126353408   127401984          28  chr1_126353408\n",
       "1       chr1     153092096   154140672           7  chr1_153092096\n",
       "2       chr1     153616384   154664960          10  chr1_153616384\n",
       "3       chr1     173015040   174063616           5  chr1_173015040\n",
       "4      chr10      78643200    79691776           1  chr10_78643200"
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes_count_intersect['window'] = ppn_ptr_divergent_windows_genes_count_intersect['window_chr'] + '_' + ppn_ptr_divergent_windows_genes_count_intersect['window_start'].astype(str)\n",
    "ppn_ptr_divergent_windows_genes_count_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "52e87837-bfea-4336-861e-8a51a8a39a56",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>minimum</th>\n",
       "      <th>mean</th>\n",
       "      <th>maximum</th>\n",
       "      <th>variance</th>\n",
       "      <th>gene_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_78643200</td>\n",
       "      <td>0.011702</td>\n",
       "      <td>0.031134</td>\n",
       "      <td>0.052345</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.033005</td>\n",
       "      <td>0.040717</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr11_22544384</td>\n",
       "      <td>0.014435</td>\n",
       "      <td>0.024658</td>\n",
       "      <td>0.050180</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.023994</td>\n",
       "      <td>0.037768</td>\n",
       "      <td>0.057039</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11_77594624</td>\n",
       "      <td>0.050357</td>\n",
       "      <td>0.076725</td>\n",
       "      <td>0.122029</td>\n",
       "      <td>0.000142</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           window   minimum      mean   maximum  variance  gene_count\n",
       "0  chr10_78643200  0.011702  0.031134  0.052345  0.000043           1\n",
       "1  chr10_87556096  0.026767  0.033005  0.040717  0.000005           8\n",
       "2  chr11_22544384  0.014435  0.024658  0.050180  0.000037           3\n",
       "3  chr11_77070336  0.023994  0.037768  0.057039  0.000029           0\n",
       "4  chr11_77594624  0.050357  0.076725  0.122029  0.000142           2"
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_stats_gene_count = pd.merge(ppn_ptr_divergent_window_stats, ppn_ptr_divergent_windows_genes_count_intersect[['window','gene_count']], on = 'window', how = 'inner')\n",
    "ppn_ptr_divergent_window_stats_gene_count.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "de27c3f0-30c4-46df-ab41-0b3c3f8284d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_stats_gene_count.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_divergent_window_stats_gene_count.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "9a113d4c-dd93-4fac-bcc9-eb00fe1370c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.3673811631053057 0.00039775579289958807\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(ppn_ptr_divergent_window_stats_gene_count['minimum'], ppn_ptr_divergent_window_stats_gene_count['gene_count'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "df009293-feb5-4130-8de7-3863b53f8b1f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.49078994548019744 1.0457737105949218e-06\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(ppn_ptr_divergent_window_stats_gene_count['mean'], ppn_ptr_divergent_window_stats_gene_count['gene_count'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "015e286c-9107-4ba0-a8e6-1c76b41c56f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.5060559479527063 4.2132088900800013e-07\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(ppn_ptr_divergent_window_stats_gene_count['maximum'], ppn_ptr_divergent_window_stats_gene_count['gene_count'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1bd054d5-f38c-44f4-b100-0b0fad47d209",
   "metadata": {},
   "source": [
    "### phastCons Elements <a class = 'anchor' id = 'phastconselements'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef26a4ca-73b4-48af-9a2d-b51c1ae417c9",
   "metadata": {},
   "source": [
    "Let's also consider conservation in these windows by the proportion of phastCons conserved elements in bonobo-chimpanzee divergent windows. Run a script to download and liftOver the conserved elements from hg38 to panTro6 and filter for elements with LOD $\\geq$ 500."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "d564b595-5bce-4613-8db9-927c033a0d96",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t613\t772\t552\n",
      " chr1\t6190\t6427\t612\n",
      " chr1\t36658\t36857\t586\n",
      " chr1\t76342\t76520\t529\n",
      " chr1\t80960\t81083\t500\n",
      " "
     ]
    }
   ],
   "source": [
    "phastCons_elements_pbtBED = pybedtools.BedTool('/wynton/group/capra/projects/pan_3d_genome/data/phastCons/phastConsElements30way_panTro6.bed')\n",
    "phastCons_elements_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "334387b8-19f8-4e6c-848d-ce4a0dfe3925",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "235405"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(phastCons_elements_pbtBED)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "b79271e4-4604-40a2-9fec-5e9b0f6f261c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>element_chr</th>\n",
       "      <th>element_start</th>\n",
       "      <th>element_end</th>\n",
       "      <th>element_score</th>\n",
       "      <th>bp_overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126367862</td>\n",
       "      <td>126368248</td>\n",
       "      <td>668</td>\n",
       "      <td>386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126371694</td>\n",
       "      <td>126371855</td>\n",
       "      <td>508</td>\n",
       "      <td>161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126394894</td>\n",
       "      <td>126395348</td>\n",
       "      <td>696</td>\n",
       "      <td>454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126411938</td>\n",
       "      <td>126412184</td>\n",
       "      <td>606</td>\n",
       "      <td>246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "      <td>chr1</td>\n",
       "      <td>126414434</td>\n",
       "      <td>126414614</td>\n",
       "      <td>515</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  window_chr  window_start  window_end element_chr  element_start  \\\n",
       "0       chr1     126353408   127401984        chr1      126367862   \n",
       "1       chr1     126353408   127401984        chr1      126371694   \n",
       "2       chr1     126353408   127401984        chr1      126394894   \n",
       "3       chr1     126353408   127401984        chr1      126411938   \n",
       "4       chr1     126353408   127401984        chr1      126414434   \n",
       "\n",
       "   element_end  element_score  bp_overlap  \n",
       "0    126368248            668         386  \n",
       "1    126371855            508         161  \n",
       "2    126395348            696         454  \n",
       "3    126412184            606         246  \n",
       "4    126414614            515         180  "
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phastCons_elements_ppn_ptr_divergent_windows_intersect = ppn_ptr_divergent_windows_pbtBED.intersect(phastCons_elements_pbtBED, wao = True).to_dataframe(names=['window_chr','window_start','window_end','element_chr','element_start','element_end','element_score','bp_overlap']) \n",
    "phastCons_elements_ppn_ptr_divergent_windows_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "39ea7073-7a96-41db-9462-b716e99b12b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>bp_overlap</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_78643200</td>\n",
       "      <td>4816</td>\n",
       "      <td>0.004593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>34393</td>\n",
       "      <td>0.032800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr11_22544384</td>\n",
       "      <td>484</td>\n",
       "      <td>0.000462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>6701</td>\n",
       "      <td>0.006391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11_77594624</td>\n",
       "      <td>12147</td>\n",
       "      <td>0.011584</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           window  bp_overlap      prop\n",
       "0  chr10_78643200        4816  0.004593\n",
       "1  chr10_87556096       34393  0.032800\n",
       "2  chr11_22544384         484  0.000462\n",
       "3  chr11_77070336        6701  0.006391\n",
       "4  chr11_77594624       12147  0.011584"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phastCons_elements_ppn_ptr_divergent_windows = phastCons_elements_ppn_ptr_divergent_windows_intersect.copy()\n",
    "phastCons_elements_ppn_ptr_divergent_windows['window'] = phastCons_elements_ppn_ptr_divergent_windows['window_chr'] + '_' + phastCons_elements_ppn_ptr_divergent_windows['window_start'].astype(str)\n",
    "phastCons_elements_ppn_ptr_divergent_windows = phastCons_elements_ppn_ptr_divergent_windows.drop(columns = ['window_chr','window_start','window_end','element_chr','element_start','element_end','element_score'])\n",
    "phastCons_elements_ppn_ptr_divergent_windows = phastCons_elements_ppn_ptr_divergent_windows[['window','bp_overlap']]\n",
    "phastCons_elements_ppn_ptr_divergent_windows = phastCons_elements_ppn_ptr_divergent_windows.groupby(['window']).sum('bp_overlap').reset_index().sort_values(['window'], ascending = True)\n",
    "phastCons_elements_ppn_ptr_divergent_windows['prop'] = phastCons_elements_ppn_ptr_divergent_windows['bp_overlap']/1048576\n",
    "phastCons_elements_ppn_ptr_divergent_windows.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dea0cef7-955e-4e54-8910-6d6490835d11",
   "metadata": {},
   "source": [
    "Let's compare this to the genome-wide background."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "14997f52-9f8c-429e-87cb-79895993042b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/wynton/home/capra/cbrand/miniconda3/envs/jupyter/lib/python3.10/site-packages/pybedtools/bedtool.py:3714: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  return pandas.read_csv(self.fn, *args, sep=\"\\t\", **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window_chr</th>\n",
       "      <th>window_start</th>\n",
       "      <th>window_end</th>\n",
       "      <th>element_chr</th>\n",
       "      <th>element_start</th>\n",
       "      <th>element_end</th>\n",
       "      <th>element_score</th>\n",
       "      <th>bp_overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1086782</td>\n",
       "      <td>1086949</td>\n",
       "      <td>545</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1087296</td>\n",
       "      <td>1087406</td>\n",
       "      <td>534</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1185992</td>\n",
       "      <td>1186177</td>\n",
       "      <td>528</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1200100</td>\n",
       "      <td>1200249</td>\n",
       "      <td>570</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1048576</td>\n",
       "      <td>2097152</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1200111</td>\n",
       "      <td>1200248</td>\n",
       "      <td>585</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  window_chr  window_start  window_end element_chr  element_start  \\\n",
       "0       chr1       1048576     2097152        chr1        1086782   \n",
       "1       chr1       1048576     2097152        chr1        1087296   \n",
       "2       chr1       1048576     2097152        chr1        1185992   \n",
       "3       chr1       1048576     2097152        chr1        1200100   \n",
       "4       chr1       1048576     2097152        chr1        1200111   \n",
       "\n",
       "   element_end element_score  bp_overlap  \n",
       "0      1086949           545         167  \n",
       "1      1087406           534         110  \n",
       "2      1186177           528         185  \n",
       "3      1200249           570         149  \n",
       "4      1200248           585         137  "
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phastCons_elements_all_windows_intersect = windows_pbtBED.intersect(phastCons_elements_pbtBED, wao = True).to_dataframe(names=['window_chr','window_start','window_end','element_chr','element_start','element_end','element_score','bp_overlap']) \n",
    "phastCons_elements_all_windows_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "c111b332-659a-4ce0-b583-eb56ab882a74",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>bp_overlap</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100139008</td>\n",
       "      <td>26802</td>\n",
       "      <td>0.025560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_100663296</td>\n",
       "      <td>19669</td>\n",
       "      <td>0.018758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_101187584</td>\n",
       "      <td>11165</td>\n",
       "      <td>0.010648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_101711872</td>\n",
       "      <td>16447</td>\n",
       "      <td>0.015685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_102236160</td>\n",
       "      <td>16215</td>\n",
       "      <td>0.015464</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window  bp_overlap      prop\n",
       "0  chr10_100139008       26802  0.025560\n",
       "1  chr10_100663296       19669  0.018758\n",
       "2  chr10_101187584       11165  0.010648\n",
       "3  chr10_101711872       16447  0.015685\n",
       "4  chr10_102236160       16215  0.015464"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phastCons_elements_all_windows = phastCons_elements_all_windows_intersect.copy()\n",
    "phastCons_elements_all_windows['window'] = phastCons_elements_all_windows['window_chr'] + '_' + phastCons_elements_all_windows['window_start'].astype(str)\n",
    "phastCons_elements_all_windows = phastCons_elements_all_windows.drop(columns = ['window_chr','window_start','window_end','element_chr','element_start','element_end','element_score'])\n",
    "phastCons_elements_all_windows = phastCons_elements_all_windows[['window','bp_overlap']]\n",
    "phastCons_elements_all_windows = phastCons_elements_all_windows.groupby(['window']).sum('bp_overlap').reset_index().sort_values(['window'], ascending = True)\n",
    "phastCons_elements_all_windows['prop'] = phastCons_elements_all_windows['bp_overlap']/1048576\n",
    "phastCons_elements_all_windows.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "ab771020-a005-40ec-8eac-710ccca3d48b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4420"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(phastCons_elements_all_windows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "561340fc-2895-491d-b828-c2a67b84ce80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.020641775044920218"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phastCons_elements_all_windows['prop'].mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ee1790b-5d85-419e-a0ea-c33cbc105fa9",
   "metadata": {},
   "source": [
    "How many bonobo-chimpanzee divergent windows are below and above this genome-wide mean?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "f58fda91-c8bc-4a3e-971e-37c88938ff39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "66"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(phastCons_elements_ppn_ptr_divergent_windows[phastCons_elements_ppn_ptr_divergent_windows['prop'] < 0.020641775044920218])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "796c8771-5aa3-4628-bb97-19b1af34e006",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23"
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(phastCons_elements_ppn_ptr_divergent_windows[phastCons_elements_ppn_ptr_divergent_windows['prop'] > 0.020641775044920218])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44e1d24f-43a9-46ee-b7c0-d6166aa15e40",
   "metadata": {},
   "source": [
    "Let's map on maximum divergence per window and examine the relationship between proportion of conserved elements and divergence."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "900a9d68-e830-47c9-abba-6bb92327208f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>bp_overlap</th>\n",
       "      <th>prop</th>\n",
       "      <th>maximum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_78643200</td>\n",
       "      <td>4816</td>\n",
       "      <td>0.004593</td>\n",
       "      <td>0.052345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>34393</td>\n",
       "      <td>0.032800</td>\n",
       "      <td>0.040717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr11_22544384</td>\n",
       "      <td>484</td>\n",
       "      <td>0.000462</td>\n",
       "      <td>0.050180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>6701</td>\n",
       "      <td>0.006391</td>\n",
       "      <td>0.057039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11_77594624</td>\n",
       "      <td>12147</td>\n",
       "      <td>0.011584</td>\n",
       "      <td>0.122029</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           window  bp_overlap      prop   maximum\n",
       "0  chr10_78643200        4816  0.004593  0.052345\n",
       "1  chr10_87556096       34393  0.032800  0.040717\n",
       "2  chr11_22544384         484  0.000462  0.050180\n",
       "3  chr11_77070336        6701  0.006391  0.057039\n",
       "4  chr11_77594624       12147  0.011584  0.122029"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_maxes_dict = dict(zip(ppn_ptr_window_stats['window'], ppn_ptr_window_stats['maximum']))\n",
    "phastCons_elements_ppn_ptr_divergent_windows['maximum'] = phastCons_elements_ppn_ptr_divergent_windows['window'].map(ppn_ptr_divergent_window_maxes_dict)\n",
    "phastCons_elements_ppn_ptr_divergent_windows.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "84f3f1eb-80fc-4a41-9eea-4a69b645e549",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "89"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(phastCons_elements_ppn_ptr_divergent_windows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "29c79eaf-bff7-4f93-9679-1396d6433423",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.1821416411304052 0.08757112628834791\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(phastCons_elements_ppn_ptr_divergent_windows['maximum'], phastCons_elements_ppn_ptr_divergent_windows['prop'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15cee602-586c-4535-86ac-b1c2f825da4e",
   "metadata": {},
   "source": [
    "Peak at the windows with the highest divergence."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "40f7df52-5975-455c-af04-751db7e4a61c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>window</th>\n",
       "      <th>bp_overlap</th>\n",
       "      <th>prop</th>\n",
       "      <th>maximum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr13_52428800</td>\n",
       "      <td>89429</td>\n",
       "      <td>0.085286</td>\n",
       "      <td>0.248103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr19_51380224</td>\n",
       "      <td>72334</td>\n",
       "      <td>0.068983</td>\n",
       "      <td>0.103107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr2B_42991616</td>\n",
       "      <td>54387</td>\n",
       "      <td>0.051867</td>\n",
       "      <td>0.020830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr3_144703488</td>\n",
       "      <td>50137</td>\n",
       "      <td>0.047814</td>\n",
       "      <td>0.112864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>chr8_113770496</td>\n",
       "      <td>39926</td>\n",
       "      <td>0.038076</td>\n",
       "      <td>0.159110</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            window  bp_overlap      prop   maximum\n",
       "12  chr13_52428800       89429  0.085286  0.248103\n",
       "26  chr19_51380224       72334  0.068983  0.103107\n",
       "39  chr2B_42991616       54387  0.051867  0.020830\n",
       "46  chr3_144703488       50137  0.047814  0.112864\n",
       "75  chr8_113770496       39926  0.038076  0.159110"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phastCons_elements_ppn_ptr_divergent_windows.sort_values('prop', ascending = False).head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "7fb32735-77eb-4e7f-87f6-88e779c48534",
   "metadata": {},
   "outputs": [],
   "source": [
    "phastCons_elements_ppn_ptr_divergent_windows.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/phastCons/phastCons_elements_ppn_ptr_divergent_windows.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24542620-b23a-4dd5-8331-18fdb6b197ef",
   "metadata": {},
   "source": [
    "## Phenotype Enrichment <a class = 'anchor' id = 'phenotypeenrichment'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8be8149-0bcc-4763-8938-675ebb407040",
   "metadata": {},
   "source": [
    "Let's examine the results of our phenotype enrichment pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "6caa75ae-6df7-4ff1-913d-0347a0ba04cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "fdr_table = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "791d3d0b-323f-4bc2-bb70-4b255abcfa8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def reportFDRcorrectedPthreshold(set_name, ontology, q_value_threshold, resolution=0.0001, minStart=0):\n",
    "    fdr_empiric = pd.read_csv(f'/wynton/group/capra/projects/pan_3d_genome/data/phenotype_enrichment/empiric_FDR/{set_name}_{ontology}_empiric_FDR.txt', sep = '\\t', header = None, index_col = 0)\n",
    "    obs = pd.read_csv(f'/wynton/group/capra/projects/pan_3d_genome/data/phenotype_enrichment/enrichment/{set_name}_{ontology}_enrichment.txt', sep = '\\t')\n",
    "\n",
    "    fdr_threshold = []\n",
    "    for i in np.arange(minStart,0.05,resolution):\n",
    "        \n",
    "        observed_positive = sum(obs['p_value'] <= i)\n",
    "        average_false_positive = (fdr_empiric <= i).sum().mean()\n",
    "        q = average_false_positive/observed_positive\n",
    "        fdr_threshold.append([set_name, ontology, q_value_threshold, i, observed_positive, average_false_positive, q])\n",
    "        \n",
    "        if (q != np.inf) & (q > q_value_threshold):\n",
    "            break\n",
    "    \n",
    "    threshold = fdr_threshold[-2]\n",
    "    fdr_table.append(threshold)\n",
    "    #fdr_threshold = pd.DataFrame(fdr_threshold, columns = ['pval_threshold','obsPos','avgFalsePos','q'])\n",
    "    #return fdr_threshold.tail(2).head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "bd328521-ed7f-4b0e-9960-c2fbcd80c20f",
   "metadata": {},
   "outputs": [],
   "source": [
    "combinations = [(set_name,ontology,q_value_threshold) for set_name in ['ppn_ptr_clustering'] for ontology in ['BP','GWAS','HPO','MP'] for q_value_threshold in [0.05,0.1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "5a4761f1-22e1-44ba-aebb-881771e36ac6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_2007303/1065567570.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[None, None, None, None, None, None, None, None]"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[reportFDRcorrectedPthreshold(set_name, ontology, q_value_threshold) for set_name, ontology, q_value_threshold in combinations]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "72652f58-742e-46d7-8cbb-cd2d3907df27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set</th>\n",
       "      <th>ontology</th>\n",
       "      <th>q_value_threshold</th>\n",
       "      <th>p_value_threshold</th>\n",
       "      <th>observed_positive</th>\n",
       "      <th>average_false_positive</th>\n",
       "      <th>q</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ppn_ptr_clustering</td>\n",
       "      <td>BP</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0078</td>\n",
       "      <td>0</td>\n",
       "      <td>6.4675</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ppn_ptr_clustering</td>\n",
       "      <td>BP</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0078</td>\n",
       "      <td>0</td>\n",
       "      <td>6.4675</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ppn_ptr_clustering</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0077</td>\n",
       "      <td>0</td>\n",
       "      <td>1.7036</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ppn_ptr_clustering</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0077</td>\n",
       "      <td>0</td>\n",
       "      <td>1.7036</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ppn_ptr_clustering</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0094</td>\n",
       "      <td>0</td>\n",
       "      <td>2.4849</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ppn_ptr_clustering</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0094</td>\n",
       "      <td>0</td>\n",
       "      <td>2.4849</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ppn_ptr_clustering</td>\n",
       "      <td>MP</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>0</td>\n",
       "      <td>1.1145</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ppn_ptr_clustering</td>\n",
       "      <td>MP</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>0</td>\n",
       "      <td>1.1145</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  set ontology  q_value_threshold  p_value_threshold  \\\n",
       "0  ppn_ptr_clustering       BP               0.05             0.0078   \n",
       "1  ppn_ptr_clustering       BP               0.10             0.0078   \n",
       "2  ppn_ptr_clustering     GWAS               0.05             0.0077   \n",
       "3  ppn_ptr_clustering     GWAS               0.10             0.0077   \n",
       "4  ppn_ptr_clustering      HPO               0.05             0.0094   \n",
       "5  ppn_ptr_clustering      HPO               0.10             0.0094   \n",
       "6  ppn_ptr_clustering       MP               0.05             0.0019   \n",
       "7  ppn_ptr_clustering       MP               0.10             0.0019   \n",
       "\n",
       "   observed_positive  average_false_positive    q  \n",
       "0                  0                  6.4675  inf  \n",
       "1                  0                  6.4675  inf  \n",
       "2                  0                  1.7036  inf  \n",
       "3                  0                  1.7036  inf  \n",
       "4                  0                  2.4849  inf  \n",
       "5                  0                  2.4849  inf  \n",
       "6                  0                  1.1145  inf  \n",
       "7                  0                  1.1145  inf  "
      ]
     },
     "execution_count": 237,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fdr_table = pd.DataFrame(fdr_table, columns = ['set', 'ontology', 'q_value_threshold', 'p_value_threshold','observed_positive','average_false_positive','q'])\n",
    "fdr_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28085115",
   "metadata": {},
   "source": [
    "### *In Silico* Mutagenesis for Bonobo-Specific Variants <a class = 'anchor' id = 'ppninsilicomutagenesis'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1cda3978",
   "metadata": {},
   "source": [
    "What individual variants are driving divergence in bonobo-chimpanzee divergent windows? Let's examine the results of *in silico* mutagenesis for bonobo-specific variants."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "283cf2e2",
   "metadata": {
    "editable": true,
    "scrolled": true,
    "slideshow": {
     "slide_type": ""
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353856</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>6.057105e-09</td>\n",
       "      <td>2.141901e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126354684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>5.636190e-10</td>\n",
       "      <td>2.657346e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126357977</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>1.889927e-09</td>\n",
       "      <td>6.916147e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126358737</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>1.360100e-09</td>\n",
       "      <td>5.963432e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126361540</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>4.040863e-10</td>\n",
       "      <td>1.929300e-09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr        pos ref alt          window           mse    divergence\n",
       "0  chr1  126353856   C   T  chr1_126353408  6.057105e-09  2.141901e-08\n",
       "1  chr1  126354684   C   T  chr1_126353408  5.636190e-10  2.657346e-09\n",
       "2  chr1  126357977   T   C  chr1_126353408  1.889927e-09  6.916147e-09\n",
       "3  chr1  126358737   C   T  chr1_126353408  1.360100e-09  5.963432e-09\n",
       "4  chr1  126361540   G   T  chr1_126353408  4.040863e-10  1.929300e-09"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_variants = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/in_silico_mutagenesis/ppn_specific_variants_in_ppn_ptr_divergent_windows_in_silico_mutagenesis.txt', sep = '\\t', header = 0)\n",
    "ppn_ptr_divergent_window_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "4d4aaff4-2c14-4660-95e4-6b0c8ef2f7b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "115191"
      ]
     },
     "execution_count": 239,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_variants[['chr','pos','ref','alt']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "7f46059a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "127075"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_variants)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b03afc6a",
   "metadata": {},
   "source": [
    "How many windows are represented by these variants?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "a064ee5e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "89"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_variants['window'].unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1d31f687",
   "metadata": {},
   "source": [
    "To identify 3d-modifying variants, let's map the lowest bonobo-chimpanzee divergence score to the above dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "cad84e83",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_mins_dict = dict(zip(ppn_ptr_window_stats['window'], ppn_ptr_window_stats['minimum']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "id": "73e65c8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_variants['min'] = ppn_ptr_divergent_window_variants['window'].map(ppn_ptr_divergent_window_mins_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "id": "7d619b5f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353856</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>6.057105e-09</td>\n",
       "      <td>2.141901e-08</td>\n",
       "      <td>0.029323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126354684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>5.636190e-10</td>\n",
       "      <td>2.657346e-09</td>\n",
       "      <td>0.029323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126357977</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>1.889927e-09</td>\n",
       "      <td>6.916147e-09</td>\n",
       "      <td>0.029323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126358737</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>1.360100e-09</td>\n",
       "      <td>5.963432e-09</td>\n",
       "      <td>0.029323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126361540</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>4.040863e-10</td>\n",
       "      <td>1.929300e-09</td>\n",
       "      <td>0.029323</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr        pos ref alt          window           mse    divergence  \\\n",
       "0  chr1  126353856   C   T  chr1_126353408  6.057105e-09  2.141901e-08   \n",
       "1  chr1  126354684   C   T  chr1_126353408  5.636190e-10  2.657346e-09   \n",
       "2  chr1  126357977   T   C  chr1_126353408  1.889927e-09  6.916147e-09   \n",
       "3  chr1  126358737   C   T  chr1_126353408  1.360100e-09  5.963432e-09   \n",
       "4  chr1  126361540   G   T  chr1_126353408  4.040863e-10  1.929300e-09   \n",
       "\n",
       "        min  \n",
       "0  0.029323  \n",
       "1  0.029323  \n",
       "2  0.029323  \n",
       "3  0.029323  \n",
       "4  0.029323  "
      ]
     },
     "execution_count": 244,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_variants.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d727b294-99ec-4635-acce-fdc687dab044",
   "metadata": {},
   "source": [
    "Now let's identify variants with a measured effect $\\geq$ than the lowest divergence score among bonobo-chimpanzee pairwise comparisons for that window."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "e17df452",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>0.011125</td>\n",
       "      <td>0.036647</td>\n",
       "      <td>0.029323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.017142</td>\n",
       "      <td>0.016506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7964</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.018322</td>\n",
       "      <td>0.031364</td>\n",
       "      <td>0.026767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7965</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.019119</td>\n",
       "      <td>0.032893</td>\n",
       "      <td>0.026767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.002093</td>\n",
       "      <td>0.025199</td>\n",
       "      <td>0.023994</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr        pos ref alt          window       mse  divergence  \\\n",
       "651     chr1  126940560   C   G  chr1_126353408  0.011125    0.036647   \n",
       "2146    chr1  153901388   A   T  chr1_153616384  0.005189    0.017142   \n",
       "7964   chr10   88057593   G   C  chr10_87556096  0.018322    0.031364   \n",
       "7965   chr10   88057595   G   T  chr10_87556096  0.019119    0.032893   \n",
       "12773  chr11   77997697   G   A  chr11_77070336  0.002093    0.025199   \n",
       "\n",
       "            min  \n",
       "651    0.029323  \n",
       "2146   0.016506  \n",
       "7964   0.026767  \n",
       "7965   0.026767  \n",
       "12773  0.023994  "
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants = ppn_ptr_divergent_window_variants[ppn_ptr_divergent_window_variants['divergence'] >= ppn_ptr_divergent_window_variants['min']]\n",
    "ppn_ptr_divergent_window_3d_modifying_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "id": "dd2310e3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "61"
      ]
     },
     "execution_count": 246,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "id": "c8c96c6c-1e74-44d2-85a9-8a8b4210bf25",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51"
      ]
     },
     "execution_count": 247,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants[['chr','pos','ref','alt']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68f4ac55-a5e7-49a5-9b42-3aabfc2a3634",
   "metadata": {},
   "source": [
    "Let's save a version of this dataframe now. We will create another with additional information later."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "id": "d8424982-284e-4267-b27a-53dc197bfaf8",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_divergent_window_3d_modifying_variants.txt', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "021b0a02-7074-4e2f-8f4d-4f57cebb3421",
   "metadata": {},
   "source": [
    "The presence of duplicates indicates the presence of following scenarios: 1) multiple 3d-modifying variants per window or 2) a 3d-modifying variant impacting adjacent windows. Let's assess."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "id": "7d9d7cf3-615d-4e9d-a6d1-86c527031c02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.002093</td>\n",
       "      <td>0.025199</td>\n",
       "      <td>0.023994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12774</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr11_77594624</td>\n",
       "      <td>0.007758</td>\n",
       "      <td>0.069116</td>\n",
       "      <td>0.050357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30827</th>\n",
       "      <td>chr14</td>\n",
       "      <td>45793476</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr14_45088768</td>\n",
       "      <td>0.011070</td>\n",
       "      <td>0.030854</td>\n",
       "      <td>0.020095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30828</th>\n",
       "      <td>chr14</td>\n",
       "      <td>45793476</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr14_45613056</td>\n",
       "      <td>0.007928</td>\n",
       "      <td>0.033563</td>\n",
       "      <td>0.019686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44464</th>\n",
       "      <td>chr19</td>\n",
       "      <td>26559939</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>chr19_25690112</td>\n",
       "      <td>0.006024</td>\n",
       "      <td>0.039807</td>\n",
       "      <td>0.029334</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr       pos ref alt          window       mse  divergence       min\n",
       "12773  chr11  77997697   G   A  chr11_77070336  0.002093    0.025199  0.023994\n",
       "12774  chr11  77997697   G   A  chr11_77594624  0.007758    0.069116  0.050357\n",
       "30827  chr14  45793476   G   T  chr14_45088768  0.011070    0.030854  0.020095\n",
       "30828  chr14  45793476   G   T  chr14_45613056  0.007928    0.033563  0.019686\n",
       "44464  chr19  26559939   A   T  chr19_25690112  0.006024    0.039807  0.029334"
      ]
     },
     "execution_count": 249,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multi_window_ppn_ptr_divergent_window_3d_modifying_variants = ppn_ptr_divergent_window_3d_modifying_variants[ppn_ptr_divergent_window_3d_modifying_variants.duplicated(['chr','pos'], keep = False)]\n",
    "multi_window_ppn_ptr_divergent_window_3d_modifying_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "id": "cf14f8b1-a542-42bc-8a43-4c1cec6ece5f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10.0"
      ]
     },
     "execution_count": 250,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(multi_window_ppn_ptr_divergent_window_3d_modifying_variants)/2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc0dea23-3daa-43a5-88c8-51c1b6b1b374",
   "metadata": {},
   "source": [
    "10 windows have a 3d-modifying variant that impacts both windows. What about windows with multiple variants?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "id": "be99763e-edd7-4cc1-9116-d05cebbb5108",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>window</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr10_87556096</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4_113770496</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                N\n",
       "window           \n",
       "chr10_87556096  2\n",
       "chr4_113770496  2"
      ]
     },
     "execution_count": 251,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_n_variants_per_window = ppn_ptr_divergent_window_3d_modifying_variants.groupby(['window']).size().to_frame('N')\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_n_variants_per_window[ppn_ptr_divergent_window_3d_modifying_variants_n_variants_per_window['N'] > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "id": "1021940d-df94-498a-8215-ef25111a8cc0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7964</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.018322</td>\n",
       "      <td>0.031364</td>\n",
       "      <td>0.026767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7965</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.019119</td>\n",
       "      <td>0.032893</td>\n",
       "      <td>0.026767</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr       pos ref alt          window       mse  divergence       min\n",
       "7964  chr10  88057593   G   C  chr10_87556096  0.018322    0.031364  0.026767\n",
       "7965  chr10  88057595   G   T  chr10_87556096  0.019119    0.032893  0.026767"
      ]
     },
     "execution_count": 252,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants[ppn_ptr_divergent_window_3d_modifying_variants['window'] == 'chr10_87556096']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "id": "eebf062d-1cff-443f-a6e0-dac1cff70a6a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88832</th>\n",
       "      <td>chr4</td>\n",
       "      <td>114115871</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr4_113770496</td>\n",
       "      <td>0.000826</td>\n",
       "      <td>0.038260</td>\n",
       "      <td>0.029619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88833</th>\n",
       "      <td>chr4</td>\n",
       "      <td>114115875</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr4_113770496</td>\n",
       "      <td>0.000969</td>\n",
       "      <td>0.045659</td>\n",
       "      <td>0.029619</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr        pos ref alt          window       mse  divergence       min\n",
       "88832  chr4  114115871   G   A  chr4_113770496  0.000826    0.038260  0.029619\n",
       "88833  chr4  114115875   C   T  chr4_113770496  0.000969    0.045659  0.029619"
      ]
     },
     "execution_count": 253,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants[ppn_ptr_divergent_window_3d_modifying_variants['window'] == 'chr4_113770496']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec7a9e70",
   "metadata": {},
   "source": [
    "How many windows are covered by 3d-modifying variants?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "9e54d5fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "59"
      ]
     },
     "execution_count": 254,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants['window'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "id": "a359205d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.6629213483146067"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "59/89"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "id": "02681a04-d397-425a-ac96-9a56a2ebc7bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.036591365416"
      ]
     },
     "execution_count": 256,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr7_83886080') & (comparisons['lineages'] == 'ppn-ptr')]['divergence'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "id": "e55ae880-03c8-4ddf-a854-a04c08454b60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0699732955899999"
      ]
     },
     "execution_count": 257,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comparisons[(comparisons['window'] == 'chr7_83886080') & (comparisons['lineages'] == 'ppn-ptr')]['divergence'].max()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3f84351-9688-4b4a-9f14-6e0dbddb096a",
   "metadata": {},
   "source": [
    "Let's create a quick BED file of these windows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "92d04cd7-aa81-48ea-b681-3a70214f7ee9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126353408</td>\n",
       "      <td>127401984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153616384</td>\n",
       "      <td>154664960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7964</th>\n",
       "      <td>chr10</td>\n",
       "      <td>87556096</td>\n",
       "      <td>88604672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77070336</td>\n",
       "      <td>78118912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12774</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77594624</td>\n",
       "      <td>78643200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr      start        end\n",
       "651     chr1  126353408  127401984\n",
       "2146    chr1  153616384  154664960\n",
       "7964   chr10   87556096   88604672\n",
       "12773  chr11   77070336   78118912\n",
       "12774  chr11   77594624   78643200"
      ]
     },
     "execution_count": 258,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_BED = ppn_ptr_divergent_window_3d_modifying_variants['window'].str.split('_', expand = True).rename(columns = {0:'chr', 1:'start'})\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED = ppn_ptr_divergent_window_3d_modifying_variants_BED.drop_duplicates()\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED['start'] = ppn_ptr_divergent_window_3d_modifying_variants_BED['start'].astype(int)\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED['end'] = ppn_ptr_divergent_window_3d_modifying_variants_BED['start'] + 1048576\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "2038226d-2c58-4608-bca5-8abda73c3b97",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "59"
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants_BED)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d5caad3",
   "metadata": {},
   "source": [
    "Are there differences in the mutation frequency of these variants?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "2394fe12",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_unique_variants = ppn_ptr_divergent_window_3d_modifying_variants[['chr','pos','ref','alt']].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "14817146-79e0-48c0-9baf-8f8ff866c581",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>N</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>0.098039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3</td>\n",
       "      <td>0.058824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.039216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0.019608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>3</td>\n",
       "      <td>0.058824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>14</td>\n",
       "      <td>0.274510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>8</td>\n",
       "      <td>0.156863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>0.098039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>0.078431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>0.098039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.019608</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ref alt   N      prop\n",
       "0    A   C   5  0.098039\n",
       "1    A   G   3  0.058824\n",
       "2    A   T   2  0.039216\n",
       "3    C   A   1  0.019608\n",
       "4    C   G   3  0.058824\n",
       "5    C   T  14  0.274510\n",
       "6    G   A   8  0.156863\n",
       "7    G   C   5  0.098039\n",
       "8    G   T   4  0.078431\n",
       "9    T   C   5  0.098039\n",
       "10   T   G   1  0.019608"
      ]
     },
     "execution_count": 261,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_unique_variant_counts = ppn_ptr_divergent_window_3d_modifying_unique_variants.groupby(['ref','alt']).size().to_frame('N').reset_index()\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variant_counts['prop'] = ppn_ptr_divergent_window_3d_modifying_unique_variant_counts['N']/ppn_ptr_divergent_window_3d_modifying_unique_variant_counts['N'].sum()\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variant_counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "58b9e4cb-3384-49b5-8816-eadd5e8bc994",
   "metadata": {},
   "source": [
    "Let's manually add the missing rows to this dataframe for visualization and save."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "6751d2d4-a5a2-4fef-8a42-d480fd1551e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_rows = [{'ref': 'A', 'alt': 'A', 'N': np.nan, 'prop': np.nan},\n",
    "            {'ref': 'C', 'alt': 'C', 'N': np.nan, 'prop': np.nan},\n",
    "            {'ref': 'G', 'alt': 'G', 'N': np.nan, 'prop': np.nan},\n",
    "            {'ref': 'T', 'alt': 'A', 'N': 0, 'prop': 0},\n",
    "            {'ref': 'T', 'alt': 'T', 'N': np.nan, 'prop': np.nan}]\n",
    "\n",
    "new_df = pd.DataFrame(new_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "id": "07e17d9f-7bae-4273-8388-898f9fb1466f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>N</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.098039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.058824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.039216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.019608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.058824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.274510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.156863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.098039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.078431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.098039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.019608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ref alt     N      prop\n",
       "11   A   A   NaN       NaN\n",
       "0    A   C   5.0  0.098039\n",
       "1    A   G   3.0  0.058824\n",
       "2    A   T   2.0  0.039216\n",
       "3    C   A   1.0  0.019608\n",
       "12   C   C   NaN       NaN\n",
       "4    C   G   3.0  0.058824\n",
       "5    C   T  14.0  0.274510\n",
       "6    G   A   8.0  0.156863\n",
       "7    G   C   5.0  0.098039\n",
       "13   G   G   NaN       NaN\n",
       "8    G   T   4.0  0.078431\n",
       "14   T   A   0.0  0.000000\n",
       "9    T   C   5.0  0.098039\n",
       "10   T   G   1.0  0.019608\n",
       "15   T   T   NaN       NaN"
      ]
     },
     "execution_count": 263,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_unique_variant_counts = pd.concat([ppn_ptr_divergent_window_3d_modifying_unique_variant_counts, new_df], ignore_index=True).sort_values(by = ['ref','alt'], ascending = True)\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variant_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "id": "e168ba65-95f9-44b8-9251-fe891e8961f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_unique_variant_counts.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_divergent_window_3d_modifying_unique_variant_counts.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "420b100b-92cf-4baf-ac3c-15c738bbabc8",
   "metadata": {},
   "source": [
    "How frequent are transitions?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "id": "abb517de-c5d1-4b14-a711-7c7786a8efa8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.5084745762711864"
      ]
     },
     "execution_count": 265,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(3+14+8+5)/59"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b5029d2-ccc5-4a8d-bd12-8a98dce0c580",
   "metadata": {},
   "source": [
    "There are quite a few C > T mutations. How many of these are CpGs? Run the CpG retrieval script and then read in the output."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "id": "4e1fa465-8993-41c8-8934-e3a546c5b0a0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>window</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>CpG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>CT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos          window ref alt  CpG\n",
       "0   chr1  126940560  chr1_126353408   C   G   CT\n",
       "1   chr1  153901388  chr1_153616384   A   T  NaN\n",
       "2  chr10   88057593  chr10_87556096   G   C  NaN\n",
       "3  chr10   88057595  chr10_87556096   G   T  NaN\n",
       "4  chr11   77997697  chr11_77070336   G   A  NaN"
      ]
     },
     "execution_count": 266,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_dinucleotides = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_divergent_window_3d_modifying_variants_dinucleotides.txt', sep = '\\t', header = 0)\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_dinucleotides.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "2f6a30d0-4242-40f1-bc33-4a20a7907959",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CpG</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CA</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CC</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CG</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CT</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     N\n",
       "CpG   \n",
       "CA   4\n",
       "CC   4\n",
       "CG   3\n",
       "CT   7"
      ]
     },
     "execution_count": 267,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CpG_counts = ppn_ptr_divergent_window_3d_modifying_variants_dinucleotides[ppn_ptr_divergent_window_3d_modifying_variants_dinucleotides['ref'] == 'C']\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CpG_counts = ppn_ptr_divergent_window_3d_modifying_variants_with_CpG_counts[['chr','pos','ref','alt','CpG']].drop_duplicates()\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CpG_counts = ppn_ptr_divergent_window_3d_modifying_variants_with_CpG_counts.groupby(['CpG']).size().to_frame('N')\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CpG_counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "474b6112-bc97-4928-a7f8-970647bff2f3",
   "metadata": {},
   "source": [
    "### Ancestral Allele <a class = 'anchor' id = 'ancestralallele'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60e2ee39-aad1-4670-ab98-f2dded785678",
   "metadata": {},
   "source": [
    "What is the ancestral allele for 3d-modifying variants? We predict that most should be derived. First, let's send a BED file to a directory with a pipeline that can retrieve ancestral alleles."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "f4bc3242-8832-476e-a892-2287c4a57438",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940559</td>\n",
       "      <td>126940560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901387</td>\n",
       "      <td>153901388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7964</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057592</td>\n",
       "      <td>88057593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7965</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057594</td>\n",
       "      <td>88057595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997696</td>\n",
       "      <td>77997697</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr      start        end\n",
       "651     chr1  126940559  126940560\n",
       "2146    chr1  153901387  153901388\n",
       "7964   chr10   88057592   88057593\n",
       "7965   chr10   88057594   88057595\n",
       "12773  chr11   77997696   77997697"
      ]
     },
     "execution_count": 268,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_BED = ppn_ptr_divergent_window_3d_modifying_variants[['chr','pos']].copy()\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED.rename(columns = {'pos':'end'}, inplace = True)\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED['start'] = ppn_ptr_divergent_window_3d_modifying_variants_BED['end'] - 1\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED = ppn_ptr_divergent_window_3d_modifying_variants_BED[['chr','start','end']]\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "10f456e0-4dae-4038-8991-d4c8d4cfed58",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_BED.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ancestral_allele/data/ppn_ptr_divergent_window_3d_modifying_variants.bed', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05e694fe-63b2-4c53-be1c-4ddff16abe81",
   "metadata": {},
   "source": [
    "Now run the pipeline and read in the output."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "id": "1bde1a53-e393-465d-b54e-b012e5657a22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ancestral_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ancestral_allele\n",
       "0   chr1  126940560                C\n",
       "1   chr1  153901388                A\n",
       "2  chr10   88057593                G\n",
       "3  chr10   88057595                G\n",
       "4  chr11   77997697                G"
      ]
     },
     "execution_count": 270,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_ancestral_alleles = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ancestral_allele/data/ppn_ptr_divergent_window_3d_modifying_variants_ancestral_allele.txt', sep = '\\t', names = ['chr','pos','ancestral_allele'])\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_ancestral_alleles.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "9eea26de-8da9-4924-a2c9-65acc623049d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>0.011125</td>\n",
       "      <td>0.036647</td>\n",
       "      <td>0.029323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.017142</td>\n",
       "      <td>0.016506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.018322</td>\n",
       "      <td>0.031364</td>\n",
       "      <td>0.026767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.019119</td>\n",
       "      <td>0.032893</td>\n",
       "      <td>0.026767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.002093</td>\n",
       "      <td>0.025199</td>\n",
       "      <td>0.023994</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ref alt ancestral_allele          window       mse  \\\n",
       "0   chr1  126940560   C   G                C  chr1_126353408  0.011125   \n",
       "1   chr1  153901388   A   T                A  chr1_153616384  0.005189   \n",
       "2  chr10   88057593   G   C                G  chr10_87556096  0.018322   \n",
       "3  chr10   88057595   G   T                G  chr10_87556096  0.019119   \n",
       "4  chr11   77997697   G   A                G  chr11_77070336  0.002093   \n",
       "\n",
       "   divergence       min  \n",
       "0    0.036647  0.029323  \n",
       "1    0.017142  0.016506  \n",
       "2    0.031364  0.026767  \n",
       "3    0.032893  0.026767  \n",
       "4    0.025199  0.023994  "
      ]
     },
     "execution_count": 271,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants = pd.merge(ppn_ptr_divergent_window_3d_modifying_variants, ppn_ptr_divergent_window_3d_modifying_variants_ancestral_alleles, on = ['chr','pos'])\n",
    "ppn_ptr_divergent_window_3d_modifying_variants = ppn_ptr_divergent_window_3d_modifying_variants.drop_duplicates()\n",
    "ppn_ptr_divergent_window_3d_modifying_variants = ppn_ptr_divergent_window_3d_modifying_variants[['chr','pos','ref','alt','ancestral_allele','window','mse','divergence','min']]\n",
    "ppn_ptr_divergent_window_3d_modifying_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "ca8085e5-ad87-427a-b782-a9c2c8c8ad71",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "61"
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "59c608a2-fe46-41d3-ad55-ecf2df35fb8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>ancestral_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ref alt ancestral_allele\n",
       "0   chr1  126940560   C   G                C\n",
       "1   chr1  153901388   A   T                A\n",
       "2  chr10   88057593   G   C                G\n",
       "3  chr10   88057595   G   T                G\n",
       "4  chr11   77997697   G   A                G"
      ]
     },
     "execution_count": 273,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_dedup = ppn_ptr_divergent_window_3d_modifying_variants[['chr','pos','ref','alt','ancestral_allele']].drop_duplicates()\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_dedup.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "id": "7dcc034f-8ac5-4ccf-a448-b36486e8a199",
   "metadata": {},
   "outputs": [],
   "source": [
    "def alt_ancestral_derived(ppn_ptr_divergent_window_3d_modifying_variants):\n",
    "    if ppn_ptr_divergent_window_3d_modifying_variants['ancestral_allele'].upper() == ppn_ptr_divergent_window_3d_modifying_variants['alt']:\n",
    "        return 'ancestral'\n",
    "    else:\n",
    "        return 'derived'\n",
    "\n",
    "ppn_ptr_divergent_window_3d_modifying_variants['alt_ancestral_derived'] = ppn_ptr_divergent_window_3d_modifying_variants.apply(alt_ancestral_derived, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "id": "9f3ae877-c293-4b79-9384-3c1c27aef0fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>alt_ancestral_derived</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>0.011125</td>\n",
       "      <td>0.036647</td>\n",
       "      <td>0.029323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.017142</td>\n",
       "      <td>0.016506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.018322</td>\n",
       "      <td>0.031364</td>\n",
       "      <td>0.026767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.019119</td>\n",
       "      <td>0.032893</td>\n",
       "      <td>0.026767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.002093</td>\n",
       "      <td>0.025199</td>\n",
       "      <td>0.023994</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ref alt ancestral_allele alt_ancestral_derived  \\\n",
       "0   chr1  126940560   C   G                C               derived   \n",
       "1   chr1  153901388   A   T                A               derived   \n",
       "2  chr10   88057593   G   C                G               derived   \n",
       "3  chr10   88057595   G   T                G               derived   \n",
       "4  chr11   77997697   G   A                G               derived   \n",
       "\n",
       "           window       mse  divergence       min  \n",
       "0  chr1_126353408  0.011125    0.036647  0.029323  \n",
       "1  chr1_153616384  0.005189    0.017142  0.016506  \n",
       "2  chr10_87556096  0.018322    0.031364  0.026767  \n",
       "3  chr10_87556096  0.019119    0.032893  0.026767  \n",
       "4  chr11_77070336  0.002093    0.025199  0.023994  "
      ]
     },
     "execution_count": 275,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants = ppn_ptr_divergent_window_3d_modifying_variants[['chr','pos','ref','alt','ancestral_allele','alt_ancestral_derived','window','mse','divergence','min']]\n",
    "ppn_ptr_divergent_window_3d_modifying_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "e25e2377-4873-4f2b-b2cf-b9b4ebb45d71",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_dedup = ppn_ptr_divergent_window_3d_modifying_variants[['chr','pos','ref','alt','ancestral_allele','alt_ancestral_derived']].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "id": "0822724b-e570-47ec-b808-d027ca948e05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51"
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants_dedup)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ee4ed7b-c142-4213-80f7-5ca8495a35a4",
   "metadata": {},
   "source": [
    "Ancestral?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "id": "5ef34096-d3a3-4e4c-9c5f-ff8d40df4fb0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 278,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants_dedup[ppn_ptr_divergent_window_3d_modifying_variants_dedup['alt_ancestral_derived'] == 'ancestral'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d89f42c7-ec52-45d7-aadd-3c1bf7490fae",
   "metadata": {},
   "source": [
    "Derived?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "4f9c0608-d378-4812-885d-ef905c0a696e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41"
      ]
     },
     "execution_count": 279,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants_dedup[ppn_ptr_divergent_window_3d_modifying_variants_dedup['alt_ancestral_derived'] == 'derived'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4bdfb70c-607a-4f2b-9364-8d2ef58bf0d2",
   "metadata": {},
   "source": [
    "Are there sites where neither bonobo nor chimpanzee are ancestral?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "id": "ca86c1f1-fe67-484d-b37d-d0de35836f97",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>alt_ancestral_derived</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr12</td>\n",
       "      <td>38458311</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr4</td>\n",
       "      <td>62682758</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>chr5</td>\n",
       "      <td>35592012</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr       pos ref alt ancestral_allele alt_ancestral_derived\n",
       "8   chr12  38458311   C   T                A               derived\n",
       "48   chr4  62682758   G   A                T               derived\n",
       "58   chr5  35592012   G   A                C               derived"
      ]
     },
     "execution_count": 280,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_dedup[(ppn_ptr_divergent_window_3d_modifying_variants_dedup['ancestral_allele'].str.upper() != ppn_ptr_divergent_window_3d_modifying_variants_dedup['alt']) & (ppn_ptr_divergent_window_3d_modifying_variants_dedup['ancestral_allele'].str.upper() != ppn_ptr_divergent_window_3d_modifying_variants_dedup['ref'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca421271-d366-490c-87c6-b7e054776c33",
   "metadata": {},
   "source": [
    "### Explained Divergence <a class = 'anchor' id = 'explaineddivergence'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a22cbe6-352b-4ac9-87ad-a0686de4cfc8",
   "metadata": {},
   "source": [
    "What is the fraction of divergence explained by a given variant?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "id": "65e2da35-d146-445b-addb-660fd5709e94",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_maxes_dict = dict(zip(ppn_ptr_window_stats['window'], ppn_ptr_window_stats['maximum']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "f4a45011-f26c-404a-b139-25385a65e642",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants['max'] = ppn_ptr_divergent_window_3d_modifying_variants['window'].map(ppn_ptr_divergent_window_maxes_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "6be3b9b8-51e9-46a3-99c6-d6c7afe0bdd5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>alt_ancestral_derived</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>0.011125</td>\n",
       "      <td>0.036647</td>\n",
       "      <td>0.029323</td>\n",
       "      <td>0.063015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.017142</td>\n",
       "      <td>0.016506</td>\n",
       "      <td>0.032285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.018322</td>\n",
       "      <td>0.031364</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.040717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.019119</td>\n",
       "      <td>0.032893</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.040717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.002093</td>\n",
       "      <td>0.025199</td>\n",
       "      <td>0.023994</td>\n",
       "      <td>0.057039</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ref alt ancestral_allele alt_ancestral_derived  \\\n",
       "0   chr1  126940560   C   G                C               derived   \n",
       "1   chr1  153901388   A   T                A               derived   \n",
       "2  chr10   88057593   G   C                G               derived   \n",
       "3  chr10   88057595   G   T                G               derived   \n",
       "4  chr11   77997697   G   A                G               derived   \n",
       "\n",
       "           window       mse  divergence       min       max  \n",
       "0  chr1_126353408  0.011125    0.036647  0.029323  0.063015  \n",
       "1  chr1_153616384  0.005189    0.017142  0.016506  0.032285  \n",
       "2  chr10_87556096  0.018322    0.031364  0.026767  0.040717  \n",
       "3  chr10_87556096  0.019119    0.032893  0.026767  0.040717  \n",
       "4  chr11_77070336  0.002093    0.025199  0.023994  0.057039  "
      ]
     },
     "execution_count": 283,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "id": "7cd02141-d4e7-4030-bbbd-89f3e68d0732",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>alt_ancestral_derived</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>divergence_explained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>0.011125</td>\n",
       "      <td>0.036647</td>\n",
       "      <td>0.029323</td>\n",
       "      <td>0.063015</td>\n",
       "      <td>0.581557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.017142</td>\n",
       "      <td>0.016506</td>\n",
       "      <td>0.032285</td>\n",
       "      <td>0.530974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.018322</td>\n",
       "      <td>0.031364</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.040717</td>\n",
       "      <td>0.770293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.019119</td>\n",
       "      <td>0.032893</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.040717</td>\n",
       "      <td>0.807843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.002093</td>\n",
       "      <td>0.025199</td>\n",
       "      <td>0.023994</td>\n",
       "      <td>0.057039</td>\n",
       "      <td>0.441779</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ref alt ancestral_allele alt_ancestral_derived  \\\n",
       "0   chr1  126940560   C   G                C               derived   \n",
       "1   chr1  153901388   A   T                A               derived   \n",
       "2  chr10   88057593   G   C                G               derived   \n",
       "3  chr10   88057595   G   T                G               derived   \n",
       "4  chr11   77997697   G   A                G               derived   \n",
       "\n",
       "           window       mse  divergence       min       max  \\\n",
       "0  chr1_126353408  0.011125    0.036647  0.029323  0.063015   \n",
       "1  chr1_153616384  0.005189    0.017142  0.016506  0.032285   \n",
       "2  chr10_87556096  0.018322    0.031364  0.026767  0.040717   \n",
       "3  chr10_87556096  0.019119    0.032893  0.026767  0.040717   \n",
       "4  chr11_77070336  0.002093    0.025199  0.023994  0.057039   \n",
       "\n",
       "   divergence_explained  \n",
       "0              0.581557  \n",
       "1              0.530974  \n",
       "2              0.770293  \n",
       "3              0.807843  \n",
       "4              0.441779  "
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants['divergence_explained'] = ppn_ptr_divergent_window_3d_modifying_variants['divergence']/ppn_ptr_divergent_window_3d_modifying_variants['max']\n",
    "ppn_ptr_divergent_window_3d_modifying_variants.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3102460b-c631-46dc-b4ba-b5b48f01216b",
   "metadata": {},
   "source": [
    "Let's consider some statistics from this distribution."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "b5b2f5c9-f81a-42d9-b0f0-012c8aa6f350",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0854815608052588"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants['divergence_explained'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "id": "027d781f-64dd-4d39-bac7-ce66c9f2914f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.5715557636188744"
      ]
     },
     "execution_count": 286,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants['divergence_explained'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "id": "45036863-fec7-4408-b377-cf766e46772d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.1739817752739712"
      ]
     },
     "execution_count": 287,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants['divergence_explained'].max()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc4b60be-c454-4e96-9866-5ad4be6c52b9",
   "metadata": {},
   "source": [
    "### CTCF <a class = 'anchor' id = 'ctcf'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ddad3e50-e5d7-4866-99e5-58e1cfea23b4",
   "metadata": {},
   "source": [
    "How often do these variants fall within a CTCF binding site. Let's use data from Schwalie et al. 2013."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "ec961866-4735-4bc1-bdd6-ffc6324952e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t5909\t6180\t6.166802\t270\t0\t0\n",
      " chr1\t32407\t32758\t12.266173\t0\t0\t350\n",
      " chr1\t35033\t35334\t7.860073\t0\t0\t300\n",
      " chr1\t58418\t59129\t24.39928\t710\t381\t421\n",
      " chr1\t61438\t61919\t28.7659185\t460\t411\t451\n",
      " "
     ]
    }
   ],
   "source": [
    "CTCF_pbtBED = pybedtools.BedTool('/wynton/group/capra/projects/pan_3d_genome/data/annotations/panTro6_CTCF.bed')\n",
    "CTCF_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a0c0afc-0cf2-4e5f-9a5d-fc3cd4cf148d",
   "metadata": {},
   "source": [
    "And convert the 3d-modifying variant dataframe into a PBT BED."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "8c580d7a-6487-4d55-b9b2-f3672606fcc8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940559</td>\n",
       "      <td>126940560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901387</td>\n",
       "      <td>153901388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057592</td>\n",
       "      <td>88057593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057594</td>\n",
       "      <td>88057595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997696</td>\n",
       "      <td>77997697</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr      start        end\n",
       "0   chr1  126940559  126940560\n",
       "1   chr1  153901387  153901388\n",
       "2  chr10   88057592   88057593\n",
       "3  chr10   88057594   88057595\n",
       "4  chr11   77997696   77997697"
      ]
     },
     "execution_count": 289,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_BED = ppn_ptr_divergent_window_3d_modifying_variants[['chr','pos']].rename(columns = {'chr':'chr', 'pos':'end'})\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED['start'] = ppn_ptr_divergent_window_3d_modifying_variants_BED['end'] - 1\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED = ppn_ptr_divergent_window_3d_modifying_variants_BED[['chr','start','end']]\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED = ppn_ptr_divergent_window_3d_modifying_variants_BED.drop_duplicates()\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_BED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "e187267d-e224-436c-b76f-fbed3e6e27f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants_BED)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "8d75ae87-b194-4fb3-8f99-ed024e5898dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_pbtBED = pybedtools.BedTool().from_dataframe(ppn_ptr_divergent_window_3d_modifying_variants_BED).sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "31d80cf7-20ee-43dd-ba64-dfc4194841cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>CTCF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos  CTCF\n",
       "0   chr1  126940560     1\n",
       "1   chr1  153901388     1\n",
       "2  chr10   88057593     1\n",
       "3  chr10   88057595     1\n",
       "4  chr11   77997697     1"
      ]
     },
     "execution_count": 292,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect = ppn_ptr_divergent_window_3d_modifying_variants_pbtBED.intersect(CTCF_pbtBED, c = True).to_dataframe(names=['chr','start','end','count'])\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect = ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect.rename(columns = {'chr':'chr', 'start':'start', 'end':'pos', 'count':'CTCF'})\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect = ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect[['chr','pos','CTCF']]\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "364b8e34-90ad-47bb-b4ed-cc095d082c56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 293,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect[ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect['CTCF'] > 0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "id": "8c669e6c-6fda-45c9-9fe9-3a6c082cfcc3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.6666666666666666"
      ]
     },
     "execution_count": 294,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "34/51"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1dfd4e69-d699-4f7a-a6f2-1eae3c203e20",
   "metadata": {},
   "source": [
    "Let's add this to the main 3d-modifying variants dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "b6defe01-7384-4da7-b792-655208219d68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>alt_ancestral_derived</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>divergence_explained</th>\n",
       "      <th>CTCF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>0.011125</td>\n",
       "      <td>0.036647</td>\n",
       "      <td>0.029323</td>\n",
       "      <td>0.063015</td>\n",
       "      <td>0.581557</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.017142</td>\n",
       "      <td>0.016506</td>\n",
       "      <td>0.032285</td>\n",
       "      <td>0.530974</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.018322</td>\n",
       "      <td>0.031364</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.040717</td>\n",
       "      <td>0.770293</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.019119</td>\n",
       "      <td>0.032893</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.040717</td>\n",
       "      <td>0.807843</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.002093</td>\n",
       "      <td>0.025199</td>\n",
       "      <td>0.023994</td>\n",
       "      <td>0.057039</td>\n",
       "      <td>0.441779</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ref alt ancestral_allele alt_ancestral_derived  \\\n",
       "0   chr1  126940560   C   G                C               derived   \n",
       "1   chr1  153901388   A   T                A               derived   \n",
       "2  chr10   88057593   G   C                G               derived   \n",
       "3  chr10   88057595   G   T                G               derived   \n",
       "4  chr11   77997697   G   A                G               derived   \n",
       "\n",
       "           window       mse  divergence       min       max  \\\n",
       "0  chr1_126353408  0.011125    0.036647  0.029323  0.063015   \n",
       "1  chr1_153616384  0.005189    0.017142  0.016506  0.032285   \n",
       "2  chr10_87556096  0.018322    0.031364  0.026767  0.040717   \n",
       "3  chr10_87556096  0.019119    0.032893  0.026767  0.040717   \n",
       "4  chr11_77070336  0.002093    0.025199  0.023994  0.057039   \n",
       "\n",
       "   divergence_explained  CTCF  \n",
       "0              0.581557     1  \n",
       "1              0.530974     1  \n",
       "2              0.770293     1  \n",
       "3              0.807843     1  \n",
       "4              0.441779     1  "
      ]
     },
     "execution_count": 295,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF = pd.merge(ppn_ptr_divergent_window_3d_modifying_variants, ppn_ptr_divergent_window_3d_modifying_unique_variants_CTCF_intersect, on = ['chr','pos'], how = 'outer')\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "69f865d9-4374-44cc-8d3e-31181ec391ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "61"
      ]
     },
     "execution_count": 296,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e57669eb-2d1b-4fb9-83ed-ac1745ddb555",
   "metadata": {},
   "source": [
    "Let's investigate if we recover more CTCF sites by adding an interval around the variant."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "id": "5cad2fcd-5376-40f3-9d46-b589e9e33e59",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t126930559\t126950560\n",
      " chr1\t153891387\t153911388\n",
      " chr10\t88047592\t88067593\n",
      " chr10\t88047594\t88067595\n",
      " chr11\t77987696\t78007697\n",
      " "
     ]
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_1kb_buffer_pbtBED = ppn_ptr_divergent_window_3d_modifying_variants_pbtBED.slop(b = 10000, g = '/wynton/group/capra/projects/pan_3d_genome/data/metadata/panTro6_chr_lengths.txt')\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_1kb_buffer_pbtBED.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "id": "c4b71f15-4d94-4209-a06d-af706e40b49a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>CTCF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126950560</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153911388</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88067593</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88067595</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>78007697</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos  CTCF\n",
       "0   chr1  126950560     4\n",
       "1   chr1  153911388     2\n",
       "2  chr10   88067593     2\n",
       "3  chr10   88067595     2\n",
       "4  chr11   78007697     1"
      ]
     },
     "execution_count": 298,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_unique_variants_1kb_buffer_CTCF_intersect = ppn_ptr_divergent_window_3d_modifying_variants_1kb_buffer_pbtBED.intersect(CTCF_pbtBED, c = True).to_dataframe(names=['chr','start','end','count'])\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variants_1kb_buffer_CTCF_intersect = ppn_ptr_divergent_window_3d_modifying_unique_variants_1kb_buffer_CTCF_intersect.rename(columns = {'chr':'chr', 'start':'start', 'end':'pos', 'count':'CTCF'})\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variants_1kb_buffer_CTCF_intersect = ppn_ptr_divergent_window_3d_modifying_unique_variants_1kb_buffer_CTCF_intersect[['chr','pos','CTCF']]\n",
    "ppn_ptr_divergent_window_3d_modifying_unique_variants_1kb_buffer_CTCF_intersect.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "id": "eb8e056f-f77f-4fde-a3e2-6a3316f7a2e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36"
      ]
     },
     "execution_count": 299,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_unique_variants_1kb_buffer_CTCF_intersect[ppn_ptr_divergent_window_3d_modifying_unique_variants_1kb_buffer_CTCF_intersect['CTCF'] > 0])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9960711d-0cb9-4ed5-b8b3-b719d925c560",
   "metadata": {},
   "source": [
    "Do CTCF sites explain more 3d divergence?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "id": "297156e7-4a88-4dc5-93d6-5fd9f7929ee9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.471638961035618"
      ]
     },
     "execution_count": 300,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "divegence_explained_no_CTCF = ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF[ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF['CTCF'] == 0]['divergence_explained']\n",
    "divegence_explained_no_CTCF.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "87a178eb-5229-4ac0-8594-a8939405d234",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.6133814019095398"
      ]
     },
     "execution_count": 301,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "divegence_explained_CTCF = ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF[ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF['CTCF'] == 1]['divergence_explained']\n",
    "divegence_explained_CTCF.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "36f7cc73-d4ec-4ae1-9ba8-4a52ed349584",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=207.0, pvalue=0.004532625650113199)"
      ]
     },
     "execution_count": 302,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(divegence_explained_no_CTCF, divegence_explained_CTCF)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5eda508a-cedc-46ba-85af-7bb366b79b0e",
   "metadata": {},
   "source": [
    "### Chromatin Contact Effect <a class = 'anchor' id = 'chromatincontacteffect'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "171ce958-fb30-415a-826d-fc6b6722fe31",
   "metadata": {},
   "source": [
    "Now let's classify the effect of the 3d-modifying variant as resulting in decreased or increased chromatin contact where we can assess."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "0061fbf1-b766-4bee-aa4a-8531d093c786",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>window</th>\n",
       "      <th>ref_map_contacts_sum</th>\n",
       "      <th>alt_map_contacts_sum</th>\n",
       "      <th>summed_contact_difference</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>-15.429675</td>\n",
       "      <td>-106.086547</td>\n",
       "      <td>-90.656872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>-13.501551</td>\n",
       "      <td>-72.696401</td>\n",
       "      <td>-59.194849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>-37.248298</td>\n",
       "      <td>-100.210362</td>\n",
       "      <td>-62.962064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>-37.248298</td>\n",
       "      <td>-101.694820</td>\n",
       "      <td>-64.446522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>41.417256</td>\n",
       "      <td>-11.054677</td>\n",
       "      <td>-52.471933</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos          window  ref_map_contacts_sum  \\\n",
       "0   chr1  126940560  chr1_126353408            -15.429675   \n",
       "1   chr1  153901388  chr1_153616384            -13.501551   \n",
       "2  chr10   88057593  chr10_87556096            -37.248298   \n",
       "3  chr10   88057595  chr10_87556096            -37.248298   \n",
       "4  chr11   77997697  chr11_77070336             41.417256   \n",
       "\n",
       "   alt_map_contacts_sum  summed_contact_difference  \n",
       "0           -106.086547                 -90.656872  \n",
       "1            -72.696401                 -59.194849  \n",
       "2           -100.210362                 -62.962064  \n",
       "3           -101.694820                 -64.446522  \n",
       "4            -11.054677                 -52.471933  "
      ]
     },
     "execution_count": 303,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_chromatin_contact_effects = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_divergent_window_3d_modifying_variant_contact_effects.txt', sep = '\\t', header = 0)\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_chromatin_contact_effects.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86725242-1614-4dfb-bbee-fd8d0d0c4ef1",
   "metadata": {},
   "source": [
    "Add a column that indicates whether the summed contact difference is negative (decrease) or positive (increase)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "abf7d275-fc23-47e8-9d9d-8b1217b707dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>window</th>\n",
       "      <th>ref_map_contacts_sum</th>\n",
       "      <th>alt_map_contacts_sum</th>\n",
       "      <th>summed_contact_difference</th>\n",
       "      <th>effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>-15.429675</td>\n",
       "      <td>-106.086547</td>\n",
       "      <td>-90.656872</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>-13.501551</td>\n",
       "      <td>-72.696401</td>\n",
       "      <td>-59.194849</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>-37.248298</td>\n",
       "      <td>-100.210362</td>\n",
       "      <td>-62.962064</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>-37.248298</td>\n",
       "      <td>-101.694820</td>\n",
       "      <td>-64.446522</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>41.417256</td>\n",
       "      <td>-11.054677</td>\n",
       "      <td>-52.471933</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos          window  ref_map_contacts_sum  \\\n",
       "0   chr1  126940560  chr1_126353408            -15.429675   \n",
       "1   chr1  153901388  chr1_153616384            -13.501551   \n",
       "2  chr10   88057593  chr10_87556096            -37.248298   \n",
       "3  chr10   88057595  chr10_87556096            -37.248298   \n",
       "4  chr11   77997697  chr11_77070336             41.417256   \n",
       "\n",
       "   alt_map_contacts_sum  summed_contact_difference    effect  \n",
       "0           -106.086547                 -90.656872  Decrease  \n",
       "1            -72.696401                 -59.194849  Decrease  \n",
       "2           -100.210362                 -62.962064  Decrease  \n",
       "3           -101.694820                 -64.446522  Decrease  \n",
       "4            -11.054677                 -52.471933  Decrease  "
      ]
     },
     "execution_count": 304,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_chromatin_contact_effects['effect'] = pd.cut(ppn_ptr_divergent_window_3d_modifying_variants_chromatin_contact_effects['summed_contact_difference'], bins=[float('-Inf'), 0, float('Inf')], labels=['Decrease', 'Increase'])\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_chromatin_contact_effects.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36944aa6-ea8e-4004-a6e5-3f3bb1b58a07",
   "metadata": {},
   "source": [
    "Let's map this onto the 3d-modifying variant dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "id": "4f0bf7d8-ae16-4ec2-b399-6f4a1316f20e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>alt_ancestral_derived</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>divergence_explained</th>\n",
       "      <th>CTCF</th>\n",
       "      <th>ref_map_contacts_sum</th>\n",
       "      <th>alt_map_contacts_sum</th>\n",
       "      <th>summed_contact_difference</th>\n",
       "      <th>effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>126940560</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_126353408</td>\n",
       "      <td>0.011125</td>\n",
       "      <td>0.036647</td>\n",
       "      <td>0.029323</td>\n",
       "      <td>0.063015</td>\n",
       "      <td>0.581557</td>\n",
       "      <td>1</td>\n",
       "      <td>-15.429675</td>\n",
       "      <td>-106.086547</td>\n",
       "      <td>-90.656872</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>153901388</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr1_153616384</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.017142</td>\n",
       "      <td>0.016506</td>\n",
       "      <td>0.032285</td>\n",
       "      <td>0.530974</td>\n",
       "      <td>1</td>\n",
       "      <td>-13.501551</td>\n",
       "      <td>-72.696401</td>\n",
       "      <td>-59.194849</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057593</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.018322</td>\n",
       "      <td>0.031364</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.040717</td>\n",
       "      <td>0.770293</td>\n",
       "      <td>1</td>\n",
       "      <td>-37.248298</td>\n",
       "      <td>-100.210362</td>\n",
       "      <td>-62.962064</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>88057595</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr10_87556096</td>\n",
       "      <td>0.019119</td>\n",
       "      <td>0.032893</td>\n",
       "      <td>0.026767</td>\n",
       "      <td>0.040717</td>\n",
       "      <td>0.807843</td>\n",
       "      <td>1</td>\n",
       "      <td>-37.248298</td>\n",
       "      <td>-101.694820</td>\n",
       "      <td>-64.446522</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>77997697</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>chr11_77070336</td>\n",
       "      <td>0.002093</td>\n",
       "      <td>0.025199</td>\n",
       "      <td>0.023994</td>\n",
       "      <td>0.057039</td>\n",
       "      <td>0.441779</td>\n",
       "      <td>1</td>\n",
       "      <td>41.417256</td>\n",
       "      <td>-11.054677</td>\n",
       "      <td>-52.471933</td>\n",
       "      <td>Decrease</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ref alt ancestral_allele alt_ancestral_derived  \\\n",
       "0   chr1  126940560   C   G                C               derived   \n",
       "1   chr1  153901388   A   T                A               derived   \n",
       "2  chr10   88057593   G   C                G               derived   \n",
       "3  chr10   88057595   G   T                G               derived   \n",
       "4  chr11   77997697   G   A                G               derived   \n",
       "\n",
       "           window       mse  divergence       min       max  \\\n",
       "0  chr1_126353408  0.011125    0.036647  0.029323  0.063015   \n",
       "1  chr1_153616384  0.005189    0.017142  0.016506  0.032285   \n",
       "2  chr10_87556096  0.018322    0.031364  0.026767  0.040717   \n",
       "3  chr10_87556096  0.019119    0.032893  0.026767  0.040717   \n",
       "4  chr11_77070336  0.002093    0.025199  0.023994  0.057039   \n",
       "\n",
       "   divergence_explained  CTCF  ref_map_contacts_sum  alt_map_contacts_sum  \\\n",
       "0              0.581557     1            -15.429675           -106.086547   \n",
       "1              0.530974     1            -13.501551            -72.696401   \n",
       "2              0.770293     1            -37.248298           -100.210362   \n",
       "3              0.807843     1            -37.248298           -101.694820   \n",
       "4              0.441779     1             41.417256            -11.054677   \n",
       "\n",
       "   summed_contact_difference    effect  \n",
       "0                 -90.656872  Decrease  \n",
       "1                 -59.194849  Decrease  \n",
       "2                 -62.962064  Decrease  \n",
       "3                 -64.446522  Decrease  \n",
       "4                 -52.471933  Decrease  "
      ]
     },
     "execution_count": 305,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects = pd.merge(ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF, ppn_ptr_divergent_window_3d_modifying_variants_chromatin_contact_effects, on = ['chr','pos','window'])\n",
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "id": "1c25fd0d-45f0-4f31-9e54-6cb6e115c118",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "61"
      ]
     },
     "execution_count": 306,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "236f62e5-ef2e-4e0f-a3da-74996d4f6d42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>effect</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Decrease</th>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Increase</th>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           N\n",
       "effect      \n",
       "Decrease  38\n",
       "Increase  20"
      ]
     },
     "execution_count": 307,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects.groupby(['effect']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4108a7cd-a7c4-4823-85ce-162e80cc03e0",
   "metadata": {},
   "source": [
    "Let's also stratify the effect of chromatin contact by CTCF overlap."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "119d2b38-7e5f-4faf-bfb9-e3ad34b28414",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>effect</th>\n",
       "      <th>CTCF</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Decrease</th>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Increase</th>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                N\n",
       "effect   CTCF    \n",
       "Decrease 0      4\n",
       "         1     34\n",
       "Increase 0     12\n",
       "         1      8"
      ]
     },
     "execution_count": 308,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects.groupby(['effect','CTCF']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5bce5c45-7685-4ea9-bf50-0841cc43a89f",
   "metadata": {},
   "source": [
    "Let's also stratify by ancestral vs. derived after excluding variants in overlapping windows that disagree in overall effect."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "687dc3c3-aa66-4026-a433-e5890eb6b681",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "48"
      ]
     },
     "execution_count": 309,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects[['chr','pos','ref','alt','alt_ancestral_derived','CTCF','effect']].drop_duplicates().reset_index().drop([11,12,28,29,40,41])\n",
    "len(temp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "id": "21a71237-a9eb-4f76-9ecf-e00f167d1d84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>effect</th>\n",
       "      <th>alt_ancestral_derived</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Decrease</th>\n",
       "      <th>ancestral</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>derived</th>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Increase</th>\n",
       "      <th>ancestral</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>derived</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 N\n",
       "effect   alt_ancestral_derived    \n",
       "Decrease ancestral               3\n",
       "         derived                27\n",
       "Increase ancestral               7\n",
       "         derived                 9"
      ]
     },
     "execution_count": 310,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp.groupby(['effect','alt_ancestral_derived']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b83d8c60-80c5-4597-a821-d9d8ca68f3bf",
   "metadata": {},
   "source": [
    "Finally, we will test the summed contact distribution between variants that decrease and increase contact overall."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "b85857aa-0ed7-4468-8d73-ce95558b6d63",
   "metadata": {},
   "outputs": [],
   "source": [
    "decrease_contact_3d_modifying_variants = ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects[ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects['effect'] == 'Decrease']['summed_contact_difference']\n",
    "increase_contact_3d_modifying_variants = ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects[ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects['effect'] == 'Increase']['summed_contact_difference']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "id": "8e2c4aa8-1efc-46b5-a198-4d8e7e664e4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41.72285694987095"
      ]
     },
     "execution_count": 312,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abs(decrease_contact_3d_modifying_variants).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "id": "c1196c48-ccef-456c-8df7-b232b4cfc2f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "30.289600045587456"
      ]
     },
     "execution_count": 313,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abs(increase_contact_3d_modifying_variants).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "cc8e0e35-6e1b-4340-b956-46d5db6a501f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=483.0, pvalue=0.09358127967826899)"
      ]
     },
     "execution_count": 314,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(abs(decrease_contact_3d_modifying_variants), abs(increase_contact_3d_modifying_variants))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4bdd6547-848b-4afa-807b-dad39ecebc71",
   "metadata": {},
   "source": [
    "Save this dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "3ebca479-0ab3-4ae3-810f-e92ce87f00a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ppn_ptr_windows/ppn_ptr_divergent_window_3d_modifying_variants_with_CTCF_and_effects.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ec560d1-4fa5-4b8e-9a16-5b5a2e5531c5",
   "metadata": {},
   "source": [
    "## *In Silico* Mutagenesis for Chimpanzee Lineage-Specific Variants <a class = 'anchor' id = 'ptrinsilicomutagenesis'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36bd0439-118c-4cbc-98e3-788e56faea92",
   "metadata": {},
   "source": [
    "Let's also consider *in silico* mutagenesis run on chimpanzee lineage-specific variants. We will search for any variants with a divergence > 0.001."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "86aa4c99-34a5-4772-b0e0-73205ab7efa3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1816164</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_1048576</td>\n",
       "      <td>1.237785e-08</td>\n",
       "      <td>6.420484e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1816164</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_1572864</td>\n",
       "      <td>1.241441e-08</td>\n",
       "      <td>3.121775e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1850356</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_1048576</td>\n",
       "      <td>1.183865e-07</td>\n",
       "      <td>3.686226e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1850356</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_1572864</td>\n",
       "      <td>9.813220e-08</td>\n",
       "      <td>1.407815e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1966708</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_1048576</td>\n",
       "      <td>9.442934e-09</td>\n",
       "      <td>3.243256e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr      pos ref alt        window           mse    divergence\n",
       "0  chr1  1816164   G   A  chr1_1048576  1.237785e-08  6.420484e-08\n",
       "1  chr1  1816164   G   A  chr1_1572864  1.241441e-08  3.121775e-08\n",
       "2  chr1  1850356   C   A  chr1_1048576  1.183865e-07  3.686226e-07\n",
       "3  chr1  1850356   C   A  chr1_1572864  9.813220e-08  1.407815e-07\n",
       "4  chr1  1966708   C   T  chr1_1048576  9.442934e-09  3.243256e-08"
      ]
     },
     "execution_count": 316,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pte_specific_variants = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/in_silico_mutagenesis/pte_specific_variants_in_silico_mutagenesis.txt', sep = '\\t', header = 0)\n",
    "pte_specific_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "id": "f77281c2-4806-404f-b5c7-1704f04cc8bd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "64657"
      ]
     },
     "execution_count": 317,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pte_specific_variants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "id": "503f0401-7bc2-4c57-a2b4-8b8c6ca98063",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34474"
      ]
     },
     "execution_count": 318,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pte_specific_variants[['chr','pos','ref','alt']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "id": "f86e4ced-17bc-4ed0-ab5e-598f9cb2d566",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18725</th>\n",
       "      <td>chr14</td>\n",
       "      <td>27334853</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>chr14_26738688</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.001643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28012</th>\n",
       "      <td>chr19</td>\n",
       "      <td>28725598</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr19_28311552</td>\n",
       "      <td>0.000252</td>\n",
       "      <td>0.001637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41141</th>\n",
       "      <td>chr3</td>\n",
       "      <td>155608581</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr3_155189248</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0.001303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51875</th>\n",
       "      <td>chr6</td>\n",
       "      <td>78105433</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr6_77594624</td>\n",
       "      <td>0.000537</td>\n",
       "      <td>0.007160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59002</th>\n",
       "      <td>chr8</td>\n",
       "      <td>44129911</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr8_43515904</td>\n",
       "      <td>0.001514</td>\n",
       "      <td>0.008608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63191</th>\n",
       "      <td>chr9</td>\n",
       "      <td>52984573</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr9_52428800</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.001082</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr        pos ref alt          window       mse  divergence\n",
       "18725  chr14   27334853   A   C  chr14_26738688  0.000018    0.001643\n",
       "28012  chr19   28725598   C   T  chr19_28311552  0.000252    0.001637\n",
       "41141   chr3  155608581   G   A  chr3_155189248  0.000384    0.001303\n",
       "51875   chr6   78105433   A   G   chr6_77594624  0.000537    0.007160\n",
       "59002   chr8   44129911   T   C   chr8_43515904  0.001514    0.008608\n",
       "63191   chr9   52984573   A   G   chr9_52428800  0.000042    0.001082"
      ]
     },
     "execution_count": 319,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pte_specific_variants[pte_specific_variants['divergence'] > 0.001]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "id": "32610ef7-1112-4f87-bd9e-1d8236d2e758",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13689020</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_13631488</td>\n",
       "      <td>1.512008e-07</td>\n",
       "      <td>4.080969e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13689020</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_13107200</td>\n",
       "      <td>7.498010e-07</td>\n",
       "      <td>1.060541e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21490047</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_20971520</td>\n",
       "      <td>1.515650e-09</td>\n",
       "      <td>2.777475e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21490047</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_20447232</td>\n",
       "      <td>1.810662e-11</td>\n",
       "      <td>1.642873e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39399758</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_38797312</td>\n",
       "      <td>7.904467e-08</td>\n",
       "      <td>2.444597e-07</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr       pos ref alt         window           mse    divergence\n",
       "0  chr1  13689020   A   G  chr1_13631488  1.512008e-07  4.080969e-07\n",
       "1  chr1  13689020   A   G  chr1_13107200  7.498010e-07  1.060541e-06\n",
       "2  chr1  21490047   T   C  chr1_20971520  1.515650e-09  2.777475e-09\n",
       "3  chr1  21490047   T   C  chr1_20447232  1.810662e-11  1.642873e-10\n",
       "4  chr1  39399758   G   A  chr1_38797312  7.904467e-08  2.444597e-07"
      ]
     },
     "execution_count": 320,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pts_specific_variants = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/in_silico_mutagenesis/pts_specific_variants_in_silico_mutagenesis.txt', sep = '\\t', header = 0)\n",
    "pts_specific_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "889037b9-b1c7-4508-b61d-4837f1739270",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "610"
      ]
     },
     "execution_count": 321,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pts_specific_variants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "id": "dfe3f871-a383-4610-8467-95a62f0af061",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "337"
      ]
     },
     "execution_count": 322,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pts_specific_variants[['chr','pos','ref','alt']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "id": "2143094d-204e-409f-b95a-b4c58c2b4158",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, pos, ref, alt, window, mse, divergence]\n",
       "Index: []"
      ]
     },
     "execution_count": 323,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pts_specific_variants[pts_specific_variants['divergence'] > 0.001]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "id": "eabddd76-2f2a-4fa7-9e51-ef1a26a29a28",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20843059</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_20447232</td>\n",
       "      <td>1.436774e-08</td>\n",
       "      <td>2.963894e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>36299509</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_35651584</td>\n",
       "      <td>3.652465e-07</td>\n",
       "      <td>8.120434e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>36299509</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_36175872</td>\n",
       "      <td>4.638181e-07</td>\n",
       "      <td>1.475675e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>51965780</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_51380224</td>\n",
       "      <td>8.828241e-09</td>\n",
       "      <td>1.742867e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>51965780</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_51904512</td>\n",
       "      <td>1.869705e-09</td>\n",
       "      <td>4.626260e-09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr       pos ref alt         window           mse    divergence\n",
       "0  chr1  20843059   G   T  chr1_20447232  1.436774e-08  2.963894e-08\n",
       "1  chr1  36299509   T   C  chr1_35651584  3.652465e-07  8.120434e-07\n",
       "2  chr1  36299509   T   C  chr1_36175872  4.638181e-07  1.475675e-06\n",
       "3  chr1  51965780   G   T  chr1_51380224  8.828241e-09  1.742867e-08\n",
       "4  chr1  51965780   G   T  chr1_51904512  1.869705e-09  4.626260e-09"
      ]
     },
     "execution_count": 324,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ptt_specific_variants = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/in_silico_mutagenesis/ptt_specific_variants_in_silico_mutagenesis.txt', sep = '\\t', header = 0)\n",
    "ptt_specific_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "id": "25b051ef-5db6-4934-aa68-7f9daa6e1109",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "150"
      ]
     },
     "execution_count": 325,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ptt_specific_variants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "id": "9cb5d359-05da-4bfb-9459-6293bd82b618",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "78"
      ]
     },
     "execution_count": 326,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ptt_specific_variants[['chr','pos','ref','alt']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "id": "8874a0f1-3c43-49bd-8187-f5184a673086",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, pos, ref, alt, window, mse, divergence]\n",
       "Index: []"
      ]
     },
     "execution_count": 327,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ptt_specific_variants[ptt_specific_variants['divergence'] > 0.001]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "id": "e1044911-cc2d-4337-8c29-58b34ea56041",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2353967</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_1572864</td>\n",
       "      <td>4.399900e-06</td>\n",
       "      <td>8.923979e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2353967</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_2097152</td>\n",
       "      <td>2.604769e-06</td>\n",
       "      <td>4.205610e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2354245</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_1572864</td>\n",
       "      <td>6.150447e-07</td>\n",
       "      <td>1.166121e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2354245</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_2097152</td>\n",
       "      <td>4.088708e-07</td>\n",
       "      <td>5.174435e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2367054</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_1572864</td>\n",
       "      <td>9.960170e-08</td>\n",
       "      <td>2.043531e-07</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr      pos ref alt        window           mse    divergence\n",
       "0  chr1  2353967   C   G  chr1_1572864  4.399900e-06  8.923979e-06\n",
       "1  chr1  2353967   C   G  chr1_2097152  2.604769e-06  4.205610e-06\n",
       "2  chr1  2354245   A   C  chr1_1572864  6.150447e-07  1.166121e-06\n",
       "3  chr1  2354245   A   C  chr1_2097152  4.088708e-07  5.174435e-07\n",
       "4  chr1  2367054   G   A  chr1_1572864  9.960170e-08  2.043531e-07"
      ]
     },
     "execution_count": 328,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ptv_specific_variants = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/in_silico_mutagenesis/ptv_specific_variants_in_silico_mutagenesis.txt', sep = '\\t', header = 0)\n",
    "ptv_specific_variants.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "id": "ed480afb-f854-497a-80d8-790019f35785",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22671"
      ]
     },
     "execution_count": 329,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ptv_specific_variants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "id": "dd50ffc7-d3d2-4ba7-9cab-2124b706a46c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11993"
      ]
     },
     "execution_count": 330,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ptv_specific_variants[['chr','pos','ref','alt']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "id": "1fb149d5-9fe6-4ba2-9ce5-906cdbee6d6c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2482</th>\n",
       "      <td>chr10</td>\n",
       "      <td>76470135</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr10_76021760</td>\n",
       "      <td>0.000448</td>\n",
       "      <td>0.002432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9236</th>\n",
       "      <td>chr18</td>\n",
       "      <td>60956003</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr18_60293120</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.001207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11320</th>\n",
       "      <td>chr2A</td>\n",
       "      <td>55039344</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr2A_54001664</td>\n",
       "      <td>0.000417</td>\n",
       "      <td>0.001204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11321</th>\n",
       "      <td>chr2A</td>\n",
       "      <td>55039344</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr2A_54525952</td>\n",
       "      <td>0.019976</td>\n",
       "      <td>0.079208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16860</th>\n",
       "      <td>chr5</td>\n",
       "      <td>45286251</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr5_44564480</td>\n",
       "      <td>0.005497</td>\n",
       "      <td>0.006336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16861</th>\n",
       "      <td>chr5</td>\n",
       "      <td>45286251</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr5_45088768</td>\n",
       "      <td>0.003198</td>\n",
       "      <td>0.009205</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr       pos ref alt          window       mse  divergence\n",
       "2482   chr10  76470135   T   C  chr10_76021760  0.000448    0.002432\n",
       "9236   chr18  60956003   A   G  chr18_60293120  0.000019    0.001207\n",
       "11320  chr2A  55039344   C   T  chr2A_54001664  0.000417    0.001204\n",
       "11321  chr2A  55039344   C   T  chr2A_54525952  0.019976    0.079208\n",
       "16860   chr5  45286251   T   C   chr5_44564480  0.005497    0.006336\n",
       "16861   chr5  45286251   T   C   chr5_45088768  0.003198    0.009205"
      ]
     },
     "execution_count": 331,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ptv_specific_variants[ptv_specific_variants['divergence'] > 0.001]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f2ce38d-d2a4-46b6-96ad-607c63eb6ed9",
   "metadata": {},
   "source": [
    "Let's merge and save the resulting dataframe for any chimpanzee subspecies-specific 3d-modifying variants."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 332,
   "id": "d7136532-bf56-41f5-889e-dacea7a2dff2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/cbrand/ipykernel_2007303/3631605322.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  pte_3d_modifying_variants['lineage'] = 'pte'\n",
      "/scratch/cbrand/ipykernel_2007303/3631605322.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ptv_3d_modifying_variants['lineage'] = 'ptv'\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>window</th>\n",
       "      <th>mse</th>\n",
       "      <th>divergence</th>\n",
       "      <th>lineage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18725</th>\n",
       "      <td>chr14</td>\n",
       "      <td>27334853</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>chr14_26738688</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.001643</td>\n",
       "      <td>pte</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28012</th>\n",
       "      <td>chr19</td>\n",
       "      <td>28725598</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr19_28311552</td>\n",
       "      <td>0.000252</td>\n",
       "      <td>0.001637</td>\n",
       "      <td>pte</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41141</th>\n",
       "      <td>chr3</td>\n",
       "      <td>155608581</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr3_155189248</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0.001303</td>\n",
       "      <td>pte</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51875</th>\n",
       "      <td>chr6</td>\n",
       "      <td>78105433</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr6_77594624</td>\n",
       "      <td>0.000537</td>\n",
       "      <td>0.007160</td>\n",
       "      <td>pte</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59002</th>\n",
       "      <td>chr8</td>\n",
       "      <td>44129911</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr8_43515904</td>\n",
       "      <td>0.001514</td>\n",
       "      <td>0.008608</td>\n",
       "      <td>pte</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63191</th>\n",
       "      <td>chr9</td>\n",
       "      <td>52984573</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr9_52428800</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>pte</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2482</th>\n",
       "      <td>chr10</td>\n",
       "      <td>76470135</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr10_76021760</td>\n",
       "      <td>0.000448</td>\n",
       "      <td>0.002432</td>\n",
       "      <td>ptv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9236</th>\n",
       "      <td>chr18</td>\n",
       "      <td>60956003</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr18_60293120</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.001207</td>\n",
       "      <td>ptv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11320</th>\n",
       "      <td>chr2A</td>\n",
       "      <td>55039344</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr2A_54001664</td>\n",
       "      <td>0.000417</td>\n",
       "      <td>0.001204</td>\n",
       "      <td>ptv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11321</th>\n",
       "      <td>chr2A</td>\n",
       "      <td>55039344</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr2A_54525952</td>\n",
       "      <td>0.019976</td>\n",
       "      <td>0.079208</td>\n",
       "      <td>ptv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16860</th>\n",
       "      <td>chr5</td>\n",
       "      <td>45286251</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr5_44564480</td>\n",
       "      <td>0.005497</td>\n",
       "      <td>0.006336</td>\n",
       "      <td>ptv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16861</th>\n",
       "      <td>chr5</td>\n",
       "      <td>45286251</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr5_45088768</td>\n",
       "      <td>0.003198</td>\n",
       "      <td>0.009205</td>\n",
       "      <td>ptv</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr        pos ref alt          window       mse  divergence lineage\n",
       "18725  chr14   27334853   A   C  chr14_26738688  0.000018    0.001643     pte\n",
       "28012  chr19   28725598   C   T  chr19_28311552  0.000252    0.001637     pte\n",
       "41141   chr3  155608581   G   A  chr3_155189248  0.000384    0.001303     pte\n",
       "51875   chr6   78105433   A   G   chr6_77594624  0.000537    0.007160     pte\n",
       "59002   chr8   44129911   T   C   chr8_43515904  0.001514    0.008608     pte\n",
       "63191   chr9   52984573   A   G   chr9_52428800  0.000042    0.001082     pte\n",
       "2482   chr10   76470135   T   C  chr10_76021760  0.000448    0.002432     ptv\n",
       "9236   chr18   60956003   A   G  chr18_60293120  0.000019    0.001207     ptv\n",
       "11320  chr2A   55039344   C   T  chr2A_54001664  0.000417    0.001204     ptv\n",
       "11321  chr2A   55039344   C   T  chr2A_54525952  0.019976    0.079208     ptv\n",
       "16860   chr5   45286251   T   C   chr5_44564480  0.005497    0.006336     ptv\n",
       "16861   chr5   45286251   T   C   chr5_45088768  0.003198    0.009205     ptv"
      ]
     },
     "execution_count": 332,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pte_3d_modifying_variants = pte_specific_variants[pte_specific_variants['divergence'] > 0.001]\n",
    "pte_3d_modifying_variants['lineage'] = 'pte'\n",
    "ptv_3d_modifying_variants = ptv_specific_variants[ptv_specific_variants['divergence'] > 0.001]\n",
    "ptv_3d_modifying_variants['lineage'] = 'ptv'\n",
    "ptr_subspecies_specific_3d_modifying_variants = pd.concat([pte_3d_modifying_variants, ptv_3d_modifying_variants])\n",
    "ptr_subspecies_specific_3d_modifying_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "id": "9c7c5047-4885-4cb5-9147-c7dae340f84c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ptr_subspecies_specific_3d_modifying_variants.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/ptr_subspecies_specific_variants/ptr_subspecies_specific_3d_modifying_variants.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "821dfed4-9b40-42c3-b1e1-5edbe10cb7fd",
   "metadata": {},
   "source": [
    "## Bonobo-Chimpanzee Gene Expression Differences <a class = 'anchor' id = 'geneexpression'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd7fd369-f6af-4b35-b8b6-2f2a0cf33f5f",
   "metadata": {},
   "source": [
    "Let's examine differences in gene expression between bonobos and chimpanzees using data from Brawand et al. 2011."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "id": "17838509-bf09-4c39-814b-d6faa41599a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "tissue_dict = dict({'SRR306811': 'prefrontal_cortex',\n",
    "                 'SRR306817': 'cerebellum',\n",
    "                 'SRR306818': 'cerebellum', \n",
    "                 'SRR306819': 'heart', \n",
    "                 'SRR306820': 'heart', \n",
    "                 'SRR306821': 'kidney', \n",
    "                 'SRR306822': 'kidney', \n",
    "                 'SRR306823': 'liver', \n",
    "                 'SRR306824': 'liver', \n",
    "                 'SRR306825': 'testis',\n",
    "                 'SRR306827': 'prefrontal_cortex',\n",
    "                 'SRR306828': 'prefrontal_cortex',\n",
    "                 'SRR306829': 'cerebellum',\n",
    "                 'SRR306830': 'cerebellum', \n",
    "                 'SRR306831': 'heart', \n",
    "                 'SRR306832': 'heart', \n",
    "                 'SRR306833': 'kidney', \n",
    "                 'SRR306834': 'kidney', \n",
    "                 'SRR306835': 'liver', \n",
    "                 'SRR306836': 'liver', \n",
    "                 'SRR306837': 'testis'})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a82e9929-4527-4995-9549-43fa2f285a21",
   "metadata": {},
   "source": [
    "Define which samples belong to which individual."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "id": "1e3c5e3a-d118-488a-bdd9-b51693c68663",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_female = ['SRR306827','SRR306829','SRR306831','SRR306833','SRR306835']\n",
    "ppn_male = ['SRR306828','SRR306830','SRR306832','SRR306834','SRR306836','SRR306837']\n",
    "ptr_female = ['SRR306811','SRR306817','SRR306819','SRR306821','SRR306823']\n",
    "ptr_male = ['SRR306818','SRR306820','SRR306822','SRR306824','SRR306825']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc6512c1-702d-475a-86c9-b536fc22c95f",
   "metadata": {},
   "source": [
    "Now let's write a function to gather the read count data per individual."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "id": "43335ed0-1610-4b58-9ca3-aa0270d7f3ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "def reads_per_individual(individual, SRR_ids_list):\n",
    "    individual_read_counts_dfs_list = []\n",
    "    \n",
    "    for i in range(len(SRR_ids_list)):\n",
    "        individual_temp_df = pd.read_csv('RNAseq/read_counts/'+SRR_ids_list[i]+'_read_counts.txt', sep = '\\t', names = ['gene',SRR_ids_list[i]])\n",
    "        individual_read_counts_dfs_list.append(individual_temp_df)\n",
    "        \n",
    "    for df in individual_read_counts_dfs_list:\n",
    "        df.set_index('gene', inplace = True)\n",
    "        \n",
    "    individual_read_counts_df = pd.concat(individual_read_counts_dfs_list, axis = 1, sort = False).reset_index()\n",
    "    \n",
    "    individual_read_counts_df = individual_read_counts_df.set_index(['gene']).stack().to_frame().reset_index()\n",
    "    individual_read_counts_df.rename(columns={ 'level_1': 'sample', 0: individual}, inplace = True)\n",
    "    individual_read_counts_df['tissue'] = individual_read_counts_df['sample'].map(tissue_dict)\n",
    "    individual_read_counts_df = individual_read_counts_df.drop('sample', axis = 1)\n",
    "    individual_read_counts_df = individual_read_counts_df[~individual_read_counts_df['gene'].str.startswith('_')]\n",
    "    individual_read_counts_df = individual_read_counts_df[~individual_read_counts_df['gene'].str.startswith('unassigned')]\n",
    "    individual_read_counts_df = individual_read_counts_df[['gene','tissue',individual]]\n",
    "    \n",
    "    return individual_read_counts_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "73c3c409-b996-4de5-b5ce-44f3cbc0ce34",
   "metadata": {},
   "source": [
    "Apply the function to all four individuals."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "id": "09cf19ff-3239-4386-a30c-80334ea911d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "ppn_female_reads = reads_per_individual('ppn_female', ppn_female).set_index(['gene','tissue'])\n",
    "ppn_male_reads = reads_per_individual('ppn_male', ppn_male).set_index(['gene','tissue'])\n",
    "ptr_female_reads = reads_per_individual('ptr_female', ptr_female).set_index(['gene','tissue'])\n",
    "ptr_male_reads = reads_per_individual('ptr_male', ptr_male).set_index(['gene','tissue'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "98d4b44e-57f0-48b9-ac54-8585cfe8ac14",
   "metadata": {},
   "source": [
    "Gather those data and check out the dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "id": "6a050d4e-37bd-4aa5-8c46-b2fc6cd54033",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>107.0</td>\n",
       "      <td>57</td>\n",
       "      <td>51.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>158.0</td>\n",
       "      <td>233</td>\n",
       "      <td>84.0</td>\n",
       "      <td>136.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>heart</td>\n",
       "      <td>24.0</td>\n",
       "      <td>20</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>kidney</td>\n",
       "      <td>149.0</td>\n",
       "      <td>98</td>\n",
       "      <td>10.0</td>\n",
       "      <td>87.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>liver</td>\n",
       "      <td>40254.0</td>\n",
       "      <td>14391</td>\n",
       "      <td>31371.0</td>\n",
       "      <td>56937.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>heart</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>kidney</td>\n",
       "      <td>220.0</td>\n",
       "      <td>278</td>\n",
       "      <td>435.0</td>\n",
       "      <td>1305.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>liver</td>\n",
       "      <td>2929.0</td>\n",
       "      <td>1876</td>\n",
       "      <td>1151.0</td>\n",
       "      <td>2853.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>A2M</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>4805.0</td>\n",
       "      <td>6004</td>\n",
       "      <td>2561.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>A2M</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1780.0</td>\n",
       "      <td>2109</td>\n",
       "      <td>3051.0</td>\n",
       "      <td>1100.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "0   A1BG  prefrontal_cortex       107.0        57        51.0       NaN\n",
       "1   A1BG         cerebellum       158.0       233        84.0     136.0\n",
       "2   A1BG              heart        24.0        20         2.0       4.0\n",
       "3   A1BG             kidney       149.0        98        10.0      87.0\n",
       "4   A1BG              liver     40254.0     14391     31371.0   56937.0\n",
       "5   A1CF  prefrontal_cortex         0.0         1         0.0       NaN\n",
       "6   A1CF         cerebellum         0.0         1         3.0       2.0\n",
       "7   A1CF              heart         0.0         1         0.0       1.0\n",
       "8   A1CF             kidney       220.0       278       435.0    1305.0\n",
       "9   A1CF              liver      2929.0      1876      1151.0    2853.0\n",
       "10   A2M  prefrontal_cortex      4805.0      6004      2561.0       NaN\n",
       "11   A2M         cerebellum      1780.0      2109      3051.0    1100.0"
      ]
     },
     "execution_count": 338,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts = pd.concat([ppn_female_reads, ppn_male_reads, ptr_female_reads, ptr_male_reads], axis = 1, sort = False).reset_index()\n",
    "all_read_counts.head(12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "id": "9c365edb-de03-415b-aa88-4068bc14defb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "208308"
      ]
     },
     "execution_count": 339,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_read_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d4596a9-d48c-483d-85a4-88b8d847fabd",
   "metadata": {},
   "source": [
    "Let's examine read counts for the first example bonobo-chimpanzee divergent window in Figure 3."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "id": "2362f9a4-4d47-44e0-a5bc-d157af34dcad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>172515</th>\n",
       "      <td>ZNF622</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>340.0</td>\n",
       "      <td>748</td>\n",
       "      <td>731.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172516</th>\n",
       "      <td>ZNF622</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>474.0</td>\n",
       "      <td>715</td>\n",
       "      <td>1042.0</td>\n",
       "      <td>486.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172517</th>\n",
       "      <td>ZNF622</td>\n",
       "      <td>heart</td>\n",
       "      <td>613.0</td>\n",
       "      <td>316</td>\n",
       "      <td>859.0</td>\n",
       "      <td>544.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172518</th>\n",
       "      <td>ZNF622</td>\n",
       "      <td>kidney</td>\n",
       "      <td>352.0</td>\n",
       "      <td>335</td>\n",
       "      <td>764.0</td>\n",
       "      <td>587.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172519</th>\n",
       "      <td>ZNF622</td>\n",
       "      <td>liver</td>\n",
       "      <td>511.0</td>\n",
       "      <td>238</td>\n",
       "      <td>1091.0</td>\n",
       "      <td>328.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208093</th>\n",
       "      <td>ZNF622</td>\n",
       "      <td>testis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>637.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "172515  ZNF622  prefrontal_cortex       340.0       748       731.0       NaN\n",
       "172516  ZNF622         cerebellum       474.0       715      1042.0     486.0\n",
       "172517  ZNF622              heart       613.0       316       859.0     544.0\n",
       "172518  ZNF622             kidney       352.0       335       764.0     587.0\n",
       "172519  ZNF622              liver       511.0       238      1091.0     328.0\n",
       "208093  ZNF622             testis         NaN       640         NaN     637.0"
      ]
     },
     "execution_count": 340,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts[all_read_counts['gene'] == 'ZNF622']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "id": "ae5ab672-a58b-4286-85ff-ebf862f3fb99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>145385</th>\n",
       "      <td>RETREG1</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>453.0</td>\n",
       "      <td>1885</td>\n",
       "      <td>398.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145386</th>\n",
       "      <td>RETREG1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>499.0</td>\n",
       "      <td>626</td>\n",
       "      <td>707.0</td>\n",
       "      <td>462.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145387</th>\n",
       "      <td>RETREG1</td>\n",
       "      <td>heart</td>\n",
       "      <td>2892.0</td>\n",
       "      <td>700</td>\n",
       "      <td>1826.0</td>\n",
       "      <td>3262.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145388</th>\n",
       "      <td>RETREG1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>652.0</td>\n",
       "      <td>461</td>\n",
       "      <td>1515.0</td>\n",
       "      <td>990.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145389</th>\n",
       "      <td>RETREG1</td>\n",
       "      <td>liver</td>\n",
       "      <td>4637.0</td>\n",
       "      <td>496</td>\n",
       "      <td>1066.0</td>\n",
       "      <td>505.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202667</th>\n",
       "      <td>RETREG1</td>\n",
       "      <td>testis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2289.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "145385  RETREG1  prefrontal_cortex       453.0      1885       398.0       NaN\n",
       "145386  RETREG1         cerebellum       499.0       626       707.0     462.0\n",
       "145387  RETREG1              heart      2892.0       700      1826.0    3262.0\n",
       "145388  RETREG1             kidney       652.0       461      1515.0     990.0\n",
       "145389  RETREG1              liver      4637.0       496      1066.0     505.0\n",
       "202667  RETREG1             testis         NaN      1360         NaN    2289.0"
      ]
     },
     "execution_count": 341,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts[all_read_counts['gene'] == 'RETREG1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "id": "b310a9bd-e967-4443-9957-dae11144ef60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>129540</th>\n",
       "      <td>MYO10</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>2656.0</td>\n",
       "      <td>1943</td>\n",
       "      <td>829.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129541</th>\n",
       "      <td>MYO10</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>609.0</td>\n",
       "      <td>833</td>\n",
       "      <td>854.0</td>\n",
       "      <td>603.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129542</th>\n",
       "      <td>MYO10</td>\n",
       "      <td>heart</td>\n",
       "      <td>357.0</td>\n",
       "      <td>495</td>\n",
       "      <td>386.0</td>\n",
       "      <td>882.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129543</th>\n",
       "      <td>MYO10</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1914.0</td>\n",
       "      <td>1383</td>\n",
       "      <td>2184.0</td>\n",
       "      <td>3619.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129544</th>\n",
       "      <td>MYO10</td>\n",
       "      <td>liver</td>\n",
       "      <td>288.0</td>\n",
       "      <td>427</td>\n",
       "      <td>264.0</td>\n",
       "      <td>538.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199498</th>\n",
       "      <td>MYO10</td>\n",
       "      <td>testis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1918.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "129540  MYO10  prefrontal_cortex      2656.0      1943       829.0       NaN\n",
       "129541  MYO10         cerebellum       609.0       833       854.0     603.0\n",
       "129542  MYO10              heart       357.0       495       386.0     882.0\n",
       "129543  MYO10             kidney      1914.0      1383      2184.0    3619.0\n",
       "129544  MYO10              liver       288.0       427       264.0     538.0\n",
       "199498  MYO10             testis         NaN       770         NaN    1918.0"
      ]
     },
     "execution_count": 342,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts[all_read_counts['gene'] == 'MYO10']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "id": "d1d49b3a-a1f1-40ca-bd21-af86ad6dbba2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6955</th>\n",
       "      <td>BASP1</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>6121.0</td>\n",
       "      <td>11050</td>\n",
       "      <td>9051.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6956</th>\n",
       "      <td>BASP1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>3102.0</td>\n",
       "      <td>2160</td>\n",
       "      <td>1754.0</td>\n",
       "      <td>1972.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6957</th>\n",
       "      <td>BASP1</td>\n",
       "      <td>heart</td>\n",
       "      <td>114.0</td>\n",
       "      <td>62</td>\n",
       "      <td>174.0</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6958</th>\n",
       "      <td>BASP1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>210.0</td>\n",
       "      <td>156</td>\n",
       "      <td>213.0</td>\n",
       "      <td>189.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6959</th>\n",
       "      <td>BASP1</td>\n",
       "      <td>liver</td>\n",
       "      <td>489.0</td>\n",
       "      <td>92</td>\n",
       "      <td>903.0</td>\n",
       "      <td>94.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174981</th>\n",
       "      <td>BASP1</td>\n",
       "      <td>testis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "6955    BASP1  prefrontal_cortex      6121.0     11050      9051.0       NaN\n",
       "6956    BASP1         cerebellum      3102.0      2160      1754.0    1972.0\n",
       "6957    BASP1              heart       114.0        62       174.0      28.0\n",
       "6958    BASP1             kidney       210.0       156       213.0     189.0\n",
       "6959    BASP1              liver       489.0        92       903.0      94.0\n",
       "174981  BASP1             testis         NaN        36         NaN      35.0"
      ]
     },
     "execution_count": 343,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts[all_read_counts['gene'] == 'BASP1']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a03c069c-7dec-4f91-b9ca-65cad554f468",
   "metadata": {},
   "source": [
    "Now for the example in Figure 4."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "id": "2584978f-447b-461b-874c-5e0ac0db02f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>155455</th>\n",
       "      <td>SRI</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>1848.0</td>\n",
       "      <td>3479</td>\n",
       "      <td>1131.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155456</th>\n",
       "      <td>SRI</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1309.0</td>\n",
       "      <td>1434</td>\n",
       "      <td>1359.0</td>\n",
       "      <td>713.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155457</th>\n",
       "      <td>SRI</td>\n",
       "      <td>heart</td>\n",
       "      <td>670.0</td>\n",
       "      <td>450</td>\n",
       "      <td>409.0</td>\n",
       "      <td>467.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155458</th>\n",
       "      <td>SRI</td>\n",
       "      <td>kidney</td>\n",
       "      <td>871.0</td>\n",
       "      <td>788</td>\n",
       "      <td>869.0</td>\n",
       "      <td>1010.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155459</th>\n",
       "      <td>SRI</td>\n",
       "      <td>liver</td>\n",
       "      <td>316.0</td>\n",
       "      <td>194</td>\n",
       "      <td>580.0</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204681</th>\n",
       "      <td>SRI</td>\n",
       "      <td>testis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>360.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "155455  SRI  prefrontal_cortex      1848.0      3479      1131.0       NaN\n",
       "155456  SRI         cerebellum      1309.0      1434      1359.0     713.0\n",
       "155457  SRI              heart       670.0       450       409.0     467.0\n",
       "155458  SRI             kidney       871.0       788       869.0    1010.0\n",
       "155459  SRI              liver       316.0       194       580.0     100.0\n",
       "204681  SRI             testis         NaN       260         NaN     360.0"
      ]
     },
     "execution_count": 344,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts[all_read_counts['gene'] == 'SRI']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "id": "77c3df43-c004-4569-9bb2-19a3a29de926",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>156170</th>\n",
       "      <td>STEAP4</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>12.0</td>\n",
       "      <td>44</td>\n",
       "      <td>21.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156171</th>\n",
       "      <td>STEAP4</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>27.0</td>\n",
       "      <td>19</td>\n",
       "      <td>30.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156172</th>\n",
       "      <td>STEAP4</td>\n",
       "      <td>heart</td>\n",
       "      <td>330.0</td>\n",
       "      <td>320</td>\n",
       "      <td>1019.0</td>\n",
       "      <td>272.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156173</th>\n",
       "      <td>STEAP4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>60.0</td>\n",
       "      <td>74</td>\n",
       "      <td>421.0</td>\n",
       "      <td>41.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156174</th>\n",
       "      <td>STEAP4</td>\n",
       "      <td>liver</td>\n",
       "      <td>983.0</td>\n",
       "      <td>399</td>\n",
       "      <td>1951.0</td>\n",
       "      <td>289.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204824</th>\n",
       "      <td>STEAP4</td>\n",
       "      <td>testis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>186.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "156170  STEAP4  prefrontal_cortex        12.0        44        21.0       NaN\n",
       "156171  STEAP4         cerebellum        27.0        19        30.0       9.0\n",
       "156172  STEAP4              heart       330.0       320      1019.0     272.0\n",
       "156173  STEAP4             kidney        60.0        74       421.0      41.0\n",
       "156174  STEAP4              liver       983.0       399      1951.0     289.0\n",
       "204824  STEAP4             testis         NaN       108         NaN     186.0"
      ]
     },
     "execution_count": 345,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts[all_read_counts['gene'] == 'STEAP4']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "id": "2cd18bae-6594-4776-ba96-3234620ff4d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>173090</th>\n",
       "      <td>ZNF804B</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>17.0</td>\n",
       "      <td>175</td>\n",
       "      <td>23.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173091</th>\n",
       "      <td>ZNF804B</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>47.0</td>\n",
       "      <td>78</td>\n",
       "      <td>22.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173092</th>\n",
       "      <td>ZNF804B</td>\n",
       "      <td>heart</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173093</th>\n",
       "      <td>ZNF804B</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173094</th>\n",
       "      <td>ZNF804B</td>\n",
       "      <td>liver</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208208</th>\n",
       "      <td>ZNF804B</td>\n",
       "      <td>testis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "173090  ZNF804B  prefrontal_cortex        17.0       175        23.0       NaN\n",
       "173091  ZNF804B         cerebellum        47.0        78        22.0      11.0\n",
       "173092  ZNF804B              heart         0.0         2         0.0       0.0\n",
       "173093  ZNF804B             kidney         1.0         2         0.0       0.0\n",
       "173094  ZNF804B              liver         0.0         0         0.0       0.0\n",
       "208208  ZNF804B             testis         NaN        11         NaN      29.0"
      ]
     },
     "execution_count": 346,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts[all_read_counts['gene'] == 'ZNF804B']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "id": "82682a8a-4fe5-4887-aaeb-9533b48a0ba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>158935</th>\n",
       "      <td>TEX47</td>\n",
       "      <td>prefrontal_cortex</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158936</th>\n",
       "      <td>TEX47</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158937</th>\n",
       "      <td>TEX47</td>\n",
       "      <td>heart</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158938</th>\n",
       "      <td>TEX47</td>\n",
       "      <td>kidney</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158939</th>\n",
       "      <td>TEX47</td>\n",
       "      <td>liver</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205377</th>\n",
       "      <td>TEX47</td>\n",
       "      <td>testis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>689.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene             tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "158935  TEX47  prefrontal_cortex         1.0         3         0.0       NaN\n",
       "158936  TEX47         cerebellum         9.0        10         0.0       3.0\n",
       "158937  TEX47              heart         0.0         0         0.0       0.0\n",
       "158938  TEX47             kidney         0.0         0         0.0       1.0\n",
       "158939  TEX47              liver         0.0         2         0.0       0.0\n",
       "205377  TEX47             testis         NaN       195         NaN     689.0"
      ]
     },
     "execution_count": 347,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_read_counts[all_read_counts['gene'] == 'TEX47']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e714f75-59d4-4aaf-a0de-569db2e78520",
   "metadata": {},
   "source": [
    "Now let's focus on all genes in bonobo-chimpanzee divergent windows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "id": "8c955d59-8d0a-438d-a759-1ad1b394b407",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>511</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1085.0</td>\n",
       "      <td>1865</td>\n",
       "      <td>1276.0</td>\n",
       "      <td>1168.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>512</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>heart</td>\n",
       "      <td>394.0</td>\n",
       "      <td>295</td>\n",
       "      <td>375.0</td>\n",
       "      <td>425.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>513</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>520.0</td>\n",
       "      <td>405</td>\n",
       "      <td>516.0</td>\n",
       "      <td>649.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>liver</td>\n",
       "      <td>289.0</td>\n",
       "      <td>238</td>\n",
       "      <td>266.0</td>\n",
       "      <td>163.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>ABO</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>2984.0</td>\n",
       "      <td>4796</td>\n",
       "      <td>2052.0</td>\n",
       "      <td>3137.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene      tissue  ppn_female  ppn_male  ptr_female  ptr_male\n",
       "511  ABL2  cerebellum      1085.0      1865      1276.0    1168.0\n",
       "512  ABL2       heart       394.0       295       375.0     425.0\n",
       "513  ABL2      kidney       520.0       405       516.0     649.0\n",
       "514  ABL2       liver       289.0       238       266.0     163.0\n",
       "531   ABO  cerebellum      2984.0      4796      2052.0    3137.0"
      ]
     },
     "execution_count": 348,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppn_ptr_divergent_windows_genes_read_counts = all_read_counts[all_read_counts['gene'].isin(ppn_ptr_divergent_windows_genes)]\n",
    "ppn_ptr_divergent_windows_genes_read_counts = ppn_ptr_divergent_windows_genes_read_counts[(ppn_ptr_divergent_windows_genes_read_counts['ppn_female'] != 0) & (ppn_ptr_divergent_windows_genes_read_counts['ppn_male'] != 0) & (ppn_ptr_divergent_windows_genes_read_counts['ptr_female'] != 0) & (ppn_ptr_divergent_windows_genes_read_counts['ptr_male'] != 0)] \n",
    "ppn_ptr_divergent_windows_genes_read_counts = ppn_ptr_divergent_windows_genes_read_counts[~ppn_ptr_divergent_windows_genes_read_counts['tissue'].isin(['prefrontal_cortex','testis'])]\n",
    "ppn_ptr_divergent_windows_genes_read_counts.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "id": "796dec1f-97c4-419c-9f4a-72e96a13edad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1361"
      ]
     },
     "execution_count": 349,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ppn_ptr_divergent_windows_genes_read_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d67b40e-2f38-4452-9da8-61f1fac897f2",
   "metadata": {},
   "source": [
    "Write a function to identify genes with a species difference in expression."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "id": "ca808e70-6a5c-449e-b69c-0d70424a5e0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_species_difference(dataframe):\n",
    "    df = dataframe.copy()\n",
    "    df['ppn_delta'] = df.apply(lambda row: abs(row['ppn_female'] - row['ppn_male']), axis=1)\n",
    "    df['ptr_delta'] = df.apply(lambda row: abs(row['ptr_female'] - row['ptr_male']), axis=1)\n",
    "    species_diff_list = []\n",
    "\n",
    "    for index, row in df.iterrows():\n",
    "        ppn_min = min(row['ppn_female'], row['ppn_male'])\n",
    "        ppn_max = max(row['ppn_female'], row['ppn_male'])\n",
    "        ptr_min = min(row['ptr_female'], row['ptr_male'])\n",
    "        ptr_max = max(row['ptr_female'], row['ptr_male'])\n",
    "\n",
    "        if ppn_min > ptr_max:\n",
    "            species_diff_list.append(ppn_min - ptr_max)\n",
    "        elif ptr_min > ppn_max:\n",
    "            species_diff_list.append(ptr_min - ppn_max)\n",
    "        else:\n",
    "            species_diff_list.append(None)\n",
    "        \n",
    "    df['species_diff'] = species_diff_list\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "id": "0b3d768b-d1a8-44b7-a034-d09b47932091",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "      <th>ppn_delta</th>\n",
       "      <th>ptr_delta</th>\n",
       "      <th>species_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>511</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1085.0</td>\n",
       "      <td>1865</td>\n",
       "      <td>1276.0</td>\n",
       "      <td>1168.0</td>\n",
       "      <td>780.0</td>\n",
       "      <td>108.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>512</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>heart</td>\n",
       "      <td>394.0</td>\n",
       "      <td>295</td>\n",
       "      <td>375.0</td>\n",
       "      <td>425.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>513</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>520.0</td>\n",
       "      <td>405</td>\n",
       "      <td>516.0</td>\n",
       "      <td>649.0</td>\n",
       "      <td>115.0</td>\n",
       "      <td>133.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>liver</td>\n",
       "      <td>289.0</td>\n",
       "      <td>238</td>\n",
       "      <td>266.0</td>\n",
       "      <td>163.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>ABO</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>2984.0</td>\n",
       "      <td>4796</td>\n",
       "      <td>2052.0</td>\n",
       "      <td>3137.0</td>\n",
       "      <td>1812.0</td>\n",
       "      <td>1085.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene      tissue  ppn_female  ppn_male  ptr_female  ptr_male  ppn_delta  \\\n",
       "511  ABL2  cerebellum      1085.0      1865      1276.0    1168.0      780.0   \n",
       "512  ABL2       heart       394.0       295       375.0     425.0       99.0   \n",
       "513  ABL2      kidney       520.0       405       516.0     649.0      115.0   \n",
       "514  ABL2       liver       289.0       238       266.0     163.0       51.0   \n",
       "531   ABO  cerebellum      2984.0      4796      2052.0    3137.0     1812.0   \n",
       "\n",
       "     ptr_delta  species_diff  \n",
       "511      108.0           NaN  \n",
       "512       50.0           NaN  \n",
       "513      133.0           NaN  \n",
       "514      103.0           NaN  \n",
       "531     1085.0           NaN  "
      ]
     },
     "execution_count": 351,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "species_different_gene_expression = calculate_species_difference(ppn_ptr_divergent_windows_genes_read_counts)\n",
    "species_different_gene_expression.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "id": "4543a146-c0df-467e-a1dd-66c825c5f58d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1361"
      ]
     },
     "execution_count": 352,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(species_different_gene_expression)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "id": "425560d8-5764-4790-94a8-043d494a710a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "442"
      ]
     },
     "execution_count": 353,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(species_different_gene_expression[species_different_gene_expression['species_diff'].notna()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "id": "cd037b26-3588-465e-8dca-4fb7e923e953",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ppn_female</th>\n",
       "      <th>ppn_male</th>\n",
       "      <th>ptr_female</th>\n",
       "      <th>ptr_male</th>\n",
       "      <th>ppn_delta</th>\n",
       "      <th>ptr_delta</th>\n",
       "      <th>species_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1282</th>\n",
       "      <td>ADAM22</td>\n",
       "      <td>heart</td>\n",
       "      <td>31.0</td>\n",
       "      <td>47</td>\n",
       "      <td>53.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283</th>\n",
       "      <td>ADAM22</td>\n",
       "      <td>kidney</td>\n",
       "      <td>198.0</td>\n",
       "      <td>181</td>\n",
       "      <td>199.0</td>\n",
       "      <td>505.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>306.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1284</th>\n",
       "      <td>ADAM22</td>\n",
       "      <td>liver</td>\n",
       "      <td>13.0</td>\n",
       "      <td>129</td>\n",
       "      <td>10.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1437</th>\n",
       "      <td>ADAMTSL2</td>\n",
       "      <td>heart</td>\n",
       "      <td>151.0</td>\n",
       "      <td>714</td>\n",
       "      <td>89.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>563.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>62.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1439</th>\n",
       "      <td>ADAMTSL2</td>\n",
       "      <td>liver</td>\n",
       "      <td>352.0</td>\n",
       "      <td>305</td>\n",
       "      <td>689.0</td>\n",
       "      <td>402.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>287.0</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2119</th>\n",
       "      <td>AGMO</td>\n",
       "      <td>liver</td>\n",
       "      <td>254.0</td>\n",
       "      <td>221</td>\n",
       "      <td>22.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>39.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2163</th>\n",
       "      <td>AGPAT5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>187.0</td>\n",
       "      <td>199</td>\n",
       "      <td>317.0</td>\n",
       "      <td>687.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>370.0</td>\n",
       "      <td>118.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2413</th>\n",
       "      <td>AK8</td>\n",
       "      <td>kidney</td>\n",
       "      <td>30.0</td>\n",
       "      <td>20</td>\n",
       "      <td>127.0</td>\n",
       "      <td>134.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2471</th>\n",
       "      <td>AKAP5</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>181.0</td>\n",
       "      <td>141</td>\n",
       "      <td>64.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>39.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2474</th>\n",
       "      <td>AKAP5</td>\n",
       "      <td>liver</td>\n",
       "      <td>37.0</td>\n",
       "      <td>55</td>\n",
       "      <td>7.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3221</th>\n",
       "      <td>ANGPT2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>102.0</td>\n",
       "      <td>45</td>\n",
       "      <td>39.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3223</th>\n",
       "      <td>ANGPT2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>35.0</td>\n",
       "      <td>25</td>\n",
       "      <td>18.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3224</th>\n",
       "      <td>ANGPT2</td>\n",
       "      <td>liver</td>\n",
       "      <td>33.0</td>\n",
       "      <td>43</td>\n",
       "      <td>21.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3232</th>\n",
       "      <td>ANGPTL1</td>\n",
       "      <td>heart</td>\n",
       "      <td>98.0</td>\n",
       "      <td>176</td>\n",
       "      <td>51.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3234</th>\n",
       "      <td>ANGPTL1</td>\n",
       "      <td>liver</td>\n",
       "      <td>144.0</td>\n",
       "      <td>121</td>\n",
       "      <td>22.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>79.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3886</th>\n",
       "      <td>AP1S2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>478.0</td>\n",
       "      <td>511</td>\n",
       "      <td>347.0</td>\n",
       "      <td>276.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>131.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3889</th>\n",
       "      <td>AP1S2</td>\n",
       "      <td>liver</td>\n",
       "      <td>193.0</td>\n",
       "      <td>114</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4352</th>\n",
       "      <td>AQP4</td>\n",
       "      <td>heart</td>\n",
       "      <td>13.0</td>\n",
       "      <td>5</td>\n",
       "      <td>24.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>134.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5086</th>\n",
       "      <td>ARMC9</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>254.0</td>\n",
       "      <td>270</td>\n",
       "      <td>227.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5087</th>\n",
       "      <td>ARMC9</td>\n",
       "      <td>heart</td>\n",
       "      <td>43.0</td>\n",
       "      <td>45</td>\n",
       "      <td>35.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5262</th>\n",
       "      <td>ARSB</td>\n",
       "      <td>heart</td>\n",
       "      <td>149.0</td>\n",
       "      <td>100</td>\n",
       "      <td>251.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5263</th>\n",
       "      <td>ARSB</td>\n",
       "      <td>kidney</td>\n",
       "      <td>445.0</td>\n",
       "      <td>584</td>\n",
       "      <td>2249.0</td>\n",
       "      <td>2629.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>380.0</td>\n",
       "      <td>1665.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5457</th>\n",
       "      <td>ASB9</td>\n",
       "      <td>heart</td>\n",
       "      <td>18.0</td>\n",
       "      <td>14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5669</th>\n",
       "      <td>ASTN2</td>\n",
       "      <td>liver</td>\n",
       "      <td>41.0</td>\n",
       "      <td>72</td>\n",
       "      <td>19.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5736</th>\n",
       "      <td>ATF1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>222.0</td>\n",
       "      <td>241</td>\n",
       "      <td>157.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>65.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5811</th>\n",
       "      <td>ATG16L1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1487.0</td>\n",
       "      <td>2151</td>\n",
       "      <td>1354.0</td>\n",
       "      <td>736.0</td>\n",
       "      <td>664.0</td>\n",
       "      <td>618.0</td>\n",
       "      <td>133.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6323</th>\n",
       "      <td>ATP6V1G3</td>\n",
       "      <td>kidney</td>\n",
       "      <td>231.0</td>\n",
       "      <td>265</td>\n",
       "      <td>653.0</td>\n",
       "      <td>466.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>201.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6488</th>\n",
       "      <td>AUH</td>\n",
       "      <td>kidney</td>\n",
       "      <td>123.0</td>\n",
       "      <td>105</td>\n",
       "      <td>158.0</td>\n",
       "      <td>338.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6579</th>\n",
       "      <td>AXDND1</td>\n",
       "      <td>liver</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6956</th>\n",
       "      <td>BASP1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>3102.0</td>\n",
       "      <td>2160</td>\n",
       "      <td>1754.0</td>\n",
       "      <td>1972.0</td>\n",
       "      <td>942.0</td>\n",
       "      <td>218.0</td>\n",
       "      <td>188.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7498</th>\n",
       "      <td>BHMT</td>\n",
       "      <td>kidney</td>\n",
       "      <td>2778.0</td>\n",
       "      <td>3455</td>\n",
       "      <td>5767.0</td>\n",
       "      <td>15958.0</td>\n",
       "      <td>677.0</td>\n",
       "      <td>10191.0</td>\n",
       "      <td>2312.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7501</th>\n",
       "      <td>BHMT2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>41.0</td>\n",
       "      <td>39</td>\n",
       "      <td>5.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7502</th>\n",
       "      <td>BHMT2</td>\n",
       "      <td>heart</td>\n",
       "      <td>366.0</td>\n",
       "      <td>152</td>\n",
       "      <td>50.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>70.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7503</th>\n",
       "      <td>BHMT2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1925.0</td>\n",
       "      <td>1317</td>\n",
       "      <td>2907.0</td>\n",
       "      <td>9406.0</td>\n",
       "      <td>608.0</td>\n",
       "      <td>6499.0</td>\n",
       "      <td>982.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7558</th>\n",
       "      <td>BIN2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>173.0</td>\n",
       "      <td>145</td>\n",
       "      <td>186.0</td>\n",
       "      <td>183.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7779</th>\n",
       "      <td>BMX</td>\n",
       "      <td>liver</td>\n",
       "      <td>52.0</td>\n",
       "      <td>31</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8192</th>\n",
       "      <td>BTAF1</td>\n",
       "      <td>heart</td>\n",
       "      <td>491.0</td>\n",
       "      <td>430</td>\n",
       "      <td>859.0</td>\n",
       "      <td>603.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8193</th>\n",
       "      <td>BTAF1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>773.0</td>\n",
       "      <td>514</td>\n",
       "      <td>1115.0</td>\n",
       "      <td>965.0</td>\n",
       "      <td>259.0</td>\n",
       "      <td>150.0</td>\n",
       "      <td>192.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9362</th>\n",
       "      <td>C1H1orf53</td>\n",
       "      <td>heart</td>\n",
       "      <td>15.0</td>\n",
       "      <td>5</td>\n",
       "      <td>18.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9364</th>\n",
       "      <td>C1H1orf53</td>\n",
       "      <td>liver</td>\n",
       "      <td>37.0</td>\n",
       "      <td>25</td>\n",
       "      <td>16.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9556</th>\n",
       "      <td>C21H21orf140</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>29.0</td>\n",
       "      <td>26</td>\n",
       "      <td>37.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9557</th>\n",
       "      <td>C21H21orf140</td>\n",
       "      <td>heart</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3</td>\n",
       "      <td>19.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9558</th>\n",
       "      <td>C21H21orf140</td>\n",
       "      <td>kidney</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2</td>\n",
       "      <td>27.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9559</th>\n",
       "      <td>C21H21orf140</td>\n",
       "      <td>liver</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1</td>\n",
       "      <td>9.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9724</th>\n",
       "      <td>C2BH2orf72</td>\n",
       "      <td>liver</td>\n",
       "      <td>2243.0</td>\n",
       "      <td>3142</td>\n",
       "      <td>460.0</td>\n",
       "      <td>1832.0</td>\n",
       "      <td>899.0</td>\n",
       "      <td>1372.0</td>\n",
       "      <td>411.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9779</th>\n",
       "      <td>C2CD4D</td>\n",
       "      <td>liver</td>\n",
       "      <td>15.0</td>\n",
       "      <td>7</td>\n",
       "      <td>19.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10367</th>\n",
       "      <td>C9H9orf16</td>\n",
       "      <td>heart</td>\n",
       "      <td>118.0</td>\n",
       "      <td>192</td>\n",
       "      <td>66.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>52.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10503</th>\n",
       "      <td>CA5B</td>\n",
       "      <td>kidney</td>\n",
       "      <td>94.0</td>\n",
       "      <td>86</td>\n",
       "      <td>150.0</td>\n",
       "      <td>287.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>56.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10532</th>\n",
       "      <td>CAB39</td>\n",
       "      <td>heart</td>\n",
       "      <td>1822.0</td>\n",
       "      <td>589</td>\n",
       "      <td>2038.0</td>\n",
       "      <td>2416.0</td>\n",
       "      <td>1233.0</td>\n",
       "      <td>378.0</td>\n",
       "      <td>216.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10533</th>\n",
       "      <td>CAB39</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1321.0</td>\n",
       "      <td>1076</td>\n",
       "      <td>1374.0</td>\n",
       "      <td>2756.0</td>\n",
       "      <td>245.0</td>\n",
       "      <td>1382.0</td>\n",
       "      <td>53.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10596</th>\n",
       "      <td>CACFD1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>540.0</td>\n",
       "      <td>606</td>\n",
       "      <td>536.0</td>\n",
       "      <td>361.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10722</th>\n",
       "      <td>CACNG6</td>\n",
       "      <td>heart</td>\n",
       "      <td>14.0</td>\n",
       "      <td>4</td>\n",
       "      <td>41.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10756</th>\n",
       "      <td>CADM2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1972.0</td>\n",
       "      <td>2314</td>\n",
       "      <td>1441.0</td>\n",
       "      <td>1284.0</td>\n",
       "      <td>342.0</td>\n",
       "      <td>157.0</td>\n",
       "      <td>531.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10757</th>\n",
       "      <td>CADM2</td>\n",
       "      <td>heart</td>\n",
       "      <td>14.0</td>\n",
       "      <td>6</td>\n",
       "      <td>27.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11531</th>\n",
       "      <td>CBLB</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>864.0</td>\n",
       "      <td>903</td>\n",
       "      <td>641.0</td>\n",
       "      <td>527.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13437</th>\n",
       "      <td>CDC42EP5</td>\n",
       "      <td>heart</td>\n",
       "      <td>296.0</td>\n",
       "      <td>324</td>\n",
       "      <td>42.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>246.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13438</th>\n",
       "      <td>CDC42EP5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>202.0</td>\n",
       "      <td>158</td>\n",
       "      <td>98.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>60.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13534</th>\n",
       "      <td>CDH11</td>\n",
       "      <td>liver</td>\n",
       "      <td>31.0</td>\n",
       "      <td>30</td>\n",
       "      <td>13.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14076</th>\n",
       "      <td>CEL</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>643.0</td>\n",
       "      <td>602</td>\n",
       "      <td>209.0</td>\n",
       "      <td>245.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>357.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14106</th>\n",
       "      <td>CELF3</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>4607.0</td>\n",
       "      <td>4593</td>\n",
       "      <td>2707.0</td>\n",
       "      <td>3722.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1015.0</td>\n",
       "      <td>871.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14462</th>\n",
       "      <td>CERCAM</td>\n",
       "      <td>heart</td>\n",
       "      <td>187.0</td>\n",
       "      <td>301</td>\n",
       "      <td>47.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>119.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14463</th>\n",
       "      <td>CERCAM</td>\n",
       "      <td>kidney</td>\n",
       "      <td>313.0</td>\n",
       "      <td>209</td>\n",
       "      <td>59.0</td>\n",
       "      <td>138.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>71.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14464</th>\n",
       "      <td>CERCAM</td>\n",
       "      <td>liver</td>\n",
       "      <td>37.0</td>\n",
       "      <td>33</td>\n",
       "      <td>47.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15242</th>\n",
       "      <td>CHRNG</td>\n",
       "      <td>heart</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15317</th>\n",
       "      <td>CHST9</td>\n",
       "      <td>heart</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "      <td>172.0</td>\n",
       "      <td>242.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>163.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15319</th>\n",
       "      <td>CHST9</td>\n",
       "      <td>liver</td>\n",
       "      <td>42.0</td>\n",
       "      <td>35</td>\n",
       "      <td>115.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15352</th>\n",
       "      <td>CHURC1</td>\n",
       "      <td>heart</td>\n",
       "      <td>997.0</td>\n",
       "      <td>425</td>\n",
       "      <td>1123.0</td>\n",
       "      <td>1329.0</td>\n",
       "      <td>572.0</td>\n",
       "      <td>206.0</td>\n",
       "      <td>126.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15353</th>\n",
       "      <td>CHURC1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>709.0</td>\n",
       "      <td>632</td>\n",
       "      <td>967.0</td>\n",
       "      <td>1470.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>503.0</td>\n",
       "      <td>258.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15493</th>\n",
       "      <td>CIZ1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1376.0</td>\n",
       "      <td>957</td>\n",
       "      <td>2208.0</td>\n",
       "      <td>1878.0</td>\n",
       "      <td>419.0</td>\n",
       "      <td>330.0</td>\n",
       "      <td>502.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15494</th>\n",
       "      <td>CIZ1</td>\n",
       "      <td>liver</td>\n",
       "      <td>365.0</td>\n",
       "      <td>263</td>\n",
       "      <td>1203.0</td>\n",
       "      <td>522.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>681.0</td>\n",
       "      <td>157.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15956</th>\n",
       "      <td>CLIC6</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>16.0</td>\n",
       "      <td>26</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15959</th>\n",
       "      <td>CLIC6</td>\n",
       "      <td>liver</td>\n",
       "      <td>8.0</td>\n",
       "      <td>11</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16158</th>\n",
       "      <td>CLTRN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>927.0</td>\n",
       "      <td>1083</td>\n",
       "      <td>1592.0</td>\n",
       "      <td>2934.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>1342.0</td>\n",
       "      <td>509.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16159</th>\n",
       "      <td>CLTRN</td>\n",
       "      <td>liver</td>\n",
       "      <td>27.0</td>\n",
       "      <td>7</td>\n",
       "      <td>40.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16308</th>\n",
       "      <td>CMYA5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>583.0</td>\n",
       "      <td>376</td>\n",
       "      <td>850.0</td>\n",
       "      <td>1066.0</td>\n",
       "      <td>207.0</td>\n",
       "      <td>216.0</td>\n",
       "      <td>267.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16471</th>\n",
       "      <td>CNOT3</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1476.0</td>\n",
       "      <td>1720</td>\n",
       "      <td>1297.0</td>\n",
       "      <td>1291.0</td>\n",
       "      <td>244.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>179.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16472</th>\n",
       "      <td>CNOT3</td>\n",
       "      <td>heart</td>\n",
       "      <td>409.0</td>\n",
       "      <td>349</td>\n",
       "      <td>290.0</td>\n",
       "      <td>312.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16474</th>\n",
       "      <td>CNOT3</td>\n",
       "      <td>liver</td>\n",
       "      <td>412.0</td>\n",
       "      <td>237</td>\n",
       "      <td>612.0</td>\n",
       "      <td>668.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>200.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17183</th>\n",
       "      <td>COQ4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1060.0</td>\n",
       "      <td>790</td>\n",
       "      <td>1353.0</td>\n",
       "      <td>2417.0</td>\n",
       "      <td>270.0</td>\n",
       "      <td>1064.0</td>\n",
       "      <td>293.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17478</th>\n",
       "      <td>CPEB3</td>\n",
       "      <td>kidney</td>\n",
       "      <td>200.0</td>\n",
       "      <td>145</td>\n",
       "      <td>254.0</td>\n",
       "      <td>489.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>235.0</td>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17479</th>\n",
       "      <td>CPEB3</td>\n",
       "      <td>liver</td>\n",
       "      <td>201.0</td>\n",
       "      <td>338</td>\n",
       "      <td>130.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19421</th>\n",
       "      <td>CYTIP</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>10.0</td>\n",
       "      <td>15</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19424</th>\n",
       "      <td>CYTIP</td>\n",
       "      <td>liver</td>\n",
       "      <td>307.0</td>\n",
       "      <td>96</td>\n",
       "      <td>77.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>211.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19467</th>\n",
       "      <td>DACH1</td>\n",
       "      <td>heart</td>\n",
       "      <td>240.0</td>\n",
       "      <td>224</td>\n",
       "      <td>99.0</td>\n",
       "      <td>186.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19598</th>\n",
       "      <td>DAZAP2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>3164.0</td>\n",
       "      <td>3021</td>\n",
       "      <td>5429.0</td>\n",
       "      <td>6962.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>1533.0</td>\n",
       "      <td>2265.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19606</th>\n",
       "      <td>DBF4</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>50.0</td>\n",
       "      <td>31</td>\n",
       "      <td>30.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19608</th>\n",
       "      <td>DBF4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>15.0</td>\n",
       "      <td>19</td>\n",
       "      <td>39.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19617</th>\n",
       "      <td>DBH</td>\n",
       "      <td>heart</td>\n",
       "      <td>6.0</td>\n",
       "      <td>16</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19666</th>\n",
       "      <td>DBX2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>7.0</td>\n",
       "      <td>8</td>\n",
       "      <td>11.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19778</th>\n",
       "      <td>DCBLD1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>150.0</td>\n",
       "      <td>155</td>\n",
       "      <td>189.0</td>\n",
       "      <td>195.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20156</th>\n",
       "      <td>DDX31</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>306.0</td>\n",
       "      <td>312</td>\n",
       "      <td>265.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>41.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20157</th>\n",
       "      <td>DDX31</td>\n",
       "      <td>heart</td>\n",
       "      <td>42.0</td>\n",
       "      <td>24</td>\n",
       "      <td>45.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20158</th>\n",
       "      <td>DDX31</td>\n",
       "      <td>kidney</td>\n",
       "      <td>125.0</td>\n",
       "      <td>135</td>\n",
       "      <td>149.0</td>\n",
       "      <td>193.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20551</th>\n",
       "      <td>DENND4A</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1049.0</td>\n",
       "      <td>1661</td>\n",
       "      <td>854.0</td>\n",
       "      <td>820.0</td>\n",
       "      <td>612.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>195.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20552</th>\n",
       "      <td>DENND4A</td>\n",
       "      <td>heart</td>\n",
       "      <td>215.0</td>\n",
       "      <td>177</td>\n",
       "      <td>400.0</td>\n",
       "      <td>332.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>117.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20553</th>\n",
       "      <td>DENND4A</td>\n",
       "      <td>kidney</td>\n",
       "      <td>319.0</td>\n",
       "      <td>267</td>\n",
       "      <td>385.0</td>\n",
       "      <td>709.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>324.0</td>\n",
       "      <td>66.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20723</th>\n",
       "      <td>DGKB</td>\n",
       "      <td>kidney</td>\n",
       "      <td>22.0</td>\n",
       "      <td>30</td>\n",
       "      <td>47.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20727</th>\n",
       "      <td>DGKD</td>\n",
       "      <td>heart</td>\n",
       "      <td>680.0</td>\n",
       "      <td>658</td>\n",
       "      <td>468.0</td>\n",
       "      <td>484.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>174.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20728</th>\n",
       "      <td>DGKD</td>\n",
       "      <td>kidney</td>\n",
       "      <td>749.0</td>\n",
       "      <td>735</td>\n",
       "      <td>644.0</td>\n",
       "      <td>692.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20729</th>\n",
       "      <td>DGKD</td>\n",
       "      <td>liver</td>\n",
       "      <td>463.0</td>\n",
       "      <td>505</td>\n",
       "      <td>426.0</td>\n",
       "      <td>197.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20987</th>\n",
       "      <td>DICER1</td>\n",
       "      <td>heart</td>\n",
       "      <td>496.0</td>\n",
       "      <td>418</td>\n",
       "      <td>521.0</td>\n",
       "      <td>754.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>233.0</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20988</th>\n",
       "      <td>DICER1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>589.0</td>\n",
       "      <td>492</td>\n",
       "      <td>786.0</td>\n",
       "      <td>1172.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>386.0</td>\n",
       "      <td>197.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21026</th>\n",
       "      <td>DIP2B</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>3304.0</td>\n",
       "      <td>4884</td>\n",
       "      <td>2891.0</td>\n",
       "      <td>2409.0</td>\n",
       "      <td>1580.0</td>\n",
       "      <td>482.0</td>\n",
       "      <td>413.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21028</th>\n",
       "      <td>DIP2B</td>\n",
       "      <td>kidney</td>\n",
       "      <td>425.0</td>\n",
       "      <td>350</td>\n",
       "      <td>477.0</td>\n",
       "      <td>823.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>346.0</td>\n",
       "      <td>52.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21068</th>\n",
       "      <td>DIS3L</td>\n",
       "      <td>kidney</td>\n",
       "      <td>344.0</td>\n",
       "      <td>323</td>\n",
       "      <td>349.0</td>\n",
       "      <td>706.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>357.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21296</th>\n",
       "      <td>DMD</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>792.0</td>\n",
       "      <td>737</td>\n",
       "      <td>704.0</td>\n",
       "      <td>499.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>33.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21297</th>\n",
       "      <td>DMD</td>\n",
       "      <td>heart</td>\n",
       "      <td>2500.0</td>\n",
       "      <td>832</td>\n",
       "      <td>3684.0</td>\n",
       "      <td>5423.0</td>\n",
       "      <td>1668.0</td>\n",
       "      <td>1739.0</td>\n",
       "      <td>1184.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21298</th>\n",
       "      <td>DMD</td>\n",
       "      <td>kidney</td>\n",
       "      <td>156.0</td>\n",
       "      <td>125</td>\n",
       "      <td>275.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>99.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21299</th>\n",
       "      <td>DMD</td>\n",
       "      <td>liver</td>\n",
       "      <td>1061.0</td>\n",
       "      <td>546</td>\n",
       "      <td>265.0</td>\n",
       "      <td>468.0</td>\n",
       "      <td>515.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>78.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21303</th>\n",
       "      <td>DMGDH</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>754</td>\n",
       "      <td>2758.0</td>\n",
       "      <td>6399.0</td>\n",
       "      <td>372.0</td>\n",
       "      <td>3641.0</td>\n",
       "      <td>1632.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21652</th>\n",
       "      <td>DNAJC28</td>\n",
       "      <td>heart</td>\n",
       "      <td>38.0</td>\n",
       "      <td>10</td>\n",
       "      <td>49.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21653</th>\n",
       "      <td>DNAJC28</td>\n",
       "      <td>kidney</td>\n",
       "      <td>30.0</td>\n",
       "      <td>15</td>\n",
       "      <td>49.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21654</th>\n",
       "      <td>DNAJC28</td>\n",
       "      <td>liver</td>\n",
       "      <td>39.0</td>\n",
       "      <td>24</td>\n",
       "      <td>23.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21772</th>\n",
       "      <td>DNM1</td>\n",
       "      <td>heart</td>\n",
       "      <td>215.0</td>\n",
       "      <td>206</td>\n",
       "      <td>126.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>80.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21958</th>\n",
       "      <td>DONSON</td>\n",
       "      <td>kidney</td>\n",
       "      <td>128.0</td>\n",
       "      <td>128</td>\n",
       "      <td>141.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22068</th>\n",
       "      <td>DPM2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>718.0</td>\n",
       "      <td>628</td>\n",
       "      <td>1101.0</td>\n",
       "      <td>719.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>382.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22932</th>\n",
       "      <td>ECEL1</td>\n",
       "      <td>heart</td>\n",
       "      <td>20.0</td>\n",
       "      <td>14</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22933</th>\n",
       "      <td>ECEL1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>32.0</td>\n",
       "      <td>27</td>\n",
       "      <td>11.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23613</th>\n",
       "      <td>EIF3H</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1404.0</td>\n",
       "      <td>1401</td>\n",
       "      <td>1676.0</td>\n",
       "      <td>2877.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1201.0</td>\n",
       "      <td>272.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23673</th>\n",
       "      <td>EIF4E2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1046.0</td>\n",
       "      <td>783</td>\n",
       "      <td>1057.0</td>\n",
       "      <td>1667.0</td>\n",
       "      <td>263.0</td>\n",
       "      <td>610.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24169</th>\n",
       "      <td>ENDOG</td>\n",
       "      <td>liver</td>\n",
       "      <td>242.0</td>\n",
       "      <td>178</td>\n",
       "      <td>285.0</td>\n",
       "      <td>370.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24461</th>\n",
       "      <td>EPHA5</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>40.0</td>\n",
       "      <td>89</td>\n",
       "      <td>314.0</td>\n",
       "      <td>230.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>141.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24463</th>\n",
       "      <td>EPHA5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5</td>\n",
       "      <td>7.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24841</th>\n",
       "      <td>ERMN</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>692.0</td>\n",
       "      <td>732</td>\n",
       "      <td>519.0</td>\n",
       "      <td>361.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24863</th>\n",
       "      <td>ERO1A</td>\n",
       "      <td>kidney</td>\n",
       "      <td>336.0</td>\n",
       "      <td>333</td>\n",
       "      <td>358.0</td>\n",
       "      <td>378.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24977</th>\n",
       "      <td>ESR2</td>\n",
       "      <td>heart</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25136</th>\n",
       "      <td>EVA1A</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3</td>\n",
       "      <td>25.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25138</th>\n",
       "      <td>EVA1A</td>\n",
       "      <td>kidney</td>\n",
       "      <td>220.0</td>\n",
       "      <td>203</td>\n",
       "      <td>401.0</td>\n",
       "      <td>577.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>181.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25139</th>\n",
       "      <td>EVA1A</td>\n",
       "      <td>liver</td>\n",
       "      <td>752.0</td>\n",
       "      <td>889</td>\n",
       "      <td>329.0</td>\n",
       "      <td>575.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>246.0</td>\n",
       "      <td>177.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25648</th>\n",
       "      <td>FAM102A</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1524.0</td>\n",
       "      <td>1827</td>\n",
       "      <td>2927.0</td>\n",
       "      <td>2381.0</td>\n",
       "      <td>303.0</td>\n",
       "      <td>546.0</td>\n",
       "      <td>554.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26073</th>\n",
       "      <td>FAM181B</td>\n",
       "      <td>kidney</td>\n",
       "      <td>58.0</td>\n",
       "      <td>52</td>\n",
       "      <td>179.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26162</th>\n",
       "      <td>FAM199X</td>\n",
       "      <td>heart</td>\n",
       "      <td>306.0</td>\n",
       "      <td>178</td>\n",
       "      <td>310.0</td>\n",
       "      <td>376.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26163</th>\n",
       "      <td>FAM199X</td>\n",
       "      <td>kidney</td>\n",
       "      <td>461.0</td>\n",
       "      <td>437</td>\n",
       "      <td>572.0</td>\n",
       "      <td>769.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>197.0</td>\n",
       "      <td>111.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26958</th>\n",
       "      <td>FAT4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>268.0</td>\n",
       "      <td>239</td>\n",
       "      <td>458.0</td>\n",
       "      <td>848.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>390.0</td>\n",
       "      <td>190.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27797</th>\n",
       "      <td>FGFBP3</td>\n",
       "      <td>heart</td>\n",
       "      <td>23.0</td>\n",
       "      <td>14</td>\n",
       "      <td>69.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27877</th>\n",
       "      <td>FHIT</td>\n",
       "      <td>heart</td>\n",
       "      <td>28.0</td>\n",
       "      <td>12</td>\n",
       "      <td>37.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27879</th>\n",
       "      <td>FHIT</td>\n",
       "      <td>liver</td>\n",
       "      <td>71.0</td>\n",
       "      <td>35</td>\n",
       "      <td>10.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28091</th>\n",
       "      <td>FLG</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>102.0</td>\n",
       "      <td>138</td>\n",
       "      <td>51.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>51.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28092</th>\n",
       "      <td>FLG</td>\n",
       "      <td>heart</td>\n",
       "      <td>22.0</td>\n",
       "      <td>26</td>\n",
       "      <td>4.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28351</th>\n",
       "      <td>FNTB</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>668.0</td>\n",
       "      <td>697</td>\n",
       "      <td>532.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>138.0</td>\n",
       "      <td>136.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28741</th>\n",
       "      <td>FRMD6</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>357.0</td>\n",
       "      <td>163</td>\n",
       "      <td>554.0</td>\n",
       "      <td>379.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29036</th>\n",
       "      <td>FUT8</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1103.0</td>\n",
       "      <td>1127</td>\n",
       "      <td>670.0</td>\n",
       "      <td>650.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>433.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29037</th>\n",
       "      <td>FUT8</td>\n",
       "      <td>heart</td>\n",
       "      <td>214.0</td>\n",
       "      <td>271</td>\n",
       "      <td>168.0</td>\n",
       "      <td>196.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29039</th>\n",
       "      <td>FUT8</td>\n",
       "      <td>liver</td>\n",
       "      <td>112.0</td>\n",
       "      <td>112</td>\n",
       "      <td>106.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29666</th>\n",
       "      <td>GAS2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>40.0</td>\n",
       "      <td>35</td>\n",
       "      <td>34.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29669</th>\n",
       "      <td>GAS2</td>\n",
       "      <td>liver</td>\n",
       "      <td>391.0</td>\n",
       "      <td>200</td>\n",
       "      <td>47.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29799</th>\n",
       "      <td>GBGT1</td>\n",
       "      <td>liver</td>\n",
       "      <td>131.0</td>\n",
       "      <td>58</td>\n",
       "      <td>150.0</td>\n",
       "      <td>247.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29879</th>\n",
       "      <td>GCFC2</td>\n",
       "      <td>liver</td>\n",
       "      <td>318.0</td>\n",
       "      <td>176</td>\n",
       "      <td>83.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>142.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>93.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30382</th>\n",
       "      <td>GIGYF2</td>\n",
       "      <td>heart</td>\n",
       "      <td>1506.0</td>\n",
       "      <td>899</td>\n",
       "      <td>1724.0</td>\n",
       "      <td>1908.0</td>\n",
       "      <td>607.0</td>\n",
       "      <td>184.0</td>\n",
       "      <td>218.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30383</th>\n",
       "      <td>GIGYF2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1478.0</td>\n",
       "      <td>1078</td>\n",
       "      <td>1666.0</td>\n",
       "      <td>2284.0</td>\n",
       "      <td>400.0</td>\n",
       "      <td>618.0</td>\n",
       "      <td>188.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30667</th>\n",
       "      <td>GLE1</td>\n",
       "      <td>heart</td>\n",
       "      <td>314.0</td>\n",
       "      <td>194</td>\n",
       "      <td>342.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30668</th>\n",
       "      <td>GLE1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>409.0</td>\n",
       "      <td>348</td>\n",
       "      <td>603.0</td>\n",
       "      <td>729.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>194.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30808</th>\n",
       "      <td>GLRX5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1568.0</td>\n",
       "      <td>1675</td>\n",
       "      <td>2047.0</td>\n",
       "      <td>2494.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>447.0</td>\n",
       "      <td>372.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30809</th>\n",
       "      <td>GLRX5</td>\n",
       "      <td>liver</td>\n",
       "      <td>1323.0</td>\n",
       "      <td>781</td>\n",
       "      <td>1589.0</td>\n",
       "      <td>1613.0</td>\n",
       "      <td>542.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>266.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31116</th>\n",
       "      <td>GNG2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>645.0</td>\n",
       "      <td>533</td>\n",
       "      <td>857.0</td>\n",
       "      <td>675.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31244</th>\n",
       "      <td>GOLGA2</td>\n",
       "      <td>liver</td>\n",
       "      <td>647.0</td>\n",
       "      <td>472</td>\n",
       "      <td>3105.0</td>\n",
       "      <td>913.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>2192.0</td>\n",
       "      <td>266.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31332</th>\n",
       "      <td>GOPC</td>\n",
       "      <td>heart</td>\n",
       "      <td>660.0</td>\n",
       "      <td>342</td>\n",
       "      <td>686.0</td>\n",
       "      <td>681.0</td>\n",
       "      <td>318.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31333</th>\n",
       "      <td>GOPC</td>\n",
       "      <td>kidney</td>\n",
       "      <td>446.0</td>\n",
       "      <td>407</td>\n",
       "      <td>495.0</td>\n",
       "      <td>852.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>357.0</td>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31541</th>\n",
       "      <td>GPD2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1260.0</td>\n",
       "      <td>1537</td>\n",
       "      <td>830.0</td>\n",
       "      <td>889.0</td>\n",
       "      <td>277.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>371.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31543</th>\n",
       "      <td>GPD2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>320.0</td>\n",
       "      <td>284</td>\n",
       "      <td>359.0</td>\n",
       "      <td>528.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>39.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31666</th>\n",
       "      <td>GPR137C</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>600.0</td>\n",
       "      <td>688</td>\n",
       "      <td>440.0</td>\n",
       "      <td>382.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>160.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31914</th>\n",
       "      <td>GPR55</td>\n",
       "      <td>liver</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32076</th>\n",
       "      <td>GPX2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>37.0</td>\n",
       "      <td>32</td>\n",
       "      <td>18.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32401</th>\n",
       "      <td>GRM7</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>3.0</td>\n",
       "      <td>63</td>\n",
       "      <td>133.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32497</th>\n",
       "      <td>GSDME</td>\n",
       "      <td>heart</td>\n",
       "      <td>256.0</td>\n",
       "      <td>251</td>\n",
       "      <td>122.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>129.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32746</th>\n",
       "      <td>GTF3C4</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1084.0</td>\n",
       "      <td>1640</td>\n",
       "      <td>863.0</td>\n",
       "      <td>899.0</td>\n",
       "      <td>556.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>185.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32747</th>\n",
       "      <td>GTF3C4</td>\n",
       "      <td>heart</td>\n",
       "      <td>226.0</td>\n",
       "      <td>195</td>\n",
       "      <td>318.0</td>\n",
       "      <td>324.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>92.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32748</th>\n",
       "      <td>GTF3C4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>247.0</td>\n",
       "      <td>282</td>\n",
       "      <td>473.0</td>\n",
       "      <td>654.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>191.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33606</th>\n",
       "      <td>HEG1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>790.0</td>\n",
       "      <td>637</td>\n",
       "      <td>569.0</td>\n",
       "      <td>343.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>226.0</td>\n",
       "      <td>68.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33607</th>\n",
       "      <td>HEG1</td>\n",
       "      <td>heart</td>\n",
       "      <td>4308.0</td>\n",
       "      <td>3188</td>\n",
       "      <td>4617.0</td>\n",
       "      <td>10642.0</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>6025.0</td>\n",
       "      <td>309.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33609</th>\n",
       "      <td>HEG1</td>\n",
       "      <td>liver</td>\n",
       "      <td>482.0</td>\n",
       "      <td>481</td>\n",
       "      <td>1367.0</td>\n",
       "      <td>609.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>758.0</td>\n",
       "      <td>127.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34886</th>\n",
       "      <td>HRNR</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>8.0</td>\n",
       "      <td>13</td>\n",
       "      <td>56.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34887</th>\n",
       "      <td>HRNR</td>\n",
       "      <td>heart</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5</td>\n",
       "      <td>11.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34888</th>\n",
       "      <td>HRNR</td>\n",
       "      <td>kidney</td>\n",
       "      <td>4.0</td>\n",
       "      <td>16</td>\n",
       "      <td>18.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35122</th>\n",
       "      <td>HSPA2</td>\n",
       "      <td>heart</td>\n",
       "      <td>645.0</td>\n",
       "      <td>320</td>\n",
       "      <td>239.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>325.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>81.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35276</th>\n",
       "      <td>HTR2B</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35278</th>\n",
       "      <td>HTR2B</td>\n",
       "      <td>kidney</td>\n",
       "      <td>16.0</td>\n",
       "      <td>16</td>\n",
       "      <td>31.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35556</th>\n",
       "      <td>IDE</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>382.0</td>\n",
       "      <td>425</td>\n",
       "      <td>376.0</td>\n",
       "      <td>157.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>219.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35558</th>\n",
       "      <td>IDE</td>\n",
       "      <td>kidney</td>\n",
       "      <td>363.0</td>\n",
       "      <td>369</td>\n",
       "      <td>387.0</td>\n",
       "      <td>520.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>133.0</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35818</th>\n",
       "      <td>IFNGR2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>816.0</td>\n",
       "      <td>559</td>\n",
       "      <td>962.0</td>\n",
       "      <td>1409.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>447.0</td>\n",
       "      <td>146.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36948</th>\n",
       "      <td>INTS14</td>\n",
       "      <td>kidney</td>\n",
       "      <td>317.0</td>\n",
       "      <td>282</td>\n",
       "      <td>411.0</td>\n",
       "      <td>579.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>168.0</td>\n",
       "      <td>94.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37413</th>\n",
       "      <td>ISPD</td>\n",
       "      <td>kidney</td>\n",
       "      <td>62.0</td>\n",
       "      <td>39</td>\n",
       "      <td>66.0</td>\n",
       "      <td>168.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37571</th>\n",
       "      <td>ITGB5</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>691.0</td>\n",
       "      <td>805</td>\n",
       "      <td>664.0</td>\n",
       "      <td>358.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>306.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37573</th>\n",
       "      <td>ITGB5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1748.0</td>\n",
       "      <td>1458</td>\n",
       "      <td>2581.0</td>\n",
       "      <td>3610.0</td>\n",
       "      <td>290.0</td>\n",
       "      <td>1029.0</td>\n",
       "      <td>833.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37713</th>\n",
       "      <td>ITSN1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>571.0</td>\n",
       "      <td>437</td>\n",
       "      <td>840.0</td>\n",
       "      <td>1123.0</td>\n",
       "      <td>134.0</td>\n",
       "      <td>283.0</td>\n",
       "      <td>269.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37901</th>\n",
       "      <td>JMY</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>953.0</td>\n",
       "      <td>1124</td>\n",
       "      <td>740.0</td>\n",
       "      <td>593.0</td>\n",
       "      <td>171.0</td>\n",
       "      <td>147.0</td>\n",
       "      <td>213.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37902</th>\n",
       "      <td>JMY</td>\n",
       "      <td>heart</td>\n",
       "      <td>318.0</td>\n",
       "      <td>199</td>\n",
       "      <td>388.0</td>\n",
       "      <td>514.0</td>\n",
       "      <td>119.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>70.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37903</th>\n",
       "      <td>JMY</td>\n",
       "      <td>kidney</td>\n",
       "      <td>398.0</td>\n",
       "      <td>271</td>\n",
       "      <td>429.0</td>\n",
       "      <td>762.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>333.0</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38266</th>\n",
       "      <td>KCNE1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>11.0</td>\n",
       "      <td>19</td>\n",
       "      <td>6.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38648</th>\n",
       "      <td>KCTD1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>752.0</td>\n",
       "      <td>442</td>\n",
       "      <td>847.0</td>\n",
       "      <td>807.0</td>\n",
       "      <td>310.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>55.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39183</th>\n",
       "      <td>KIF11</td>\n",
       "      <td>kidney</td>\n",
       "      <td>27.0</td>\n",
       "      <td>26</td>\n",
       "      <td>40.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39527</th>\n",
       "      <td>KLF13</td>\n",
       "      <td>heart</td>\n",
       "      <td>1209.0</td>\n",
       "      <td>1397</td>\n",
       "      <td>1487.0</td>\n",
       "      <td>1452.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>55.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39528</th>\n",
       "      <td>KLF13</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1820.0</td>\n",
       "      <td>1525</td>\n",
       "      <td>2679.0</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>295.0</td>\n",
       "      <td>287.0</td>\n",
       "      <td>859.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39529</th>\n",
       "      <td>KLF13</td>\n",
       "      <td>liver</td>\n",
       "      <td>906.0</td>\n",
       "      <td>776</td>\n",
       "      <td>2104.0</td>\n",
       "      <td>985.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>1119.0</td>\n",
       "      <td>79.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40558</th>\n",
       "      <td>KYAT1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>696.0</td>\n",
       "      <td>533</td>\n",
       "      <td>728.0</td>\n",
       "      <td>1505.0</td>\n",
       "      <td>163.0</td>\n",
       "      <td>777.0</td>\n",
       "      <td>32.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40559</th>\n",
       "      <td>KYAT1</td>\n",
       "      <td>liver</td>\n",
       "      <td>466.0</td>\n",
       "      <td>344</td>\n",
       "      <td>596.0</td>\n",
       "      <td>585.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>119.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40821</th>\n",
       "      <td>LARP4</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>441.0</td>\n",
       "      <td>554</td>\n",
       "      <td>405.0</td>\n",
       "      <td>305.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41206</th>\n",
       "      <td>LENG1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>153.0</td>\n",
       "      <td>202</td>\n",
       "      <td>234.0</td>\n",
       "      <td>209.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41211</th>\n",
       "      <td>LENG8</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>9409.0</td>\n",
       "      <td>12783</td>\n",
       "      <td>9346.0</td>\n",
       "      <td>8642.0</td>\n",
       "      <td>3374.0</td>\n",
       "      <td>704.0</td>\n",
       "      <td>63.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41212</th>\n",
       "      <td>LENG8</td>\n",
       "      <td>heart</td>\n",
       "      <td>2738.0</td>\n",
       "      <td>2130</td>\n",
       "      <td>1726.0</td>\n",
       "      <td>2045.0</td>\n",
       "      <td>608.0</td>\n",
       "      <td>319.0</td>\n",
       "      <td>85.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41214</th>\n",
       "      <td>LENG8</td>\n",
       "      <td>liver</td>\n",
       "      <td>1748.0</td>\n",
       "      <td>1053</td>\n",
       "      <td>3541.0</td>\n",
       "      <td>2210.0</td>\n",
       "      <td>695.0</td>\n",
       "      <td>1331.0</td>\n",
       "      <td>462.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41216</th>\n",
       "      <td>LENG9</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>24.0</td>\n",
       "      <td>19</td>\n",
       "      <td>16.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41217</th>\n",
       "      <td>LENG9</td>\n",
       "      <td>heart</td>\n",
       "      <td>35.0</td>\n",
       "      <td>31</td>\n",
       "      <td>23.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41219</th>\n",
       "      <td>LENG9</td>\n",
       "      <td>liver</td>\n",
       "      <td>44.0</td>\n",
       "      <td>9</td>\n",
       "      <td>283.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>224.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41527</th>\n",
       "      <td>LILRB3</td>\n",
       "      <td>heart</td>\n",
       "      <td>27.0</td>\n",
       "      <td>20</td>\n",
       "      <td>41.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41528</th>\n",
       "      <td>LILRB3</td>\n",
       "      <td>kidney</td>\n",
       "      <td>28.0</td>\n",
       "      <td>14</td>\n",
       "      <td>73.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41656</th>\n",
       "      <td>LINGO4</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>24.0</td>\n",
       "      <td>58</td>\n",
       "      <td>267.0</td>\n",
       "      <td>226.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>168.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41657</th>\n",
       "      <td>LINGO4</td>\n",
       "      <td>heart</td>\n",
       "      <td>70.0</td>\n",
       "      <td>12</td>\n",
       "      <td>164.0</td>\n",
       "      <td>119.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41658</th>\n",
       "      <td>LINGO4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>22.0</td>\n",
       "      <td>26</td>\n",
       "      <td>153.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>52.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41659</th>\n",
       "      <td>LINGO4</td>\n",
       "      <td>liver</td>\n",
       "      <td>41.0</td>\n",
       "      <td>30</td>\n",
       "      <td>197.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44053</th>\n",
       "      <td>LOC100612575</td>\n",
       "      <td>kidney</td>\n",
       "      <td>8.0</td>\n",
       "      <td>7</td>\n",
       "      <td>12.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44556</th>\n",
       "      <td>LOC100613592</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>15.0</td>\n",
       "      <td>14</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44557</th>\n",
       "      <td>LOC100613592</td>\n",
       "      <td>heart</td>\n",
       "      <td>5.0</td>\n",
       "      <td>7</td>\n",
       "      <td>15.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47211</th>\n",
       "      <td>LOC101059176</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>157.0</td>\n",
       "      <td>131</td>\n",
       "      <td>64.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>60.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47212</th>\n",
       "      <td>LOC101059176</td>\n",
       "      <td>heart</td>\n",
       "      <td>30.0</td>\n",
       "      <td>12</td>\n",
       "      <td>8.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48396</th>\n",
       "      <td>LOC104001534</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51763</th>\n",
       "      <td>LOC104003432</td>\n",
       "      <td>kidney</td>\n",
       "      <td>28.0</td>\n",
       "      <td>20</td>\n",
       "      <td>39.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51764</th>\n",
       "      <td>LOC104003432</td>\n",
       "      <td>liver</td>\n",
       "      <td>13.0</td>\n",
       "      <td>12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58606</th>\n",
       "      <td>LOC104007525</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>120.0</td>\n",
       "      <td>74</td>\n",
       "      <td>50.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86693</th>\n",
       "      <td>LOC107976709</td>\n",
       "      <td>kidney</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4</td>\n",
       "      <td>10.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86694</th>\n",
       "      <td>LOC107976709</td>\n",
       "      <td>liver</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110621</th>\n",
       "      <td>LOC453134</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>4.0</td>\n",
       "      <td>9</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110624</th>\n",
       "      <td>LOC453134</td>\n",
       "      <td>liver</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110681</th>\n",
       "      <td>LOC453285</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>87.0</td>\n",
       "      <td>116</td>\n",
       "      <td>16.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>71.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111017</th>\n",
       "      <td>LOC456283</td>\n",
       "      <td>heart</td>\n",
       "      <td>20.0</td>\n",
       "      <td>13</td>\n",
       "      <td>101.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111516</th>\n",
       "      <td>LOC460522</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>8.0</td>\n",
       "      <td>6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111524</th>\n",
       "      <td>LOC460532</td>\n",
       "      <td>liver</td>\n",
       "      <td>42.0</td>\n",
       "      <td>32</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112122</th>\n",
       "      <td>LOC464914</td>\n",
       "      <td>heart</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112236</th>\n",
       "      <td>LOC465512</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>534.0</td>\n",
       "      <td>532</td>\n",
       "      <td>188.0</td>\n",
       "      <td>466.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>278.0</td>\n",
       "      <td>66.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112239</th>\n",
       "      <td>LOC465512</td>\n",
       "      <td>liver</td>\n",
       "      <td>200.0</td>\n",
       "      <td>143</td>\n",
       "      <td>262.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112974</th>\n",
       "      <td>LOC467545</td>\n",
       "      <td>liver</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113711</th>\n",
       "      <td>LOC470670</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116148</th>\n",
       "      <td>LOC739470</td>\n",
       "      <td>kidney</td>\n",
       "      <td>104.0</td>\n",
       "      <td>149</td>\n",
       "      <td>217.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117298</th>\n",
       "      <td>LOC742641</td>\n",
       "      <td>kidney</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2</td>\n",
       "      <td>25.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120587</th>\n",
       "      <td>LRRC8A</td>\n",
       "      <td>heart</td>\n",
       "      <td>1541.0</td>\n",
       "      <td>1330</td>\n",
       "      <td>712.0</td>\n",
       "      <td>1285.0</td>\n",
       "      <td>211.0</td>\n",
       "      <td>573.0</td>\n",
       "      <td>45.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120589</th>\n",
       "      <td>LRRC8A</td>\n",
       "      <td>liver</td>\n",
       "      <td>839.0</td>\n",
       "      <td>992</td>\n",
       "      <td>2096.0</td>\n",
       "      <td>1308.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>788.0</td>\n",
       "      <td>316.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121748</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>327.0</td>\n",
       "      <td>330</td>\n",
       "      <td>442.0</td>\n",
       "      <td>543.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121749</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>liver</td>\n",
       "      <td>198.0</td>\n",
       "      <td>123</td>\n",
       "      <td>410.0</td>\n",
       "      <td>369.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>171.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122108</th>\n",
       "      <td>MARCH5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>318.0</td>\n",
       "      <td>296</td>\n",
       "      <td>345.0</td>\n",
       "      <td>548.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122296</th>\n",
       "      <td>MAX</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>1389.0</td>\n",
       "      <td>1337</td>\n",
       "      <td>793.0</td>\n",
       "      <td>1114.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>321.0</td>\n",
       "      <td>223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123046</th>\n",
       "      <td>MEOX2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>13.0</td>\n",
       "      <td>13</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123049</th>\n",
       "      <td>MEOX2</td>\n",
       "      <td>liver</td>\n",
       "      <td>57.0</td>\n",
       "      <td>29</td>\n",
       "      <td>15.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127716</th>\n",
       "      <td>MPP6</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>391.0</td>\n",
       "      <td>413</td>\n",
       "      <td>283.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>108.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127718</th>\n",
       "      <td>MPP6</td>\n",
       "      <td>kidney</td>\n",
       "      <td>145.0</td>\n",
       "      <td>163</td>\n",
       "      <td>230.0</td>\n",
       "      <td>357.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>67.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128001</th>\n",
       "      <td>MRPL19</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>547.0</td>\n",
       "      <td>549</td>\n",
       "      <td>237.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128004</th>\n",
       "      <td>MRPL19</td>\n",
       "      <td>liver</td>\n",
       "      <td>776.0</td>\n",
       "      <td>526</td>\n",
       "      <td>368.0</td>\n",
       "      <td>250.0</td>\n",
       "      <td>250.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>158.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128186</th>\n",
       "      <td>MRPL9</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>903.0</td>\n",
       "      <td>978</td>\n",
       "      <td>899.0</td>\n",
       "      <td>668.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>231.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128321</th>\n",
       "      <td>MRPS6</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>158.0</td>\n",
       "      <td>219</td>\n",
       "      <td>278.0</td>\n",
       "      <td>223.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128323</th>\n",
       "      <td>MRPS6</td>\n",
       "      <td>kidney</td>\n",
       "      <td>455.0</td>\n",
       "      <td>362</td>\n",
       "      <td>522.0</td>\n",
       "      <td>794.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>272.0</td>\n",
       "      <td>67.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128793</th>\n",
       "      <td>MTHFD1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1528.0</td>\n",
       "      <td>1448</td>\n",
       "      <td>1759.0</td>\n",
       "      <td>4139.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>2380.0</td>\n",
       "      <td>231.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129543</th>\n",
       "      <td>MYO10</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1914.0</td>\n",
       "      <td>1383</td>\n",
       "      <td>2184.0</td>\n",
       "      <td>3619.0</td>\n",
       "      <td>531.0</td>\n",
       "      <td>1435.0</td>\n",
       "      <td>270.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129992</th>\n",
       "      <td>NAIF1</td>\n",
       "      <td>heart</td>\n",
       "      <td>56.0</td>\n",
       "      <td>68</td>\n",
       "      <td>53.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129993</th>\n",
       "      <td>NAIF1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>88.0</td>\n",
       "      <td>77</td>\n",
       "      <td>166.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>57.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129994</th>\n",
       "      <td>NAIF1</td>\n",
       "      <td>liver</td>\n",
       "      <td>54.0</td>\n",
       "      <td>48</td>\n",
       "      <td>180.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130626</th>\n",
       "      <td>NDUFA3</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>667.0</td>\n",
       "      <td>496</td>\n",
       "      <td>483.0</td>\n",
       "      <td>451.0</td>\n",
       "      <td>171.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130986</th>\n",
       "      <td>NEK7</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>426.0</td>\n",
       "      <td>382</td>\n",
       "      <td>191.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>191.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130988</th>\n",
       "      <td>NEK7</td>\n",
       "      <td>kidney</td>\n",
       "      <td>547.0</td>\n",
       "      <td>412</td>\n",
       "      <td>577.0</td>\n",
       "      <td>1065.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>488.0</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131027</th>\n",
       "      <td>NELL2</td>\n",
       "      <td>heart</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131028</th>\n",
       "      <td>NELL2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "      <td>41.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>32.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131029</th>\n",
       "      <td>NELL2</td>\n",
       "      <td>liver</td>\n",
       "      <td>4.0</td>\n",
       "      <td>50</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131264</th>\n",
       "      <td>NFIL3</td>\n",
       "      <td>liver</td>\n",
       "      <td>4524.0</td>\n",
       "      <td>498</td>\n",
       "      <td>440.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>4026.0</td>\n",
       "      <td>197.0</td>\n",
       "      <td>58.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131787</th>\n",
       "      <td>NLRP12</td>\n",
       "      <td>heart</td>\n",
       "      <td>18.0</td>\n",
       "      <td>11</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131789</th>\n",
       "      <td>NLRP12</td>\n",
       "      <td>liver</td>\n",
       "      <td>66.0</td>\n",
       "      <td>17</td>\n",
       "      <td>11.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132714</th>\n",
       "      <td>NRG3</td>\n",
       "      <td>liver</td>\n",
       "      <td>15.0</td>\n",
       "      <td>6</td>\n",
       "      <td>38.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133411</th>\n",
       "      <td>NUS1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>169.0</td>\n",
       "      <td>184</td>\n",
       "      <td>121.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>48.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133413</th>\n",
       "      <td>NUS1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>160.0</td>\n",
       "      <td>183</td>\n",
       "      <td>190.0</td>\n",
       "      <td>456.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>266.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133606</th>\n",
       "      <td>OAZ3</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>80.0</td>\n",
       "      <td>114</td>\n",
       "      <td>79.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133608</th>\n",
       "      <td>OAZ3</td>\n",
       "      <td>kidney</td>\n",
       "      <td>9.0</td>\n",
       "      <td>7</td>\n",
       "      <td>34.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133702</th>\n",
       "      <td>ODF2</td>\n",
       "      <td>heart</td>\n",
       "      <td>268.0</td>\n",
       "      <td>260</td>\n",
       "      <td>414.0</td>\n",
       "      <td>361.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>93.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133703</th>\n",
       "      <td>ODF2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>430.0</td>\n",
       "      <td>400</td>\n",
       "      <td>1064.0</td>\n",
       "      <td>862.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>432.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135136</th>\n",
       "      <td>OSBPL11</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>384.0</td>\n",
       "      <td>340</td>\n",
       "      <td>293.0</td>\n",
       "      <td>196.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>47.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135138</th>\n",
       "      <td>OSBPL11</td>\n",
       "      <td>kidney</td>\n",
       "      <td>269.0</td>\n",
       "      <td>193</td>\n",
       "      <td>274.0</td>\n",
       "      <td>343.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135152</th>\n",
       "      <td>OSBPL3</td>\n",
       "      <td>heart</td>\n",
       "      <td>137.0</td>\n",
       "      <td>93</td>\n",
       "      <td>148.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135182</th>\n",
       "      <td>OSCAR</td>\n",
       "      <td>heart</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135357</th>\n",
       "      <td>OTUD7A</td>\n",
       "      <td>heart</td>\n",
       "      <td>23.0</td>\n",
       "      <td>36</td>\n",
       "      <td>17.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135358</th>\n",
       "      <td>OTUD7A</td>\n",
       "      <td>kidney</td>\n",
       "      <td>54.0</td>\n",
       "      <td>47</td>\n",
       "      <td>76.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135658</th>\n",
       "      <td>PACRG</td>\n",
       "      <td>kidney</td>\n",
       "      <td>148.0</td>\n",
       "      <td>91</td>\n",
       "      <td>180.0</td>\n",
       "      <td>152.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135659</th>\n",
       "      <td>PACRG</td>\n",
       "      <td>liver</td>\n",
       "      <td>60.0</td>\n",
       "      <td>21</td>\n",
       "      <td>7.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135953</th>\n",
       "      <td>PAPD4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>376.0</td>\n",
       "      <td>306</td>\n",
       "      <td>482.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>223.0</td>\n",
       "      <td>106.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135987</th>\n",
       "      <td>PAPPA</td>\n",
       "      <td>heart</td>\n",
       "      <td>56.0</td>\n",
       "      <td>79</td>\n",
       "      <td>31.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135988</th>\n",
       "      <td>PAPPA</td>\n",
       "      <td>kidney</td>\n",
       "      <td>746.0</td>\n",
       "      <td>325</td>\n",
       "      <td>1940.0</td>\n",
       "      <td>807.0</td>\n",
       "      <td>421.0</td>\n",
       "      <td>1133.0</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136549</th>\n",
       "      <td>PCDH9</td>\n",
       "      <td>liver</td>\n",
       "      <td>8.0</td>\n",
       "      <td>54</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138267</th>\n",
       "      <td>PHYHD1</td>\n",
       "      <td>heart</td>\n",
       "      <td>584.0</td>\n",
       "      <td>260</td>\n",
       "      <td>101.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>324.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>159.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138268</th>\n",
       "      <td>PHYHD1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1224.0</td>\n",
       "      <td>966</td>\n",
       "      <td>406.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>258.0</td>\n",
       "      <td>232.0</td>\n",
       "      <td>560.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138269</th>\n",
       "      <td>PHYHD1</td>\n",
       "      <td>liver</td>\n",
       "      <td>2916.0</td>\n",
       "      <td>1728</td>\n",
       "      <td>1077.0</td>\n",
       "      <td>1636.0</td>\n",
       "      <td>1188.0</td>\n",
       "      <td>559.0</td>\n",
       "      <td>92.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138708</th>\n",
       "      <td>PIR</td>\n",
       "      <td>kidney</td>\n",
       "      <td>66.0</td>\n",
       "      <td>61</td>\n",
       "      <td>85.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138907</th>\n",
       "      <td>PKN3</td>\n",
       "      <td>heart</td>\n",
       "      <td>448.0</td>\n",
       "      <td>620</td>\n",
       "      <td>173.0</td>\n",
       "      <td>236.0</td>\n",
       "      <td>172.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>212.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138908</th>\n",
       "      <td>PKN3</td>\n",
       "      <td>kidney</td>\n",
       "      <td>345.0</td>\n",
       "      <td>318</td>\n",
       "      <td>218.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139342</th>\n",
       "      <td>PLEKHG3</td>\n",
       "      <td>heart</td>\n",
       "      <td>301.0</td>\n",
       "      <td>373</td>\n",
       "      <td>216.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>73.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139344</th>\n",
       "      <td>PLEKHG3</td>\n",
       "      <td>liver</td>\n",
       "      <td>264.0</td>\n",
       "      <td>315</td>\n",
       "      <td>1439.0</td>\n",
       "      <td>512.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>927.0</td>\n",
       "      <td>197.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139506</th>\n",
       "      <td>PLOD2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>194.0</td>\n",
       "      <td>224</td>\n",
       "      <td>160.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139608</th>\n",
       "      <td>PLSCR1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>332.0</td>\n",
       "      <td>365</td>\n",
       "      <td>572.0</td>\n",
       "      <td>562.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>197.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139624</th>\n",
       "      <td>PLSCR4</td>\n",
       "      <td>liver</td>\n",
       "      <td>327.0</td>\n",
       "      <td>201</td>\n",
       "      <td>179.0</td>\n",
       "      <td>168.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139626</th>\n",
       "      <td>PLSCR5</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>13.0</td>\n",
       "      <td>18</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140903</th>\n",
       "      <td>PPP1R36</td>\n",
       "      <td>kidney</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5</td>\n",
       "      <td>22.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140921</th>\n",
       "      <td>PPP1R3C</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>275.0</td>\n",
       "      <td>241</td>\n",
       "      <td>224.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>133.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140923</th>\n",
       "      <td>PPP1R3C</td>\n",
       "      <td>kidney</td>\n",
       "      <td>111.0</td>\n",
       "      <td>123</td>\n",
       "      <td>127.0</td>\n",
       "      <td>193.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140924</th>\n",
       "      <td>PPP1R3C</td>\n",
       "      <td>liver</td>\n",
       "      <td>1162.0</td>\n",
       "      <td>4940</td>\n",
       "      <td>197.0</td>\n",
       "      <td>499.0</td>\n",
       "      <td>3778.0</td>\n",
       "      <td>302.0</td>\n",
       "      <td>663.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141271</th>\n",
       "      <td>PRCP</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>807.0</td>\n",
       "      <td>876</td>\n",
       "      <td>674.0</td>\n",
       "      <td>343.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>331.0</td>\n",
       "      <td>133.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141274</th>\n",
       "      <td>PRCP</td>\n",
       "      <td>liver</td>\n",
       "      <td>2096.0</td>\n",
       "      <td>986</td>\n",
       "      <td>928.0</td>\n",
       "      <td>673.0</td>\n",
       "      <td>1110.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>58.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141581</th>\n",
       "      <td>PRKCG</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>273.0</td>\n",
       "      <td>1222</td>\n",
       "      <td>3013.0</td>\n",
       "      <td>1228.0</td>\n",
       "      <td>949.0</td>\n",
       "      <td>1785.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141647</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>heart</td>\n",
       "      <td>420.0</td>\n",
       "      <td>55</td>\n",
       "      <td>811.0</td>\n",
       "      <td>519.0</td>\n",
       "      <td>365.0</td>\n",
       "      <td>292.0</td>\n",
       "      <td>99.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141648</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>244.0</td>\n",
       "      <td>122</td>\n",
       "      <td>631.0</td>\n",
       "      <td>561.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>317.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141649</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>liver</td>\n",
       "      <td>127.0</td>\n",
       "      <td>182</td>\n",
       "      <td>283.0</td>\n",
       "      <td>253.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>71.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141873</th>\n",
       "      <td>PRPF31</td>\n",
       "      <td>kidney</td>\n",
       "      <td>824.0</td>\n",
       "      <td>587</td>\n",
       "      <td>1292.0</td>\n",
       "      <td>1151.0</td>\n",
       "      <td>237.0</td>\n",
       "      <td>141.0</td>\n",
       "      <td>327.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141874</th>\n",
       "      <td>PRPF31</td>\n",
       "      <td>liver</td>\n",
       "      <td>488.0</td>\n",
       "      <td>284</td>\n",
       "      <td>1450.0</td>\n",
       "      <td>526.0</td>\n",
       "      <td>204.0</td>\n",
       "      <td>924.0</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142546</th>\n",
       "      <td>PSMC6</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>289.0</td>\n",
       "      <td>383</td>\n",
       "      <td>261.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142548</th>\n",
       "      <td>PSMC6</td>\n",
       "      <td>kidney</td>\n",
       "      <td>209.0</td>\n",
       "      <td>232</td>\n",
       "      <td>258.0</td>\n",
       "      <td>466.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142553</th>\n",
       "      <td>PSMD1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1229.0</td>\n",
       "      <td>1041</td>\n",
       "      <td>1448.0</td>\n",
       "      <td>1432.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>203.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142801</th>\n",
       "      <td>PTGDR</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>26.0</td>\n",
       "      <td>16</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142804</th>\n",
       "      <td>PTGDR</td>\n",
       "      <td>liver</td>\n",
       "      <td>24.0</td>\n",
       "      <td>30</td>\n",
       "      <td>14.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142823</th>\n",
       "      <td>PTGER2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>37.0</td>\n",
       "      <td>52</td>\n",
       "      <td>80.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142843</th>\n",
       "      <td>PTGES2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>2598.0</td>\n",
       "      <td>1703</td>\n",
       "      <td>1686.0</td>\n",
       "      <td>1505.0</td>\n",
       "      <td>895.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143092</th>\n",
       "      <td>PTPRC</td>\n",
       "      <td>heart</td>\n",
       "      <td>171.0</td>\n",
       "      <td>134</td>\n",
       "      <td>261.0</td>\n",
       "      <td>215.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143591</th>\n",
       "      <td>RAB11A</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>908.0</td>\n",
       "      <td>1157</td>\n",
       "      <td>869.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>249.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>39.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143641</th>\n",
       "      <td>RAB15</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>4394.0</td>\n",
       "      <td>6163</td>\n",
       "      <td>1996.0</td>\n",
       "      <td>3644.0</td>\n",
       "      <td>1769.0</td>\n",
       "      <td>1648.0</td>\n",
       "      <td>750.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143642</th>\n",
       "      <td>RAB15</td>\n",
       "      <td>heart</td>\n",
       "      <td>48.0</td>\n",
       "      <td>103</td>\n",
       "      <td>124.0</td>\n",
       "      <td>242.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143644</th>\n",
       "      <td>RAB15</td>\n",
       "      <td>liver</td>\n",
       "      <td>89.0</td>\n",
       "      <td>222</td>\n",
       "      <td>998.0</td>\n",
       "      <td>269.0</td>\n",
       "      <td>133.0</td>\n",
       "      <td>729.0</td>\n",
       "      <td>47.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144231</th>\n",
       "      <td>RALGPS2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>456.0</td>\n",
       "      <td>566</td>\n",
       "      <td>322.0</td>\n",
       "      <td>282.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>134.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144232</th>\n",
       "      <td>RALGPS2</td>\n",
       "      <td>heart</td>\n",
       "      <td>135.0</td>\n",
       "      <td>56</td>\n",
       "      <td>155.0</td>\n",
       "      <td>183.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144233</th>\n",
       "      <td>RALGPS2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>81.0</td>\n",
       "      <td>107</td>\n",
       "      <td>167.0</td>\n",
       "      <td>260.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>60.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144234</th>\n",
       "      <td>RALGPS2</td>\n",
       "      <td>liver</td>\n",
       "      <td>743.0</td>\n",
       "      <td>784</td>\n",
       "      <td>221.0</td>\n",
       "      <td>350.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>129.0</td>\n",
       "      <td>393.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144468</th>\n",
       "      <td>RASAL2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>443.0</td>\n",
       "      <td>354</td>\n",
       "      <td>472.0</td>\n",
       "      <td>664.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145021</th>\n",
       "      <td>RCAN1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>2300.0</td>\n",
       "      <td>2296</td>\n",
       "      <td>1329.0</td>\n",
       "      <td>789.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>540.0</td>\n",
       "      <td>967.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145388</th>\n",
       "      <td>RETREG1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>652.0</td>\n",
       "      <td>461</td>\n",
       "      <td>1515.0</td>\n",
       "      <td>990.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>525.0</td>\n",
       "      <td>338.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145434</th>\n",
       "      <td>REXO4</td>\n",
       "      <td>liver</td>\n",
       "      <td>244.0</td>\n",
       "      <td>225</td>\n",
       "      <td>795.0</td>\n",
       "      <td>289.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>506.0</td>\n",
       "      <td>45.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146867</th>\n",
       "      <td>ROR2</td>\n",
       "      <td>heart</td>\n",
       "      <td>34.0</td>\n",
       "      <td>22</td>\n",
       "      <td>13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146868</th>\n",
       "      <td>ROR2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>40.0</td>\n",
       "      <td>27</td>\n",
       "      <td>19.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147237</th>\n",
       "      <td>RPL7A</td>\n",
       "      <td>heart</td>\n",
       "      <td>47.0</td>\n",
       "      <td>43</td>\n",
       "      <td>31.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147547</th>\n",
       "      <td>RPS9</td>\n",
       "      <td>heart</td>\n",
       "      <td>1983.0</td>\n",
       "      <td>1871</td>\n",
       "      <td>1110.0</td>\n",
       "      <td>1587.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>477.0</td>\n",
       "      <td>284.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147968</th>\n",
       "      <td>RTRAF</td>\n",
       "      <td>kidney</td>\n",
       "      <td>723.0</td>\n",
       "      <td>718</td>\n",
       "      <td>1056.0</td>\n",
       "      <td>1296.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>240.0</td>\n",
       "      <td>333.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148012</th>\n",
       "      <td>RUNDC3B</td>\n",
       "      <td>heart</td>\n",
       "      <td>56.0</td>\n",
       "      <td>28</td>\n",
       "      <td>8.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148013</th>\n",
       "      <td>RUNDC3B</td>\n",
       "      <td>kidney</td>\n",
       "      <td>11.0</td>\n",
       "      <td>14</td>\n",
       "      <td>17.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148014</th>\n",
       "      <td>RUNDC3B</td>\n",
       "      <td>liver</td>\n",
       "      <td>50.0</td>\n",
       "      <td>72</td>\n",
       "      <td>36.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148157</th>\n",
       "      <td>S100A10</td>\n",
       "      <td>heart</td>\n",
       "      <td>1385.0</td>\n",
       "      <td>1259</td>\n",
       "      <td>1482.0</td>\n",
       "      <td>1398.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148162</th>\n",
       "      <td>S100A11</td>\n",
       "      <td>heart</td>\n",
       "      <td>668.0</td>\n",
       "      <td>1372</td>\n",
       "      <td>430.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>704.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>238.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148163</th>\n",
       "      <td>S100A11</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1927.0</td>\n",
       "      <td>2475</td>\n",
       "      <td>1751.0</td>\n",
       "      <td>1014.0</td>\n",
       "      <td>548.0</td>\n",
       "      <td>737.0</td>\n",
       "      <td>176.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148412</th>\n",
       "      <td>SAMD5</td>\n",
       "      <td>heart</td>\n",
       "      <td>24.0</td>\n",
       "      <td>47</td>\n",
       "      <td>16.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148542</th>\n",
       "      <td>SASH1</td>\n",
       "      <td>heart</td>\n",
       "      <td>1249.0</td>\n",
       "      <td>1821</td>\n",
       "      <td>1146.0</td>\n",
       "      <td>1224.0</td>\n",
       "      <td>572.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148543</th>\n",
       "      <td>SASH1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>990.0</td>\n",
       "      <td>832</td>\n",
       "      <td>1377.0</td>\n",
       "      <td>1212.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>222.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149298</th>\n",
       "      <td>SEC16B</td>\n",
       "      <td>kidney</td>\n",
       "      <td>192.0</td>\n",
       "      <td>162</td>\n",
       "      <td>314.0</td>\n",
       "      <td>419.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>122.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149832</th>\n",
       "      <td>SERPINA5</td>\n",
       "      <td>heart</td>\n",
       "      <td>16.0</td>\n",
       "      <td>15</td>\n",
       "      <td>127.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150012</th>\n",
       "      <td>SET</td>\n",
       "      <td>heart</td>\n",
       "      <td>1587.0</td>\n",
       "      <td>876</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>1957.0</td>\n",
       "      <td>711.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>114.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150013</th>\n",
       "      <td>SET</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1469.0</td>\n",
       "      <td>1430</td>\n",
       "      <td>2040.0</td>\n",
       "      <td>2179.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>571.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151218</th>\n",
       "      <td>SLC11A2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>694.0</td>\n",
       "      <td>318</td>\n",
       "      <td>844.0</td>\n",
       "      <td>1032.0</td>\n",
       "      <td>376.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>150.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151219</th>\n",
       "      <td>SLC11A2</td>\n",
       "      <td>liver</td>\n",
       "      <td>174.0</td>\n",
       "      <td>184</td>\n",
       "      <td>774.0</td>\n",
       "      <td>262.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>512.0</td>\n",
       "      <td>78.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151259</th>\n",
       "      <td>SLC12A8</td>\n",
       "      <td>liver</td>\n",
       "      <td>98.0</td>\n",
       "      <td>110</td>\n",
       "      <td>232.0</td>\n",
       "      <td>183.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>73.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151657</th>\n",
       "      <td>SLC24A1</td>\n",
       "      <td>heart</td>\n",
       "      <td>99.0</td>\n",
       "      <td>127</td>\n",
       "      <td>81.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151988</th>\n",
       "      <td>SLC27A4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>863.0</td>\n",
       "      <td>942</td>\n",
       "      <td>1096.0</td>\n",
       "      <td>1147.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>154.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152092</th>\n",
       "      <td>SLC2A6</td>\n",
       "      <td>heart</td>\n",
       "      <td>100.0</td>\n",
       "      <td>137</td>\n",
       "      <td>96.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152284</th>\n",
       "      <td>SLC35F1</td>\n",
       "      <td>liver</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153053</th>\n",
       "      <td>SLCO3A1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>324.0</td>\n",
       "      <td>278</td>\n",
       "      <td>250.0</td>\n",
       "      <td>254.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153261</th>\n",
       "      <td>SMAGP</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>30.0</td>\n",
       "      <td>27</td>\n",
       "      <td>20.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153262</th>\n",
       "      <td>SMAGP</td>\n",
       "      <td>heart</td>\n",
       "      <td>58.0</td>\n",
       "      <td>83</td>\n",
       "      <td>28.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153263</th>\n",
       "      <td>SMAGP</td>\n",
       "      <td>kidney</td>\n",
       "      <td>142.0</td>\n",
       "      <td>116</td>\n",
       "      <td>176.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153463</th>\n",
       "      <td>SMIM11A</td>\n",
       "      <td>kidney</td>\n",
       "      <td>226.0</td>\n",
       "      <td>195</td>\n",
       "      <td>237.0</td>\n",
       "      <td>492.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153541</th>\n",
       "      <td>SMIM34A</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153787</th>\n",
       "      <td>SNAPC5</td>\n",
       "      <td>heart</td>\n",
       "      <td>412.0</td>\n",
       "      <td>170</td>\n",
       "      <td>465.0</td>\n",
       "      <td>502.0</td>\n",
       "      <td>242.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>53.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153788</th>\n",
       "      <td>SNAPC5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>281.0</td>\n",
       "      <td>278</td>\n",
       "      <td>376.0</td>\n",
       "      <td>532.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>95.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153848</th>\n",
       "      <td>SNORC</td>\n",
       "      <td>kidney</td>\n",
       "      <td>23.0</td>\n",
       "      <td>49</td>\n",
       "      <td>50.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153849</th>\n",
       "      <td>SNORC</td>\n",
       "      <td>liver</td>\n",
       "      <td>15.0</td>\n",
       "      <td>30</td>\n",
       "      <td>40.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154123</th>\n",
       "      <td>SNX4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>488.0</td>\n",
       "      <td>474</td>\n",
       "      <td>793.0</td>\n",
       "      <td>1073.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>305.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154237</th>\n",
       "      <td>SON</td>\n",
       "      <td>heart</td>\n",
       "      <td>2435.0</td>\n",
       "      <td>2252</td>\n",
       "      <td>2956.0</td>\n",
       "      <td>2984.0</td>\n",
       "      <td>183.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>521.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154238</th>\n",
       "      <td>SON</td>\n",
       "      <td>kidney</td>\n",
       "      <td>2783.0</td>\n",
       "      <td>2465</td>\n",
       "      <td>3135.0</td>\n",
       "      <td>6061.0</td>\n",
       "      <td>318.0</td>\n",
       "      <td>2926.0</td>\n",
       "      <td>352.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154428</th>\n",
       "      <td>SP100</td>\n",
       "      <td>kidney</td>\n",
       "      <td>544.0</td>\n",
       "      <td>417</td>\n",
       "      <td>903.0</td>\n",
       "      <td>738.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>194.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154438</th>\n",
       "      <td>SP140</td>\n",
       "      <td>kidney</td>\n",
       "      <td>36.0</td>\n",
       "      <td>46</td>\n",
       "      <td>24.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154523</th>\n",
       "      <td>SPACA9</td>\n",
       "      <td>kidney</td>\n",
       "      <td>117.0</td>\n",
       "      <td>106</td>\n",
       "      <td>167.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155278</th>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>6955.0</td>\n",
       "      <td>6613</td>\n",
       "      <td>11840.0</td>\n",
       "      <td>11352.0</td>\n",
       "      <td>342.0</td>\n",
       "      <td>488.0</td>\n",
       "      <td>4397.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155647</th>\n",
       "      <td>SS18</td>\n",
       "      <td>heart</td>\n",
       "      <td>265.0</td>\n",
       "      <td>215</td>\n",
       "      <td>459.0</td>\n",
       "      <td>269.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>190.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155648</th>\n",
       "      <td>SS18</td>\n",
       "      <td>kidney</td>\n",
       "      <td>493.0</td>\n",
       "      <td>454</td>\n",
       "      <td>849.0</td>\n",
       "      <td>891.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>356.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155907</th>\n",
       "      <td>ST6GALNAC4</td>\n",
       "      <td>heart</td>\n",
       "      <td>453.0</td>\n",
       "      <td>309</td>\n",
       "      <td>306.0</td>\n",
       "      <td>224.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156324</th>\n",
       "      <td>STKLD1</td>\n",
       "      <td>liver</td>\n",
       "      <td>19.0</td>\n",
       "      <td>25</td>\n",
       "      <td>12.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156608</th>\n",
       "      <td>STXBP5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>189.0</td>\n",
       "      <td>196</td>\n",
       "      <td>273.0</td>\n",
       "      <td>286.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>77.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156857</th>\n",
       "      <td>SURF1</td>\n",
       "      <td>heart</td>\n",
       "      <td>465.0</td>\n",
       "      <td>314</td>\n",
       "      <td>505.0</td>\n",
       "      <td>536.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156858</th>\n",
       "      <td>SURF1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>868.0</td>\n",
       "      <td>665</td>\n",
       "      <td>1542.0</td>\n",
       "      <td>2099.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>557.0</td>\n",
       "      <td>674.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156861</th>\n",
       "      <td>SURF2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>499.0</td>\n",
       "      <td>465</td>\n",
       "      <td>771.0</td>\n",
       "      <td>542.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156863</th>\n",
       "      <td>SURF2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>321.0</td>\n",
       "      <td>275</td>\n",
       "      <td>611.0</td>\n",
       "      <td>382.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156864</th>\n",
       "      <td>SURF2</td>\n",
       "      <td>liver</td>\n",
       "      <td>150.0</td>\n",
       "      <td>140</td>\n",
       "      <td>933.0</td>\n",
       "      <td>183.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>750.0</td>\n",
       "      <td>33.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156868</th>\n",
       "      <td>SURF4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>2830.0</td>\n",
       "      <td>1901</td>\n",
       "      <td>4162.0</td>\n",
       "      <td>3921.0</td>\n",
       "      <td>929.0</td>\n",
       "      <td>241.0</td>\n",
       "      <td>1091.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156873</th>\n",
       "      <td>SURF6</td>\n",
       "      <td>kidney</td>\n",
       "      <td>452.0</td>\n",
       "      <td>439</td>\n",
       "      <td>729.0</td>\n",
       "      <td>651.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>199.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156874</th>\n",
       "      <td>SURF6</td>\n",
       "      <td>liver</td>\n",
       "      <td>189.0</td>\n",
       "      <td>165</td>\n",
       "      <td>627.0</td>\n",
       "      <td>335.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>292.0</td>\n",
       "      <td>146.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156927</th>\n",
       "      <td>SV2B</td>\n",
       "      <td>heart</td>\n",
       "      <td>37.0</td>\n",
       "      <td>51</td>\n",
       "      <td>1740.0</td>\n",
       "      <td>5575.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>3835.0</td>\n",
       "      <td>1689.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156952</th>\n",
       "      <td>SVIP</td>\n",
       "      <td>heart</td>\n",
       "      <td>111.0</td>\n",
       "      <td>38</td>\n",
       "      <td>168.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>57.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156953</th>\n",
       "      <td>SVIP</td>\n",
       "      <td>kidney</td>\n",
       "      <td>192.0</td>\n",
       "      <td>234</td>\n",
       "      <td>338.0</td>\n",
       "      <td>510.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>172.0</td>\n",
       "      <td>104.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156954</th>\n",
       "      <td>SVIP</td>\n",
       "      <td>liver</td>\n",
       "      <td>194.0</td>\n",
       "      <td>129</td>\n",
       "      <td>90.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>39.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156974</th>\n",
       "      <td>SWI5</td>\n",
       "      <td>liver</td>\n",
       "      <td>283.0</td>\n",
       "      <td>222</td>\n",
       "      <td>178.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157056</th>\n",
       "      <td>SYK</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>51.0</td>\n",
       "      <td>72</td>\n",
       "      <td>370.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>266.0</td>\n",
       "      <td>32.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157057</th>\n",
       "      <td>SYK</td>\n",
       "      <td>heart</td>\n",
       "      <td>101.0</td>\n",
       "      <td>97</td>\n",
       "      <td>203.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157058</th>\n",
       "      <td>SYK</td>\n",
       "      <td>kidney</td>\n",
       "      <td>702.0</td>\n",
       "      <td>533</td>\n",
       "      <td>918.0</td>\n",
       "      <td>841.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157114</th>\n",
       "      <td>SYNE3</td>\n",
       "      <td>liver</td>\n",
       "      <td>69.0</td>\n",
       "      <td>61</td>\n",
       "      <td>162.0</td>\n",
       "      <td>146.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>77.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157403</th>\n",
       "      <td>TACR1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157513</th>\n",
       "      <td>TAF4B</td>\n",
       "      <td>kidney</td>\n",
       "      <td>84.0</td>\n",
       "      <td>84</td>\n",
       "      <td>93.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157889</th>\n",
       "      <td>TBC1D13</td>\n",
       "      <td>liver</td>\n",
       "      <td>355.0</td>\n",
       "      <td>264</td>\n",
       "      <td>627.0</td>\n",
       "      <td>442.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>185.0</td>\n",
       "      <td>87.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158371</th>\n",
       "      <td>TCHH</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5</td>\n",
       "      <td>33.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158373</th>\n",
       "      <td>TCHH</td>\n",
       "      <td>kidney</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158572</th>\n",
       "      <td>TDRKH</td>\n",
       "      <td>heart</td>\n",
       "      <td>27.0</td>\n",
       "      <td>4</td>\n",
       "      <td>46.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158573</th>\n",
       "      <td>TDRKH</td>\n",
       "      <td>kidney</td>\n",
       "      <td>47.0</td>\n",
       "      <td>35</td>\n",
       "      <td>83.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158901</th>\n",
       "      <td>TEX35</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>31.0</td>\n",
       "      <td>45</td>\n",
       "      <td>12.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159021</th>\n",
       "      <td>TFCP2</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>745.0</td>\n",
       "      <td>935</td>\n",
       "      <td>707.0</td>\n",
       "      <td>461.0</td>\n",
       "      <td>190.0</td>\n",
       "      <td>246.0</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159343</th>\n",
       "      <td>THEM5</td>\n",
       "      <td>kidney</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9</td>\n",
       "      <td>60.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159547</th>\n",
       "      <td>TIGD1</td>\n",
       "      <td>heart</td>\n",
       "      <td>18.0</td>\n",
       "      <td>32</td>\n",
       "      <td>69.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159548</th>\n",
       "      <td>TIGD1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>102.0</td>\n",
       "      <td>85</td>\n",
       "      <td>145.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160259</th>\n",
       "      <td>TMEM117</td>\n",
       "      <td>liver</td>\n",
       "      <td>159.0</td>\n",
       "      <td>77</td>\n",
       "      <td>70.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161157</th>\n",
       "      <td>TMEM50B</td>\n",
       "      <td>heart</td>\n",
       "      <td>402.0</td>\n",
       "      <td>401</td>\n",
       "      <td>529.0</td>\n",
       "      <td>564.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>127.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161581</th>\n",
       "      <td>TMX3</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>727.0</td>\n",
       "      <td>845</td>\n",
       "      <td>701.0</td>\n",
       "      <td>407.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>294.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161838</th>\n",
       "      <td>TNKS2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>639.0</td>\n",
       "      <td>587</td>\n",
       "      <td>1169.0</td>\n",
       "      <td>1486.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>317.0</td>\n",
       "      <td>530.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162533</th>\n",
       "      <td>TRAM1L1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>16.0</td>\n",
       "      <td>17</td>\n",
       "      <td>58.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162818</th>\n",
       "      <td>TRIM32</td>\n",
       "      <td>kidney</td>\n",
       "      <td>230.0</td>\n",
       "      <td>181</td>\n",
       "      <td>234.0</td>\n",
       "      <td>375.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>141.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165288</th>\n",
       "      <td>TRUB2</td>\n",
       "      <td>kidney</td>\n",
       "      <td>331.0</td>\n",
       "      <td>393</td>\n",
       "      <td>418.0</td>\n",
       "      <td>864.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>446.0</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165297</th>\n",
       "      <td>TSC1</td>\n",
       "      <td>heart</td>\n",
       "      <td>660.0</td>\n",
       "      <td>460</td>\n",
       "      <td>731.0</td>\n",
       "      <td>1059.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>328.0</td>\n",
       "      <td>71.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165298</th>\n",
       "      <td>TSC1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>1171.0</td>\n",
       "      <td>786</td>\n",
       "      <td>1405.0</td>\n",
       "      <td>1956.0</td>\n",
       "      <td>385.0</td>\n",
       "      <td>551.0</td>\n",
       "      <td>234.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165991</th>\n",
       "      <td>TTYH1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>2303.0</td>\n",
       "      <td>2501</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>787.0</td>\n",
       "      <td>198.0</td>\n",
       "      <td>1228.0</td>\n",
       "      <td>288.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165992</th>\n",
       "      <td>TTYH1</td>\n",
       "      <td>heart</td>\n",
       "      <td>24.0</td>\n",
       "      <td>17</td>\n",
       "      <td>7.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166291</th>\n",
       "      <td>TXNDC16</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>398.0</td>\n",
       "      <td>481</td>\n",
       "      <td>315.0</td>\n",
       "      <td>171.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>83.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166294</th>\n",
       "      <td>TXNDC16</td>\n",
       "      <td>liver</td>\n",
       "      <td>69.0</td>\n",
       "      <td>48</td>\n",
       "      <td>32.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167156</th>\n",
       "      <td>UMPS</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>382.0</td>\n",
       "      <td>593</td>\n",
       "      <td>345.0</td>\n",
       "      <td>297.0</td>\n",
       "      <td>211.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167158</th>\n",
       "      <td>UMPS</td>\n",
       "      <td>kidney</td>\n",
       "      <td>280.0</td>\n",
       "      <td>303</td>\n",
       "      <td>359.0</td>\n",
       "      <td>559.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>56.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167228</th>\n",
       "      <td>UNC5D</td>\n",
       "      <td>kidney</td>\n",
       "      <td>148.0</td>\n",
       "      <td>59</td>\n",
       "      <td>240.0</td>\n",
       "      <td>217.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>69.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167383</th>\n",
       "      <td>UQCRFS1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>570.0</td>\n",
       "      <td>688</td>\n",
       "      <td>689.0</td>\n",
       "      <td>1306.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>617.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167384</th>\n",
       "      <td>UQCRFS1</td>\n",
       "      <td>liver</td>\n",
       "      <td>440.0</td>\n",
       "      <td>275</td>\n",
       "      <td>258.0</td>\n",
       "      <td>204.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167792</th>\n",
       "      <td>UTP23</td>\n",
       "      <td>heart</td>\n",
       "      <td>80.0</td>\n",
       "      <td>83</td>\n",
       "      <td>136.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>41.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167793</th>\n",
       "      <td>UTP23</td>\n",
       "      <td>kidney</td>\n",
       "      <td>86.0</td>\n",
       "      <td>79</td>\n",
       "      <td>121.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168054</th>\n",
       "      <td>VEGFD</td>\n",
       "      <td>liver</td>\n",
       "      <td>112.0</td>\n",
       "      <td>43</td>\n",
       "      <td>12.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168792</th>\n",
       "      <td>WDR34</td>\n",
       "      <td>heart</td>\n",
       "      <td>360.0</td>\n",
       "      <td>269</td>\n",
       "      <td>255.0</td>\n",
       "      <td>249.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168794</th>\n",
       "      <td>WDR34</td>\n",
       "      <td>liver</td>\n",
       "      <td>898.0</td>\n",
       "      <td>759</td>\n",
       "      <td>439.0</td>\n",
       "      <td>658.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>219.0</td>\n",
       "      <td>101.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169956</th>\n",
       "      <td>ZBTB1</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>189.0</td>\n",
       "      <td>308</td>\n",
       "      <td>135.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>119.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169957</th>\n",
       "      <td>ZBTB1</td>\n",
       "      <td>heart</td>\n",
       "      <td>266.0</td>\n",
       "      <td>248</td>\n",
       "      <td>292.0</td>\n",
       "      <td>302.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169959</th>\n",
       "      <td>ZBTB1</td>\n",
       "      <td>liver</td>\n",
       "      <td>379.0</td>\n",
       "      <td>289</td>\n",
       "      <td>240.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170022</th>\n",
       "      <td>ZBTB25</td>\n",
       "      <td>heart</td>\n",
       "      <td>90.0</td>\n",
       "      <td>45</td>\n",
       "      <td>159.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170023</th>\n",
       "      <td>ZBTB25</td>\n",
       "      <td>kidney</td>\n",
       "      <td>88.0</td>\n",
       "      <td>56</td>\n",
       "      <td>94.0</td>\n",
       "      <td>119.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170024</th>\n",
       "      <td>ZBTB25</td>\n",
       "      <td>liver</td>\n",
       "      <td>49.0</td>\n",
       "      <td>30</td>\n",
       "      <td>72.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170383</th>\n",
       "      <td>ZDHHC12</td>\n",
       "      <td>kidney</td>\n",
       "      <td>602.0</td>\n",
       "      <td>341</td>\n",
       "      <td>1006.0</td>\n",
       "      <td>642.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>364.0</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170498</th>\n",
       "      <td>ZER1</td>\n",
       "      <td>kidney</td>\n",
       "      <td>2029.0</td>\n",
       "      <td>1198</td>\n",
       "      <td>2316.0</td>\n",
       "      <td>2144.0</td>\n",
       "      <td>831.0</td>\n",
       "      <td>172.0</td>\n",
       "      <td>115.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171061</th>\n",
       "      <td>ZNF148</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>956.0</td>\n",
       "      <td>1230</td>\n",
       "      <td>891.0</td>\n",
       "      <td>645.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>246.0</td>\n",
       "      <td>65.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171062</th>\n",
       "      <td>ZNF148</td>\n",
       "      <td>heart</td>\n",
       "      <td>462.0</td>\n",
       "      <td>314</td>\n",
       "      <td>641.0</td>\n",
       "      <td>820.0</td>\n",
       "      <td>148.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>179.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171063</th>\n",
       "      <td>ZNF148</td>\n",
       "      <td>kidney</td>\n",
       "      <td>391.0</td>\n",
       "      <td>420</td>\n",
       "      <td>586.0</td>\n",
       "      <td>990.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>404.0</td>\n",
       "      <td>166.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171717</th>\n",
       "      <td>ZNF385D</td>\n",
       "      <td>heart</td>\n",
       "      <td>29.0</td>\n",
       "      <td>25</td>\n",
       "      <td>33.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172518</th>\n",
       "      <td>ZNF622</td>\n",
       "      <td>kidney</td>\n",
       "      <td>352.0</td>\n",
       "      <td>335</td>\n",
       "      <td>764.0</td>\n",
       "      <td>587.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>235.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173086</th>\n",
       "      <td>ZNF804A</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>119.0</td>\n",
       "      <td>89</td>\n",
       "      <td>50.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173088</th>\n",
       "      <td>ZNF804A</td>\n",
       "      <td>kidney</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173091</th>\n",
       "      <td>ZNF804B</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>47.0</td>\n",
       "      <td>78</td>\n",
       "      <td>22.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173387</th>\n",
       "      <td>ZRSR2</td>\n",
       "      <td>heart</td>\n",
       "      <td>318.0</td>\n",
       "      <td>205</td>\n",
       "      <td>329.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                gene      tissue  ppn_female  ppn_male  ptr_female  ptr_male  \\\n",
       "1282          ADAM22       heart        31.0        47        53.0      56.0   \n",
       "1283          ADAM22      kidney       198.0       181       199.0     505.0   \n",
       "1284          ADAM22       liver        13.0       129        10.0      11.0   \n",
       "1437        ADAMTSL2       heart       151.0       714        89.0      55.0   \n",
       "1439        ADAMTSL2       liver       352.0       305       689.0     402.0   \n",
       "2119            AGMO       liver       254.0       221        22.0     182.0   \n",
       "2163          AGPAT5      kidney       187.0       199       317.0     687.0   \n",
       "2413             AK8      kidney        30.0        20       127.0     134.0   \n",
       "2471           AKAP5  cerebellum       181.0       141        64.0     102.0   \n",
       "2474           AKAP5       liver        37.0        55         7.0      21.0   \n",
       "3221          ANGPT2  cerebellum       102.0        45        39.0      17.0   \n",
       "3223          ANGPT2      kidney        35.0        25        18.0      20.0   \n",
       "3224          ANGPT2       liver        33.0        43        21.0      26.0   \n",
       "3232         ANGPTL1       heart        98.0       176        51.0      78.0   \n",
       "3234         ANGPTL1       liver       144.0       121        22.0      42.0   \n",
       "3886           AP1S2  cerebellum       478.0       511       347.0     276.0   \n",
       "3889           AP1S2       liver       193.0       114        71.0      25.0   \n",
       "4352            AQP4       heart        13.0         5        24.0     158.0   \n",
       "5086           ARMC9  cerebellum       254.0       270       227.0     145.0   \n",
       "5087           ARMC9       heart        43.0        45        35.0      24.0   \n",
       "5262            ARSB       heart       149.0       100       251.0     180.0   \n",
       "5263            ARSB      kidney       445.0       584      2249.0    2629.0   \n",
       "5457            ASB9       heart        18.0        14         1.0       6.0   \n",
       "5669           ASTN2       liver        41.0        72        19.0      37.0   \n",
       "5736            ATF1  cerebellum       222.0       241       157.0     112.0   \n",
       "5811         ATG16L1  cerebellum      1487.0      2151      1354.0     736.0   \n",
       "6323        ATP6V1G3      kidney       231.0       265       653.0     466.0   \n",
       "6488             AUH      kidney       123.0       105       158.0     338.0   \n",
       "6579          AXDND1       liver         8.0         8         1.0       3.0   \n",
       "6956           BASP1  cerebellum      3102.0      2160      1754.0    1972.0   \n",
       "7498            BHMT      kidney      2778.0      3455      5767.0   15958.0   \n",
       "7501           BHMT2  cerebellum        41.0        39         5.0      12.0   \n",
       "7502           BHMT2       heart       366.0       152        50.0      82.0   \n",
       "7503           BHMT2      kidney      1925.0      1317      2907.0    9406.0   \n",
       "7558            BIN2      kidney       173.0       145       186.0     183.0   \n",
       "7779             BMX       liver        52.0        31        15.0      11.0   \n",
       "8192           BTAF1       heart       491.0       430       859.0     603.0   \n",
       "8193           BTAF1      kidney       773.0       514      1115.0     965.0   \n",
       "9362       C1H1orf53       heart        15.0         5        18.0      31.0   \n",
       "9364       C1H1orf53       liver        37.0        25        16.0      19.0   \n",
       "9556    C21H21orf140  cerebellum        29.0        26        37.0      47.0   \n",
       "9557    C21H21orf140       heart         4.0         3        19.0      20.0   \n",
       "9558    C21H21orf140      kidney        12.0         2        27.0      38.0   \n",
       "9559    C21H21orf140       liver         3.0         1         9.0       6.0   \n",
       "9724      C2BH2orf72       liver      2243.0      3142       460.0    1832.0   \n",
       "9779          C2CD4D       liver        15.0         7        19.0      27.0   \n",
       "10367      C9H9orf16       heart       118.0       192        66.0      50.0   \n",
       "10503           CA5B      kidney        94.0        86       150.0     287.0   \n",
       "10532          CAB39       heart      1822.0       589      2038.0    2416.0   \n",
       "10533          CAB39      kidney      1321.0      1076      1374.0    2756.0   \n",
       "10596         CACFD1  cerebellum       540.0       606       536.0     361.0   \n",
       "10722         CACNG6       heart        14.0         4        41.0      28.0   \n",
       "10756          CADM2  cerebellum      1972.0      2314      1441.0    1284.0   \n",
       "10757          CADM2       heart        14.0         6        27.0      47.0   \n",
       "11531           CBLB  cerebellum       864.0       903       641.0     527.0   \n",
       "13437       CDC42EP5       heart       296.0       324        42.0      50.0   \n",
       "13438       CDC42EP5      kidney       202.0       158        98.0      50.0   \n",
       "13534          CDH11       liver        31.0        30        13.0       4.0   \n",
       "14076            CEL  cerebellum       643.0       602       209.0     245.0   \n",
       "14106          CELF3  cerebellum      4607.0      4593      2707.0    3722.0   \n",
       "14462         CERCAM       heart       187.0       301        47.0      68.0   \n",
       "14463         CERCAM      kidney       313.0       209        59.0     138.0   \n",
       "14464         CERCAM       liver        37.0        33        47.0      53.0   \n",
       "15242          CHRNG       heart         1.0         1         2.0       2.0   \n",
       "15317          CHST9       heart         3.0         9       172.0     242.0   \n",
       "15319          CHST9       liver        42.0        35       115.0      52.0   \n",
       "15352         CHURC1       heart       997.0       425      1123.0    1329.0   \n",
       "15353         CHURC1      kidney       709.0       632       967.0    1470.0   \n",
       "15493           CIZ1      kidney      1376.0       957      2208.0    1878.0   \n",
       "15494           CIZ1       liver       365.0       263      1203.0     522.0   \n",
       "15956          CLIC6  cerebellum        16.0        26         1.0       4.0   \n",
       "15959          CLIC6       liver         8.0        11         1.0       5.0   \n",
       "16158          CLTRN      kidney       927.0      1083      1592.0    2934.0   \n",
       "16159          CLTRN       liver        27.0         7        40.0      69.0   \n",
       "16308          CMYA5      kidney       583.0       376       850.0    1066.0   \n",
       "16471          CNOT3  cerebellum      1476.0      1720      1297.0    1291.0   \n",
       "16472          CNOT3       heart       409.0       349       290.0     312.0   \n",
       "16474          CNOT3       liver       412.0       237       612.0     668.0   \n",
       "17183           COQ4      kidney      1060.0       790      1353.0    2417.0   \n",
       "17478          CPEB3      kidney       200.0       145       254.0     489.0   \n",
       "17479          CPEB3       liver       201.0       338       130.0     182.0   \n",
       "19421          CYTIP  cerebellum        10.0        15         7.0       7.0   \n",
       "19424          CYTIP       liver       307.0        96        77.0      38.0   \n",
       "19467          DACH1       heart       240.0       224        99.0     186.0   \n",
       "19598         DAZAP2      kidney      3164.0      3021      5429.0    6962.0   \n",
       "19606           DBF4  cerebellum        50.0        31        30.0      11.0   \n",
       "19608           DBF4      kidney        15.0        19        39.0      39.0   \n",
       "19617            DBH       heart         6.0        16         3.0       5.0   \n",
       "19666           DBX2  cerebellum         7.0         8        11.0       9.0   \n",
       "19778         DCBLD1      kidney       150.0       155       189.0     195.0   \n",
       "20156          DDX31  cerebellum       306.0       312       265.0     178.0   \n",
       "20157          DDX31       heart        42.0        24        45.0      56.0   \n",
       "20158          DDX31      kidney       125.0       135       149.0     193.0   \n",
       "20551        DENND4A  cerebellum      1049.0      1661       854.0     820.0   \n",
       "20552        DENND4A       heart       215.0       177       400.0     332.0   \n",
       "20553        DENND4A      kidney       319.0       267       385.0     709.0   \n",
       "20723           DGKB      kidney        22.0        30        47.0     116.0   \n",
       "20727           DGKD       heart       680.0       658       468.0     484.0   \n",
       "20728           DGKD      kidney       749.0       735       644.0     692.0   \n",
       "20729           DGKD       liver       463.0       505       426.0     197.0   \n",
       "20987         DICER1       heart       496.0       418       521.0     754.0   \n",
       "20988         DICER1      kidney       589.0       492       786.0    1172.0   \n",
       "21026          DIP2B  cerebellum      3304.0      4884      2891.0    2409.0   \n",
       "21028          DIP2B      kidney       425.0       350       477.0     823.0   \n",
       "21068          DIS3L      kidney       344.0       323       349.0     706.0   \n",
       "21296            DMD  cerebellum       792.0       737       704.0     499.0   \n",
       "21297            DMD       heart      2500.0       832      3684.0    5423.0   \n",
       "21298            DMD      kidney       156.0       125       275.0     255.0   \n",
       "21299            DMD       liver      1061.0       546       265.0     468.0   \n",
       "21303          DMGDH      kidney      1126.0       754      2758.0    6399.0   \n",
       "21652        DNAJC28       heart        38.0        10        49.0      60.0   \n",
       "21653        DNAJC28      kidney        30.0        15        49.0      87.0   \n",
       "21654        DNAJC28       liver        39.0        24        23.0       8.0   \n",
       "21772           DNM1       heart       215.0       206       126.0      73.0   \n",
       "21958         DONSON      kidney       128.0       128       141.0     181.0   \n",
       "22068           DPM2      kidney       718.0       628      1101.0     719.0   \n",
       "22932          ECEL1       heart        20.0        14         5.0       1.0   \n",
       "22933          ECEL1      kidney        32.0        27        11.0       4.0   \n",
       "23613          EIF3H      kidney      1404.0      1401      1676.0    2877.0   \n",
       "23673         EIF4E2      kidney      1046.0       783      1057.0    1667.0   \n",
       "24169          ENDOG       liver       242.0       178       285.0     370.0   \n",
       "24461          EPHA5  cerebellum        40.0        89       314.0     230.0   \n",
       "24463          EPHA5      kidney         4.0         5         7.0      11.0   \n",
       "24841           ERMN  cerebellum       692.0       732       519.0     361.0   \n",
       "24863          ERO1A      kidney       336.0       333       358.0     378.0   \n",
       "24977           ESR2       heart         2.0         1         5.0       4.0   \n",
       "25136          EVA1A  cerebellum         1.0         3        25.0       4.0   \n",
       "25138          EVA1A      kidney       220.0       203       401.0     577.0   \n",
       "25139          EVA1A       liver       752.0       889       329.0     575.0   \n",
       "25648        FAM102A      kidney      1524.0      1827      2927.0    2381.0   \n",
       "26073        FAM181B      kidney        58.0        52       179.0      66.0   \n",
       "26162        FAM199X       heart       306.0       178       310.0     376.0   \n",
       "26163        FAM199X      kidney       461.0       437       572.0     769.0   \n",
       "26958           FAT4      kidney       268.0       239       458.0     848.0   \n",
       "27797         FGFBP3       heart        23.0        14        69.0      24.0   \n",
       "27877           FHIT       heart        28.0        12        37.0      49.0   \n",
       "27879           FHIT       liver        71.0        35        10.0      24.0   \n",
       "28091            FLG  cerebellum       102.0       138        51.0      46.0   \n",
       "28092            FLG       heart        22.0        26         4.0      14.0   \n",
       "28351           FNTB  cerebellum       668.0       697       532.0     394.0   \n",
       "28741          FRMD6  cerebellum       357.0       163       554.0     379.0   \n",
       "29036           FUT8  cerebellum      1103.0      1127       670.0     650.0   \n",
       "29037           FUT8       heart       214.0       271       168.0     196.0   \n",
       "29039           FUT8       liver       112.0       112       106.0      22.0   \n",
       "29666           GAS2  cerebellum        40.0        35        34.0      14.0   \n",
       "29669           GAS2       liver       391.0       200        47.0     165.0   \n",
       "29799          GBGT1       liver       131.0        58       150.0     247.0   \n",
       "29879          GCFC2       liver       318.0       176        83.0      59.0   \n",
       "30382         GIGYF2       heart      1506.0       899      1724.0    1908.0   \n",
       "30383         GIGYF2      kidney      1478.0      1078      1666.0    2284.0   \n",
       "30667           GLE1       heart       314.0       194       342.0     394.0   \n",
       "30668           GLE1      kidney       409.0       348       603.0     729.0   \n",
       "30808          GLRX5      kidney      1568.0      1675      2047.0    2494.0   \n",
       "30809          GLRX5       liver      1323.0       781      1589.0    1613.0   \n",
       "31116           GNG2  cerebellum       645.0       533       857.0     675.0   \n",
       "31244         GOLGA2       liver       647.0       472      3105.0     913.0   \n",
       "31332           GOPC       heart       660.0       342       686.0     681.0   \n",
       "31333           GOPC      kidney       446.0       407       495.0     852.0   \n",
       "31541           GPD2  cerebellum      1260.0      1537       830.0     889.0   \n",
       "31543           GPD2      kidney       320.0       284       359.0     528.0   \n",
       "31666        GPR137C  cerebellum       600.0       688       440.0     382.0   \n",
       "31914          GPR55       liver         6.0         5         2.0       2.0   \n",
       "32076           GPX2  cerebellum        37.0        32        18.0       6.0   \n",
       "32401           GRM7  cerebellum         3.0        63       133.0      94.0   \n",
       "32497          GSDME       heart       256.0       251       122.0      59.0   \n",
       "32746         GTF3C4  cerebellum      1084.0      1640       863.0     899.0   \n",
       "32747         GTF3C4       heart       226.0       195       318.0     324.0   \n",
       "32748         GTF3C4      kidney       247.0       282       473.0     654.0   \n",
       "33606           HEG1  cerebellum       790.0       637       569.0     343.0   \n",
       "33607           HEG1       heart      4308.0      3188      4617.0   10642.0   \n",
       "33609           HEG1       liver       482.0       481      1367.0     609.0   \n",
       "34886           HRNR  cerebellum         8.0        13        56.0      29.0   \n",
       "34887           HRNR       heart         5.0         5        11.0       8.0   \n",
       "34888           HRNR      kidney         4.0        16        18.0      46.0   \n",
       "35122          HSPA2       heart       645.0       320       239.0     177.0   \n",
       "35276          HTR2B  cerebellum         8.0         8         2.0       1.0   \n",
       "35278          HTR2B      kidney        16.0        16        31.0      38.0   \n",
       "35556            IDE  cerebellum       382.0       425       376.0     157.0   \n",
       "35558            IDE      kidney       363.0       369       387.0     520.0   \n",
       "35818         IFNGR2      kidney       816.0       559       962.0    1409.0   \n",
       "36948         INTS14      kidney       317.0       282       411.0     579.0   \n",
       "37413           ISPD      kidney        62.0        39        66.0     168.0   \n",
       "37571          ITGB5  cerebellum       691.0       805       664.0     358.0   \n",
       "37573          ITGB5      kidney      1748.0      1458      2581.0    3610.0   \n",
       "37713          ITSN1      kidney       571.0       437       840.0    1123.0   \n",
       "37901            JMY  cerebellum       953.0      1124       740.0     593.0   \n",
       "37902            JMY       heart       318.0       199       388.0     514.0   \n",
       "37903            JMY      kidney       398.0       271       429.0     762.0   \n",
       "38266          KCNE1  cerebellum        11.0        19         6.0       9.0   \n",
       "38648          KCTD1      kidney       752.0       442       847.0     807.0   \n",
       "39183          KIF11      kidney        27.0        26        40.0      38.0   \n",
       "39527          KLF13       heart      1209.0      1397      1487.0    1452.0   \n",
       "39528          KLF13      kidney      1820.0      1525      2679.0    2966.0   \n",
       "39529          KLF13       liver       906.0       776      2104.0     985.0   \n",
       "40558          KYAT1      kidney       696.0       533       728.0    1505.0   \n",
       "40559          KYAT1       liver       466.0       344       596.0     585.0   \n",
       "40821          LARP4  cerebellum       441.0       554       405.0     305.0   \n",
       "41206          LENG1  cerebellum       153.0       202       234.0     209.0   \n",
       "41211          LENG8  cerebellum      9409.0     12783      9346.0    8642.0   \n",
       "41212          LENG8       heart      2738.0      2130      1726.0    2045.0   \n",
       "41214          LENG8       liver      1748.0      1053      3541.0    2210.0   \n",
       "41216          LENG9  cerebellum        24.0        19        16.0       6.0   \n",
       "41217          LENG9       heart        35.0        31        23.0      13.0   \n",
       "41219          LENG9       liver        44.0         9       283.0      59.0   \n",
       "41527         LILRB3       heart        27.0        20        41.0      31.0   \n",
       "41528         LILRB3      kidney        28.0        14        73.0      65.0   \n",
       "41656         LINGO4  cerebellum        24.0        58       267.0     226.0   \n",
       "41657         LINGO4       heart        70.0        12       164.0     119.0   \n",
       "41658         LINGO4      kidney        22.0        26       153.0      78.0   \n",
       "41659         LINGO4       liver        41.0        30       197.0      46.0   \n",
       "44053   LOC100612575      kidney         8.0         7        12.0      23.0   \n",
       "44556   LOC100613592  cerebellum        15.0        14         8.0       2.0   \n",
       "44557   LOC100613592       heart         5.0         7        15.0       9.0   \n",
       "47211   LOC101059176  cerebellum       157.0       131        64.0      71.0   \n",
       "47212   LOC101059176       heart        30.0        12         8.0       6.0   \n",
       "48396   LOC104001534  cerebellum         5.0         4         2.0       2.0   \n",
       "51763   LOC104003432      kidney        28.0        20        39.0      67.0   \n",
       "51764   LOC104003432       liver        13.0        12         2.0       4.0   \n",
       "58606   LOC104007525  cerebellum       120.0        74        50.0      35.0   \n",
       "86693   LOC107976709      kidney         3.0         4        10.0      13.0   \n",
       "86694   LOC107976709       liver         4.0         4         2.0       1.0   \n",
       "110621     LOC453134  cerebellum         4.0         9         2.0       1.0   \n",
       "110624     LOC453134       liver        10.0         9         5.0       1.0   \n",
       "110681     LOC453285  cerebellum        87.0       116        16.0       3.0   \n",
       "111017     LOC456283       heart        20.0        13       101.0      23.0   \n",
       "111516     LOC460522  cerebellum         8.0         6         1.0       4.0   \n",
       "111524     LOC460532       liver        42.0        32         3.0       1.0   \n",
       "112122     LOC464914       heart         4.0         1         5.0       9.0   \n",
       "112236     LOC465512  cerebellum       534.0       532       188.0     466.0   \n",
       "112239     LOC465512       liver       200.0       143       262.0     243.0   \n",
       "112974     LOC467545       liver         1.0         1         2.0       2.0   \n",
       "113711     LOC470670  cerebellum         5.0         9         3.0       1.0   \n",
       "116148     LOC739470      kidney       104.0       149       217.0     210.0   \n",
       "117298     LOC742641      kidney        11.0         2        25.0      15.0   \n",
       "120587        LRRC8A       heart      1541.0      1330       712.0    1285.0   \n",
       "120589        LRRC8A       liver       839.0       992      2096.0    1308.0   \n",
       "121748        MAP2K1      kidney       327.0       330       442.0     543.0   \n",
       "121749        MAP2K1       liver       198.0       123       410.0     369.0   \n",
       "122108        MARCH5      kidney       318.0       296       345.0     548.0   \n",
       "122296           MAX  cerebellum      1389.0      1337       793.0    1114.0   \n",
       "123046         MEOX2  cerebellum        13.0        13         4.0       2.0   \n",
       "123049         MEOX2       liver        57.0        29        15.0       7.0   \n",
       "127716          MPP6  cerebellum       391.0       413       283.0      89.0   \n",
       "127718          MPP6      kidney       145.0       163       230.0     357.0   \n",
       "128001        MRPL19  cerebellum       547.0       549       237.0     188.0   \n",
       "128004        MRPL19       liver       776.0       526       368.0     250.0   \n",
       "128186         MRPL9  cerebellum       903.0       978       899.0     668.0   \n",
       "128321         MRPS6  cerebellum       158.0       219       278.0     223.0   \n",
       "128323         MRPS6      kidney       455.0       362       522.0     794.0   \n",
       "128793        MTHFD1      kidney      1528.0      1448      1759.0    4139.0   \n",
       "129543         MYO10      kidney      1914.0      1383      2184.0    3619.0   \n",
       "129992         NAIF1       heart        56.0        68        53.0      31.0   \n",
       "129993         NAIF1      kidney        88.0        77       166.0     145.0   \n",
       "129994         NAIF1       liver        54.0        48       180.0      66.0   \n",
       "130626        NDUFA3  cerebellum       667.0       496       483.0     451.0   \n",
       "130986          NEK7  cerebellum       426.0       382       191.0      77.0   \n",
       "130988          NEK7      kidney       547.0       412       577.0    1065.0   \n",
       "131027         NELL2       heart        10.0         9         6.0       3.0   \n",
       "131028         NELL2      kidney         2.0         9        41.0      75.0   \n",
       "131029         NELL2       liver         4.0        50         3.0       2.0   \n",
       "131264         NFIL3       liver      4524.0       498       440.0     243.0   \n",
       "131787        NLRP12       heart        18.0        11         6.0       4.0   \n",
       "131789        NLRP12       liver        66.0        17        11.0       5.0   \n",
       "132714          NRG3       liver        15.0         6        38.0      20.0   \n",
       "133411          NUS1  cerebellum       169.0       184       121.0     109.0   \n",
       "133413          NUS1      kidney       160.0       183       190.0     456.0   \n",
       "133606          OAZ3  cerebellum        80.0       114        79.0      74.0   \n",
       "133608          OAZ3      kidney         9.0         7        34.0      20.0   \n",
       "133702          ODF2       heart       268.0       260       414.0     361.0   \n",
       "133703          ODF2      kidney       430.0       400      1064.0     862.0   \n",
       "135136       OSBPL11  cerebellum       384.0       340       293.0     196.0   \n",
       "135138       OSBPL11      kidney       269.0       193       274.0     343.0   \n",
       "135152        OSBPL3       heart       137.0        93       148.0     144.0   \n",
       "135182         OSCAR       heart        10.0         6         5.0       3.0   \n",
       "135357        OTUD7A       heart        23.0        36        17.0      12.0   \n",
       "135358        OTUD7A      kidney        54.0        47        76.0      64.0   \n",
       "135658         PACRG      kidney       148.0        91       180.0     152.0   \n",
       "135659         PACRG       liver        60.0        21         7.0       5.0   \n",
       "135953         PAPD4      kidney       376.0       306       482.0     705.0   \n",
       "135987         PAPPA       heart        56.0        79        31.0      38.0   \n",
       "135988         PAPPA      kidney       746.0       325      1940.0     807.0   \n",
       "136549         PCDH9       liver         8.0        54         2.0       6.0   \n",
       "138267        PHYHD1       heart       584.0       260       101.0      73.0   \n",
       "138268        PHYHD1      kidney      1224.0       966       406.0     174.0   \n",
       "138269        PHYHD1       liver      2916.0      1728      1077.0    1636.0   \n",
       "138708           PIR      kidney        66.0        61        85.0     127.0   \n",
       "138907          PKN3       heart       448.0       620       173.0     236.0   \n",
       "138908          PKN3      kidney       345.0       318       218.0     181.0   \n",
       "139342       PLEKHG3       heart       301.0       373       216.0     228.0   \n",
       "139344       PLEKHG3       liver       264.0       315      1439.0     512.0   \n",
       "139506         PLOD2  cerebellum       194.0       224       160.0      56.0   \n",
       "139608        PLSCR1      kidney       332.0       365       572.0     562.0   \n",
       "139624        PLSCR4       liver       327.0       201       179.0     168.0   \n",
       "139626        PLSCR5  cerebellum        13.0        18         4.0       4.0   \n",
       "140903       PPP1R36      kidney        10.0         5        22.0      17.0   \n",
       "140921       PPP1R3C  cerebellum       275.0       241       224.0      91.0   \n",
       "140923       PPP1R3C      kidney       111.0       123       127.0     193.0   \n",
       "140924       PPP1R3C       liver      1162.0      4940       197.0     499.0   \n",
       "141271          PRCP  cerebellum       807.0       876       674.0     343.0   \n",
       "141274          PRCP       liver      2096.0       986       928.0     673.0   \n",
       "141581         PRKCG  cerebellum       273.0      1222      3013.0    1228.0   \n",
       "141647          PRKN       heart       420.0        55       811.0     519.0   \n",
       "141648          PRKN      kidney       244.0       122       631.0     561.0   \n",
       "141649          PRKN       liver       127.0       182       283.0     253.0   \n",
       "141873        PRPF31      kidney       824.0       587      1292.0    1151.0   \n",
       "141874        PRPF31       liver       488.0       284      1450.0     526.0   \n",
       "142546         PSMC6  cerebellum       289.0       383       261.0     202.0   \n",
       "142548         PSMC6      kidney       209.0       232       258.0     466.0   \n",
       "142553         PSMD1      kidney      1229.0      1041      1448.0    1432.0   \n",
       "142801         PTGDR  cerebellum        26.0        16         6.0       1.0   \n",
       "142804         PTGDR       liver        24.0        30        14.0      16.0   \n",
       "142823        PTGER2      kidney        37.0        52        80.0      66.0   \n",
       "142843        PTGES2      kidney      2598.0      1703      1686.0    1505.0   \n",
       "143092         PTPRC       heart       171.0       134       261.0     215.0   \n",
       "143591        RAB11A  cerebellum       908.0      1157       869.0     705.0   \n",
       "143641         RAB15  cerebellum      4394.0      6163      1996.0    3644.0   \n",
       "143642         RAB15       heart        48.0       103       124.0     242.0   \n",
       "143644         RAB15       liver        89.0       222       998.0     269.0   \n",
       "144231       RALGPS2  cerebellum       456.0       566       322.0     282.0   \n",
       "144232       RALGPS2       heart       135.0        56       155.0     183.0   \n",
       "144233       RALGPS2      kidney        81.0       107       167.0     260.0   \n",
       "144234       RALGPS2       liver       743.0       784       221.0     350.0   \n",
       "144468        RASAL2      kidney       443.0       354       472.0     664.0   \n",
       "145021         RCAN1  cerebellum      2300.0      2296      1329.0     789.0   \n",
       "145388       RETREG1      kidney       652.0       461      1515.0     990.0   \n",
       "145434         REXO4       liver       244.0       225       795.0     289.0   \n",
       "146867          ROR2       heart        34.0        22        13.0       3.0   \n",
       "146868          ROR2      kidney        40.0        27        19.0      18.0   \n",
       "147237         RPL7A       heart        47.0        43        31.0      35.0   \n",
       "147547          RPS9       heart      1983.0      1871      1110.0    1587.0   \n",
       "147968         RTRAF      kidney       723.0       718      1056.0    1296.0   \n",
       "148012       RUNDC3B       heart        56.0        28         8.0      20.0   \n",
       "148013       RUNDC3B      kidney        11.0        14        17.0      64.0   \n",
       "148014       RUNDC3B       liver        50.0        72        36.0      49.0   \n",
       "148157       S100A10       heart      1385.0      1259      1482.0    1398.0   \n",
       "148162       S100A11       heart       668.0      1372       430.0     205.0   \n",
       "148163       S100A11      kidney      1927.0      2475      1751.0    1014.0   \n",
       "148412         SAMD5       heart        24.0        47        16.0      23.0   \n",
       "148542         SASH1       heart      1249.0      1821      1146.0    1224.0   \n",
       "148543         SASH1      kidney       990.0       832      1377.0    1212.0   \n",
       "149298        SEC16B      kidney       192.0       162       314.0     419.0   \n",
       "149832      SERPINA5       heart        16.0        15       127.0      17.0   \n",
       "150012           SET       heart      1587.0       876      1701.0    1957.0   \n",
       "150013           SET      kidney      1469.0      1430      2040.0    2179.0   \n",
       "151218       SLC11A2      kidney       694.0       318       844.0    1032.0   \n",
       "151219       SLC11A2       liver       174.0       184       774.0     262.0   \n",
       "151259       SLC12A8       liver        98.0       110       232.0     183.0   \n",
       "151657       SLC24A1       heart        99.0       127        81.0      87.0   \n",
       "151988       SLC27A4      kidney       863.0       942      1096.0    1147.0   \n",
       "152092        SLC2A6       heart       100.0       137        96.0      67.0   \n",
       "152284       SLC35F1       liver         2.0         2         1.0       1.0   \n",
       "153053       SLCO3A1      kidney       324.0       278       250.0     254.0   \n",
       "153261         SMAGP  cerebellum        30.0        27        20.0      14.0   \n",
       "153262         SMAGP       heart        58.0        83        28.0      56.0   \n",
       "153263         SMAGP      kidney       142.0       116       176.0     151.0   \n",
       "153463       SMIM11A      kidney       226.0       195       237.0     492.0   \n",
       "153541       SMIM34A  cerebellum         2.0         4        15.0      11.0   \n",
       "153787        SNAPC5       heart       412.0       170       465.0     502.0   \n",
       "153788        SNAPC5      kidney       281.0       278       376.0     532.0   \n",
       "153848         SNORC      kidney        23.0        49        50.0      61.0   \n",
       "153849         SNORC       liver        15.0        30        40.0      45.0   \n",
       "154123          SNX4      kidney       488.0       474       793.0    1073.0   \n",
       "154237           SON       heart      2435.0      2252      2956.0    2984.0   \n",
       "154238           SON      kidney      2783.0      2465      3135.0    6061.0   \n",
       "154428         SP100      kidney       544.0       417       903.0     738.0   \n",
       "154438         SP140      kidney        36.0        46        24.0      23.0   \n",
       "154523        SPACA9      kidney       117.0       106       167.0     256.0   \n",
       "155278        SPTAN1      kidney      6955.0      6613     11840.0   11352.0   \n",
       "155647          SS18       heart       265.0       215       459.0     269.0   \n",
       "155648          SS18      kidney       493.0       454       849.0     891.0   \n",
       "155907    ST6GALNAC4       heart       453.0       309       306.0     224.0   \n",
       "156324        STKLD1       liver        19.0        25        12.0      15.0   \n",
       "156608        STXBP5      kidney       189.0       196       273.0     286.0   \n",
       "156857         SURF1       heart       465.0       314       505.0     536.0   \n",
       "156858         SURF1      kidney       868.0       665      1542.0    2099.0   \n",
       "156861         SURF2  cerebellum       499.0       465       771.0     542.0   \n",
       "156863         SURF2      kidney       321.0       275       611.0     382.0   \n",
       "156864         SURF2       liver       150.0       140       933.0     183.0   \n",
       "156868         SURF4      kidney      2830.0      1901      4162.0    3921.0   \n",
       "156873         SURF6      kidney       452.0       439       729.0     651.0   \n",
       "156874         SURF6       liver       189.0       165       627.0     335.0   \n",
       "156927          SV2B       heart        37.0        51      1740.0    5575.0   \n",
       "156952          SVIP       heart       111.0        38       168.0     192.0   \n",
       "156953          SVIP      kidney       192.0       234       338.0     510.0   \n",
       "156954          SVIP       liver       194.0       129        90.0      50.0   \n",
       "156974          SWI5       liver       283.0       222       178.0     117.0   \n",
       "157056           SYK  cerebellum        51.0        72       370.0     104.0   \n",
       "157057           SYK       heart       101.0        97       203.0     145.0   \n",
       "157058           SYK      kidney       702.0       533       918.0     841.0   \n",
       "157114         SYNE3       liver        69.0        61       162.0     146.0   \n",
       "157403         TACR1      kidney         2.0         2         3.0       5.0   \n",
       "157513         TAF4B      kidney        84.0        84        93.0     210.0   \n",
       "157889       TBC1D13       liver       355.0       264       627.0     442.0   \n",
       "158371          TCHH  cerebellum         4.0         5        33.0      11.0   \n",
       "158373          TCHH      kidney         5.0         5         3.0       4.0   \n",
       "158572         TDRKH       heart        27.0         4        46.0      31.0   \n",
       "158573         TDRKH      kidney        47.0        35        83.0      52.0   \n",
       "158901         TEX35  cerebellum        31.0        45        12.0      18.0   \n",
       "159021         TFCP2  cerebellum       745.0       935       707.0     461.0   \n",
       "159343         THEM5      kidney         9.0         9        60.0      35.0   \n",
       "159547         TIGD1       heart        18.0        32        69.0      37.0   \n",
       "159548         TIGD1      kidney       102.0        85       145.0     116.0   \n",
       "160259       TMEM117       liver       159.0        77        70.0      50.0   \n",
       "161157       TMEM50B       heart       402.0       401       529.0     564.0   \n",
       "161581          TMX3  cerebellum       727.0       845       701.0     407.0   \n",
       "161838         TNKS2      kidney       639.0       587      1169.0    1486.0   \n",
       "162533       TRAM1L1      kidney        16.0        17        58.0      39.0   \n",
       "162818        TRIM32      kidney       230.0       181       234.0     375.0   \n",
       "165288         TRUB2      kidney       331.0       393       418.0     864.0   \n",
       "165297          TSC1       heart       660.0       460       731.0    1059.0   \n",
       "165298          TSC1      kidney      1171.0       786      1405.0    1956.0   \n",
       "165991         TTYH1  cerebellum      2303.0      2501      2015.0     787.0   \n",
       "165992         TTYH1       heart        24.0        17         7.0       6.0   \n",
       "166291       TXNDC16  cerebellum       398.0       481       315.0     171.0   \n",
       "166294       TXNDC16       liver        69.0        48        32.0      42.0   \n",
       "167156          UMPS  cerebellum       382.0       593       345.0     297.0   \n",
       "167158          UMPS      kidney       280.0       303       359.0     559.0   \n",
       "167228         UNC5D      kidney       148.0        59       240.0     217.0   \n",
       "167383       UQCRFS1      kidney       570.0       688       689.0    1306.0   \n",
       "167384       UQCRFS1       liver       440.0       275       258.0     204.0   \n",
       "167792         UTP23       heart        80.0        83       136.0     124.0   \n",
       "167793         UTP23      kidney        86.0        79       121.0     175.0   \n",
       "168054         VEGFD       liver       112.0        43        12.0       9.0   \n",
       "168792         WDR34       heart       360.0       269       255.0     249.0   \n",
       "168794         WDR34       liver       898.0       759       439.0     658.0   \n",
       "169956         ZBTB1  cerebellum       189.0       308       135.0     124.0   \n",
       "169957         ZBTB1       heart       266.0       248       292.0     302.0   \n",
       "169959         ZBTB1       liver       379.0       289       240.0     120.0   \n",
       "170022        ZBTB25       heart        90.0        45       159.0     110.0   \n",
       "170023        ZBTB25      kidney        88.0        56        94.0     119.0   \n",
       "170024        ZBTB25       liver        49.0        30        72.0      55.0   \n",
       "170383       ZDHHC12      kidney       602.0       341      1006.0     642.0   \n",
       "170498          ZER1      kidney      2029.0      1198      2316.0    2144.0   \n",
       "171061        ZNF148  cerebellum       956.0      1230       891.0     645.0   \n",
       "171062        ZNF148       heart       462.0       314       641.0     820.0   \n",
       "171063        ZNF148      kidney       391.0       420       586.0     990.0   \n",
       "171717       ZNF385D       heart        29.0        25        33.0      61.0   \n",
       "172518        ZNF622      kidney       352.0       335       764.0     587.0   \n",
       "173086       ZNF804A  cerebellum       119.0        89        50.0      65.0   \n",
       "173088       ZNF804A      kidney         3.0         1         6.0       5.0   \n",
       "173091       ZNF804B  cerebellum        47.0        78        22.0      11.0   \n",
       "173387         ZRSR2       heart       318.0       205       329.0     326.0   \n",
       "\n",
       "        ppn_delta  ptr_delta  species_diff  \n",
       "1282         16.0        3.0           6.0  \n",
       "1283         17.0      306.0           1.0  \n",
       "1284        116.0        1.0           2.0  \n",
       "1437        563.0       34.0          62.0  \n",
       "1439         47.0      287.0          50.0  \n",
       "2119         33.0      160.0          39.0  \n",
       "2163         12.0      370.0         118.0  \n",
       "2413         10.0        7.0          97.0  \n",
       "2471         40.0       38.0          39.0  \n",
       "2474         18.0       14.0          16.0  \n",
       "3221         57.0       22.0           6.0  \n",
       "3223         10.0        2.0           5.0  \n",
       "3224         10.0        5.0           7.0  \n",
       "3232         78.0       27.0          20.0  \n",
       "3234         23.0       20.0          79.0  \n",
       "3886         33.0       71.0         131.0  \n",
       "3889         79.0       46.0          43.0  \n",
       "4352          8.0      134.0          11.0  \n",
       "5086         16.0       82.0          27.0  \n",
       "5087          2.0       11.0           8.0  \n",
       "5262         49.0       71.0          31.0  \n",
       "5263        139.0      380.0        1665.0  \n",
       "5457          4.0        5.0           8.0  \n",
       "5669         31.0       18.0           4.0  \n",
       "5736         19.0       45.0          65.0  \n",
       "5811        664.0      618.0         133.0  \n",
       "6323         34.0      187.0         201.0  \n",
       "6488         18.0      180.0          35.0  \n",
       "6579          0.0        2.0           5.0  \n",
       "6956        942.0      218.0         188.0  \n",
       "7498        677.0    10191.0        2312.0  \n",
       "7501          2.0        7.0          27.0  \n",
       "7502        214.0       32.0          70.0  \n",
       "7503        608.0     6499.0         982.0  \n",
       "7558         28.0        3.0          10.0  \n",
       "7779         21.0        4.0          16.0  \n",
       "8192         61.0      256.0         112.0  \n",
       "8193        259.0      150.0         192.0  \n",
       "9362         10.0       13.0           3.0  \n",
       "9364         12.0        3.0           6.0  \n",
       "9556          3.0       10.0           8.0  \n",
       "9557          1.0        1.0          15.0  \n",
       "9558         10.0       11.0          15.0  \n",
       "9559          2.0        3.0           3.0  \n",
       "9724        899.0     1372.0         411.0  \n",
       "9779          8.0        8.0           4.0  \n",
       "10367        74.0       16.0          52.0  \n",
       "10503         8.0      137.0          56.0  \n",
       "10532      1233.0      378.0         216.0  \n",
       "10533       245.0     1382.0          53.0  \n",
       "10596        66.0      175.0           4.0  \n",
       "10722        10.0       13.0          14.0  \n",
       "10756       342.0      157.0         531.0  \n",
       "10757         8.0       20.0          13.0  \n",
       "11531        39.0      114.0         223.0  \n",
       "13437        28.0        8.0         246.0  \n",
       "13438        44.0       48.0          60.0  \n",
       "13534         1.0        9.0          17.0  \n",
       "14076        41.0       36.0         357.0  \n",
       "14106        14.0     1015.0         871.0  \n",
       "14462       114.0       21.0         119.0  \n",
       "14463       104.0       79.0          71.0  \n",
       "14464         4.0        6.0          10.0  \n",
       "15242         0.0        0.0           1.0  \n",
       "15317         6.0       70.0         163.0  \n",
       "15319         7.0       63.0          10.0  \n",
       "15352       572.0      206.0         126.0  \n",
       "15353        77.0      503.0         258.0  \n",
       "15493       419.0      330.0         502.0  \n",
       "15494       102.0      681.0         157.0  \n",
       "15956        10.0        3.0          12.0  \n",
       "15959         3.0        4.0           3.0  \n",
       "16158       156.0     1342.0         509.0  \n",
       "16159        20.0       29.0          13.0  \n",
       "16308       207.0      216.0         267.0  \n",
       "16471       244.0        6.0         179.0  \n",
       "16472        60.0       22.0          37.0  \n",
       "16474       175.0       56.0         200.0  \n",
       "17183       270.0     1064.0         293.0  \n",
       "17478        55.0      235.0          54.0  \n",
       "17479       137.0       52.0          19.0  \n",
       "19421         5.0        0.0           3.0  \n",
       "19424       211.0       39.0          19.0  \n",
       "19467        16.0       87.0          38.0  \n",
       "19598       143.0     1533.0        2265.0  \n",
       "19606        19.0       19.0           1.0  \n",
       "19608         4.0        0.0          20.0  \n",
       "19617        10.0        2.0           1.0  \n",
       "19666         1.0        2.0           1.0  \n",
       "19778         5.0        6.0          34.0  \n",
       "20156         6.0       87.0          41.0  \n",
       "20157        18.0       11.0           3.0  \n",
       "20158        10.0       44.0          14.0  \n",
       "20551       612.0       34.0         195.0  \n",
       "20552        38.0       68.0         117.0  \n",
       "20553        52.0      324.0          66.0  \n",
       "20723         8.0       69.0          17.0  \n",
       "20727        22.0       16.0         174.0  \n",
       "20728        14.0       48.0          43.0  \n",
       "20729        42.0      229.0          37.0  \n",
       "20987        78.0      233.0          25.0  \n",
       "20988        97.0      386.0         197.0  \n",
       "21026      1580.0      482.0         413.0  \n",
       "21028        75.0      346.0          52.0  \n",
       "21068        21.0      357.0           5.0  \n",
       "21296        55.0      205.0          33.0  \n",
       "21297      1668.0     1739.0        1184.0  \n",
       "21298        31.0       20.0          99.0  \n",
       "21299       515.0      203.0          78.0  \n",
       "21303       372.0     3641.0        1632.0  \n",
       "21652        28.0       11.0          11.0  \n",
       "21653        15.0       38.0          19.0  \n",
       "21654        15.0       15.0           1.0  \n",
       "21772         9.0       53.0          80.0  \n",
       "21958         0.0       40.0          13.0  \n",
       "22068        90.0      382.0           1.0  \n",
       "22932         6.0        4.0           9.0  \n",
       "22933         5.0        7.0          16.0  \n",
       "23613         3.0     1201.0         272.0  \n",
       "23673       263.0      610.0          11.0  \n",
       "24169        64.0       85.0          43.0  \n",
       "24461        49.0       84.0         141.0  \n",
       "24463         1.0        4.0           2.0  \n",
       "24841        40.0      158.0         173.0  \n",
       "24863         3.0       20.0          22.0  \n",
       "24977         1.0        1.0           2.0  \n",
       "25136         2.0       21.0           1.0  \n",
       "25138        17.0      176.0         181.0  \n",
       "25139       137.0      246.0         177.0  \n",
       "25648       303.0      546.0         554.0  \n",
       "26073         6.0      113.0           8.0  \n",
       "26162       128.0       66.0           4.0  \n",
       "26163        24.0      197.0         111.0  \n",
       "26958        29.0      390.0         190.0  \n",
       "27797         9.0       45.0           1.0  \n",
       "27877        16.0       12.0           9.0  \n",
       "27879        36.0       14.0          11.0  \n",
       "28091        36.0        5.0          51.0  \n",
       "28092         4.0       10.0           8.0  \n",
       "28351        29.0      138.0         136.0  \n",
       "28741       194.0      175.0          22.0  \n",
       "29036        24.0       20.0         433.0  \n",
       "29037        57.0       28.0          18.0  \n",
       "29039         0.0       84.0           6.0  \n",
       "29666         5.0       20.0           1.0  \n",
       "29669       191.0      118.0          35.0  \n",
       "29799        73.0       97.0          19.0  \n",
       "29879       142.0       24.0          93.0  \n",
       "30382       607.0      184.0         218.0  \n",
       "30383       400.0      618.0         188.0  \n",
       "30667       120.0       52.0          28.0  \n",
       "30668        61.0      126.0         194.0  \n",
       "30808       107.0      447.0         372.0  \n",
       "30809       542.0       24.0         266.0  \n",
       "31116       112.0      182.0          30.0  \n",
       "31244       175.0     2192.0         266.0  \n",
       "31332       318.0        5.0          21.0  \n",
       "31333        39.0      357.0          49.0  \n",
       "31541       277.0       59.0         371.0  \n",
       "31543        36.0      169.0          39.0  \n",
       "31666        88.0       58.0         160.0  \n",
       "31914         1.0        0.0           3.0  \n",
       "32076         5.0       12.0          14.0  \n",
       "32401        60.0       39.0          31.0  \n",
       "32497         5.0       63.0         129.0  \n",
       "32746       556.0       36.0         185.0  \n",
       "32747        31.0        6.0          92.0  \n",
       "32748        35.0      181.0         191.0  \n",
       "33606       153.0      226.0          68.0  \n",
       "33607      1120.0     6025.0         309.0  \n",
       "33609         1.0      758.0         127.0  \n",
       "34886         5.0       27.0          16.0  \n",
       "34887         0.0        3.0           3.0  \n",
       "34888        12.0       28.0           2.0  \n",
       "35122       325.0       62.0          81.0  \n",
       "35276         0.0        1.0           6.0  \n",
       "35278         0.0        7.0          15.0  \n",
       "35556        43.0      219.0           6.0  \n",
       "35558         6.0      133.0          18.0  \n",
       "35818       257.0      447.0         146.0  \n",
       "36948        35.0      168.0          94.0  \n",
       "37413        23.0      102.0           4.0  \n",
       "37571       114.0      306.0          27.0  \n",
       "37573       290.0     1029.0         833.0  \n",
       "37713       134.0      283.0         269.0  \n",
       "37901       171.0      147.0         213.0  \n",
       "37902       119.0      126.0          70.0  \n",
       "37903       127.0      333.0          31.0  \n",
       "38266         8.0        3.0           2.0  \n",
       "38648       310.0       40.0          55.0  \n",
       "39183         1.0        2.0          11.0  \n",
       "39527       188.0       35.0          55.0  \n",
       "39528       295.0      287.0         859.0  \n",
       "39529       130.0     1119.0          79.0  \n",
       "40558       163.0      777.0          32.0  \n",
       "40559       122.0       11.0         119.0  \n",
       "40821       113.0      100.0          36.0  \n",
       "41206        49.0       25.0           7.0  \n",
       "41211      3374.0      704.0          63.0  \n",
       "41212       608.0      319.0          85.0  \n",
       "41214       695.0     1331.0         462.0  \n",
       "41216         5.0       10.0           3.0  \n",
       "41217         4.0       10.0           8.0  \n",
       "41219        35.0      224.0          15.0  \n",
       "41527         7.0       10.0           4.0  \n",
       "41528        14.0        8.0          37.0  \n",
       "41656        34.0       41.0         168.0  \n",
       "41657        58.0       45.0          49.0  \n",
       "41658         4.0       75.0          52.0  \n",
       "41659        11.0      151.0           5.0  \n",
       "44053         1.0       11.0           4.0  \n",
       "44556         1.0        6.0           6.0  \n",
       "44557         2.0        6.0           2.0  \n",
       "47211        26.0        7.0          60.0  \n",
       "47212        18.0        2.0           4.0  \n",
       "48396         1.0        0.0           2.0  \n",
       "51763         8.0       28.0          11.0  \n",
       "51764         1.0        2.0           8.0  \n",
       "58606        46.0       15.0          24.0  \n",
       "86693         1.0        3.0           6.0  \n",
       "86694         0.0        1.0           2.0  \n",
       "110621        5.0        1.0           2.0  \n",
       "110624        1.0        4.0           4.0  \n",
       "110681       29.0       13.0          71.0  \n",
       "111017        7.0       78.0           3.0  \n",
       "111516        2.0        3.0           2.0  \n",
       "111524       10.0        2.0          29.0  \n",
       "112122        3.0        4.0           1.0  \n",
       "112236        2.0      278.0          66.0  \n",
       "112239       57.0       19.0          43.0  \n",
       "112974        0.0        0.0           1.0  \n",
       "113711        4.0        2.0           2.0  \n",
       "116148       45.0        7.0          61.0  \n",
       "117298        9.0       10.0           4.0  \n",
       "120587      211.0      573.0          45.0  \n",
       "120589      153.0      788.0         316.0  \n",
       "121748        3.0      101.0         112.0  \n",
       "121749       75.0       41.0         171.0  \n",
       "122108       22.0      203.0          27.0  \n",
       "122296       52.0      321.0         223.0  \n",
       "123046        0.0        2.0           9.0  \n",
       "123049       28.0        8.0          14.0  \n",
       "127716       22.0      194.0         108.0  \n",
       "127718       18.0      127.0          67.0  \n",
       "128001        2.0       49.0         310.0  \n",
       "128004      250.0      118.0         158.0  \n",
       "128186       75.0      231.0           4.0  \n",
       "128321       61.0       55.0           4.0  \n",
       "128323       93.0      272.0          67.0  \n",
       "128793       80.0     2380.0         231.0  \n",
       "129543      531.0     1435.0         270.0  \n",
       "129992       12.0       22.0           3.0  \n",
       "129993       11.0       21.0          57.0  \n",
       "129994        6.0      114.0          12.0  \n",
       "130626      171.0       32.0          13.0  \n",
       "130986       44.0      114.0         191.0  \n",
       "130988      135.0      488.0          30.0  \n",
       "131027        1.0        3.0           3.0  \n",
       "131028        7.0       34.0          32.0  \n",
       "131029       46.0        1.0           1.0  \n",
       "131264     4026.0      197.0          58.0  \n",
       "131787        7.0        2.0           5.0  \n",
       "131789       49.0        6.0           6.0  \n",
       "132714        9.0       18.0           5.0  \n",
       "133411       15.0       12.0          48.0  \n",
       "133413       23.0      266.0           7.0  \n",
       "133606       34.0        5.0           1.0  \n",
       "133608        2.0       14.0          11.0  \n",
       "133702        8.0       53.0          93.0  \n",
       "133703       30.0      202.0         432.0  \n",
       "135136       44.0       97.0          47.0  \n",
       "135138       76.0       69.0           5.0  \n",
       "135152       44.0        4.0           7.0  \n",
       "135182        4.0        2.0           1.0  \n",
       "135357       13.0        5.0           6.0  \n",
       "135358        7.0       12.0          10.0  \n",
       "135658       57.0       28.0           4.0  \n",
       "135659       39.0        2.0          14.0  \n",
       "135953       70.0      223.0         106.0  \n",
       "135987       23.0        7.0          18.0  \n",
       "135988      421.0     1133.0          61.0  \n",
       "136549       46.0        4.0           2.0  \n",
       "138267      324.0       28.0         159.0  \n",
       "138268      258.0      232.0         560.0  \n",
       "138269     1188.0      559.0          92.0  \n",
       "138708        5.0       42.0          19.0  \n",
       "138907      172.0       63.0         212.0  \n",
       "138908       27.0       37.0         100.0  \n",
       "139342       72.0       12.0          73.0  \n",
       "139344       51.0      927.0         197.0  \n",
       "139506       30.0      104.0          34.0  \n",
       "139608       33.0       10.0         197.0  \n",
       "139624      126.0       11.0          22.0  \n",
       "139626        5.0        0.0           9.0  \n",
       "140903        5.0        5.0           7.0  \n",
       "140921       34.0      133.0          17.0  \n",
       "140923       12.0       66.0           4.0  \n",
       "140924     3778.0      302.0         663.0  \n",
       "141271       69.0      331.0         133.0  \n",
       "141274     1110.0      255.0          58.0  \n",
       "141581      949.0     1785.0           6.0  \n",
       "141647      365.0      292.0          99.0  \n",
       "141648      122.0       70.0         317.0  \n",
       "141649       55.0       30.0          71.0  \n",
       "141873      237.0      141.0         327.0  \n",
       "141874      204.0      924.0          38.0  \n",
       "142546       94.0       59.0          28.0  \n",
       "142548       23.0      208.0          26.0  \n",
       "142553      188.0       16.0         203.0  \n",
       "142801       10.0        5.0          10.0  \n",
       "142804        6.0        2.0           8.0  \n",
       "142823       15.0       14.0          14.0  \n",
       "142843      895.0      181.0          17.0  \n",
       "143092       37.0       46.0          44.0  \n",
       "143591      249.0      164.0          39.0  \n",
       "143641     1769.0     1648.0         750.0  \n",
       "143642       55.0      118.0          21.0  \n",
       "143644      133.0      729.0          47.0  \n",
       "144231      110.0       40.0         134.0  \n",
       "144232       79.0       28.0          20.0  \n",
       "144233       26.0       93.0          60.0  \n",
       "144234       41.0      129.0         393.0  \n",
       "144468       89.0      192.0          29.0  \n",
       "145021        4.0      540.0         967.0  \n",
       "145388      191.0      525.0         338.0  \n",
       "145434       19.0      506.0          45.0  \n",
       "146867       12.0       10.0           9.0  \n",
       "146868       13.0        1.0           8.0  \n",
       "147237        4.0        4.0           8.0  \n",
       "147547      112.0      477.0         284.0  \n",
       "147968        5.0      240.0         333.0  \n",
       "148012       28.0       12.0           8.0  \n",
       "148013        3.0       47.0           3.0  \n",
       "148014       22.0       13.0           1.0  \n",
       "148157      126.0       84.0          13.0  \n",
       "148162      704.0      225.0         238.0  \n",
       "148163      548.0      737.0         176.0  \n",
       "148412       23.0        7.0           1.0  \n",
       "148542      572.0       78.0          25.0  \n",
       "148543      158.0      165.0         222.0  \n",
       "149298       30.0      105.0         122.0  \n",
       "149832        1.0      110.0           1.0  \n",
       "150012      711.0      256.0         114.0  \n",
       "150013       39.0      139.0         571.0  \n",
       "151218      376.0      188.0         150.0  \n",
       "151219       10.0      512.0          78.0  \n",
       "151259       12.0       49.0          73.0  \n",
       "151657       28.0        6.0          12.0  \n",
       "151988       79.0       51.0         154.0  \n",
       "152092       37.0       29.0           4.0  \n",
       "152284        0.0        0.0           1.0  \n",
       "153053       46.0        4.0          24.0  \n",
       "153261        3.0        6.0           7.0  \n",
       "153262       25.0       28.0           2.0  \n",
       "153263       26.0       25.0           9.0  \n",
       "153463       31.0      255.0          11.0  \n",
       "153541        2.0        4.0           7.0  \n",
       "153787      242.0       37.0          53.0  \n",
       "153788        3.0      156.0          95.0  \n",
       "153848       26.0       11.0           1.0  \n",
       "153849       15.0        5.0          10.0  \n",
       "154123       14.0      280.0         305.0  \n",
       "154237      183.0       28.0         521.0  \n",
       "154238      318.0     2926.0         352.0  \n",
       "154428      127.0      165.0         194.0  \n",
       "154438       10.0        1.0          12.0  \n",
       "154523       11.0       89.0          50.0  \n",
       "155278      342.0      488.0        4397.0  \n",
       "155647       50.0      190.0           4.0  \n",
       "155648       39.0       42.0         356.0  \n",
       "155907      144.0       82.0           3.0  \n",
       "156324        6.0        3.0           4.0  \n",
       "156608        7.0       13.0          77.0  \n",
       "156857      151.0       31.0          40.0  \n",
       "156858      203.0      557.0         674.0  \n",
       "156861       34.0      229.0          43.0  \n",
       "156863       46.0      229.0          61.0  \n",
       "156864       10.0      750.0          33.0  \n",
       "156868      929.0      241.0        1091.0  \n",
       "156873       13.0       78.0         199.0  \n",
       "156874       24.0      292.0         146.0  \n",
       "156927       14.0     3835.0        1689.0  \n",
       "156952       73.0       24.0          57.0  \n",
       "156953       42.0      172.0         104.0  \n",
       "156954       65.0       40.0          39.0  \n",
       "156974       61.0       61.0          44.0  \n",
       "157056       21.0      266.0          32.0  \n",
       "157057        4.0       58.0          44.0  \n",
       "157058      169.0       77.0         139.0  \n",
       "157114        8.0       16.0          77.0  \n",
       "157403        0.0        2.0           1.0  \n",
       "157513        0.0      117.0           9.0  \n",
       "157889       91.0      185.0          87.0  \n",
       "158371        1.0       22.0           6.0  \n",
       "158373        0.0        1.0           1.0  \n",
       "158572       23.0       15.0           4.0  \n",
       "158573       12.0       31.0           5.0  \n",
       "158901       14.0        6.0          13.0  \n",
       "159021      190.0      246.0          38.0  \n",
       "159343        0.0       25.0          26.0  \n",
       "159547       14.0       32.0           5.0  \n",
       "159548       17.0       29.0          14.0  \n",
       "160259       82.0       20.0           7.0  \n",
       "161157        1.0       35.0         127.0  \n",
       "161581      118.0      294.0          26.0  \n",
       "161838       52.0      317.0         530.0  \n",
       "162533        1.0       19.0          22.0  \n",
       "162818       49.0      141.0           4.0  \n",
       "165288       62.0      446.0          25.0  \n",
       "165297      200.0      328.0          71.0  \n",
       "165298      385.0      551.0         234.0  \n",
       "165991      198.0     1228.0         288.0  \n",
       "165992        7.0        1.0          10.0  \n",
       "166291       83.0      144.0          83.0  \n",
       "166294       21.0       10.0           6.0  \n",
       "167156      211.0       48.0          37.0  \n",
       "167158       23.0      200.0          56.0  \n",
       "167228       89.0       23.0          69.0  \n",
       "167383      118.0      617.0           1.0  \n",
       "167384      165.0       54.0          17.0  \n",
       "167792        3.0       12.0          41.0  \n",
       "167793        7.0       54.0          35.0  \n",
       "168054       69.0        3.0          31.0  \n",
       "168792       91.0        6.0          14.0  \n",
       "168794      139.0      219.0         101.0  \n",
       "169956      119.0       11.0          54.0  \n",
       "169957       18.0       10.0          26.0  \n",
       "169959       90.0      120.0          49.0  \n",
       "170022       45.0       49.0          20.0  \n",
       "170023       32.0       25.0           6.0  \n",
       "170024       19.0       17.0           6.0  \n",
       "170383      261.0      364.0          40.0  \n",
       "170498      831.0      172.0         115.0  \n",
       "171061      274.0      246.0          65.0  \n",
       "171062      148.0      179.0         179.0  \n",
       "171063       29.0      404.0         166.0  \n",
       "171717        4.0       28.0           4.0  \n",
       "172518       17.0      177.0         235.0  \n",
       "173086       30.0       15.0          24.0  \n",
       "173088        2.0        1.0           2.0  \n",
       "173091       31.0       11.0          25.0  \n",
       "173387      113.0        3.0           8.0  "
      ]
     },
     "execution_count": 354,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "species_different_gene_expression[species_different_gene_expression['species_diff'].notna()]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2fd2852c",
   "metadata": {},
   "source": [
    "## Cell Type Comparison <a class = 'anchor' id = 'celltypecomparison'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "67991449",
   "metadata": {},
   "source": [
    "Are the cell type specific predictions variable across cell types for the reference sequence?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "id": "1ef39ec4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#reference_comparisons_header = ['cell_type_1','cell_type_2','chr','window_start','mse','spearman']\n",
    "#reference_comparisons = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/comparisons/reference/all_reference_comparisons.txt', sep = '\\t', names = reference_comparisons_header)\n",
    "#reference_comparisons['window'] = reference_comparisons['chr'] + '_' + reference_comparisons['window_start'].astype(str)\n",
    "#reference_comparisons = reference_comparisons[['cell_type_1','cell_type_2','chr','window_start','window','mse','spearman']]\n",
    "#reference_comparisons.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "id": "c61a3bc0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#excluded_header = ['chr','start','end','N_missing']\n",
    "#excluded = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/metadata/panTro6_excluded_windows.bed', sep = '\\t', header = None, names = excluded_header)\n",
    "#excluded['window'] = excluded['chr'] + '_' + excluded['start'].astype(str)\n",
    "#excluded_windows = excluded['window'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "id": "ca9a879d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#reference_comparisons = reference_comparisons[~(reference_comparisons['window'].isin(excluded_windows))]\n",
    "#reference_comparisons.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "abba6eb8",
   "metadata": {},
   "source": [
    "Let's save this dataframe for plotting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "id": "f0364991",
   "metadata": {},
   "outputs": [],
   "source": [
    "#reference_comparisons.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/dataframes/reference_cell_type_comparisons.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "id": "38eac4fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(reference_comparisons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "id": "29c0f6a7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#reference_comparisons.groupby(['cell_type_1','cell_type_2'])['spearman'].mean().to_frame('Mean Rho')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7b975427",
   "metadata": {},
   "source": [
    "Most of the maps for the different cell types are nearly identical when predicting the reference sequence."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4bc8711",
   "metadata": {},
   "source": [
    "Let's compare sample predictions from HFF to GM12878. Load the data and run a correlation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "id": "b8f21a3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#GM12878_comparisons = pd.read_csv('/wynton/group/capra/projects/pan_3d_genome/data/dataframes/GM12878_comparisons.txt', sep = '\\t', header = 0)\n",
    "#GM12878_comparisons.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "id": "0a5d6118",
   "metadata": {},
   "outputs": [],
   "source": [
    "#rho, p = spearmanr(comparisons['divergence'], GM12878_comparisons['divergence'])\n",
    "#print(rho, p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "id": "6a4bf53f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#HFF_divergence = comparisons[['ind1','ind2','window','divergence']].copy()\n",
    "#HFF_divergence['cell_type'] = 'HFF'\n",
    "#GM12878_divergence = GM12878_comparisons[['ind1','ind2','window','divergence']].copy()\n",
    "#GM12878_divergence['cell_type'] = 'GM12878'\n",
    "#cell_type_correlation = HFF_divergence.merge(GM12878_divergence, how = 'outer', on = ['ind1','ind2','window'])\n",
    "#cell_type_correlation = cell_type_correlation[['ind1','ind2','window','cell_type_x','divergence_x','cell_type_y','divergence_y']]\n",
    "#cell_type_correlation.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "id": "5d422ff9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(cell_type_correlation)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18f8199e",
   "metadata": {},
   "source": [
    "Save this dataframe for plotting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "id": "449aadbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#cell_type_correlation.to_csv('/wynton/group/capra/projects/pan_3d_genome/data/dataframes/cell_type_correlation.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "199bdbea",
   "metadata": {},
   "source": [
    "After plotting, there is a noticeable cluster of windows with low HFF divergence but high GM12878 divergence. Let's take a look."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "id": "ba766484",
   "metadata": {},
   "outputs": [],
   "source": [
    "#high_GM12878_windows = cell_type_correlation[(cell_type_correlation['divergence_x'] < 0.075) & (cell_type_correlation['divergence_y'] > 0.1)]\n",
    "#high_GM12878_windows.head(200)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "id": "e8e4e315",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(high_GM12878_windows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 368,
   "id": "3e1caaf6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#high_GM12878_windows.groupby('window')['divergence_y'].count().to_frame()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "987e1ef8",
   "metadata": {},
   "source": [
    "Two windows stick out in particular. Are these maps very different in the reference sequence?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "id": "0ee41509",
   "metadata": {},
   "outputs": [],
   "source": [
    "#reference_comparisons[(reference_comparisons['cell_type_1'] == 'GM12878') & (reference_comparisons['cell_type_2'] == 'HFF') & (reference_comparisons['window'] == 'chr4_94371840')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "id": "b0e1f351",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#reference_comparisons[(reference_comparisons['cell_type_1'] == 'GM12878') & (reference_comparisons['cell_type_2'] == 'HFF') & (reference_comparisons['window'] == 'chr11_23068672')]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
